--- decompiled\net\minecraft\block\state\pattern\BlockStateHelper.java
+++ src\main\java\net\minecraft\block\state\pattern\BlockStateHelper.java
@@ -1,19 +1,22 @@
 package net.minecraft.block.state.pattern;
 
-import com.google.common.base.Predicate;
-import com.google.common.collect.Maps;
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Map.Entry;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.state.BlockState;
 import net.minecraft.block.state.IBlockState;
 
+import com.google.common.base.Predicate;
+import com.google.common.collect.Maps;
+
 public class BlockStateHelper implements Predicate
 {
+
     private final BlockState blockstate;
     private final Map propertyPredicates = Maps.newHashMap();
     private static final String __OBFID = "CL_00002019";
 
     private BlockStateHelper(BlockState blockStateIn)
