--- decompiled\net\minecraft\block\state\pattern\BlockPattern.java
+++ src\main\java\net\minecraft\block\state\pattern\BlockPattern.java
@@ -1,19 +1,22 @@
 package net.minecraft.block.state.pattern;
 
-import com.google.common.base.Predicate;
-import com.google.common.cache.CacheBuilder;
-import com.google.common.cache.LoadingCache;
 import java.util.Iterator;
+
 import net.minecraft.block.state.BlockWorldState;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.Vec3i;
 import net.minecraft.world.World;
 
+import com.google.common.base.Predicate;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.LoadingCache;
+
 public class BlockPattern
 {
+
     private final Predicate[][][] blockMatches;
     private final int fingerLength;
     private final int thumbLength;
     private final int palmLength;
     private static final String __OBFID = "CL_00002024";
@@ -125,10 +128,11 @@
         }
     }
 
     static class CacheLoader extends com.google.common.cache.CacheLoader
         {
+
             private final World world;
             private static final String __OBFID = "CL_00002023";
 
             public CacheLoader(World worldIn)
             {
@@ -146,10 +150,11 @@
             }
         }
 
     public static class PatternHelper
         {
+
             private final BlockPos pos;
             private final EnumFacing finger;
             private final EnumFacing thumb;
             private final LoadingCache lcache;
             private static final String __OBFID = "CL_00002022";
