--- decompiled\net\minecraft\block\state\pattern\FactoryBlockPattern.java
+++ src\main\java\net\minecraft\block\state\pattern\FactoryBlockPattern.java
@@ -1,23 +1,26 @@
 package net.minecraft.block.state.pattern;
 
-import com.google.common.base.Joiner;
-import com.google.common.base.Predicate;
-import com.google.common.base.Predicates;
-import com.google.common.collect.Lists;
-import com.google.common.collect.Maps;
 import java.lang.reflect.Array;
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
+
 import org.apache.commons.lang3.ArrayUtils;
 import org.apache.commons.lang3.StringUtils;
 
+import com.google.common.base.Joiner;
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
 public class FactoryBlockPattern
 {
+
     private static final Joiner COMMA_JOIN = Joiner.on(",");
     private final List depth = Lists.newArrayList();
     private final Map symbolMap = Maps.newHashMap();
     private int aisleHeight;
     private int rowWidth;
@@ -26,11 +29,11 @@
     private FactoryBlockPattern()
     {
         this.symbolMap.put(' ', Predicates.alwaysTrue());
     }
 
-    public FactoryBlockPattern aisle(String ... aisle)
+    public FactoryBlockPattern aisle(String... aisle)
     {
         if (!ArrayUtils.isEmpty(aisle) && !StringUtils.isEmpty(aisle[0]))
         {
             if (this.depth.isEmpty())
             {
