--- decompiled\net\minecraft\server\ServerEula.java
+++ src\main\java\net\minecraft\server\ServerEula.java
@@ -2,76 +2,60 @@
 
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.util.Properties;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
+
 import org.apache.commons.io.IOUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-@SideOnly(Side.SERVER)
-public class ServerEula
-{
+public class ServerEula {
+
     private static final Logger LOG = LogManager.getLogger();
     private final File eulaFile;
     private final boolean acceptedEULA;
     private static final String __OBFID = "CL_00001911";
 
-    public ServerEula(File eulaFile)
-    {
+    public ServerEula(File eulaFile) {
         this.eulaFile = eulaFile;
         this.acceptedEULA = this.loadEULAFile(eulaFile);
     }
 
-    private boolean loadEULAFile(File inFile)
-    {
+    private boolean loadEULAFile(File inFile) {
         FileInputStream fileinputstream = null;
         boolean flag = false;
 
-        try
-        {
+        try {
             Properties properties = new Properties();
             fileinputstream = new FileInputStream(inFile);
             properties.load(fileinputstream);
             flag = Boolean.parseBoolean(properties.getProperty("eula", "false"));
-        }
-        catch (Exception exception)
-        {
+        } catch (Exception exception) {
             LOG.warn("Failed to load " + inFile);
             this.createEULAFile();
-        }
-        finally
-        {
+        } finally {
             IOUtils.closeQuietly(fileinputstream);
         }
 
         return flag;
     }
 
-    public boolean hasAcceptedEULA()
-    {
+    public boolean hasAcceptedEULA() {
         return this.acceptedEULA;
     }
 
-    public void createEULAFile()
-    {
+    public void createEULAFile() {
         FileOutputStream fileoutputstream = null;
 
-        try
-        {
+        try {
             Properties properties = new Properties();
             fileoutputstream = new FileOutputStream(this.eulaFile);
             properties.setProperty("eula", "false");
             properties.store(fileoutputstream, "By changing the setting below to TRUE you are indicating your agreement to our EULA (https://account.mojang.com/documents/minecraft_eula).");
-        }
-        catch (Exception exception)
-        {
+        } catch (Exception exception) {
             LOG.warn("Failed to save " + this.eulaFile, exception);
-        }
-        finally
-        {
+        } finally {
             IOUtils.closeQuietly(fileoutputstream);
         }
     }
 }
