--- decompiled\net\minecraft\world\gen\structure\MapGenStronghold.java
+++ src\main\java\net\minecraft\world\gen\structure\MapGenStronghold.java
@@ -1,23 +1,26 @@
 package net.minecraft.world.gen.structure;
 
-import com.google.common.collect.Lists;
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
-import java.util.Random;
 import java.util.Map.Entry;
+import java.util.Random;
+
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.ChunkCoordIntPair;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.BiomeGenBase;
 
+import com.google.common.collect.Lists;
+
 public class MapGenStronghold extends MapGenStructure
 {
-    private List field_151546_e;
+
+    public final List field_151546_e;
     private boolean ranBiomeCheck;
     private ChunkCoordIntPair[] structureCoords;
     private double field_82671_h;
     private int field_82672_i;
     private static final String __OBFID = "CL_00000481";
@@ -153,13 +156,16 @@
         return start;
     }
 
     public static class Start extends StructureStart
         {
+
             private static final String __OBFID = "CL_00000482";
 
-            public Start() {}
+            public Start()
+            {
+            }
 
             public Start(World worldIn, Random p_i2067_2_, int p_i2067_3_, int p_i2067_4_)
             {
                 super(p_i2067_3_, p_i2067_4_);
                 StructureStrongholdPieces.prepareStructurePieces();
