--- decompiled\net\minecraft\world\gen\structure\MapGenMineshaft.java
+++ src\main\java\net\minecraft\world\gen\structure\MapGenMineshaft.java
@@ -1,18 +1,22 @@
 package net.minecraft.world.gen.structure;
 
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Map.Entry;
+
 import net.minecraft.util.MathHelper;
 
 public class MapGenMineshaft extends MapGenStructure
 {
+
     private double field_82673_e = 0.004D;
     private static final String __OBFID = "CL_00000443";
 
-    public MapGenMineshaft() {}
+    public MapGenMineshaft()
+    {
+    }
 
     public String getStructureName()
     {
         return "Mineshaft";
     }
