--- decompiled\net\minecraft\world\gen\structure\StructureStart.java
+++ src\main\java\net\minecraft\world\gen\structure\StructureStart.java
@@ -1,24 +1,28 @@
 package net.minecraft.world.gen.structure;
 
 import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.Random;
+
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.world.ChunkCoordIntPair;
 import net.minecraft.world.World;
 
 public abstract class StructureStart
 {
+
     protected LinkedList components = new LinkedList();
     protected StructureBoundingBox boundingBox;
     private int field_143024_c;
     private int field_143023_d;
     private static final String __OBFID = "CL_00000513";
 
-    public StructureStart() {}
+    public StructureStart()
+    {
+    }
 
     public StructureStart(int p_i43002_1_, int p_i43002_2_)
     {
         this.field_143024_c = p_i43002_1_;
         this.field_143023_d = p_i43002_2_;
@@ -80,11 +84,13 @@
         nbttagcompound.setTag("Children", nbttaglist);
         this.func_143022_a(nbttagcompound);
         return nbttagcompound;
     }
 
-    public void func_143022_a(NBTTagCompound p_143022_1_) {}
+    public void func_143022_a(NBTTagCompound p_143022_1_)
+    {
+    }
 
     public void func_143020_a(World worldIn, NBTTagCompound p_143020_2_)
     {
         this.field_143024_c = p_143020_2_.getInteger("ChunkX");
         this.field_143023_d = p_143020_2_.getInteger("ChunkZ");
@@ -102,11 +108,13 @@
         }
 
         this.func_143017_b(p_143020_2_);
     }
 
-    public void func_143017_b(NBTTagCompound p_143017_1_) {}
+    public void func_143017_b(NBTTagCompound p_143017_1_)
+    {
+    }
 
     protected void markAvailableHeight(World worldIn, Random p_75067_2_, int p_75067_3_)
     {
         int j = 63 - p_75067_3_;
         int k = this.boundingBox.getYSize() + 1;
@@ -161,11 +169,13 @@
     public boolean func_175788_a(ChunkCoordIntPair p_175788_1_)
     {
         return true;
     }
 
-    public void func_175787_b(ChunkCoordIntPair p_175787_1_) {}
+    public void func_175787_b(ChunkCoordIntPair p_175787_1_)
+    {
+    }
 
     public int func_143019_e()
     {
         return this.field_143024_c;
     }
