--- decompiled\net\minecraft\world\gen\structure\StructureOceanMonumentPieces.java
+++ src\main\java\net\minecraft\world\gen\structure\StructureOceanMonumentPieces.java
@@ -1,25 +1,28 @@
 package net.minecraft.world.gen.structure;
 
-import com.google.common.collect.Lists;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Random;
+
 import net.minecraft.block.BlockPrismarine;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.IEntityLivingData;
 import net.minecraft.entity.monster.EntityGuardian;
 import net.minecraft.init.Blocks;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
+import com.google.common.collect.Lists;
+
 public class StructureOceanMonumentPieces
 {
+
     private static final String __OBFID = "CL_00001994";
 
     public static void func_175970_a()
     {
         MapGenStructureIO.registerStructureComponent(StructureOceanMonumentPieces.MonumentBuilding.class, "OMB");
@@ -35,13 +38,16 @@
         MapGenStructureIO.registerStructureComponent(StructureOceanMonumentPieces.SimpleTopRoom.class, "OMSimpleT");
     }
 
     public static class DoubleXRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private static final String __OBFID = "CL_00001983";
 
-            public DoubleXRoom() {}
+            public DoubleXRoom()
+            {
+            }
 
             public DoubleXRoom(EnumFacing p_i45597_1_, StructureOceanMonumentPieces.RoomDefinition p_i45597_2_, Random p_i45597_3_)
             {
                 super(1, p_i45597_1_, p_i45597_2_, 2, 1, 1);
             }
@@ -119,13 +125,16 @@
             }
         }
 
     public static class DoubleXYRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private static final String __OBFID = "CL_00001982";
 
-            public DoubleXYRoom() {}
+            public DoubleXYRoom()
+            {
+            }
 
             public DoubleXYRoom(EnumFacing p_i45596_1_, StructureOceanMonumentPieces.RoomDefinition p_i45596_2_, Random p_i45596_3_)
             {
                 super(1, p_i45596_1_, p_i45596_2_, 2, 2, 1);
             }
@@ -256,13 +265,16 @@
             }
         }
 
     public static class DoubleYRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private static final String __OBFID = "CL_00001981";
 
-            public DoubleYRoom() {}
+            public DoubleYRoom()
+            {
+            }
 
             public DoubleYRoom(EnumFacing p_i45595_1_, StructureOceanMonumentPieces.RoomDefinition p_i45595_2_, Random p_i45595_3_)
             {
                 super(1, p_i45595_1_, p_i45595_2_, 1, 2, 1);
             }
@@ -360,13 +372,16 @@
             }
         }
 
     public static class DoubleYZRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private static final String __OBFID = "CL_00001980";
 
-            public DoubleYZRoom() {}
+            public DoubleYZRoom()
+            {
+            }
 
             public DoubleYZRoom(EnumFacing p_i45594_1_, StructureOceanMonumentPieces.RoomDefinition p_i45594_2_, Random p_i45594_3_)
             {
                 super(1, p_i45594_1_, p_i45594_2_, 1, 2, 2);
             }
@@ -500,13 +515,16 @@
             }
         }
 
     public static class DoubleZRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private static final String __OBFID = "CL_00001979";
 
-            public DoubleZRoom() {}
+            public DoubleZRoom()
+            {
+            }
 
             public DoubleZRoom(EnumFacing p_i45593_1_, StructureOceanMonumentPieces.RoomDefinition p_i45593_2_, Random p_i45593_3_)
             {
                 super(1, p_i45593_1_, p_i45593_2_, 1, 1, 2);
             }
@@ -603,13 +621,16 @@
             }
         }
 
     public static class EntryRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private static final String __OBFID = "CL_00001978";
 
-            public EntryRoom() {}
+            public EntryRoom()
+            {
+            }
 
             public EntryRoom(EnumFacing p_i45592_1_, StructureOceanMonumentPieces.RoomDefinition p_i45592_2_)
             {
                 super(1, p_i45592_1_, p_i45592_2_, 1, 1, 1);
             }
@@ -645,13 +666,16 @@
             }
         }
 
     static class FitSimpleRoomHelper implements StructureOceanMonumentPieces.MonumentRoomFitHelper
         {
+
             private static final String __OBFID = "CL_00001987";
 
-            private FitSimpleRoomHelper() {}
+            private FitSimpleRoomHelper()
+            {
+            }
 
             public boolean func_175969_a(StructureOceanMonumentPieces.RoomDefinition p_175969_1_)
             {
                 return true;
             }
@@ -668,13 +692,16 @@
             }
         }
 
     static class FitSimpleRoomTopHelper implements StructureOceanMonumentPieces.MonumentRoomFitHelper
         {
+
             private static final String __OBFID = "CL_00001986";
 
-            private FitSimpleRoomTopHelper() {}
+            private FitSimpleRoomTopHelper()
+            {
+            }
 
             public boolean func_175969_a(StructureOceanMonumentPieces.RoomDefinition p_175969_1_)
             {
                 return !p_175969_1_.field_175966_c[EnumFacing.WEST.getIndex()] && !p_175969_1_.field_175966_c[EnumFacing.EAST.getIndex()] && !p_175969_1_.field_175966_c[EnumFacing.NORTH.getIndex()] && !p_175969_1_.field_175966_c[EnumFacing.SOUTH.getIndex()] && !p_175969_1_.field_175966_c[EnumFacing.UP.getIndex()];
             }
@@ -691,16 +718,19 @@
             }
         }
 
     public static class MonumentBuilding extends StructureOceanMonumentPieces.Piece
         {
+
             private StructureOceanMonumentPieces.RoomDefinition field_175845_o;
             private StructureOceanMonumentPieces.RoomDefinition field_175844_p;
             private List field_175843_q = Lists.newArrayList();
             private static final String __OBFID = "CL_00001985";
 
-            public MonumentBuilding() {}
+            public MonumentBuilding()
+            {
+            }
 
             public MonumentBuilding(Random p_i45599_1_, int p_i45599_2_, int p_i45599_3_, EnumFacing p_i45599_4_)
             {
                 super(0);
                 this.coordBaseMode = p_i45599_4_;
@@ -718,30 +748,30 @@
                 List list = this.func_175836_a(p_i45599_1_);
                 this.field_175845_o.field_175963_d = true;
                 this.field_175843_q.add(new StructureOceanMonumentPieces.EntryRoom(this.coordBaseMode, this.field_175845_o));
                 this.field_175843_q.add(new StructureOceanMonumentPieces.MonumentCoreRoom(this.coordBaseMode, this.field_175844_p, p_i45599_1_));
                 ArrayList arraylist = Lists.newArrayList();
-                arraylist.add(new StructureOceanMonumentPieces.XYDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing)null));
-                arraylist.add(new StructureOceanMonumentPieces.YZDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing)null));
-                arraylist.add(new StructureOceanMonumentPieces.ZDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing)null));
-                arraylist.add(new StructureOceanMonumentPieces.XDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing)null));
-                arraylist.add(new StructureOceanMonumentPieces.YDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing)null));
-                arraylist.add(new StructureOceanMonumentPieces.FitSimpleRoomTopHelper((StructureOceanMonumentPieces.SwitchEnumFacing)null));
-                arraylist.add(new StructureOceanMonumentPieces.FitSimpleRoomHelper((StructureOceanMonumentPieces.SwitchEnumFacing)null));
+                arraylist.add(new StructureOceanMonumentPieces.XYDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing) null));
+                arraylist.add(new StructureOceanMonumentPieces.YZDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing) null));
+                arraylist.add(new StructureOceanMonumentPieces.ZDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing) null));
+                arraylist.add(new StructureOceanMonumentPieces.XDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing) null));
+                arraylist.add(new StructureOceanMonumentPieces.YDoubleRoomFitHelper((StructureOceanMonumentPieces.SwitchEnumFacing) null));
+                arraylist.add(new StructureOceanMonumentPieces.FitSimpleRoomTopHelper((StructureOceanMonumentPieces.SwitchEnumFacing) null));
+                arraylist.add(new StructureOceanMonumentPieces.FitSimpleRoomHelper((StructureOceanMonumentPieces.SwitchEnumFacing) null));
                 Iterator iterator = list.iterator();
 
                 while (iterator.hasNext())
                 {
-                    StructureOceanMonumentPieces.RoomDefinition roomdefinition = (StructureOceanMonumentPieces.RoomDefinition)iterator.next();
+                    StructureOceanMonumentPieces.RoomDefinition roomdefinition = (StructureOceanMonumentPieces.RoomDefinition) iterator.next();
 
                     if (!roomdefinition.field_175963_d && !roomdefinition.func_175961_b())
                     {
                         Iterator iterator1 = arraylist.iterator();
 
                         while (iterator1.hasNext())
                         {
-                            StructureOceanMonumentPieces.MonumentRoomFitHelper monumentroomfithelper = (StructureOceanMonumentPieces.MonumentRoomFitHelper)iterator1.next();
+                            StructureOceanMonumentPieces.MonumentRoomFitHelper monumentroomfithelper = (StructureOceanMonumentPieces.MonumentRoomFitHelper) iterator1.next();
 
                             if (monumentroomfithelper.func_175969_a(roomdefinition))
                             {
                                 this.field_175843_q.add(monumentroomfithelper.func_175968_a(this.coordBaseMode, roomdefinition, p_i45599_1_));
                                 break;
@@ -755,11 +785,11 @@
                 int j1 = this.getZWithOffset(9, 22);
                 Iterator iterator2 = this.field_175843_q.iterator();
 
                 while (iterator2.hasNext())
                 {
-                    StructureOceanMonumentPieces.Piece piece = (StructureOceanMonumentPieces.Piece)iterator2.next();
+                    StructureOceanMonumentPieces.Piece piece = (StructureOceanMonumentPieces.Piece) iterator2.next();
                     piece.getBoundingBox().offset(i1, l, j1);
                 }
 
                 StructureBoundingBox structureboundingbox1 = StructureBoundingBox.func_175899_a(this.getXWithOffset(1, 1), this.getYWithOffset(1), this.getZWithOffset(1, 1), this.getXWithOffset(23, 21), this.getYWithOffset(8), this.getZWithOffset(23, 21));
                 StructureBoundingBox structureboundingbox2 = StructureBoundingBox.func_175899_a(this.getXWithOffset(34, 1), this.getYWithOffset(1), this.getZWithOffset(34, 1), this.getXWithOffset(56, 21), this.getYWithOffset(8), this.getZWithOffset(56, 21));
@@ -896,11 +926,11 @@
                 int k2 = 1;
                 Iterator iterator = arraylist.iterator();
 
                 while (iterator.hasNext())
                 {
-                    StructureOceanMonumentPieces.RoomDefinition roomdefinition3 = (StructureOceanMonumentPieces.RoomDefinition)iterator.next();
+                    StructureOceanMonumentPieces.RoomDefinition roomdefinition3 = (StructureOceanMonumentPieces.RoomDefinition) iterator.next();
                     int l2 = 0;
                     j1 = 0;
 
                     while (l2 < 2 && j1 < 5)
                     {
@@ -988,11 +1018,11 @@
 
                 Iterator iterator = this.field_175843_q.iterator();
 
                 while (iterator.hasNext())
                 {
-                    StructureOceanMonumentPieces.Piece piece = (StructureOceanMonumentPieces.Piece)iterator.next();
+                    StructureOceanMonumentPieces.Piece piece = (StructureOceanMonumentPieces.Piece) iterator.next();
 
                     if (piece.getBoundingBox().intersectsWith(p_74875_3_))
                     {
                         piece.addComponentParts(worldIn, p_74875_2_, p_74875_3_);
                     }
@@ -1377,13 +1407,16 @@
             }
         }
 
     public static class MonumentCoreRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private static final String __OBFID = "CL_00001984";
 
-            public MonumentCoreRoom() {}
+            public MonumentCoreRoom()
+            {
+            }
 
             public MonumentCoreRoom(EnumFacing p_i45598_1_, StructureOceanMonumentPieces.RoomDefinition p_i45598_2_, Random p_i45598_3_)
             {
                 super(1, p_i45598_1_, p_i45598_2_, 2, 2, 2);
             }
@@ -1465,20 +1498,24 @@
             }
         }
 
     interface MonumentRoomFitHelper
     {
+
         boolean func_175969_a(StructureOceanMonumentPieces.RoomDefinition p_175969_1_);
 
         StructureOceanMonumentPieces.Piece func_175968_a(EnumFacing p_175968_1_, StructureOceanMonumentPieces.RoomDefinition p_175968_2_, Random p_175968_3_);
     }
 
     public static class Penthouse extends StructureOceanMonumentPieces.Piece
         {
+
             private static final String __OBFID = "CL_00001977";
 
-            public Penthouse() {}
+            public Penthouse()
+            {
+            }
 
             public Penthouse(EnumFacing p_i45591_1_, StructureBoundingBox p_i45591_2_)
             {
                 super(p_i45591_1_, p_i45591_2_);
             }
@@ -1533,10 +1570,11 @@
             }
         }
 
     public abstract static class Piece extends StructureComponent
         {
+
             protected static final IBlockState field_175828_a = Blocks.prismarine.getStateFromMeta(BlockPrismarine.ROUGH_META);
             protected static final IBlockState field_175826_b = Blocks.prismarine.getStateFromMeta(BlockPrismarine.BRICKS_META);
             protected static final IBlockState field_175827_c = Blocks.prismarine.getStateFromMeta(BlockPrismarine.DARK_META);
             protected static final IBlockState field_175824_d = field_175826_b;
             protected static final IBlockState field_175825_e = Blocks.sea_lantern.getDefaultState();
@@ -1603,13 +1641,17 @@
                     default:
                         this.boundingBox.offset(k1 * 8, l1 * 4, j1 * 8);
                 }
             }
 
-            protected void writeStructureToNBT(NBTTagCompound p_143012_1_) {}
+            protected void writeStructureToNBT(NBTTagCompound p_143012_1_)
+            {
+            }
 
-            protected void readStructureFromNBT(NBTTagCompound p_143011_1_) {}
+            protected void readStructureFromNBT(NBTTagCompound p_143011_1_)
+            {
+            }
 
             protected void func_175821_a(World worldIn, StructureBoundingBox p_175821_2_, int p_175821_3_, int p_175821_4_, boolean p_175821_5_)
             {
                 if (p_175821_5_)
                 {
@@ -1663,12 +1705,12 @@
                 if (p_175817_2_.func_175898_b(new BlockPos(l, i1, j1)))
                 {
                     EntityGuardian entityguardian = new EntityGuardian(worldIn);
                     entityguardian.func_175467_a(true);
                     entityguardian.heal(entityguardian.getMaxHealth());
-                    entityguardian.setLocationAndAngles((double)l + 0.5D, (double)i1, (double)j1 + 0.5D, 0.0F, 0.0F);
-                    entityguardian.func_180482_a(worldIn.getDifficultyForLocation(new BlockPos(entityguardian)), (IEntityLivingData)null);
+                    entityguardian.setLocationAndAngles((double) l + 0.5D, (double) i1, (double) j1 + 0.5D, 0.0F, 0.0F);
+                    entityguardian.func_180482_a(worldIn.getDifficultyForLocation(new BlockPos(entityguardian)), (IEntityLivingData) null);
                     worldIn.spawnEntityInWorld(entityguardian);
                     return true;
                 }
                 else
                 {
@@ -1677,10 +1719,11 @@
             }
         }
 
     static class RoomDefinition
         {
+
             int field_175967_a;
             StructureOceanMonumentPieces.RoomDefinition[] field_175965_b = new StructureOceanMonumentPieces.RoomDefinition[6];
             boolean[] field_175966_c = new boolean[6];
             boolean field_175963_d;
             boolean field_175964_e;
@@ -1749,14 +1792,17 @@
             }
         }
 
     public static class SimpleRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private int field_175833_o;
             private static final String __OBFID = "CL_00001975";
 
-            public SimpleRoom() {}
+            public SimpleRoom()
+            {
+            }
 
             public SimpleRoom(EnumFacing p_i45587_1_, StructureOceanMonumentPieces.RoomDefinition p_i45587_2_, Random p_i45587_3_)
             {
                 super(1, p_i45587_1_, p_i45587_2_, 1, 1, 1);
                 this.field_175833_o = p_i45587_3_.nextInt(3);
@@ -1949,13 +1995,16 @@
             }
         }
 
     public static class SimpleTopRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private static final String __OBFID = "CL_00001974";
 
-            public SimpleTopRoom() {}
+            public SimpleTopRoom()
+            {
+            }
 
             public SimpleTopRoom(EnumFacing p_i45586_1_, StructureOceanMonumentPieces.RoomDefinition p_i45586_2_, Random p_i45586_3_)
             {
                 super(1, p_i45586_1_, p_i45586_2_, 1, 1, 1);
             }
@@ -2010,10 +2059,11 @@
             }
         }
 
     static final class SwitchEnumFacing
         {
+
             static final int[] field_175971_a = new int[EnumFacing.values().length];
             private static final String __OBFID = "CL_00001993";
 
             static
             {
@@ -2046,14 +2096,17 @@
             }
         }
 
     public static class WingRoom extends StructureOceanMonumentPieces.Piece
         {
+
             private int field_175834_o;
             private static final String __OBFID = "CL_00001973";
 
-            public WingRoom() {}
+            public WingRoom()
+            {
+            }
 
             public WingRoom(EnumFacing p_i45585_1_, StructureBoundingBox p_i45585_2_, int p_i45585_3_)
             {
                 super(p_i45585_1_, p_i45585_2_);
                 this.field_175834_o = p_i45585_3_ & 1;
@@ -2164,13 +2217,16 @@
             }
         }
 
     static class XDoubleRoomFitHelper implements StructureOceanMonumentPieces.MonumentRoomFitHelper
         {
+
             private static final String __OBFID = "CL_00001992";
 
-            private XDoubleRoomFitHelper() {}
+            private XDoubleRoomFitHelper()
+            {
+            }
 
             public boolean func_175969_a(StructureOceanMonumentPieces.RoomDefinition p_175969_1_)
             {
                 return p_175969_1_.field_175966_c[EnumFacing.EAST.getIndex()] && !p_175969_1_.field_175965_b[EnumFacing.EAST.getIndex()].field_175963_d;
             }
@@ -2188,13 +2244,16 @@
             }
         }
 
     static class XYDoubleRoomFitHelper implements StructureOceanMonumentPieces.MonumentRoomFitHelper
         {
+
             private static final String __OBFID = "CL_00001991";
 
-            private XYDoubleRoomFitHelper() {}
+            private XYDoubleRoomFitHelper()
+            {
+            }
 
             public boolean func_175969_a(StructureOceanMonumentPieces.RoomDefinition p_175969_1_)
             {
                 if (p_175969_1_.field_175966_c[EnumFacing.EAST.getIndex()] && !p_175969_1_.field_175965_b[EnumFacing.EAST.getIndex()].field_175963_d && p_175969_1_.field_175966_c[EnumFacing.UP.getIndex()] && !p_175969_1_.field_175965_b[EnumFacing.UP.getIndex()].field_175963_d)
                 {
@@ -2222,13 +2281,16 @@
             }
         }
 
     static class YDoubleRoomFitHelper implements StructureOceanMonumentPieces.MonumentRoomFitHelper
         {
+
             private static final String __OBFID = "CL_00001990";
 
-            private YDoubleRoomFitHelper() {}
+            private YDoubleRoomFitHelper()
+            {
+            }
 
             public boolean func_175969_a(StructureOceanMonumentPieces.RoomDefinition p_175969_1_)
             {
                 return p_175969_1_.field_175966_c[EnumFacing.UP.getIndex()] && !p_175969_1_.field_175965_b[EnumFacing.UP.getIndex()].field_175963_d;
             }
@@ -2246,13 +2308,16 @@
             }
         }
 
     static class YZDoubleRoomFitHelper implements StructureOceanMonumentPieces.MonumentRoomFitHelper
         {
+
             private static final String __OBFID = "CL_00001989";
 
-            private YZDoubleRoomFitHelper() {}
+            private YZDoubleRoomFitHelper()
+            {
+            }
 
             public boolean func_175969_a(StructureOceanMonumentPieces.RoomDefinition p_175969_1_)
             {
                 if (p_175969_1_.field_175966_c[EnumFacing.NORTH.getIndex()] && !p_175969_1_.field_175965_b[EnumFacing.NORTH.getIndex()].field_175963_d && p_175969_1_.field_175966_c[EnumFacing.UP.getIndex()] && !p_175969_1_.field_175965_b[EnumFacing.UP.getIndex()].field_175963_d)
                 {
@@ -2280,13 +2345,16 @@
             }
         }
 
     static class ZDoubleRoomFitHelper implements StructureOceanMonumentPieces.MonumentRoomFitHelper
         {
+
             private static final String __OBFID = "CL_00001988";
 
-            private ZDoubleRoomFitHelper() {}
+            private ZDoubleRoomFitHelper()
+            {
+            }
 
             public boolean func_175969_a(StructureOceanMonumentPieces.RoomDefinition p_175969_1_)
             {
                 return p_175969_1_.field_175966_c[EnumFacing.NORTH.getIndex()] && !p_175969_1_.field_175965_b[EnumFacing.NORTH.getIndex()].field_175963_d;
             }
