--- decompiled\net\minecraft\world\gen\structure\StructureComponent.java
+++ src\main\java\net\minecraft\world\gen\structure\StructureComponent.java
@@ -1,10 +1,11 @@
 package net.minecraft.world.gen.structure;
 
 import java.util.Iterator;
 import java.util.List;
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockDirectional;
 import net.minecraft.block.BlockDoor;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
@@ -19,16 +20,19 @@
 import net.minecraft.util.WeightedRandomChestContent;
 import net.minecraft.world.World;
 
 public abstract class StructureComponent
 {
+
     protected StructureBoundingBox boundingBox;
     protected EnumFacing coordBaseMode;
     protected int componentType;
     private static final String __OBFID = "CL_00000511";
 
-    public StructureComponent() {}
+    public StructureComponent()
+    {
+    }
 
     protected StructureComponent(int p_i2091_1_)
     {
         this.componentType = p_i2091_1_;
     }
@@ -59,11 +63,13 @@
         this.readStructureFromNBT(p_143009_2_);
     }
 
     protected abstract void readStructureFromNBT(NBTTagCompound p_143011_1_);
 
-    public void buildComponent(StructureComponent p_74861_1_, List p_74861_2_, Random p_74861_3_) {}
+    public void buildComponent(StructureComponent p_74861_1_, List p_74861_2_, Random p_74861_3_)
+    {
+    }
 
     public abstract boolean addComponentParts(World worldIn, Random p_74875_2_, StructureBoundingBox p_74875_3_);
 
     public StructureBoundingBox getBoundingBox()
     {
@@ -772,10 +778,11 @@
         }
     }
 
     public abstract static class BlockSelector
         {
+
             protected IBlockState field_151562_a;
             private static final String __OBFID = "CL_00000512";
 
             protected BlockSelector()
             {
@@ -790,10 +797,11 @@
             }
         }
 
     static final class SwitchEnumFacing
         {
+
             static final int[] field_176100_a = new int[EnumFacing.values().length];
             private static final String __OBFID = "CL_00001969";
 
             static
             {
