--- decompiled\net\minecraft\world\gen\structure\StructureBoundingBox.java
+++ src\main\java\net\minecraft\world\gen\structure\StructureBoundingBox.java
@@ -1,24 +1,28 @@
 package net.minecraft.world.gen.structure;
 
-import com.google.common.base.Objects;
 import net.minecraft.nbt.NBTTagIntArray;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.Vec3i;
 
+import com.google.common.base.Objects;
+
 public class StructureBoundingBox
 {
+
     public int minX;
     public int minY;
     public int minZ;
     public int maxX;
     public int maxY;
     public int maxZ;
     private static final String __OBFID = "CL_00000442";
 
-    public StructureBoundingBox() {}
+    public StructureBoundingBox()
+    {
+    }
 
     public StructureBoundingBox(int[] p_i43000_1_)
     {
         if (p_i43000_1_.length == 6)
         {
@@ -168,10 +172,11 @@
         return new NBTTagIntArray(new int[] {this.minX, this.minY, this.minZ, this.maxX, this.maxY, this.maxZ});
     }
 
     static final class SwitchEnumFacing
         {
+
             static final int[] field_175895_a = new int[EnumFacing.values().length];
             private static final String __OBFID = "CL_00001999";
 
             static
             {
