--- decompiled\net\minecraft\entity\ai\attributes\BaseAttribute.java
+++ src\main\java\net\minecraft\entity\ai\attributes\BaseAttribute.java
@@ -1,9 +1,10 @@
 package net.minecraft.entity.ai.attributes;
 
 public abstract class BaseAttribute implements IAttribute
 {
+
     private final IAttribute field_180373_a;
     private final String unlocalizedName;
     private final double defaultValue;
     private boolean shouldWatch;
     private static final String __OBFID = "CL_00001565";
@@ -51,8 +52,8 @@
         return this.unlocalizedName.hashCode();
     }
 
     public boolean equals(Object p_equals_1_)
     {
-        return p_equals_1_ instanceof IAttribute && this.unlocalizedName.equals(((IAttribute)p_equals_1_).getAttributeUnlocalizedName());
+        return p_equals_1_ instanceof IAttribute && this.unlocalizedName.equals(((IAttribute) p_equals_1_).getAttributeUnlocalizedName());
     }
 }
