--- decompiled\net\minecraft\entity\ai\attributes\IAttributeInstance.java
+++ src\main\java\net\minecraft\entity\ai\attributes\IAttributeInstance.java
@@ -1,14 +1,14 @@
 package net.minecraft.entity.ai.attributes;
 
 import java.util.Collection;
 import java.util.UUID;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public interface IAttributeInstance
 {
+
     IAttribute getAttribute();
 
     double getBaseValue();
 
     void setBaseValue(double p_111128_1_);
@@ -23,10 +23,7 @@
 
     void applyModifier(AttributeModifier p_111121_1_);
 
     void removeModifier(AttributeModifier p_111124_1_);
 
-    @SideOnly(Side.CLIENT)
-    void removeAllModifiers();
-
     double getAttributeValue();
 }
