--- decompiled\net\minecraft\stats\Achievement.java
+++ src\main\java\net\minecraft\stats\Achievement.java
@@ -4,22 +4,19 @@
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.ChatComponentTranslation;
 import net.minecraft.util.EnumChatFormatting;
 import net.minecraft.util.IChatComponent;
-import net.minecraft.util.StatCollector;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class Achievement extends StatBase
 {
+
     public final int displayColumn;
     public final int displayRow;
     public final Achievement parentAchievement;
     private final String achievementDescription;
-    @SideOnly(Side.CLIENT)
-    private IStatStringFormat statStringFormatter;
     public final ItemStack theItemStack;
     private boolean isSpecial;
     private static final String __OBFID = "CL_00001466";
 
     public Achievement(String p_i46327_1_, String p_i46327_2_, int column, int row, Item p_i46327_5_, Achievement p_i46327_6_)
@@ -97,23 +94,11 @@
     public Achievement func_180787_a(Class p_180787_1_)
     {
         return (Achievement)super.func_150953_b(p_180787_1_);
     }
 
-    @SideOnly(Side.CLIENT)
-    public String getDescription()
-    {
-        return this.statStringFormatter != null ? this.statStringFormatter.formatString(StatCollector.translateToLocal(this.achievementDescription)) : StatCollector.translateToLocal(this.achievementDescription);
-    }
 
-    @SideOnly(Side.CLIENT)
-    public Achievement setStatStringFormatter(IStatStringFormat p_75988_1_)
-    {
-        this.statStringFormatter = p_75988_1_;
-        return this;
-    }
-
     public boolean getSpecial()
     {
         return this.isSpecial;
     }
 
