--- decompiled\net\minecraft\stats\AchievementList.java
+++ src\main\java\net\minecraft\stats\AchievementList.java
@@ -1,16 +1,19 @@
 package net.minecraft.stats;
 
-import com.google.common.collect.Lists;
 import java.util.List;
+
 import net.minecraft.init.Blocks;
 import net.minecraft.init.Items;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.JsonSerializableSet;
 
+import com.google.common.collect.Lists;
+
 public class AchievementList
 {
+
     public static int minDisplayColumn;
     public static int minDisplayRow;
     public static int maxDisplayColumn;
     public static int maxDisplayRow;
     public static List achievementList = Lists.newArrayList();
@@ -48,7 +51,9 @@
     public static Achievement fullBeacon = (new Achievement("achievement.fullBeacon", "fullBeacon", 7, 8, Blocks.beacon, killWither)).setSpecial().func_180788_c();
     public static Achievement exploreAllBiomes = (new Achievement("achievement.exploreAllBiomes", "exploreAllBiomes", 4, 8, Items.diamond_boots, theEnd)).func_180787_a(JsonSerializableSet.class).setSpecial().func_180788_c();
     public static Achievement overpowered = (new Achievement("achievement.overpowered", "overpowered", 6, 4, Items.golden_apple, buildBetterPickaxe)).setSpecial().func_180788_c();
     private static final String __OBFID = "CL_00001467";
 
-    public static void init() {}
+    public static void init()
+    {
+    }
 }
