--- decompiled\net\minecraft\stats\StatFileWriter.java
+++ src\main\java\net\minecraft\stats\StatFileWriter.java
@@ -1,17 +1,21 @@
 package net.minecraft.stats;
 
-import com.google.common.collect.Maps;
 import java.util.Map;
+
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.IJsonSerializable;
 import net.minecraft.util.TupleIntJsonSerializable;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.event.Cancellable;
+
+import com.google.common.collect.Maps;
+
 public class StatFileWriter
 {
+
     protected final Map statsData = Maps.newConcurrentMap();
     private static final String __OBFID = "CL_00001481";
 
     public boolean hasAchievementUnlocked(Achievement p_77443_1_)
     {
@@ -25,31 +29,20 @@
 
     public void func_150871_b(EntityPlayer p_150871_1_, StatBase p_150871_2_, int p_150871_3_)
     {
         if (!p_150871_2_.isAchievement() || this.canUnlockAchievement((Achievement)p_150871_2_))
         {
-            this.func_150873_a(p_150871_1_, p_150871_2_, this.readStat(p_150871_2_) + p_150871_3_);
-        }
-    }
+            // CraftBukkit start - fire Statistic events
+            Cancellable cancellable = CraftEventFactory.handleStatisticsIncrease(p_150871_1_, p_150871_2_, this.readStat(p_150871_2_), p_150871_3_);
 
-    @SideOnly(Side.CLIENT)
-    public int func_150874_c(Achievement p_150874_1_)
-    {
-        if (this.hasAchievementUnlocked(p_150874_1_))
-        {
-            return 0;
-        }
-        else
-        {
-            int i = 0;
-
-            for (Achievement achievement1 = p_150874_1_.parentAchievement; achievement1 != null && !this.hasAchievementUnlocked(achievement1); ++i)
+            if (cancellable != null && cancellable.isCancelled())
             {
-                achievement1 = achievement1.parentAchievement;
+                return;
             }
 
-            return i;
+            // CraftBukkit end
+            this.func_150873_a(p_150871_1_, p_150871_2_, this.readStat(p_150871_2_) + p_150871_3_);
         }
     }
 
     public void func_150873_a(EntityPlayer p_150873_1_, StatBase p_150873_2_, int p_150873_3_)
     {
