--- decompiled\net\minecraft\stats\StatBase.java
+++ src\main\java\net\minecraft\stats\StatBase.java
@@ -1,70 +1,44 @@
 package net.minecraft.stats;
 
 import java.text.DecimalFormat;
 import java.text.NumberFormat;
 import java.util.Locale;
+
 import net.minecraft.event.HoverEvent;
 import net.minecraft.scoreboard.IScoreObjectiveCriteria;
 import net.minecraft.util.ChatComponentText;
 import net.minecraft.util.EnumChatFormatting;
 import net.minecraft.util.IChatComponent;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class StatBase
 {
+
     public final String statId;
     private final IChatComponent statName;
     public boolean isIndependent;
     private final IStatType type;
     private final IScoreObjectiveCriteria field_150957_c;
     private Class field_150956_d;
     private static NumberFormat numberFormat = NumberFormat.getIntegerInstance(Locale.US);
     public static IStatType simpleStatType = new IStatType()
     {
         private static final String __OBFID = "CL_00001473";
-        @SideOnly(Side.CLIENT)
-        public String format(int p_75843_1_)
-        {
-            return StatBase.numberFormat.format((long)p_75843_1_);
-        }
     };
     private static DecimalFormat decimalFormat = new DecimalFormat("########0.00");
     public static IStatType timeStatType = new IStatType()
     {
         private static final String __OBFID = "CL_00001474";
-        @SideOnly(Side.CLIENT)
-        public String format(int p_75843_1_)
-        {
-            double d0 = (double)p_75843_1_ / 20.0D;
-            double d1 = d0 / 60.0D;
-            double d2 = d1 / 60.0D;
-            double d3 = d2 / 24.0D;
-            double d4 = d3 / 365.0D;
-            return d4 > 0.5D ? StatBase.decimalFormat.format(d4) + " y" : (d3 > 0.5D ? StatBase.decimalFormat.format(d3) + " d" : (d2 > 0.5D ? StatBase.decimalFormat.format(d2) + " h" : (d1 > 0.5D ? StatBase.decimalFormat.format(d1) + " m" : d0 + " s")));
-        }
     };
     public static IStatType distanceStatType = new IStatType()
     {
         private static final String __OBFID = "CL_00001475";
-        @SideOnly(Side.CLIENT)
-        public String format(int p_75843_1_)
-        {
-            double d0 = (double)p_75843_1_ / 100.0D;
-            double d1 = d0 / 1000.0D;
-            return d1 > 0.5D ? StatBase.decimalFormat.format(d1) + " km" : (d0 > 0.5D ? StatBase.decimalFormat.format(d0) + " m" : p_75843_1_ + " cm");
-        }
     };
     public static IStatType field_111202_k = new IStatType()
     {
         private static final String __OBFID = "CL_00001476";
-        @SideOnly(Side.CLIENT)
-        public String format(int p_75843_1_)
-        {
-            return StatBase.decimalFormat.format((double)p_75843_1_ * 0.1D);
-        }
     };
     private static final String __OBFID = "CL_00001472";
 
     public StatBase(String p_i45307_1_, IChatComponent p_i45307_2_, IStatType p_i45307_3_)
     {
@@ -103,16 +77,10 @@
     public boolean isAchievement()
     {
         return false;
     }
 
-    @SideOnly(Side.CLIENT)
-    public String func_75968_a(int p_75968_1_)
-    {
-        return this.type.format(p_75968_1_);
-    }
-
     public IChatComponent getStatName()
     {
         IChatComponent ichatcomponent = this.statName.createCopy();
         ichatcomponent.getChatStyle().setColor(EnumChatFormatting.GRAY);
         ichatcomponent.getChatStyle().setChatHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_ACHIEVEMENT, new ChatComponentText(this.statId)));
