--- decompiled\net\minecraft\world\pathfinder\WalkNodeProcessor.java
+++ src\main\java\net\minecraft\world\pathfinder\WalkNodeProcessor.java
@@ -14,10 +14,11 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.IBlockAccess;
 
 public class WalkNodeProcessor extends NodeProcessor
 {
+
     private boolean field_176180_f;
     private boolean field_176181_g;
     private boolean field_176183_h;
     private boolean field_176184_i;
     private boolean field_176182_j;
