--- decompiled\net\minecraft\entity\projectile\EntitySmallFireball.java
+++ src\main\java\net\minecraft\entity\projectile\EntitySmallFireball.java
@@ -6,12 +6,15 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraft.world.World;
 
+import org.bukkit.event.entity.EntityCombustByEntityEvent;
+
 public class EntitySmallFireball extends EntityFireball
 {
+
     private static final String __OBFID = "CL_00001721";
 
     public EntitySmallFireball(World worldIn)
     {
         super(worldIn);
@@ -44,11 +47,20 @@
                 {
                     this.func_174815_a(this.shootingEntity, movingObject.entityHit);
 
                     if (!movingObject.entityHit.isImmuneToFire())
                     {
-                        movingObject.entityHit.setFire(5);
+                        // CraftBukkit start - Entity damage by entity event + combust event
+                        EntityCombustByEntityEvent event = new EntityCombustByEntityEvent((org.bukkit.entity.Projectile)this.getBukkitEntity(), movingObject.entityHit.getBukkitEntity(), 5);
+                        movingObject.entityHit.worldObj.getServer().getPluginManager().callEvent(event);
+
+                        if (!event.isCancelled())
+                        {
+                            movingObject.entityHit.setFire(event.getDuration());
+                        }
+
+                        // CraftBukkit end
                     }
                 }
             }
             else
             {
@@ -63,11 +75,17 @@
                 {
                     BlockPos blockpos = movingObject.getBlockPos().offset(movingObject.sideHit);
 
                     if (this.worldObj.isAirBlock(blockpos))
                     {
-                        this.worldObj.setBlockState(blockpos, Blocks.fire.getDefaultState());
+                        // CraftBukkit start
+                        if (isIncendiary && !org.bukkit.craftbukkit.event.CraftEventFactory.callBlockIgniteEvent(worldObj, blockpos.getX(), blockpos.getY(), blockpos.getZ(), this).isCancelled())
+                        {
+                            this.worldObj.setBlockState(blockpos, Blocks.fire.getDefaultState());
+                        }
+
+                        // CraftBukkit end
                     }
                 }
             }
 
             this.setDead();
