--- decompiled\net\minecraft\entity\projectile\EntitySnowball.java
+++ src\main\java\net\minecraft\entity\projectile\EntitySnowball.java
@@ -7,10 +7,11 @@
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraft.world.World;
 
 public class EntitySnowball extends EntityThrowable
 {
+
     private static final String __OBFID = "CL_00001722";
 
     public EntitySnowball(World worldIn)
     {
         super(worldIn);
