--- decompiled\net\minecraft\entity\projectile\EntityEgg.java
+++ src\main\java\net\minecraft\entity\projectile\EntityEgg.java
@@ -1,18 +1,28 @@
 package net.minecraft.entity.projectile;
 
+import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.passive.EntityChicken;
+import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.init.Items;
 import net.minecraft.item.Item;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraft.world.World;
 
+import org.bukkit.Location;
+import org.bukkit.entity.Ageable;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Player;
+import org.bukkit.event.player.PlayerEggThrowEvent;
+
+
 public class EntityEgg extends EntityThrowable
 {
+
     private static final String __OBFID = "CL_00001724";
 
     public EntityEgg(World worldIn)
     {
         super(worldIn);
@@ -33,10 +43,47 @@
         if (p_70184_1_.entityHit != null)
         {
             p_70184_1_.entityHit.attackEntityFrom(DamageSource.causeThrownDamage(this, this.getThrower()), 0.0F);
         }
 
+        // CraftBukkit start - Fire PlayerEggThrowEvent
+        boolean hatching = !this.worldObj.isRemote && this.rand.nextInt(8) == 0;
+        int numHatching = (this.rand.nextInt(32) == 0) ? 4 : 1;
+
+        if (!hatching)
+        {
+            numHatching = 0;
+        }
+
+        EntityType hatchingType = EntityType.CHICKEN;
+        Entity shooter = this.getThrower();
+
+        if (shooter instanceof EntityPlayerMP)
+        {
+            Player player = (shooter == null) ? null : (Player)shooter.getBukkitEntity();
+            PlayerEggThrowEvent event = new PlayerEggThrowEvent(player, (org.bukkit.entity.Egg)this.getBukkitEntity(), hatching, (byte)numHatching, hatchingType);
+            this.worldObj.getServer().getPluginManager().callEvent(event);
+            hatching = event.isHatching();
+            numHatching = event.getNumHatches();
+            hatchingType = event.getHatchingType();
+        }
+
+        if (hatching)
+        {
+            for (int k = 0; k < numHatching; k++)
+            {
+                org.bukkit.entity.Entity entity = worldObj.getWorld().spawn(new Location(worldObj.getWorld(), this.posX, this.posY, this.posZ, this.rotationYaw, 0.0F), hatchingType.getEntityClass(), org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.EGG);
+
+                if (entity instanceof Ageable)
+                {
+                    ((Ageable)entity).setBaby();
+                }
+            }
+        }
+
+        // CraftBukkit end
+
         if (!this.worldObj.isRemote && this.rand.nextInt(8) == 0)
         {
             byte b0 = 1;
 
             if (this.rand.nextInt(32) == 0)
