--- decompiled\net\minecraft\world\chunk\IChunkProvider.java
+++ src\main\java\net\minecraft\world\chunk\IChunkProvider.java
@@ -1,15 +1,17 @@
 package net.minecraft.world.chunk;
 
 import java.util.List;
+
 import net.minecraft.entity.EnumCreatureType;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.IProgressUpdate;
 import net.minecraft.world.World;
 
 public interface IChunkProvider
 {
+
     boolean chunkExists(int x, int z);
 
     Chunk provideChunk(int x, int z);
 
     Chunk provideChunk(BlockPos blockPosIn);
