--- decompiled\net\minecraft\world\chunk\NibbleArray.java
+++ src\main\java\net\minecraft\world\chunk\NibbleArray.java
@@ -1,72 +1,58 @@
 package net.minecraft.world.chunk;
 
-public class NibbleArray
-{
+public class NibbleArray {
+
     private final byte[] data;
     private static final String __OBFID = "CL_00000371";
 
-    public NibbleArray()
-    {
+    public NibbleArray() {
         this.data = new byte[2048];
     }
 
-    public NibbleArray(byte[] storageArray)
-    {
+    public NibbleArray(byte[] storageArray) {
         this.data = storageArray;
 
-        if (storageArray.length != 2048)
-        {
+        if (storageArray.length != 2048) {
             throw new IllegalArgumentException("ChunkNibbleArrays should be 2048 bytes not: " + storageArray.length);
         }
     }
 
-    public int get(int x, int y, int z)
-    {
+    public int get(int x, int y, int z) {
         return this.getFromIndex(this.getCoordinateIndex(x, y, z));
     }
 
-    public void set(int x, int y, int z, int value)
-    {
+    public void set(int x, int y, int z, int value) {
         this.setIndex(this.getCoordinateIndex(x, y, z), value);
     }
 
-    private int getCoordinateIndex(int x, int y, int z)
-    {
+    private int getCoordinateIndex(int x, int y, int z) {
         return y << 8 | z << 4 | x;
     }
 
-    public int getFromIndex(int index)
-    {
+    public int getFromIndex(int index) {
         int j = this.func_177478_c(index);
         return this.func_177479_b(index) ? this.data[j] & 15 : this.data[j] >> 4 & 15;
     }
 
-    public void setIndex(int index, int value)
-    {
+    public void setIndex(int index, int value) {
         int k = this.func_177478_c(index);
 
-        if (this.func_177479_b(index))
-        {
-            this.data[k] = (byte)(this.data[k] & 240 | value & 15);
+        if (this.func_177479_b(index)) {
+            this.data[k] = (byte) (this.data[k] & 240 | value & 15);
+        } else {
+            this.data[k] = (byte) (this.data[k] & 15 | (value & 15) << 4);
         }
-        else
-        {
-            this.data[k] = (byte)(this.data[k] & 15 | (value & 15) << 4);
-        }
     }
 
-    private boolean func_177479_b(int p_177479_1_)
-    {
+    private boolean func_177479_b(int p_177479_1_) {
         return (p_177479_1_ & 1) == 0;
     }
 
-    private int func_177478_c(int p_177478_1_)
-    {
+    private int func_177478_c(int p_177478_1_) {
         return p_177478_1_ >> 1;
     }
 
-    public byte[] getData()
-    {
+    public byte[] getData() {
         return this.data;
     }
 }
