--- decompiled\net\minecraft\block\material\Material.java
+++ src\main\java\net\minecraft\block\material\Material.java
@@ -1,9 +1,10 @@
 package net.minecraft.block.material;
 
 public class Material
 {
+
     public static final Material air = new MaterialTransparent(MapColor.airColor);
     public static final Material grass = new Material(MapColor.grassColor);
     public static final Material ground = new Material(MapColor.dirtColor);
     public static final Material wood = (new Material(MapColor.woodColor)).setBurning();
     public static final Material rock = (new Material(MapColor.stoneColor)).setRequiresTool();
