--- decompiled\net\minecraft\block\material\MaterialTransparent.java
+++ src\main\java\net\minecraft\block\material\MaterialTransparent.java
@@ -1,27 +1,23 @@
 package net.minecraft.block.material;
 
-public class MaterialTransparent extends Material
-{
+public class MaterialTransparent extends Material {
+
     private static final String __OBFID = "CL_00000540";
 
-    public MaterialTransparent(MapColor color)
-    {
+    public MaterialTransparent(MapColor color) {
         super(color);
         this.setReplaceable();
     }
 
-    public boolean isSolid()
-    {
+    public boolean isSolid() {
         return false;
     }
 
-    public boolean blocksLight()
-    {
+    public boolean blocksLight() {
         return false;
     }
 
-    public boolean blocksMovement()
-    {
+    public boolean blocksMovement() {
         return false;
     }
 }
