--- decompiled\net\minecraft\init\Blocks.java
+++ src\main\java\net\minecraft\init\Blocks.java
@@ -34,10 +34,11 @@
 import net.minecraft.block.BlockTripWireHook;
 import net.minecraft.util.ResourceLocation;
 
 public class Blocks
 {
+
     public static final Block air;
     public static final Block stone;
     public static final BlockGrass grass;
     public static final Block dirt;
     public static final Block cobblestone;
