--- decompiled\net\minecraft\init\Items.java
+++ src\main\java\net\minecraft\init\Items.java
@@ -12,10 +12,11 @@
 import net.minecraft.item.ItemShears;
 import net.minecraft.util.ResourceLocation;
 
 public class Items
 {
+
     public static final Item iron_shovel;
     public static final Item iron_pickaxe;
     public static final Item iron_axe;
     public static final Item flint_and_steel;
     public static final Item apple;
