--- decompiled\net\minecraft\world\gen\layer\GenLayerShore.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerShore.java
@@ -4,10 +4,11 @@
 import net.minecraft.world.biome.BiomeGenJungle;
 import net.minecraft.world.biome.BiomeGenMesa;
 
 public class GenLayerShore extends GenLayer
 {
+
     private static final String __OBFID = "CL_00000568";
 
     public GenLayerShore(long p_i2130_1_, GenLayer p_i2130_3_)
     {
         super(p_i2130_1_);
