--- decompiled\net\minecraft\world\gen\layer\GenLayerSmooth.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerSmooth.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.gen.layer;
 
 public class GenLayerSmooth extends GenLayer
 {
+
     private static final String __OBFID = "CL_00000569";
 
     public GenLayerSmooth(long p_i2131_1_, GenLayer p_i2131_3_)
     {
         super(p_i2131_1_);
