--- decompiled\net\minecraft\world\gen\layer\GenLayerAddMushroomIsland.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerAddMushroomIsland.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.world.biome.BiomeGenBase;
 
 public class GenLayerAddMushroomIsland extends GenLayer
 {
+
     private static final String __OBFID = "CL_00000552";
 
     public GenLayerAddMushroomIsland(long p_i2120_1_, GenLayer p_i2120_3_)
     {
         super(p_i2120_1_);
