--- decompiled\net\minecraft\world\gen\layer\GenLayerRiver.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerRiver.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.world.biome.BiomeGenBase;
 
 public class GenLayerRiver extends GenLayer
 {
+
     private static final String __OBFID = "CL_00000566";
 
     public GenLayerRiver(long p_i2128_1_, GenLayer p_i2128_3_)
     {
         super(p_i2128_1_);
