--- decompiled\net\minecraft\world\gen\layer\GenLayerRareBiome.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerRareBiome.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.world.biome.BiomeGenBase;
 
 public class GenLayerRareBiome extends GenLayer
 {
+
     private static final String __OBFID = "CL_00000562";
 
     public GenLayerRareBiome(long p_i45478_1_, GenLayer p_i45478_3_)
     {
         super(p_i45478_1_);
