--- decompiled\net\minecraft\world\gen\layer\GenLayerEdge.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerEdge.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.gen.layer;
 
 public class GenLayerEdge extends GenLayer
 {
+
     private final GenLayerEdge.Mode field_151627_c;
     private static final String __OBFID = "CL_00000547";
 
     public GenLayerEdge(long p_i45474_1_, GenLayer p_i45474_3_, GenLayerEdge.Mode p_i45474_4_)
     {
@@ -134,10 +135,11 @@
         private static final String __OBFID = "CL_00000549";
     }
 
     static final class SwitchMode
         {
+
             static final int[] field_151642_a = new int[GenLayerEdge.Mode.values().length];
             private static final String __OBFID = "CL_00000548";
 
             static
             {
