--- decompiled\net\minecraft\world\gen\layer\GenLayerBiome.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerBiome.java
@@ -4,10 +4,11 @@
 import net.minecraft.world.biome.BiomeGenBase;
 import net.minecraft.world.gen.ChunkProviderSettings;
 
 public class GenLayerBiome extends GenLayer
 {
+
     private BiomeGenBase[] field_151623_c;
     private BiomeGenBase[] field_151621_d;
     private BiomeGenBase[] field_151622_e;
     private BiomeGenBase[] field_151620_f;
     private final ChunkProviderSettings field_175973_g;
