--- decompiled\net\minecraft\world\gen\layer\GenLayerHills.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerHills.java
@@ -1,13 +1,15 @@
 package net.minecraft.world.gen.layer;
 
 import net.minecraft.world.biome.BiomeGenBase;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class GenLayerHills extends GenLayer
 {
+
     private static final Logger logger = LogManager.getLogger();
     private GenLayer field_151628_d;
     private static final String __OBFID = "CL_00000563";
 
     public GenLayerHills(long p_i45479_1_, GenLayer p_i45479_3_, GenLayer p_i45479_4_)
