--- decompiled\net\minecraft\world\gen\layer\GenLayerRiverMix.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerRiverMix.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.world.biome.BiomeGenBase;
 
 public class GenLayerRiverMix extends GenLayer
 {
+
     private GenLayer biomePatternGeneratorChain;
     private GenLayer riverPatternGeneratorChain;
     private static final String __OBFID = "CL_00000567";
 
     public GenLayerRiverMix(long p_i2129_1_, GenLayer p_i2129_3_, GenLayer p_i2129_4_)
