--- decompiled\net\minecraft\world\gen\layer\GenLayerAddSnow.java
+++ src\main\java\net\minecraft\world\gen\layer\GenLayerAddSnow.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.gen.layer;
 
 public class GenLayerAddSnow extends GenLayer
 {
+
     private static final String __OBFID = "CL_00000553";
 
     public GenLayerAddSnow(long p_i2121_1_, GenLayer p_i2121_3_)
     {
         super(p_i2121_1_);
