--- decompiled\net\minecraft\server\dedicated\DedicatedPlayerList.java
+++ src\main\java\net\minecraft\server\dedicated\DedicatedPlayerList.java
@@ -1,19 +1,19 @@
 package net.minecraft.server.dedicated;
 
-import com.mojang.authlib.GameProfile;
 import java.io.IOException;
-import net.minecraft.server.MinecraftServer;
+
 import net.minecraft.server.management.ServerConfigurationManager;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-@SideOnly(Side.SERVER)
+import com.mojang.authlib.GameProfile;
+
 public class DedicatedPlayerList extends ServerConfigurationManager
 {
+
     private static final Logger LOGGER = LogManager.getLogger();
     private static final String __OBFID = "CL_00001783";
 
     public DedicatedPlayerList(DedicatedServer server)
     {
