--- decompiled\net\minecraft\server\dedicated\ServerHangWatchdog.java
+++ src\main\java\net\minecraft\server\dedicated\ServerHangWatchdog.java
@@ -6,21 +6,21 @@
 import java.lang.management.ThreadMXBean;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 import java.util.Timer;
 import java.util.TimerTask;
+
 import net.minecraft.crash.CrashReport;
 import net.minecraft.crash.CrashReportCategory;
 import net.minecraft.server.MinecraftServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-@SideOnly(Side.SERVER)
 public class ServerHangWatchdog implements Runnable
 {
+
     private static final Logger LOGGER = LogManager.getLogger();
     private final DedicatedServer server;
     private final long maxTickTime;
     private static final String __OBFID = "CL_00002634";
 
