--- decompiled\net\minecraft\scoreboard\GoalColor.java
+++ src\main\java\net\minecraft\scoreboard\GoalColor.java
@@ -1,12 +1,14 @@
 package net.minecraft.scoreboard;
 
 import java.util.List;
+
 import net.minecraft.util.EnumChatFormatting;
 
 public class GoalColor implements IScoreObjectiveCriteria
 {
+
     private final String goalName;
     private static final String __OBFID = "CL_00001961";
 
     public GoalColor(String p_i45549_1_, EnumChatFormatting p_i45549_2_)
     {
