--- decompiled\net\minecraft\scoreboard\ScoreObjective.java
+++ src\main\java\net\minecraft\scoreboard\ScoreObjective.java
@@ -1,12 +1,11 @@
 package net.minecraft.scoreboard;
 
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
 public class ScoreObjective
 {
+
     private final Scoreboard theScoreboard;
     private final String name;
     private final IScoreObjectiveCriteria objectiveCriteria;
     private IScoreObjectiveCriteria.EnumRenderType renderType;
     private String displayName;
@@ -19,16 +18,10 @@
         this.objectiveCriteria = p_i2307_3_;
         this.displayName = p_i2307_2_;
         this.renderType = p_i2307_3_.getRenderType();
     }
 
-    @SideOnly(Side.CLIENT)
-    public Scoreboard getScoreboard()
-    {
-        return this.theScoreboard;
-    }
-
     public String getName()
     {
         return this.name;
     }
 
