--- decompiled\net\minecraft\scoreboard\ScorePlayerTeam.java
+++ src\main\java\net\minecraft\scoreboard\ScorePlayerTeam.java
@@ -1,16 +1,17 @@
 package net.minecraft.scoreboard;
 
-import com.google.common.collect.Sets;
 import java.util.Collection;
 import java.util.Set;
+
 import net.minecraft.util.EnumChatFormatting;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.collect.Sets;
+
 public class ScorePlayerTeam extends Team
 {
+
     private final Scoreboard theScoreboard;
     private final String registeredName;
     private final Set membershipSet = Sets.newHashSet();
     private String teamNameSPT;
     private String namePrefixSPT = "";
@@ -158,17 +159,10 @@
         }
 
         return i;
     }
 
-    @SideOnly(Side.CLIENT)
-    public void func_98298_a(int p_98298_1_)
-    {
-        this.setAllowFriendlyFire((p_98298_1_ & 1) > 0);
-        this.setSeeFriendlyInvisiblesEnabled((p_98298_1_ & 2) > 0);
-    }
-
     public void func_178774_a(EnumChatFormatting p_178774_1_)
     {
         this.field_178777_k = p_178774_1_;
     }
 
