--- decompiled\net\minecraft\scoreboard\ScoreboardSaveData.java
+++ src\main\java\net\minecraft\scoreboard\ScoreboardSaveData.java
@@ -1,19 +1,22 @@
 package net.minecraft.scoreboard;
 
 import java.util.Collection;
 import java.util.Iterator;
+
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.nbt.NBTTagString;
 import net.minecraft.util.EnumChatFormatting;
 import net.minecraft.world.WorldSavedData;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class ScoreboardSaveData extends WorldSavedData
 {
+
     private static final Logger logger = LogManager.getLogger();
     private Scoreboard theScoreboard;
     private NBTTagCompound delayedInitNbt;
     private static final String __OBFID = "CL_00000620";
 
