--- decompiled\net\minecraft\scoreboard\Team.java
+++ src\main\java\net\minecraft\scoreboard\Team.java
@@ -1,15 +1,15 @@
 package net.minecraft.scoreboard;
 
-import com.google.common.collect.Maps;
 import java.util.Collection;
 import java.util.Map;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.collect.Maps;
+
 public abstract class Team
 {
+
     private static final String __OBFID = "CL_00000621";
 
     public boolean isSameTeam(Team other)
     {
         return other == null ? false : this == other;
@@ -17,18 +17,12 @@
 
     public abstract String getRegisteredName();
 
     public abstract String formatString(String input);
 
-    @SideOnly(Side.CLIENT)
-    public abstract boolean func_98297_h();
-
     public abstract boolean getAllowFriendlyFire();
 
-    @SideOnly(Side.CLIENT)
-    public abstract Team.EnumVisible func_178770_i();
-
     public abstract Collection getMembershipCollection();
 
     public abstract Team.EnumVisible func_178771_j();
 
     public static enum EnumVisible
