--- decompiled\net\minecraft\network\status\server\S01PacketPong.java
+++ src\main\java\net\minecraft\network\status\server\S01PacketPong.java
@@ -1,19 +1,23 @@
 package net.minecraft.network.status.server;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.status.INetHandlerStatusClient;
 
 public class S01PacketPong implements Packet
 {
+
     private long clientTime;
     private static final String __OBFID = "CL_00001383";
 
-    public S01PacketPong() {}
+    public S01PacketPong()
+    {
+    }
 
     public S01PacketPong(long time)
     {
         this.clientTime = time;
     }
@@ -33,8 +37,8 @@
         handler.handlePong(this);
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerStatusClient)handler);
+        this.processPacket((INetHandlerStatusClient) handler);
     }
 }
