--- decompiled\net\minecraft\world\chunk\storage\NibbleArrayReader.java
+++ src\main\java\net\minecraft\world\chunk\storage\NibbleArrayReader.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.chunk.storage;
 
 public class NibbleArrayReader
 {
+
     public final byte[] data;
     private final int depthBits;
     private final int depthBitsPlusFour;
     private static final String __OBFID = "CL_00000376";
 
