--- decompiled\net\minecraft\dispenser\IBlockSource.java
+++ src\main\java\net\minecraft\dispenser\IBlockSource.java
@@ -2,12 +2,12 @@
 
 import net.minecraft.block.Block;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.BlockPos;
 
-public interface IBlockSource extends ILocatableSource
-{
+public interface IBlockSource extends ILocatableSource {
+
     double getX();
 
     double getY();
 
     double getZ();
