--- decompiled\net\minecraft\nbt\JsonToNBT.java
+++ src\main\java\net\minecraft\nbt\JsonToNBT.java
@@ -1,18 +1,21 @@
 package net.minecraft.nbt;
 
-import com.google.common.base.Splitter;
-import com.google.common.collect.Iterables;
-import com.google.common.collect.Lists;
 import java.util.Iterator;
 import java.util.Stack;
 import java.util.regex.Pattern;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
 public class JsonToNBT
 {
+
     private static final Logger logger = LogManager.getLogger();
     private static final Pattern field_179273_b = Pattern.compile("\\[[-+\\d|,\\s]+\\]");
     private static final String __OBFID = "CL_00001232";
 
     public static NBTTagCompound func_180713_a(String p_180713_0_) throws NBTException
@@ -100,11 +103,11 @@
 
             return i;
         }
     }
 
-    static JsonToNBT.Any func_179272_a(String ... p_179272_0_) throws NBTException
+    static JsonToNBT.Any func_179272_a(String... p_179272_0_) throws NBTException
     {
         return func_150316_a(p_179272_0_[0], p_179272_0_[1]);
     }
 
     static JsonToNBT.Any func_150316_a(String p_150316_0_, String p_150316_1_) throws NBTException
@@ -386,18 +389,20 @@
         return p_179271_1_ > 0 && p_179271_0_.charAt(p_179271_1_ - 1) == 92 && !func_179271_b(p_179271_0_, p_179271_1_ - 1);
     }
 
     abstract static class Any
         {
+
             protected String field_150490_a;
             private static final String __OBFID = "CL_00001233";
 
             public abstract NBTBase func_150489_a();
         }
 
     static class Compound extends JsonToNBT.Any
         {
+
             protected java.util.List field_150491_b = Lists.newArrayList();
             private static final String __OBFID = "CL_00001234";
 
             public Compound(String p_i45137_1_)
             {
@@ -419,10 +424,11 @@
             }
         }
 
     static class List extends JsonToNBT.Any
         {
+
             protected java.util.List field_150492_b = Lists.newArrayList();
             private static final String __OBFID = "CL_00001235";
 
             public List(String p_i45138_1_)
             {
@@ -444,10 +450,11 @@
             }
         }
 
     static class Primitive extends JsonToNBT.Any
         {
+
             private static final Pattern field_179265_c = Pattern.compile("[-+]?[0-9]*\\.?[0-9]+[d|D]");
             private static final Pattern field_179263_d = Pattern.compile("[-+]?[0-9]*\\.?[0-9]+[f|F]");
             private static final Pattern field_179264_e = Pattern.compile("[-+]?[0-9]+[b|B]");
             private static final Pattern field_179261_f = Pattern.compile("[-+]?[0-9]+[l|L]");
             private static final Pattern field_179262_g = Pattern.compile("[-+]?[0-9]+[s|S]");
