--- decompiled\net\minecraft\nbt\NBTBase.java
+++ src\main\java\net\minecraft\nbt\NBTBase.java
@@ -4,10 +4,11 @@
 import java.io.DataOutput;
 import java.io.IOException;
 
 public abstract class NBTBase
 {
+
     public static final String[] NBT_TYPES = new String[] {"END", "BYTE", "SHORT", "INT", "LONG", "FLOAT", "DOUBLE", "BYTE[]", "STRING", "LIST", "COMPOUND", "INT[]"};
     private static final String __OBFID = "CL_00001229";
 
     abstract void write(DataOutput output) throws IOException;
 
@@ -80,10 +81,11 @@
         return this.toString();
     }
 
     public abstract static class NBTPrimitive extends NBTBase
         {
+
             private static final String __OBFID = "CL_00001230";
 
             public abstract long getLong();
 
             public abstract int getInt();
