--- decompiled\net\minecraft\nbt\NBTSizeTracker.java
+++ src\main\java\net\minecraft\nbt\NBTSizeTracker.java
@@ -1,13 +1,16 @@
 package net.minecraft.nbt;
 
 public class NBTSizeTracker
 {
+
     public static final NBTSizeTracker INFINITE = new NBTSizeTracker(0L)
     {
         private static final String __OBFID = "CL_00001902";
-        public void read(long bits) {}
+        public void read(long bits)
+        {
+        }
     };
     private final long max;
     private long read;
     private static final String __OBFID = "CL_00001903";
 
