--- decompiled\net\minecraft\nbt\NBTTagByteArray.java
+++ src\main\java\net\minecraft\nbt\NBTTagByteArray.java
@@ -5,14 +5,17 @@
 import java.io.IOException;
 import java.util.Arrays;
 
 public class NBTTagByteArray extends NBTBase
 {
+
     private byte[] data;
     private static final String __OBFID = "CL_00001213";
 
-    NBTTagByteArray() {}
+    NBTTagByteArray()
+    {
+    }
 
     public NBTTagByteArray(byte[] data)
     {
         this.data = data;
     }
