--- decompiled\net\minecraft\nbt\NBTTagLong.java
+++ src\main\java\net\minecraft\nbt\NBTTagLong.java
@@ -4,14 +4,17 @@
 import java.io.DataOutput;
 import java.io.IOException;
 
 public class NBTTagLong extends NBTBase.NBTPrimitive
 {
+
     private long data;
     private static final String __OBFID = "CL_00001225";
 
-    NBTTagLong() {}
+    NBTTagLong()
+    {
+    }
 
     public NBTTagLong(long data)
     {
         this.data = data;
     }
