--- decompiled\net\minecraft\nbt\NBTTagFloat.java
+++ src\main\java\net\minecraft\nbt\NBTTagFloat.java
@@ -1,18 +1,22 @@
 package net.minecraft.nbt;
 
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
+
 import net.minecraft.util.MathHelper;
 
 public class NBTTagFloat extends NBTBase.NBTPrimitive
 {
+
     private float data;
     private static final String __OBFID = "CL_00001220";
 
-    NBTTagFloat() {}
+    NBTTagFloat()
+    {
+    }
 
     public NBTTagFloat(float data)
     {
         this.data = data;
     }
