--- decompiled\net\minecraft\nbt\NBTTagIntArray.java
+++ src\main\java\net\minecraft\nbt\NBTTagIntArray.java
@@ -5,14 +5,17 @@
 import java.io.IOException;
 import java.util.Arrays;
 
 public class NBTTagIntArray extends NBTBase
 {
+
     private int[] intArray;
     private static final String __OBFID = "CL_00001221";
 
-    NBTTagIntArray() {}
+    NBTTagIntArray()
+    {
+    }
 
     public NBTTagIntArray(int[] p_i45132_1_)
     {
         this.intArray = p_i45132_1_;
     }
