--- decompiled\net\minecraft\nbt\NBTTagShort.java
+++ src\main\java\net\minecraft\nbt\NBTTagShort.java
@@ -4,14 +4,17 @@
 import java.io.DataOutput;
 import java.io.IOException;
 
 public class NBTTagShort extends NBTBase.NBTPrimitive
 {
+
     private short data;
     private static final String __OBFID = "CL_00001227";
 
-    public NBTTagShort() {}
+    public NBTTagShort()
+    {
+    }
 
     public NBTTagShort(short data)
     {
         this.data = data;
     }
