--- decompiled\net\minecraft\nbt\NBTTagByte.java
+++ src\main\java\net\minecraft\nbt\NBTTagByte.java
@@ -4,14 +4,17 @@
 import java.io.DataOutput;
 import java.io.IOException;
 
 public class NBTTagByte extends NBTBase.NBTPrimitive
 {
+
     private byte data;
     private static final String __OBFID = "CL_00001214";
 
-    NBTTagByte() {}
+    NBTTagByte()
+    {
+    }
 
     public NBTTagByte(byte data)
     {
         this.data = data;
     }
