--- decompiled\net\minecraft\nbt\NBTTagInt.java
+++ src\main\java\net\minecraft\nbt\NBTTagInt.java
@@ -4,14 +4,17 @@
 import java.io.DataOutput;
 import java.io.IOException;
 
 public class NBTTagInt extends NBTBase.NBTPrimitive
 {
+
     private int data;
     private static final String __OBFID = "CL_00001223";
 
-    NBTTagInt() {}
+    NBTTagInt()
+    {
+    }
 
     public NBTTagInt(int data)
     {
         this.data = data;
     }
