--- decompiled\net\minecraft\nbt\NBTTagDouble.java
+++ src\main\java\net\minecraft\nbt\NBTTagDouble.java
@@ -1,18 +1,22 @@
 package net.minecraft.nbt;
 
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
+
 import net.minecraft.util.MathHelper;
 
 public class NBTTagDouble extends NBTBase.NBTPrimitive
 {
+
     private double data;
     private static final String __OBFID = "CL_00001218";
 
-    NBTTagDouble() {}
+    NBTTagDouble()
+    {
+    }
 
     public NBTTagDouble(double data)
     {
         this.data = data;
     }
