--- decompiled\net\minecraft\nbt\NBTTagFloat.java
+++ src\main\java\net\minecraft\nbt\NBTTagFloat.java
@@ -1,93 +1,78 @@
 package net.minecraft.nbt;
 
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
+
 import net.minecraft.util.MathHelper;
 
-public class NBTTagFloat extends NBTBase.NBTPrimitive
-{
+public class NBTTagFloat extends NBTBase.NBTPrimitive {
+
     private float data;
     private static final String __OBFID = "CL_00001220";
 
-    NBTTagFloat() {}
+    NBTTagFloat() {
+    }
 
-    public NBTTagFloat(float data)
-    {
+    public NBTTagFloat(float data) {
         this.data = data;
     }
 
-    void write(DataOutput output) throws IOException
-    {
+    void write(DataOutput output) throws IOException {
         output.writeFloat(this.data);
     }
 
-    void read(DataInput input, int depth, NBTSizeTracker sizeTracker) throws IOException
-    {
+    void read(DataInput input, int depth, NBTSizeTracker sizeTracker) throws IOException {
         sizeTracker.read(32L);
         this.data = input.readFloat();
     }
 
-    public byte getId()
-    {
-        return (byte)5;
+    public byte getId() {
+        return (byte) 5;
     }
 
-    public String toString()
-    {
+    public String toString() {
         return "" + this.data + "f";
     }
 
-    public NBTBase copy()
-    {
+    public NBTBase copy() {
         return new NBTTagFloat(this.data);
     }
 
-    public boolean equals(Object p_equals_1_)
-    {
-        if (super.equals(p_equals_1_))
-        {
-            NBTTagFloat nbttagfloat = (NBTTagFloat)p_equals_1_;
+    public boolean equals(Object p_equals_1_) {
+        if (super.equals(p_equals_1_)) {
+            NBTTagFloat nbttagfloat = (NBTTagFloat) p_equals_1_;
             return this.data == nbttagfloat.data;
-        }
-        else
-        {
+        } else {
             return false;
         }
     }
 
-    public int hashCode()
-    {
+    public int hashCode() {
         return super.hashCode() ^ Float.floatToIntBits(this.data);
     }
 
-    public long getLong()
-    {
-        return (long)this.data;
+    public long getLong() {
+        return (long) this.data;
     }
 
-    public int getInt()
-    {
+    public int getInt() {
         return MathHelper.floor_float(this.data);
     }
 
-    public short getShort()
-    {
-        return (short)(MathHelper.floor_float(this.data) & 65535);
+    public short getShort() {
+        return (short) (MathHelper.floor_float(this.data) & 65535);
     }
 
-    public byte getByte()
-    {
-        return (byte)(MathHelper.floor_float(this.data) & 255);
+    public byte getByte() {
+        return (byte) (MathHelper.floor_float(this.data) & 255);
     }
 
-    public double getDouble()
-    {
-        return (double)this.data;
+    public double getDouble() {
+        return (double) this.data;
     }
 
-    public float getFloat()
-    {
+    public float getFloat() {
         return this.data;
     }
 }
