--- decompiled\net\minecraft\nbt\NBTTagCompound.java
+++ src\main\java\net\minecraft\nbt\NBTTagCompound.java
@@ -1,23 +1,27 @@
 package net.minecraft.nbt;
 
-import com.google.common.collect.Maps;
 import java.io.DataInput;
 import java.io.DataOutput;
 import java.io.IOException;
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.Callable;
+
 import net.minecraft.crash.CrashReport;
 import net.minecraft.crash.CrashReportCategory;
 import net.minecraft.util.ReportedException;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import com.google.common.collect.Maps;
+
 public class NBTTagCompound extends NBTBase
 {
+
     private static final Logger logger = LogManager.getLogger();
     private Map tagMap = Maps.newHashMap();
     private static final String __OBFID = "CL_00001215";
 
     void write(DataOutput output) throws IOException
