--- decompiled\net\minecraft\nbt\NBTUtil.java
+++ src\main\java\net\minecraft\nbt\NBTUtil.java
@@ -1,15 +1,18 @@
 package net.minecraft.nbt;
 
-import com.mojang.authlib.GameProfile;
-import com.mojang.authlib.properties.Property;
 import java.util.Iterator;
 import java.util.UUID;
+
 import net.minecraft.util.StringUtils;
 
+import com.mojang.authlib.GameProfile;
+import com.mojang.authlib.properties.Property;
+
 public final class NBTUtil
 {
+
     private static final String __OBFID = "CL_00001901";
 
     public static GameProfile readGameProfileFromNBT(NBTTagCompound compound)
     {
         String s = null;
