--- decompiled\net\minecraft\network\login\server\S02PacketLoginSuccess.java
+++ src\main\java\net\minecraft\network\login\server\S02PacketLoginSuccess.java
@@ -1,23 +1,26 @@
 package net.minecraft.network.login.server;
 
-import com.mojang.authlib.GameProfile;
 import java.io.IOException;
 import java.util.UUID;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.login.INetHandlerLoginClient;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.mojang.authlib.GameProfile;
+
 public class S02PacketLoginSuccess implements Packet
 {
+
     private GameProfile profile;
     private static final String __OBFID = "CL_00001375";
 
-    public S02PacketLoginSuccess() {}
+    public S02PacketLoginSuccess()
+    {
+    }
 
     public S02PacketLoginSuccess(GameProfile profileIn)
     {
         this.profile = profileIn;
     }
@@ -40,16 +43,10 @@
     public void processPacket(INetHandlerLoginClient handler)
     {
         handler.handleLoginSuccess(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public GameProfile getProfile()
-    {
-        return this.profile;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerLoginClient)handler);
+        this.processPacket((INetHandlerLoginClient) handler);
     }
 }
