--- decompiled\net\minecraft\entity\player\PlayerCapabilities.java
+++ src\main\java\net\minecraft\entity\player\PlayerCapabilities.java
@@ -1,20 +1,20 @@
 package net.minecraft.entity.player;
 
 import net.minecraft.nbt.NBTTagCompound;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class PlayerCapabilities
 {
+
     public boolean disableDamage;
     public boolean isFlying;
     public boolean allowFlying;
     public boolean isCreativeMode;
     public boolean allowEdit = true;
-    private float flySpeed = 0.05F;
-    private float walkSpeed = 0.1F;
+    public float flySpeed = 0.05F; // SpigotMCP -- Apply access transformation
+    public float walkSpeed = 0.1F; // SpigotMCP -- Apply access transformation
     private static final String __OBFID = "CL_00001708";
 
     public void writeCapabilitiesToNBT(NBTTagCompound p_75091_1_)
     {
         NBTTagCompound nbttagcompound1 = new NBTTagCompound();
@@ -54,22 +54,11 @@
     public float getFlySpeed()
     {
         return this.flySpeed;
     }
 
-    @SideOnly(Side.CLIENT)
-    public void setFlySpeed(float speed)
-    {
-        this.flySpeed = speed;
-    }
-
     public float getWalkSpeed()
     {
         return this.walkSpeed;
     }
 
-    @SideOnly(Side.CLIENT)
-    public void setPlayerWalkSpeed(float speed)
-    {
-        this.walkSpeed = speed;
-    }
 }
