--- decompiled\net\minecraft\world\biome\BiomeGenStoneBeach.java
+++ src\main\java\net\minecraft\world\biome\BiomeGenStoneBeach.java
@@ -1,15 +1,14 @@
 package net.minecraft.world.biome;
 
 import net.minecraft.init.Blocks;
 
-public class BiomeGenStoneBeach extends BiomeGenBase
-{
+public class BiomeGenStoneBeach extends BiomeGenBase {
+
     private static final String __OBFID = "CL_00000184";
 
-    public BiomeGenStoneBeach(int p_i45384_1_)
-    {
+    public BiomeGenStoneBeach(int p_i45384_1_) {
         super(p_i45384_1_);
         this.spawnableCreatureList.clear();
         this.topBlock = Blocks.stone.getDefaultState();
         this.fillerBlock = Blocks.stone.getDefaultState();
         this.theBiomeDecorator.treesPerChunk = -999;
