--- decompiled\net\minecraft\world\biome\BiomeGenSwamp.java
+++ src\main\java\net\minecraft\world\biome\BiomeGenSwamp.java
@@ -1,21 +1,22 @@
 package net.minecraft.world.biome;
 
 import java.util.Random;
+
 import net.minecraft.block.BlockFlower;
 import net.minecraft.block.material.Material;
 import net.minecraft.entity.monster.EntitySlime;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 import net.minecraft.world.chunk.ChunkPrimer;
 import net.minecraft.world.gen.feature.WorldGenAbstractTree;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class BiomeGenSwamp extends BiomeGenBase
 {
+
     private static final String __OBFID = "CL_00000185";
 
     protected BiomeGenSwamp(int p_i1988_1_)
     {
         super(p_i1988_1_);
@@ -71,19 +72,6 @@
             }
         }
 
         this.generateBiomeTerrain(worldIn, p_180622_2_, p_180622_3_, p_180622_4_, p_180622_5_, p_180622_6_);
     }
-
-    @SideOnly(Side.CLIENT)
-    public int getGrassColorAtPos(BlockPos p_180627_1_)
-    {
-        double d0 = field_180281_af.func_151601_a((double)p_180627_1_.getX() * 0.0225D, (double)p_180627_1_.getZ() * 0.0225D);
-        return d0 < -0.1D ? 5011004 : 6975545;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int getFoliageColorAtPos(BlockPos p_180625_1_)
-    {
-        return 6975545;
-    }
 }
