--- decompiled\net\minecraft\world\biome\BiomeGenBeach.java
+++ src\main\java\net\minecraft\world\biome\BiomeGenBeach.java
@@ -1,15 +1,14 @@
 package net.minecraft.world.biome;
 
 import net.minecraft.init.Blocks;
 
-public class BiomeGenBeach extends BiomeGenBase
-{
+public class BiomeGenBeach extends BiomeGenBase {
+
     private static final String __OBFID = "CL_00000157";
 
-    public BiomeGenBeach(int p_i1969_1_)
-    {
+    public BiomeGenBeach(int p_i1969_1_) {
         super(p_i1969_1_);
         this.spawnableCreatureList.clear();
         this.topBlock = Blocks.sand.getDefaultState();
         this.fillerBlock = Blocks.sand.getDefaultState();
         this.theBiomeDecorator.treesPerChunk = -999;
