--- decompiled\net\minecraft\world\biome\BiomeGenForest.java
+++ src\main\java\net\minecraft\world\biome\BiomeGenForest.java
@@ -1,23 +1,24 @@
 package net.minecraft.world.biome;
 
 import java.util.Random;
+
 import net.minecraft.block.BlockDoublePlant;
 import net.minecraft.block.BlockFlower;
 import net.minecraft.entity.passive.EntityWolf;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 import net.minecraft.world.gen.feature.WorldGenAbstractTree;
 import net.minecraft.world.gen.feature.WorldGenBigMushroom;
 import net.minecraft.world.gen.feature.WorldGenCanopyTree;
 import net.minecraft.world.gen.feature.WorldGenForest;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class BiomeGenForest extends BiomeGenBase
 {
+
     private int field_150632_aF;
     protected static final WorldGenForest field_150629_aC = new WorldGenForest(false, true);
     protected static final WorldGenForest field_150630_aD = new WorldGenForest(false, false);
     protected static final WorldGenCanopyTree field_150631_aE = new WorldGenCanopyTree(false);
     private static final String __OBFID = "CL_00000170";
@@ -55,11 +56,11 @@
         {
             this.theBiomeDecorator.treesPerChunk = -999;
         }
     }
 
-    protected BiomeGenBase func_150557_a(int p_150557_1_, boolean p_150557_2_)
+    public BiomeGenBase func_150557_a(int p_150557_1_, boolean p_150557_2_)
     {
         if (this.field_150632_aF == 2)
         {
             this.field_150609_ah = 353825;
             this.color = p_150557_1_;
@@ -181,19 +182,12 @@
         }
 
         super.decorate(worldIn, p_180624_2_, p_180624_3_);
     }
 
-    @SideOnly(Side.CLIENT)
-    public int getGrassColorAtPos(BlockPos p_180627_1_)
+    public BiomeGenBase createMutatedBiome(final int p_180277_1_)
     {
-        int i = super.getGrassColorAtPos(p_180627_1_);
-        return this.field_150632_aF == 3 ? (i & 16711422) + 2634762 >> 1 : i;
-    }
-
-    protected BiomeGenBase createMutatedBiome(final int p_180277_1_)
-    {
         if (this.biomeID == BiomeGenBase.forest.biomeID)
         {
             BiomeGenForest biomegenforest = new BiomeGenForest(p_180277_1_, 1);
             biomegenforest.setHeight(new BiomeGenBase.Height(this.minHeight, this.maxHeight + 0.2F));
             biomegenforest.setBiomeName("Flower Forest");
@@ -208,11 +202,11 @@
                 private static final String __OBFID = "CL_00001861";
                 public void decorate(World worldIn, Random p_180624_2_, BlockPos p_180624_3_)
                 {
                     this.baseBiome.decorate(worldIn, p_180624_2_, p_180624_3_);
                 }
-            }: new BiomeGenMutated(p_180277_1_, this)
+            } : new BiomeGenMutated(p_180277_1_, this)
             {
                 private static final String __OBFID = "CL_00000172";
                 public WorldGenAbstractTree genBigTreeChance(Random p_150567_1_)
                 {
                     return p_150567_1_.nextBoolean() ? BiomeGenForest.field_150629_aC : BiomeGenForest.field_150630_aD;
