--- decompiled\net\minecraft\world\biome\BiomeGenEnd.java
+++ src\main\java\net\minecraft\world\biome\BiomeGenEnd.java
@@ -1,14 +1,14 @@
 package net.minecraft.world.biome;
 
 import net.minecraft.entity.monster.EntityEnderman;
 import net.minecraft.init.Blocks;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class BiomeGenEnd extends BiomeGenBase
 {
+
     private static final String __OBFID = "CL_00000187";
 
     public BiomeGenEnd(int p_i1990_1_)
     {
         super(p_i1990_1_);
@@ -19,12 +19,6 @@
         this.spawnableMonsterList.add(new BiomeGenBase.SpawnListEntry(EntityEnderman.class, 10, 4, 4));
         this.topBlock = Blocks.dirt.getDefaultState();
         this.fillerBlock = Blocks.dirt.getDefaultState();
         this.theBiomeDecorator = new BiomeEndDecorator();
     }
-
-    @SideOnly(Side.CLIENT)
-    public int getSkyColorByTemp(float p_76731_1_)
-    {
-        return 0;
-    }
 }
