--- decompiled\net\minecraft\world\biome\BiomeGenPlains.java
+++ src\main\java\net\minecraft\world\biome\BiomeGenPlains.java
@@ -1,20 +1,22 @@
 package net.minecraft.world.biome;
 
 import java.util.Random;
+
 import net.minecraft.block.BlockDoublePlant;
 import net.minecraft.block.BlockFlower;
 import net.minecraft.entity.passive.EntityHorse;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class BiomeGenPlains extends BiomeGenBase
 {
+
     protected boolean field_150628_aC;
     private static final String __OBFID = "CL_00000180";
 
-    protected BiomeGenPlains(int p_i1986_1_)
+    public BiomeGenPlains(int p_i1986_1_)
     {
         super(p_i1986_1_);
         this.setTemperatureRainfall(0.8F, 0.4F);
         this.setHeight(height_LowPlains);
         this.spawnableCreatureList.add(new BiomeGenBase.SpawnListEntry(EntityHorse.class, 5, 2, 6));
@@ -98,11 +100,11 @@
         }
 
         super.decorate(worldIn, p_180624_2_, p_180624_3_);
     }
 
-    protected BiomeGenBase createMutatedBiome(int p_180277_1_)
+    public BiomeGenBase createMutatedBiome(int p_180277_1_)
     {
         BiomeGenPlains biomegenplains = new BiomeGenPlains(p_180277_1_);
         biomegenplains.setBiomeName("Sunflower Plains");
         biomegenplains.field_150628_aC = true;
         biomegenplains.setColor(9286496);
