--- decompiled\net\minecraft\world\biome\BiomeDecorator.java
+++ src\main\java\net\minecraft\world\biome\BiomeDecorator.java
@@ -1,8 +1,9 @@
 package net.minecraft.world.biome;
 
 import java.util.Random;
+
 import net.minecraft.block.BlockFlower;
 import net.minecraft.block.BlockStone;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
@@ -23,47 +24,48 @@
 import net.minecraft.world.gen.feature.WorldGenWaterlily;
 import net.minecraft.world.gen.feature.WorldGenerator;
 
 public class BiomeDecorator
 {
-    protected World currentWorld;
-    protected Random randomGenerator;
-    protected BlockPos field_180294_c;
-    protected ChunkProviderSettings chunkProviderSettings;
-    protected WorldGenerator clayGen = new WorldGenClay(4);
-    protected WorldGenerator sandGen;
-    protected WorldGenerator gravelAsSandGen;
-    protected WorldGenerator dirtGen;
-    protected WorldGenerator gravelGen;
-    protected WorldGenerator graniteGen;
-    protected WorldGenerator dioriteGen;
-    protected WorldGenerator andesiteGen;
-    protected WorldGenerator coalGen;
-    protected WorldGenerator ironGen;
-    protected WorldGenerator goldGen;
-    protected WorldGenerator redstoneGen;
-    protected WorldGenerator diamondGen;
-    protected WorldGenerator lapisGen;
-    protected WorldGenFlowers yellowFlowerGen;
-    protected WorldGenerator mushroomBrownGen;
-    protected WorldGenerator mushroomRedGen;
-    protected WorldGenerator bigMushroomGen;
-    protected WorldGenerator reedGen;
-    protected WorldGenerator cactusGen;
-    protected WorldGenerator waterlilyGen;
-    protected int waterlilyPerChunk;
-    protected int treesPerChunk;
-    protected int flowersPerChunk;
-    protected int grassPerChunk;
-    protected int deadBushPerChunk;
-    protected int mushroomsPerChunk;
-    protected int reedsPerChunk;
-    protected int cactiPerChunk;
-    protected int sandPerChunk;
-    protected int sandPerChunk2;
-    protected int clayPerChunk;
-    protected int bigMushroomsPerChunk;
+
+    public World currentWorld;
+    public Random randomGenerator;
+    public BlockPos field_180294_c;
+    public ChunkProviderSettings chunkProviderSettings;
+    public WorldGenerator clayGen = new WorldGenClay(4);
+    public WorldGenerator sandGen;
+    public WorldGenerator gravelAsSandGen;
+    public WorldGenerator dirtGen;
+    public WorldGenerator gravelGen;
+    public WorldGenerator graniteGen;
+    public WorldGenerator dioriteGen;
+    public WorldGenerator andesiteGen;
+    public WorldGenerator coalGen;
+    public WorldGenerator ironGen;
+    public WorldGenerator goldGen;
+    public WorldGenerator redstoneGen;
+    public WorldGenerator diamondGen;
+    public WorldGenerator lapisGen;
+    public WorldGenFlowers yellowFlowerGen;
+    public WorldGenerator mushroomBrownGen;
+    public WorldGenerator mushroomRedGen;
+    public WorldGenerator bigMushroomGen;
+    public WorldGenerator reedGen;
+    public WorldGenerator cactusGen;
+    public WorldGenerator waterlilyGen;
+    public int waterlilyPerChunk;
+    public int treesPerChunk;
+    public int flowersPerChunk;
+    public int grassPerChunk;
+    public int deadBushPerChunk;
+    public int mushroomsPerChunk;
+    public int reedsPerChunk;
+    public int cactiPerChunk;
+    public int sandPerChunk;
+    public int sandPerChunk2;
+    public int clayPerChunk;
+    public int bigMushroomsPerChunk;
     public boolean generateLakes;
     private static final String __OBFID = "CL_00000164";
 
     public BiomeDecorator()
     {
@@ -165,11 +167,11 @@
         {
             k = this.randomGenerator.nextInt(16) + 8;
             l = this.randomGenerator.nextInt(16) + 8;
             WorldGenAbstractTree worldgenabstracttree = p_150513_1_.genBigTreeChance(this.randomGenerator);
             worldgenabstracttree.func_175904_e();
-            blockpos = this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l));
+            blockpos = this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)); // CraftBukkit - Prevent crash
 
             if (worldgenabstracttree.generate(this.currentWorld, this.randomGenerator, blockpos))
             {
                 worldgenabstracttree.func_180711_a(this.currentWorld, this.randomGenerator, blockpos);
             }
@@ -177,20 +179,20 @@
 
         for (j = 0; j < this.bigMushroomsPerChunk; ++j)
         {
             k = this.randomGenerator.nextInt(16) + 8;
             l = this.randomGenerator.nextInt(16) + 8;
-            this.bigMushroomGen.generate(this.currentWorld, this.randomGenerator, this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l)));
+            this.bigMushroomGen.generate(this.currentWorld, this.randomGenerator, this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)));  // CraftBukkit - Prevent crash
         }
 
         int i1;
 
         for (j = 0; j < this.flowersPerChunk; ++j)
         {
             k = this.randomGenerator.nextInt(16) + 8;
             l = this.randomGenerator.nextInt(16) + 8;
-            i1 = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l)).getY() + 32);
+            i1 = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)).getY() + 32); // CraftBukkit - Prevent crash
             blockpos = this.field_180294_c.add(k, i1, l);
             BlockFlower.EnumFlowerType enumflowertype = p_150513_1_.pickRandomFlower(this.randomGenerator, blockpos);
             BlockFlower blockflower = enumflowertype.getBlockType().getBlock();
 
             if (blockflower.getMaterial() != Material.air)
@@ -202,29 +204,29 @@
 
         for (j = 0; j < this.grassPerChunk; ++j)
         {
             k = this.randomGenerator.nextInt(16) + 8;
             l = this.randomGenerator.nextInt(16) + 8;
-            i1 = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l)).getY() * 2);
+            i1 = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)).getY() * 2); // CraftBukkit - Prevent crash
             p_150513_1_.getRandomWorldGenForGrass(this.randomGenerator).generate(this.currentWorld, this.randomGenerator, this.field_180294_c.add(k, i1, l));
         }
 
         for (j = 0; j < this.deadBushPerChunk; ++j)
         {
             k = this.randomGenerator.nextInt(16) + 8;
             l = this.randomGenerator.nextInt(16) + 8;
-            i1 = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l)).getY() * 2);
+            i1 = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)).getY() * 2); // CraftBukkit - Prevent crash
             (new WorldGenDeadBush()).generate(this.currentWorld, this.randomGenerator, this.field_180294_c.add(k, i1, l));
         }
 
         j = 0;
 
         while (j < this.waterlilyPerChunk)
         {
             k = this.randomGenerator.nextInt(16) + 8;
             l = this.randomGenerator.nextInt(16) + 8;
-            i1 = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l)).getY() * 2);
+            i1 = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)).getY() * 2); // CraftBukkit - Prevent crash
             blockpos = this.field_180294_c.add(k, i1, l);
 
             while (true)
             {
                 if (blockpos.getY() > 0)
@@ -248,69 +250,69 @@
         {
             if (this.randomGenerator.nextInt(4) == 0)
             {
                 k = this.randomGenerator.nextInt(16) + 8;
                 l = this.randomGenerator.nextInt(16) + 8;
-                BlockPos blockpos2 = this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l));
+                BlockPos blockpos2 = this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)); // CraftBukkit - Prevent crash
                 this.mushroomBrownGen.generate(this.currentWorld, this.randomGenerator, blockpos2);
             }
 
             if (this.randomGenerator.nextInt(8) == 0)
             {
                 k = this.randomGenerator.nextInt(16) + 8;
                 l = this.randomGenerator.nextInt(16) + 8;
-                i1 = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l)).getY() * 2);
+                i1 = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)).getY() * 2);  // CraftBukkit - Prevent crash
                 blockpos = this.field_180294_c.add(k, i1, l);
                 this.mushroomRedGen.generate(this.currentWorld, this.randomGenerator, blockpos);
             }
         }
 
         if (this.randomGenerator.nextInt(4) == 0)
         {
             j = this.randomGenerator.nextInt(16) + 8;
             k = this.randomGenerator.nextInt(16) + 8;
-            l = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(j, 0, k)).getY() * 2);
+            l = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(j, 0, k)).getY() * 2); // CraftBukkit - Prevent crash
             this.mushroomBrownGen.generate(this.currentWorld, this.randomGenerator, this.field_180294_c.add(j, l, k));
         }
 
         if (this.randomGenerator.nextInt(8) == 0)
         {
             j = this.randomGenerator.nextInt(16) + 8;
             k = this.randomGenerator.nextInt(16) + 8;
-            l = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(j, 0, k)).getY() * 2);
+            l = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(j, 0, k)).getY() * 2); // CraftBukkit - Prevent crash
             this.mushroomRedGen.generate(this.currentWorld, this.randomGenerator, this.field_180294_c.add(j, l, k));
         }
 
         for (j = 0; j < this.reedsPerChunk; ++j)
         {
             k = this.randomGenerator.nextInt(16) + 8;
             l = this.randomGenerator.nextInt(16) + 8;
-            i1 = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l)).getY() * 2);
+            i1 = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)).getY() * 2); // CraftBukkit - Prevent crash
             this.reedGen.generate(this.currentWorld, this.randomGenerator, this.field_180294_c.add(k, i1, l));
         }
 
         for (j = 0; j < 10; ++j)
         {
             k = this.randomGenerator.nextInt(16) + 8;
             l = this.randomGenerator.nextInt(16) + 8;
-            i1 = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l)).getY() * 2);
+            i1 = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)).getY() * 2); // CraftBukkit - Prevent crash
             this.reedGen.generate(this.currentWorld, this.randomGenerator, this.field_180294_c.add(k, i1, l));
         }
 
         if (this.randomGenerator.nextInt(32) == 0)
         {
             j = this.randomGenerator.nextInt(16) + 8;
             k = this.randomGenerator.nextInt(16) + 8;
-            l = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(j, 0, k)).getY() * 2);
+            l = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(j, 0, k)).getY() * 2); // CraftBukkit - Prevent crash
             (new WorldGenPumpkin()).generate(this.currentWorld, this.randomGenerator, this.field_180294_c.add(j, l, k));
         }
 
         for (j = 0; j < this.cactiPerChunk; ++j)
         {
             k = this.randomGenerator.nextInt(16) + 8;
             l = this.randomGenerator.nextInt(16) + 8;
-            i1 = this.randomGenerator.nextInt(this.currentWorld.getHorizon(this.field_180294_c.add(k, 0, l)).getY() * 2);
+            i1 = this.randomGenerator.nextInt(this.getHighestBlockYAt(this.field_180294_c.add(k, 0, l)).getY() * 2); // CraftBukkit - Prevent crash
             this.cactusGen.generate(this.currentWorld, this.randomGenerator, this.field_180294_c.add(k, i1, l));
         }
 
         if (this.generateLakes)
         {
@@ -328,10 +330,25 @@
                 (new WorldGenLiquids(Blocks.flowing_lava)).generate(this.currentWorld, this.randomGenerator, blockpos1);
             }
         }
     }
 
+
+    // CraftBukkit start - The heightMap can be wrong calculated and return Y = 0 which will crash the Server
+    private BlockPos getHighestBlockYAt(BlockPos blockPosition)
+    {
+        BlockPos returnBlockPosition = this.currentWorld.getHorizon(blockPosition);
+
+        if (returnBlockPosition.getY() == 0)
+        {
+            returnBlockPosition = returnBlockPosition.up(1);
+        }
+
+        return returnBlockPosition;
+    }
+    // CraftBukkit end
+
     protected void genStandardOre1(int p_76795_1_, WorldGenerator p_76795_2_, int p_76795_3_, int p_76795_4_)
     {
         int l;
 
         if (p_76795_4_ < p_76795_3_)
