--- decompiled\net\minecraft\world\biome\BiomeGenHell.java
+++ src\main\java\net\minecraft\world\biome\BiomeGenHell.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.monster.EntityMagmaCube;
 import net.minecraft.entity.monster.EntityPigZombie;
 
 public class BiomeGenHell extends BiomeGenBase
 {
+
     private static final String __OBFID = "CL_00000173";
 
     public BiomeGenHell(int p_i1981_1_)
     {
         super(p_i1981_1_);
