--- decompiled\net\minecraft\world\biome\BiomeGenTaiga.java
+++ src\main\java\net\minecraft\world\biome\BiomeGenTaiga.java
@@ -1,8 +1,9 @@
 package net.minecraft.world.biome;
 
 import java.util.Random;
+
 import net.minecraft.block.BlockDirt;
 import net.minecraft.block.BlockDoublePlant;
 import net.minecraft.block.BlockTallGrass;
 import net.minecraft.entity.passive.EntityWolf;
 import net.minecraft.init.Blocks;
@@ -17,10 +18,11 @@
 import net.minecraft.world.gen.feature.WorldGenTallGrass;
 import net.minecraft.world.gen.feature.WorldGenerator;
 
 public class BiomeGenTaiga extends BiomeGenBase
 {
+
     private static final WorldGenTaiga1 field_150639_aC = new WorldGenTaiga1();
     private static final WorldGenTaiga2 field_150640_aD = new WorldGenTaiga2(false);
     private static final WorldGenMegaPineTree field_150641_aE = new WorldGenMegaPineTree(false, false);
     private static final WorldGenMegaPineTree field_150642_aF = new WorldGenMegaPineTree(false, true);
     private static final WorldGenBlockBlob field_150643_aG = new WorldGenBlockBlob(Blocks.mossy_cobblestone, 0);
@@ -108,10 +110,10 @@
         }
 
         this.generateBiomeTerrain(worldIn, p_180622_2_, p_180622_3_, p_180622_4_, p_180622_5_, p_180622_6_);
     }
 
-    protected BiomeGenBase createMutatedBiome(int p_180277_1_)
+    public BiomeGenBase createMutatedBiome(int p_180277_1_)
     {
         return this.biomeID == BiomeGenBase.megaTaiga.biomeID ? (new BiomeGenTaiga(p_180277_1_, 2)).func_150557_a(5858897, true).setBiomeName("Mega Spruce Taiga").setFillerBlockMetadata(5159473).setTemperatureRainfall(0.25F, 0.8F).setHeight(new BiomeGenBase.Height(this.minHeight, this.maxHeight)) : super.createMutatedBiome(p_180277_1_);
     }
 }
