--- decompiled\net\minecraft\world\biome\BiomeGenSavanna.java
+++ src\main\java\net\minecraft\world\biome\BiomeGenSavanna.java
@@ -1,8 +1,9 @@
 package net.minecraft.world.biome;
 
 import java.util.Random;
+
 import net.minecraft.block.BlockDirt;
 import net.minecraft.block.BlockDoublePlant;
 import net.minecraft.entity.passive.EntityHorse;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
@@ -11,14 +12,15 @@
 import net.minecraft.world.gen.feature.WorldGenAbstractTree;
 import net.minecraft.world.gen.feature.WorldGenSavannaTree;
 
 public class BiomeGenSavanna extends BiomeGenBase
 {
+
     private static final WorldGenSavannaTree field_150627_aC = new WorldGenSavannaTree(false);
     private static final String __OBFID = "CL_00000182";
 
-    protected BiomeGenSavanna(int p_i45383_1_)
+    public BiomeGenSavanna(int p_i45383_1_)
     {
         super(p_i45383_1_);
         this.spawnableCreatureList.add(new BiomeGenBase.SpawnListEntry(EntityHorse.class, 1, 2, 6));
         this.theBiomeDecorator.treesPerChunk = 1;
         this.theBiomeDecorator.flowersPerChunk = 4;
@@ -28,11 +30,11 @@
     public WorldGenAbstractTree genBigTreeChance(Random p_150567_1_)
     {
         return (WorldGenAbstractTree)(p_150567_1_.nextInt(5) > 0 ? field_150627_aC : this.worldGeneratorTrees);
     }
 
-    protected BiomeGenBase createMutatedBiome(int p_180277_1_)
+    public BiomeGenBase createMutatedBiome(int p_180277_1_)
     {
         BiomeGenSavanna.Mutated mutated = new BiomeGenSavanna.Mutated(p_180277_1_, this);
         mutated.temperature = (this.temperature + 1.0F) * 0.5F;
         mutated.minHeight = this.minHeight * 0.5F + 0.3F;
         mutated.maxHeight = this.maxHeight * 0.5F + 1.2F;
@@ -54,10 +56,11 @@
         super.decorate(worldIn, p_180624_2_, p_180624_3_);
     }
 
     public static class Mutated extends BiomeGenMutated
         {
+
             private static final String __OBFID = "CL_00000183";
 
             public Mutated(int p_i45382_1_, BiomeGenBase p_i45382_2_)
             {
                 super(p_i45382_1_, p_i45382_2_);
