--- decompiled\net\minecraft\entity\passive\EntityRabbit.java
+++ src\main\java\net\minecraft\entity\passive\EntityRabbit.java
@@ -1,8 +1,7 @@
 package net.minecraft.entity.passive;
 
-import com.google.common.base.Predicate;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockCarrot;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityAgeable;
@@ -35,15 +34,16 @@
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.util.StatCollector;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.DifficultyInstance;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.base.Predicate;
+
 public class EntityRabbit extends EntityAnimal
 {
+
     private EntityRabbit.AIAvoidEntity field_175539_bk;
     private int field_175540_bm = 0;
     private int field_175535_bn = 0;
     private boolean field_175536_bo = false;
     private boolean field_175537_bp = false;
@@ -61,10 +61,17 @@
         this.field_175543_bt = null;
         this.setSize(0.6F, 0.7F);
         this.jumpHelper = new EntityRabbit.RabbitJumpHelper(this);
         this.moveHelper = new EntityRabbit.RabbitMoveHelper();
         ((PathNavigateGround)this.getNavigator()).func_179690_a(true);
+        this.initializePathFinderGoals(); // CraftBukkit - moved code
+        this.func_175515_b(0.0D);
+    }
+
+    // CraftBukkit start - code from constructor
+    public void initializePathFinderGoals()
+    {
         this.navigator.func_179678_a(2.5F);
         this.tasks.addTask(1, new EntityAISwimming(this));
         this.tasks.addTask(1, new EntityRabbit.AIPanic(1.33D));
         this.tasks.addTask(2, new EntityAITempt(this, 1.0D, Items.carrot, false));
         this.tasks.addTask(3, new EntityAIMate(this, 0.8D));
@@ -82,12 +89,12 @@
             {
                 return this.func_180086_a((Entity)p_apply_1_);
             }
         }, 16.0F, 1.33D, 1.33D);
         this.tasks.addTask(4, this.field_175539_bk);
-        this.func_175515_b(0.0D);
     }
+    // CraftBukkit end
 
     protected float func_175134_bD()
     {
         return this.moveHelper.isUpdating() && this.moveHelper.func_179919_e() > this.posY + 0.5D ? 0.5F : this.moveType.func_180074_b();
     }
@@ -95,16 +102,10 @@
     public void func_175522_a(EntityRabbit.EnumMoveType p_175522_1_)
     {
         this.moveType = p_175522_1_;
     }
 
-    @SideOnly(Side.CLIENT)
-    public float func_175521_o(float p_175521_1_)
-    {
-        return this.field_175535_bn == 0 ? 0.0F : ((float)this.field_175540_bm + p_175521_1_) / (float)this.field_175535_bn;
-    }
-
     public void func_175515_b(double p_175515_1_)
     {
         this.getNavigator().setSpeed(p_175515_1_);
         this.moveHelper.setMoveTo(this.moveHelper.func_179917_d(), this.moveHelper.func_179919_e(), this.moveHelper.func_179918_f(), p_175515_1_);
     }
@@ -219,11 +220,13 @@
         }
 
         this.field_175537_bp = this.onGround;
     }
 
-    public void spawnRunningParticles() {}
+    public void spawnRunningParticles()
+    {
+    }
 
     private void func_175533_a(double p_175533_1_, double p_175533_3_)
     {
         this.rotationYaw = (float)(Math.atan2(p_175533_3_ - this.posZ, p_175533_1_ - this.posX) * 180.0D / Math.PI) - 90.0F;
     }
@@ -449,32 +452,18 @@
     {
         this.worldObj.spawnParticle(EnumParticleTypes.BLOCK_DUST, this.posX + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, this.posY + 0.5D + (double)(this.rand.nextFloat() * this.height), this.posZ + (double)(this.rand.nextFloat() * this.width * 2.0F) - (double)this.width, 0.0D, 0.0D, 0.0D, new int[] {Block.getStateId(Blocks.carrots.getStateFromMeta(7))});
         this.carrotTicks = 100;
     }
 
-    @SideOnly(Side.CLIENT)
-    public void handleHealthUpdate(byte p_70103_1_)
-    {
-        if (p_70103_1_ == 1)
-        {
-            this.createRunningParticles();
-            this.field_175535_bn = 10;
-            this.field_175540_bm = 0;
-        }
-        else
-        {
-            super.handleHealthUpdate(p_70103_1_);
-        }
-    }
-
     public EntityAgeable createChild(EntityAgeable ageable)
     {
         return this.func_175526_b(ageable);
     }
 
     class AIAvoidEntity extends EntityAIAvoidEntity
     {
+
         private EntityRabbit entityInstance = EntityRabbit.this;
         private static final String __OBFID = "CL_00002238";
 
         public AIAvoidEntity(Predicate avoidPredicate, float p_i45865_3_, double p_i45865_4_, double p_i45865_6_)
         {
@@ -487,10 +476,11 @@
         }
     }
 
     class AIEvilAttack extends EntityAIAttackOnCollide
     {
+
         private static final String __OBFID = "CL_00002240";
 
         public AIEvilAttack()
         {
             super(EntityRabbit.this, EntityLivingBase.class, 1.4D, true);
@@ -502,10 +492,11 @@
         }
     }
 
     class AIPanic extends EntityAIPanic
     {
+
         private EntityRabbit field_179486_b = EntityRabbit.this;
         private static final String __OBFID = "CL_00002234";
 
         public AIPanic(double p_i45861_2_)
         {
@@ -519,10 +510,11 @@
         }
     }
 
     class AIRaidFarm extends EntityAIMoveToBlock
     {
+
         private boolean field_179498_d;
         private boolean field_179499_e = false;
         private static final String __OBFID = "CL_00002233";
 
         public AIRaidFarm()
@@ -649,10 +641,11 @@
         }
     }
 
     public class RabbitJumpHelper extends EntityJumpHelper
     {
+
         private EntityRabbit field_180070_c;
         private boolean field_180068_d = false;
         private static final String __OBFID = "CL_00002236";
 
         public RabbitJumpHelper(EntityRabbit p_i45863_2_)
@@ -686,10 +679,11 @@
         }
     }
 
     class RabbitMoveHelper extends EntityMoveHelper
     {
+
         private EntityRabbit field_179929_g = EntityRabbit.this;
         private static final String __OBFID = "CL_00002235";
 
         public RabbitMoveHelper()
         {
@@ -707,10 +701,11 @@
         }
     }
 
     public static class RabbitTypeData implements IEntityLivingData
         {
+
             public int field_179427_a;
             private static final String __OBFID = "CL_00002237";
 
             public RabbitTypeData(int p_i45864_1_)
             {
