--- decompiled\net\minecraft\entity\passive\EntityBat.java
+++ src\main\java\net\minecraft\entity\passive\EntityBat.java
@@ -1,8 +1,9 @@
 package net.minecraft.entity.passive;
 
 import java.util.Calendar;
+
 import net.minecraft.block.Block;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.nbt.NBTTagCompound;
@@ -11,10 +12,11 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
 public class EntityBat extends EntityAmbientCreature
 {
+
     private BlockPos spawnPosition;
     private static final String __OBFID = "CL_00001637";
 
     public EntityBat(World worldIn)
     {
@@ -57,13 +59,17 @@
     public boolean canBePushed()
     {
         return false;
     }
 
-    protected void collideWithEntity(Entity p_82167_1_) {}
+    protected void collideWithEntity(Entity p_82167_1_)
+    {
+    }
 
-    protected void collideWithNearbyEntities() {}
+    protected void collideWithNearbyEntities()
+    {
+    }
 
     protected void applyEntityAttributes()
     {
         super.applyEntityAttributes();
         this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setBaseValue(6.0D);
@@ -163,13 +169,17 @@
     protected boolean canTriggerWalking()
     {
         return false;
     }
 
-    public void fall(float distance, float damageMultiplier) {}
+    public void fall(float distance, float damageMultiplier)
+    {
+    }
 
-    protected void func_180433_a(double p_180433_1_, boolean p_180433_3_, Block p_180433_4_, BlockPos p_180433_5_) {}
+    protected void func_180433_a(double p_180433_1_, boolean p_180433_3_, Block p_180433_4_, BlockPos p_180433_5_)
+    {
+    }
 
     public boolean doesEntityNotTriggerPressurePlate()
     {
         return true;
     }
