--- decompiled\net\minecraft\entity\passive\EntityCow.java
+++ src\main\java\net\minecraft\entity\passive\EntityCow.java
@@ -17,12 +17,16 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.pathfinding.PathNavigateGround;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.craftbukkit.inventory.CraftItemStack;
+
 public class EntityCow extends EntityAnimal
 {
+
     private static final String __OBFID = "CL_00001640";
 
     public EntityCow(World worldIn)
     {
         super(worldIn);
@@ -104,19 +108,31 @@
     {
         ItemStack itemstack = player.inventory.getCurrentItem();
 
         if (itemstack != null && itemstack.getItem() == Items.bucket && !player.capabilities.isCreativeMode)
         {
-            if (itemstack.stackSize-- == 1)
+            // CraftBukkit start - Got milk?
+            org.bukkit.Location loc = this.getBukkitEntity().getLocation();
+            org.bukkit.event.player.PlayerBucketFillEvent event = CraftEventFactory.callPlayerBucketFillEvent(player, loc.getBlockX(), loc.getBlockY(), loc.getBlockZ(), null, itemstack, Items.milk_bucket);
+
+            if (event.isCancelled())
             {
-                player.inventory.setInventorySlotContents(player.inventory.currentItem, new ItemStack(Items.milk_bucket));
+                return false;
             }
-            else if (!player.inventory.addItemStackToInventory(new ItemStack(Items.milk_bucket)))
+
+            ItemStack result = CraftItemStack.asNMSCopy(event.getItemStack());
+
+            if (--itemstack.stackSize <= 0)
             {
-                player.dropPlayerItemWithRandomChoice(new ItemStack(Items.milk_bucket, 1, 0), false);
+                player.inventory.setInventorySlotContents(player.inventory.currentItem, result);
             }
+            else if (!player.inventory.addItemStackToInventory(result))
+            {
+                player.dropPlayerItemWithRandomChoice(result, false);
+            }
 
+            // CraftBukkit end
             return true;
         }
         else
         {
             return super.interact(player);
