--- decompiled\net\minecraft\entity\passive\EntityPig.java
+++ src\main\java\net\minecraft\entity\passive\EntityPig.java
@@ -22,12 +22,15 @@
 import net.minecraft.pathfinding.PathNavigateGround;
 import net.minecraft.stats.AchievementList;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+
 public class EntityPig extends EntityAnimal
 {
+
     private final EntityAIControlledByPlayer aiControlledByPlayer;
     private static final String __OBFID = "CL_00001647";
 
     public EntityPig(World worldIn)
     {
@@ -161,13 +164,22 @@
     public void onStruckByLightning(EntityLightningBolt lightningBolt)
     {
         if (!this.worldObj.isRemote)
         {
             EntityPigZombie entitypigzombie = new EntityPigZombie(this.worldObj);
+
+            // CraftBukkit start
+            if (CraftEventFactory.callPigZapEvent(this, lightningBolt, entitypigzombie).isCancelled())
+            {
+                return;
+            }
+
+            // CraftBukkit end
             entitypigzombie.setCurrentItemOrArmor(0, new ItemStack(Items.golden_sword));
             entitypigzombie.setLocationAndAngles(this.posX, this.posY, this.posZ, this.rotationYaw, this.rotationPitch);
-            this.worldObj.spawnEntityInWorld(entitypigzombie);
+            // CraftBukkit - added a reason for spawning this creature
+            this.worldObj.addEntity(entitypigzombie, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.LIGHTNING);
             this.setDead();
         }
     }
 
     public void fall(float distance, float damageMultiplier)
