--- decompiled\net\minecraft\tileentity\IHopper.java
+++ src\main\java\net\minecraft\tileentity\IHopper.java
@@ -1,12 +1,12 @@
 package net.minecraft.tileentity;
 
 import net.minecraft.inventory.IInventory;
 import net.minecraft.world.World;
 
-public interface IHopper extends IInventory
-{
+public interface IHopper extends IInventory {
+
     World getWorld();
 
     double getXPos();
 
     double getYPos();
