--- decompiled\net\minecraft\tileentity\MobSpawnerBaseLogic.java
+++ src\main\java\net\minecraft\tileentity\MobSpawnerBaseLogic.java
@@ -1,30 +1,34 @@
 package net.minecraft.tileentity;
 
-import com.google.common.collect.Lists;
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityList;
 import net.minecraft.entity.EntityLiving;
-import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.IEntityLivingData;
 import net.minecraft.entity.item.EntityMinecart;
 import net.minecraft.nbt.NBTBase;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.util.AxisAlignedBB;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.util.WeightedRandom;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.event.entity.CreatureSpawnEvent;
+import org.bukkit.event.entity.SpawnerSpawnEvent;
+
+import com.google.common.collect.Lists;
+
 public abstract class MobSpawnerBaseLogic
 {
-    private int spawnDelay = 20;
+
+    public int spawnDelay = 20; // SpigotMCP -- Apply access transformation
     private String mobID = "Pig";
     private final List minecartToSpawn = Lists.newArrayList();
     private MobSpawnerBaseLogic.WeightedRandomMinecart randomEntity;
     private double mobRotation;
     private double prevMobRotation;
@@ -35,14 +39,21 @@
     private int maxNearbyEntities = 6;
     private int activatingRangeFromPlayer = 16;
     private int spawnRange = 4;
     private static final String __OBFID = "CL_00000129";
 
-    private String getEntityNameToSpawn()
+    public String getEntityNameToSpawn()   // SpigotMCP -- Apply access transformation
     {
         if (this.getRandomEntity() == null)
         {
+            // CraftBukkit start - fix NPE
+            if (this.mobID == null)
+            {
+                this.mobID = "Pig";
+            }
+
+            // CraftBukkit end
             if (this.mobID.equals("Minecart"))
             {
                 this.mobID = "MinecartRideable";
             }
 
@@ -165,11 +176,26 @@
 
             p_180613_1_.readFromNBT(nbttagcompound);
 
             if (p_180613_1_.worldObj != null && p_180613_2_)
             {
-                p_180613_1_.worldObj.spawnEntityInWorld(p_180613_1_);
+                // CraftBukkit start - call SpawnerSpawnEvent, abort if cancelled // SpigotMCP NOTE -- Really spigot, because craftbukkit doesn't implement its own events
+                SpawnerSpawnEvent event = CraftEventFactory.callSpawnerSpawnEvent(p_180613_1_, this.func_177221_b().getX(), this.func_177221_b().getY(), this.func_177221_b().getZ());
+
+                if (!event.isCancelled())
+                {
+                    p_180613_1_.worldObj.addEntity(p_180613_1_, CreatureSpawnEvent.SpawnReason.SPAWNER); // CraftBukkit
+                    // Spigot Start
+                    if (p_180613_1_.worldObj.spigotConfig.nerfSpawnerMobs)
+                    {
+                        p_180613_1_.fromMobSpawner = true;
+                    }
+
+                    // Spigot End
+                }
+
+                // CraftBukkit end
             }
 
             NBTTagCompound nbttagcompound2;
 
             for (Entity entity1 = p_180613_1_; nbttagcompound.hasKey("Riding", 10); nbttagcompound = nbttagcompound2)
@@ -190,26 +216,50 @@
                         nbttagcompound1.setTag(s1, nbtbase1.copy());
                     }
 
                     entity2.readFromNBT(nbttagcompound1);
                     entity2.setLocationAndAngles(entity1.posX, entity1.posY, entity1.posZ, entity1.rotationYaw, entity1.rotationPitch);
+                    // CraftBukkit start - call SpawnerSpawnEvent, skip if cancelled
+                    SpawnerSpawnEvent event = CraftEventFactory.callSpawnerSpawnEvent(entity2, this.func_177221_b().getX(), this.func_177221_b().getY(), this.func_177221_b().getZ());
 
+                    if (event.isCancelled())
+                    {
+                        continue;
+                    }
+
+                    // CraftBukkit end
+
                     if (p_180613_1_.worldObj != null && p_180613_2_)
                     {
-                        p_180613_1_.worldObj.spawnEntityInWorld(entity2);
+                        p_180613_1_.worldObj.addEntity(entity2, CreatureSpawnEvent.SpawnReason.SPAWNER); // CraftBukkit
                     }
 
                     entity1.mountEntity(entity2);
                 }
 
                 entity1 = entity2;
             }
         }
-        else if (p_180613_1_ instanceof EntityLivingBase && p_180613_1_.worldObj != null && p_180613_2_)
+        else if (p_180613_1_ instanceof EntityLiving && p_180613_1_.worldObj != null && p_180613_2_)  // CraftBukkit - EntityLiving -> EntityInsentient
         {
             ((EntityLiving)p_180613_1_).func_180482_a(p_180613_1_.worldObj.getDifficultyForLocation(new BlockPos(p_180613_1_)), (IEntityLivingData)null);
-            p_180613_1_.worldObj.spawnEntityInWorld(p_180613_1_);
+            // Spigot start - call SpawnerSpawnEvent, abort if cancelled
+            SpawnerSpawnEvent event = CraftEventFactory.callSpawnerSpawnEvent(p_180613_1_, this.func_177221_b().getX(), this.func_177221_b().getY(), this.func_177221_b().getZ());
+
+            if (!event.isCancelled())
+            {
+                p_180613_1_.worldObj.addEntity(p_180613_1_, CreatureSpawnEvent.SpawnReason.SPAWNER); // CraftBukkit
+                // Spigot Start
+                if (p_180613_1_.worldObj.spigotConfig.nerfSpawnerMobs)
+                {
+                    p_180613_1_.fromMobSpawner = true;
+                }
+
+                // Spigot End
+            }
+
+            // Spigot end
         }
 
         return p_180613_1_;
     }
 
@@ -332,27 +382,10 @@
         {
             return false;
         }
     }
 
-    @SideOnly(Side.CLIENT)
-    public Entity func_180612_a(World worldIn)
-    {
-        if (this.cachedEntity == null)
-        {
-            Entity entity = EntityList.createEntityByName(this.getEntityNameToSpawn(), worldIn);
-
-            if (entity != null)
-            {
-                entity = this.func_180613_a(entity, false);
-                this.cachedEntity = entity;
-            }
-        }
-
-        return this.cachedEntity;
-    }
-
     private MobSpawnerBaseLogic.WeightedRandomMinecart getRandomEntity()
     {
         return this.randomEntity;
     }
 
@@ -365,24 +398,13 @@
 
     public abstract World getSpawnerWorld();
 
     public abstract BlockPos func_177221_b();
 
-    @SideOnly(Side.CLIENT)
-    public double getMobRotation()
-    {
-        return this.mobRotation;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public double getPrevMobRotation()
-    {
-        return this.prevMobRotation;
-    }
-
     public class WeightedRandomMinecart extends WeightedRandom.Item
     {
+
         private final NBTTagCompound field_98222_b;
         private final String entityType;
         private static final String __OBFID = "CL_00000130";
 
         public WeightedRandomMinecart(NBTTagCompound p_i1945_2_)
