--- decompiled\net\minecraft\tileentity\TileEntityMobSpawner.java
+++ src\main\java\net\minecraft\tileentity\TileEntityMobSpawner.java
@@ -8,10 +8,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class TileEntityMobSpawner extends TileEntity implements IUpdatePlayerListBox
 {
+
     private final MobSpawnerBaseLogic spawnerLogic = new MobSpawnerBaseLogic()
     {
         private static final String __OBFID = "CL_00000361";
         public void func_98267_a(int p_98267_1_)
         {
