--- decompiled\net\minecraft\tileentity\TileEntityNote.java
+++ src\main\java\net\minecraft\tileentity\TileEntityNote.java
@@ -5,12 +5,16 @@
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.event.block.NotePlayEvent;
+
 public class TileEntityNote extends TileEntity
 {
+
     public byte note;
     public boolean previousRedstoneState;
     private static final String __OBFID = "CL_00000362";
 
     public void writeToNBT(NBTTagCompound compound)
@@ -57,9 +61,17 @@
             if (material == Material.wood)
             {
                 b0 = 4;
             }
 
-            worldIn.addBlockEvent(p_175108_2_, Blocks.noteblock, b0, this.note);
+            // CraftBukkit start
+            NotePlayEvent event = CraftEventFactory.callNotePlayEvent(this.worldObj, p_175108_2_.getX(), p_175108_2_.getY(), p_175108_2_.getZ(), b0, this.note);
+
+            if (!event.isCancelled())
+            {
+                worldIn.addBlockEvent(p_175108_2_, Blocks.noteblock, event.getInstrument().getType(), event.getNote().getId());
+            }
+
+            // CraftBukkit end
         }
     }
 }
