--- decompiled\net\minecraft\tileentity\TileEntityFlowerPot.java
+++ src\main\java\net\minecraft\tileentity\TileEntityFlowerPot.java
@@ -6,15 +6,18 @@
 import net.minecraft.network.play.server.S35PacketUpdateTileEntity;
 import net.minecraft.util.ResourceLocation;
 
 public class TileEntityFlowerPot extends TileEntity
 {
+
     private Item flowerPotItem;
     private int flowerPotData;
     private static final String __OBFID = "CL_00000356";
 
-    public TileEntityFlowerPot() {}
+    public TileEntityFlowerPot()
+    {
+    }
 
     public TileEntityFlowerPot(Item potItem, int potData)
     {
         this.flowerPotItem = potItem;
         this.flowerPotData = potData;
