--- decompiled\net\minecraft\tileentity\TileEntityMobSpawner.java
+++ src\main\java\net\minecraft\tileentity\TileEntityMobSpawner.java
@@ -6,69 +6,62 @@
 import net.minecraft.network.play.server.S35PacketUpdateTileEntity;
 import net.minecraft.server.gui.IUpdatePlayerListBox;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
-public class TileEntityMobSpawner extends TileEntity implements IUpdatePlayerListBox
-{
-    private final MobSpawnerBaseLogic spawnerLogic = new MobSpawnerBaseLogic()
-    {
+public class TileEntityMobSpawner extends TileEntity implements IUpdatePlayerListBox {
+
+    private final MobSpawnerBaseLogic spawnerLogic = new MobSpawnerBaseLogic() {
+
         private static final String __OBFID = "CL_00000361";
-        public void func_98267_a(int p_98267_1_)
-        {
+
+        public void func_98267_a(int p_98267_1_) {
             TileEntityMobSpawner.this.worldObj.addBlockEvent(TileEntityMobSpawner.this.pos, Blocks.mob_spawner, p_98267_1_, 0);
         }
-        public World getSpawnerWorld()
-        {
+
+        public World getSpawnerWorld() {
             return TileEntityMobSpawner.this.worldObj;
         }
-        public BlockPos func_177221_b()
-        {
+
+        public BlockPos func_177221_b() {
             return TileEntityMobSpawner.this.pos;
         }
-        public void setRandomEntity(MobSpawnerBaseLogic.WeightedRandomMinecart p_98277_1_)
-        {
+
+        public void setRandomEntity(MobSpawnerBaseLogic.WeightedRandomMinecart p_98277_1_) {
             super.setRandomEntity(p_98277_1_);
 
-            if (this.getSpawnerWorld() != null)
-            {
+            if (this.getSpawnerWorld() != null) {
                 this.getSpawnerWorld().markBlockForUpdate(TileEntityMobSpawner.this.pos);
             }
         }
     };
     private static final String __OBFID = "CL_00000360";
 
-    public void readFromNBT(NBTTagCompound compound)
-    {
+    public void readFromNBT(NBTTagCompound compound) {
         super.readFromNBT(compound);
         this.spawnerLogic.readFromNBT(compound);
     }
 
-    public void writeToNBT(NBTTagCompound compound)
-    {
+    public void writeToNBT(NBTTagCompound compound) {
         super.writeToNBT(compound);
         this.spawnerLogic.writeToNBT(compound);
     }
 
-    public void update()
-    {
+    public void update() {
         this.spawnerLogic.updateSpawner();
     }
 
-    public Packet getDescriptionPacket()
-    {
+    public Packet getDescriptionPacket() {
         NBTTagCompound nbttagcompound = new NBTTagCompound();
         this.writeToNBT(nbttagcompound);
         nbttagcompound.removeTag("SpawnPotentials");
         return new S35PacketUpdateTileEntity(this.pos, 1, nbttagcompound);
     }
 
-    public boolean receiveClientEvent(int id, int type)
-    {
+    public boolean receiveClientEvent(int id, int type) {
         return this.spawnerLogic.setDelayToMin(id) ? true : super.receiveClientEvent(id, type);
     }
 
-    public MobSpawnerBaseLogic getSpawnerBaseLogic()
-    {
+    public MobSpawnerBaseLogic getSpawnerBaseLogic() {
         return this.spawnerLogic;
     }
 }
