--- decompiled\net\minecraft\tileentity\TileEntityEnderChest.java
+++ src\main\java\net\minecraft\tileentity\TileEntityEnderChest.java
@@ -4,10 +4,11 @@
 import net.minecraft.init.Blocks;
 import net.minecraft.server.gui.IUpdatePlayerListBox;
 
 public class TileEntityEnderChest extends TileEntity implements IUpdatePlayerListBox
 {
+
     public float field_145972_a;
     public float prevLidAngle;
     public int field_145973_j;
     private int field_145974_k;
     private static final String __OBFID = "CL_00000355";
