--- decompiled\net\minecraft\tileentity\TileEntityLockable.java
+++ src\main\java\net\minecraft\tileentity\TileEntityLockable.java
@@ -8,10 +8,11 @@
 import net.minecraft.world.ILockableContainer;
 import net.minecraft.world.LockCode;
 
 public abstract class TileEntityLockable extends TileEntity implements IInteractionObject, ILockableContainer
 {
+
     private LockCode code;
     private static final String __OBFID = "CL_00002040";
 
     public TileEntityLockable()
     {
