--- decompiled\net\minecraft\entity\EntityFlying.java
+++ src\main\java\net\minecraft\entity\EntityFlying.java
@@ -5,20 +5,25 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
 public abstract class EntityFlying extends EntityLiving
 {
+
     private static final String __OBFID = "CL_00001545";
 
     public EntityFlying(World worldIn)
     {
         super(worldIn);
     }
 
-    public void fall(float distance, float damageMultiplier) {}
+    public void fall(float distance, float damageMultiplier)
+    {
+    }
 
-    protected void func_180433_a(double p_180433_1_, boolean p_180433_3_, Block p_180433_4_, BlockPos p_180433_5_) {}
+    protected void func_180433_a(double p_180433_1_, boolean p_180433_3_, Block p_180433_4_, BlockPos p_180433_5_)
+    {
+    }
 
     public void moveEntityWithHeading(float p_70612_1_, float p_70612_2_)
     {
         if (this.isInWater())
         {
