--- decompiled\net\minecraft\entity\EntityAgeable.java
+++ src\main\java\net\minecraft\entity\EntityAgeable.java
@@ -8,17 +8,47 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
 public abstract class EntityAgeable extends EntityCreature
 {
+
     protected int field_175504_a;
     protected int field_175502_b;
     protected int field_175503_c;
     private float field_98056_d = -1.0F;
     private float field_98057_e;
+    public boolean ageLocked = false; // CraftBukkit
     private static final String __OBFID = "CL_00001530";
 
+    // Spigot start
+    @Override
+    public void inactiveTick()
+    {
+        super.inactiveTick();
+
+        if (this.worldObj.isRemote || this.ageLocked)   // CraftBukkit
+        {
+            this.setScaleForAge(this.isChild());
+        }
+        else
+        {
+            int i = this.getGrowingAge();
+
+            if (i < 0)
+            {
+                ++i;
+                this.setGrowingAge(i);
+            }
+            else if (i > 0)
+            {
+                --i;
+                this.setGrowingAge(i);
+            }
+        }
+    }
+    // Spigot end
+
     public EntityAgeable(World worldIn)
     {
         super(worldIn);
     }
 
@@ -40,22 +70,22 @@
 
                     if (entityageable != null)
                     {
                         entityageable.setGrowingAge(-24000);
                         entityageable.setLocationAndAngles(this.posX, this.posY, this.posZ, 0.0F, 0.0F);
-                        this.worldObj.spawnEntityInWorld(entityageable);
+                        this.worldObj.addEntity(entityageable, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.SPAWNER_EGG); // CraftBukkit
 
                         if (itemstack.hasDisplayName())
                         {
                             entityageable.setCustomNameTag(itemstack.getDisplayName());
                         }
 
                         if (!player.capabilities.isCreativeMode)
                         {
                             --itemstack.stackSize;
 
-                            if (itemstack.stackSize <= 0)
+                            if (itemstack.stackSize == 0)   // CraftBukkit - allow less than 0 stacks as "infinite"                            {
                             {
                                 player.inventory.setInventorySlotContents(player.inventory.currentItem, (ItemStack)null);
                             }
                         }
                     }
@@ -131,24 +161,26 @@
     public void writeEntityToNBT(NBTTagCompound tagCompound)
     {
         super.writeEntityToNBT(tagCompound);
         tagCompound.setInteger("Age", this.getGrowingAge());
         tagCompound.setInteger("ForcedAge", this.field_175502_b);
+        tagCompound.setBoolean("AgeLocked", this.ageLocked); // CraftBukkit
     }
 
     public void readEntityFromNBT(NBTTagCompound tagCompund)
     {
         super.readEntityFromNBT(tagCompund);
         this.setGrowingAge(tagCompund.getInteger("Age"));
         this.field_175502_b = tagCompund.getInteger("ForcedAge");
+        this.ageLocked = tagCompund.getBoolean("AgeLocked"); // CraftBukkit
     }
 
     public void onLivingUpdate()
     {
         super.onLivingUpdate();
 
-        if (this.worldObj.isRemote)
+        if (this.worldObj.isRemote || ageLocked)   // CraftBukkit
         {
             if (this.field_175503_c > 0)
             {
                 if (this.field_175503_c % 4 == 0)
                 {
@@ -180,11 +212,13 @@
                 this.setGrowingAge(i);
             }
         }
     }
 
-    protected void func_175500_n() {}
+    protected void func_175500_n()
+    {
+    }
 
     public boolean isChild()
     {
         return this.getGrowingAge() < 0;
     }
@@ -192,11 +226,11 @@
     public void setScaleForAge(boolean p_98054_1_)
     {
         this.setScale(p_98054_1_ ? 0.5F : 1.0F);
     }
 
-    protected final void setSize(float width, float height)
+    public final void setSize(float width, float height) // CraftBukkit - protected to public
     {
         boolean flag = this.field_98056_d > 0.0F;
         this.field_98056_d = width;
         this.field_98057_e = height;
 
