--- decompiled\net\minecraft\entity\EntityMinecartCommandBlock.java
+++ src\main\java\net\minecraft\entity\EntityMinecartCommandBlock.java
@@ -1,58 +1,62 @@
 package net.minecraft.entity;
 
-import io.netty.buffer.ByteBuf;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.command.server.CommandBlockLogic;
 import net.minecraft.entity.item.EntityMinecart;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Blocks;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.IChatComponent;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
 public class EntityMinecartCommandBlock extends EntityMinecart
 {
-    private final CommandBlockLogic commandBlockLogic = new CommandBlockLogic()
+
+    // SpigotMCP Start
+    private final CommandBlockLogic commandBlockLogic = new EntityMinecartCommandBlockListener();
+
+    public class EntityMinecartCommandBlockListener extends CommandBlockLogic
     {
+
+        {
+            this.sender = (org.bukkit.craftbukkit.entity.CraftMinecartCommand)EntityMinecartCommandBlock.this.getBukkitEntity(); // CraftBukkit - Set the sender
+        }
+
         private static final String __OBFID = "CL_00001673";
+
         public void func_145756_e()
         {
             EntityMinecartCommandBlock.this.getDataWatcher().updateObject(23, this.getCustomName());
             EntityMinecartCommandBlock.this.getDataWatcher().updateObject(24, IChatComponent.Serializer.componentToJson(this.getLastOutput()));
         }
-        @SideOnly(Side.CLIENT)
-        public int func_145751_f()
-        {
-            return 1;
-        }
-        @SideOnly(Side.CLIENT)
-        public void func_145757_a(ByteBuf p_145757_1_)
-        {
-            p_145757_1_.writeInt(EntityMinecartCommandBlock.this.getEntityId());
-        }
+
         public BlockPos getPosition()
         {
             return new BlockPos(EntityMinecartCommandBlock.this.posX, EntityMinecartCommandBlock.this.posY + 0.5D, EntityMinecartCommandBlock.this.posZ);
         }
+
         public Vec3 getPositionVector()
         {
             return new Vec3(EntityMinecartCommandBlock.this.posX, EntityMinecartCommandBlock.this.posY, EntityMinecartCommandBlock.this.posZ);
         }
+
         public World getEntityWorld()
         {
             return EntityMinecartCommandBlock.this.worldObj;
         }
+
         public Entity getCommandSenderEntity()
         {
             return EntityMinecartCommandBlock.this;
         }
-    };
+    }
+
+    ;
+    // SpigotMCP end
     private int activatorRailCooldown = 0;
     private static final String __OBFID = "CL_00001672";
 
     public EntityMinecartCommandBlock(World worldIn)
     {
