--- decompiled\net\minecraft\server\gui\MinecraftServerGui.java
+++ src\main\java\net\minecraft\server\gui\MinecraftServerGui.java
@@ -1,42 +1,33 @@
 package net.minecraft.server.gui;
 
-import com.mojang.util.QueueLogAppender;
-import java.awt.BorderLayout;
-import java.awt.Component;
-import java.awt.Dimension;
-import java.awt.Font;
+import java.awt.*;
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 import java.awt.event.FocusAdapter;
 import java.awt.event.FocusEvent;
 import java.awt.event.WindowAdapter;
 import java.awt.event.WindowEvent;
-import javax.swing.JComponent;
-import javax.swing.JFrame;
-import javax.swing.JPanel;
-import javax.swing.JScrollBar;
-import javax.swing.JScrollPane;
-import javax.swing.JTextArea;
-import javax.swing.JTextField;
-import javax.swing.SwingUtilities;
-import javax.swing.UIManager;
+
+import javax.swing.*;
 import javax.swing.border.EtchedBorder;
 import javax.swing.border.TitledBorder;
 import javax.swing.text.AttributeSet;
 import javax.swing.text.BadLocationException;
 import javax.swing.text.Document;
+
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.dedicated.DedicatedServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-@SideOnly(Side.SERVER)
+import com.mojang.util.QueueLogAppender;
+
 public class MinecraftServerGui extends JComponent
 {
+
     private static final Font serverGuiFont = new Font("Monospaced", 0, 12);
     private static final Logger LOGGER = LogManager.getLogger();
     private DedicatedServer server;
     private static final String __OBFID = "CL_00001789";
 
@@ -139,11 +130,13 @@
             }
         });
         jtextarea.addFocusListener(new FocusAdapter()
         {
             private static final String __OBFID = "CL_00001794";
-            public void focusGained(FocusEvent p_focusGained_1_) {}
+            public void focusGained(FocusEvent p_focusGained_1_)
+            {
+            }
         });
         jpanel.add(jscrollpane, "Center");
         jpanel.add(jtextfield, "South");
         jpanel.setBorder(new TitledBorder(new EtchedBorder(), "Log and chat"));
         Thread thread = new Thread(new Runnable()
