--- decompiled\net\minecraft\server\gui\PlayerListComponent.java
+++ src\main\java\net\minecraft\server\gui\PlayerListComponent.java
@@ -1,17 +1,18 @@
 package net.minecraft.server.gui;
 
 import java.util.Vector;
-import javax.swing.JList;
+
+import javax.swing.*;
+
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.server.MinecraftServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
-@SideOnly(Side.SERVER)
+
 public class PlayerListComponent extends JList implements IUpdatePlayerListBox
 {
+
     private MinecraftServer server;
     private int ticks;
     private static final String __OBFID = "CL_00001795";
 
     public PlayerListComponent(MinecraftServer server)
