--- decompiled\net\minecraft\network\play\client\C00PacketKeepAlive.java
+++ src\main\java\net\minecraft\network\play\client\C00PacketKeepAlive.java
@@ -1,26 +1,23 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C00PacketKeepAlive implements Packet
 {
+
     private int key;
     private static final String __OBFID = "CL_00001359";
 
-    public C00PacketKeepAlive() {}
-
-    @SideOnly(Side.CLIENT)
-    public C00PacketKeepAlive(int key)
+    public C00PacketKeepAlive()
     {
-        this.key = key;
     }
 
     public void processPacket(INetHandlerPlayServer handler)
     {
         handler.processKeepAlive(this);
@@ -41,8 +38,8 @@
         return this.key;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
