--- decompiled\net\minecraft\network\play\client\C01PacketChatMessage.java
+++ src\main\java\net\minecraft\network\play\client\C01PacketChatMessage.java
@@ -1,19 +1,27 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
 public class C01PacketChatMessage implements Packet
 {
+
     private String message;
     private static final String __OBFID = "CL_00001347";
 
-    public C01PacketChatMessage() {}
+    public C01PacketChatMessage()
+    {
+    }
 
     public C01PacketChatMessage(String messageIn)
     {
         if (messageIn.length() > 100)
         {
@@ -31,20 +39,37 @@
     public void writePacketData(PacketBuffer buf) throws IOException
     {
         buf.writeString(this.message);
     }
 
+    // Spigot Start
+    private static final ExecutorService executors = Executors.newCachedThreadPool(new ThreadFactoryBuilder().setDaemon(true).setNameFormat("Async Chat Thread - #%d").build());
+
     public void processPacket(INetHandlerPlayServer handler)
     {
         handler.processChatMessage(this);
     }
 
     public String getMessage()
     {
         return this.message;
     }
 
-    public void processPacket(INetHandler handler)
+    public void processPacket(final INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        if (!message.startsWith("/"))
+        {
+            executors.submit(new Runnable()
+            {
+                @Override
+                public void run()
+                {
+                    C01PacketChatMessage.this.processPacket((INetHandlerPlayServer) handler);
+                }
+            });
+            return;
+        }
+
+        // Spigot End
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
