--- decompiled\net\minecraft\network\play\client\C07PacketPlayerDigging.java
+++ src\main\java\net\minecraft\network\play\client\C07PacketPlayerDigging.java
@@ -1,37 +1,32 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C07PacketPlayerDigging implements Packet
 {
+
     private BlockPos position;
     private EnumFacing facing;
     private C07PacketPlayerDigging.Action status;
     private static final String __OBFID = "CL_00001365";
 
-    public C07PacketPlayerDigging() {}
-
-    @SideOnly(Side.CLIENT)
-    public C07PacketPlayerDigging(C07PacketPlayerDigging.Action statusIn, BlockPos posIn, EnumFacing facingIn)
+    public C07PacketPlayerDigging()
     {
-        this.status = statusIn;
-        this.position = posIn;
-        this.facing = facingIn;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
-        this.status = (C07PacketPlayerDigging.Action)buf.readEnumValue(C07PacketPlayerDigging.Action.class);
+        this.status = (C07PacketPlayerDigging.Action) buf.readEnumValue(C07PacketPlayerDigging.Action.class);
         this.position = buf.readBlockPos();
         this.facing = EnumFacing.getFront(buf.readUnsignedByte());
     }
 
     public void writePacketData(PacketBuffer buf) throws IOException
@@ -61,11 +56,11 @@
         return this.status;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 
     public static enum Action
     {
         START_DESTROY_BLOCK,
