--- decompiled\net\minecraft\network\play\client\C08PacketPlayerBlockPlacement.java
+++ src\main\java\net\minecraft\network\play\client\C08PacketPlayerBlockPlacement.java
@@ -1,28 +1,34 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.item.ItemStack;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 import net.minecraft.util.BlockPos;
 
 public class C08PacketPlayerBlockPlacement implements Packet
 {
+
     private static final BlockPos field_179726_a = new BlockPos(-1, -1, -1);
     private BlockPos position;
     private int placedBlockDirection;
     private ItemStack stack;
     private float facingX;
     private float facingY;
     private float facingZ;
     private static final String __OBFID = "CL_00001371";
 
-    public C08PacketPlayerBlockPlacement() {}
+    public long timestamp; // CraftBukkit
 
+    public C08PacketPlayerBlockPlacement()
+    {
+    }
+
     public C08PacketPlayerBlockPlacement(ItemStack p_i45930_1_)
     {
         this(field_179726_a, 255, p_i45930_1_, 0.0F, 0.0F, 0.0F);
     }
 
@@ -36,16 +42,17 @@
         this.facingZ = p_i45931_6_;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
+        timestamp = System.currentTimeMillis(); // CraftBukkit
         this.position = buf.readBlockPos();
         this.placedBlockDirection = buf.readUnsignedByte();
         this.stack = buf.readItemStackFromBuffer();
-        this.facingX = (float)buf.readUnsignedByte() / 16.0F;
-        this.facingY = (float)buf.readUnsignedByte() / 16.0F;
-        this.facingZ = (float)buf.readUnsignedByte() / 16.0F;
+        this.facingX = (float) buf.readUnsignedByte() / 16.0F;
+        this.facingY = (float) buf.readUnsignedByte() / 16.0F;
+        this.facingZ = (float) buf.readUnsignedByte() / 16.0F;
     }
 
     public void writePacketData(PacketBuffer buf) throws IOException
     {
         buf.writeBlockPos(this.position);
@@ -91,8 +98,19 @@
         return this.facingZ;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
+
+    // SpigotMCP Start -- Remove as already fixed by mcp
+    /*
+    // CraftBukkit start - fix decompile error
+    @Override
+    public void processPacket(INetHandler pl) {
+        processPacket((PacketListenerPlayIn)pl);
+    }
+    // CraftBukkit end
+    */
+    // SpigotMCP End
 }
