--- decompiled\net\minecraft\network\play\client\C09PacketHeldItemChange.java
+++ src\main\java\net\minecraft\network\play\client\C09PacketHeldItemChange.java
@@ -1,26 +1,23 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C09PacketHeldItemChange implements Packet
 {
+
     private int slotId;
     private static final String __OBFID = "CL_00001368";
 
-    public C09PacketHeldItemChange() {}
-
-    @SideOnly(Side.CLIENT)
-    public C09PacketHeldItemChange(int slotId)
+    public C09PacketHeldItemChange()
     {
-        this.slotId = slotId;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
         this.slotId = buf.readShort();
@@ -41,8 +38,8 @@
         return this.slotId;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
