--- decompiled\net\minecraft\network\play\client\C0APacketAnimation.java
+++ src\main\java\net\minecraft\network\play\client\C0APacketAnimation.java
@@ -1,26 +1,27 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 
-public class C0APacketAnimation implements Packet
-{
+public class C0APacketAnimation implements Packet {
+
     private static final String __OBFID = "CL_00001345";
 
-    public void readPacketData(PacketBuffer buf) throws IOException {}
+    public void readPacketData(PacketBuffer buf) throws IOException {
+    }
 
-    public void writePacketData(PacketBuffer buf) throws IOException {}
+    public void writePacketData(PacketBuffer buf) throws IOException {
+    }
 
-    public void handle(INetHandlerPlayServer handler)
-    {
+    public void handle(INetHandlerPlayServer handler) {
         handler.handleAnimation(this);
     }
 
-    public void processPacket(INetHandler handler)
-    {
-        this.handle((INetHandlerPlayServer)handler);
+    public void processPacket(INetHandler handler) {
+        this.handle((INetHandlerPlayServer) handler);
     }
 }
