--- decompiled\net\minecraft\network\play\client\C0CPacketInput.java
+++ src\main\java\net\minecraft\network\play\client\C0CPacketInput.java
@@ -1,32 +1,26 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C0CPacketInput implements Packet
 {
+
     private float strafeSpeed;
     private float forwardSpeed;
     private boolean jumping;
     private boolean sneaking;
     private static final String __OBFID = "CL_00001367";
 
-    public C0CPacketInput() {}
-
-    @SideOnly(Side.CLIENT)
-    public C0CPacketInput(float strafeSpeed, float forwardSpeed, boolean jumping, boolean sneaking)
+    public C0CPacketInput()
     {
-        this.strafeSpeed = strafeSpeed;
-        this.forwardSpeed = forwardSpeed;
-        this.jumping = jumping;
-        this.sneaking = sneaking;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
         this.strafeSpeed = buf.readFloat();
@@ -80,8 +74,8 @@
         return this.sneaking;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
