--- decompiled\net\minecraft\network\play\client\C0EPacketClickWindow.java
+++ src\main\java\net\minecraft\network\play\client\C0EPacketClickWindow.java
@@ -1,37 +1,29 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.item.ItemStack;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C0EPacketClickWindow implements Packet
 {
+
     private int windowId;
     private int slotId;
     private int usedButton;
     private short actionNumber;
     private ItemStack clickedItem;
     private int mode;
     private static final String __OBFID = "CL_00001353";
 
-    public C0EPacketClickWindow() {}
-
-    @SideOnly(Side.CLIENT)
-    public C0EPacketClickWindow(int windowId, int slotId, int usedButton, int mode, ItemStack clickedItem, short actionNumber)
+    public C0EPacketClickWindow()
     {
-        this.windowId = windowId;
-        this.slotId = slotId;
-        this.usedButton = usedButton;
-        this.clickedItem = clickedItem != null ? clickedItem.copy() : null;
-        this.actionNumber = actionNumber;
-        this.mode = mode;
     }
 
     public void processPacket(INetHandlerPlayServer handler)
     {
         handler.processClickWindow(this);
@@ -87,8 +79,8 @@
         return this.mode;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
