--- decompiled\net\minecraft\network\play\client\C0FPacketConfirmTransaction.java
+++ src\main\java\net\minecraft\network\play\client\C0FPacketConfirmTransaction.java
@@ -1,30 +1,25 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C0FPacketConfirmTransaction implements Packet
 {
+
     private int windowId;
     private short uid;
     private boolean accepted;
     private static final String __OBFID = "CL_00001351";
 
-    public C0FPacketConfirmTransaction() {}
-
-    @SideOnly(Side.CLIENT)
-    public C0FPacketConfirmTransaction(int windowId, short uid, boolean accepted)
+    public C0FPacketConfirmTransaction()
     {
-        this.windowId = windowId;
-        this.uid = uid;
-        this.accepted = accepted;
     }
 
     public void processPacket(INetHandlerPlayServer handler)
     {
         handler.processConfirmTransaction(this);
@@ -54,8 +49,8 @@
         return this.uid;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
