--- decompiled\net\minecraft\network\play\client\C12PacketUpdateSign.java
+++ src\main\java\net\minecraft\network\play\client\C12PacketUpdateSign.java
@@ -1,30 +1,26 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.IChatComponent;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C12PacketUpdateSign implements Packet
 {
+
     private BlockPos pos;
     private IChatComponent[] lines;
     private static final String __OBFID = "CL_00001370";
 
-    public C12PacketUpdateSign() {}
-
-    @SideOnly(Side.CLIENT)
-    public C12PacketUpdateSign(BlockPos pos, IChatComponent[] lines)
+    public C12PacketUpdateSign()
     {
-        this.pos = pos;
-        this.lines = new IChatComponent[] {lines[0], lines[1], lines[2], lines[3]};
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
         this.pos = buf.readBlockPos();
@@ -61,8 +57,8 @@
         return this.lines;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
