--- decompiled\net\minecraft\network\play\client\C13PacketPlayerAbilities.java
+++ src\main\java\net\minecraft\network\play\client\C13PacketPlayerAbilities.java
@@ -1,25 +1,29 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.entity.player.PlayerCapabilities;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 
 public class C13PacketPlayerAbilities implements Packet
 {
+
     private boolean invulnerable;
     private boolean flying;
     private boolean allowFlying;
     private boolean creativeMode;
     private float flySpeed;
     private float walkSpeed;
     private static final String __OBFID = "CL_00001364";
 
-    public C13PacketPlayerAbilities() {}
+    public C13PacketPlayerAbilities()
+    {
+    }
 
     public C13PacketPlayerAbilities(PlayerCapabilities capabilities)
     {
         this.setInvulnerable(capabilities.disableDamage);
         this.setFlying(capabilities.isFlying);
@@ -124,8 +128,8 @@
         this.walkSpeed = walkSpeedIn;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
