--- decompiled\net\minecraft\network\play\client\C14PacketTabComplete.java
+++ src\main\java\net\minecraft\network\play\client\C14PacketTabComplete.java
@@ -1,26 +1,31 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 import net.minecraft.util.BlockPos;
+
 import org.apache.commons.lang3.StringUtils;
 
 public class C14PacketTabComplete implements Packet
 {
+
     private String message;
     private BlockPos targetBlock;
     private static final String __OBFID = "CL_00001346";
 
-    public C14PacketTabComplete() {}
+    public C14PacketTabComplete()
+    {
+    }
 
     public C14PacketTabComplete(String msg)
     {
-        this(msg, (BlockPos)null);
+        this(msg, (BlockPos) null);
     }
 
     public C14PacketTabComplete(String msg, BlockPos target)
     {
         this.message = msg;
@@ -65,8 +70,8 @@
         return this.targetBlock;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
