--- decompiled\net\minecraft\network\play\client\C15PacketClientSettings.java
+++ src\main\java\net\minecraft\network\play\client\C15PacketClientSettings.java
@@ -1,35 +1,28 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C15PacketClientSettings implements Packet
 {
+
     private String lang;
     private int view;
     private EntityPlayer.EnumChatVisibility chatVisibility;
     private boolean enableColors;
     private int modelPartFlags;
     private static final String __OBFID = "CL_00001350";
 
-    public C15PacketClientSettings() {}
-
-    @SideOnly(Side.CLIENT)
-    public C15PacketClientSettings(String p_i45946_1_, int p_i45946_2_, EntityPlayer.EnumChatVisibility p_i45946_3_, boolean p_i45946_4_, int p_i45946_5_)
+    public C15PacketClientSettings()
     {
-        this.lang = p_i45946_1_;
-        this.view = p_i45946_2_;
-        this.chatVisibility = p_i45946_3_;
-        this.enableColors = p_i45946_4_;
-        this.modelPartFlags = p_i45946_5_;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
         this.lang = buf.readStringFromBuffer(7);
@@ -73,8 +66,8 @@
         return this.modelPartFlags;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayServer)handler);
+        this.processPacket((INetHandlerPlayServer) handler);
     }
 }
