--- decompiled\net\minecraft\network\play\client\C16PacketClientStatus.java
+++ src\main\java\net\minecraft\network\play\client\C16PacketClientStatus.java
@@ -1,28 +1,32 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
 
 public class C16PacketClientStatus implements Packet
 {
+
     private C16PacketClientStatus.EnumState status;
     private static final String __OBFID = "CL_00001348";
 
-    public C16PacketClientStatus() {}
+    public C16PacketClientStatus()
+    {
+    }
 
     public C16PacketClientStatus(C16PacketClientStatus.EnumState statusIn)
     {
         this.status = statusIn;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
-        this.status = (C16PacketClientStatus.EnumState)buf.readEnumValue(C16PacketClientStatus.EnumState.class);
+        this.status = (C16PacketClientStatus.EnumState) buf.readEnumValue(C16PacketClientStatus.EnumState.class);
     }
 
     public void writePacketData(PacketBuffer buf) throws IOException
     {
         buf.writeEnumValue(this.status);
@@ -38,11 +42,11 @@
         return this.status;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.func_180758_a((INetHandlerPlayServer)handler);
+        this.func_180758_a((INetHandlerPlayServer) handler);
     }
 
     public static enum EnumState
     {
         PERFORM_RESPAWN,
