--- decompiled\net\minecraft\network\play\client\C19PacketResourcePackStatus.java
+++ src\main\java\net\minecraft\network\play\client\C19PacketResourcePackStatus.java
@@ -1,39 +1,30 @@
 package net.minecraft.network.play.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C19PacketResourcePackStatus implements Packet
 {
+
     private String hash;
     private C19PacketResourcePackStatus.Action status;
     private static final String __OBFID = "CL_00002282";
 
-    public C19PacketResourcePackStatus() {}
-
-    @SideOnly(Side.CLIENT)
-    public C19PacketResourcePackStatus(String p_i45935_1_, C19PacketResourcePackStatus.Action p_i45935_2_)
+    public C19PacketResourcePackStatus()
     {
-        if (p_i45935_1_.length() > 40)
-        {
-            p_i45935_1_ = p_i45935_1_.substring(0, 40);
-        }
-
-        this.hash = p_i45935_1_;
-        this.status = p_i45935_2_;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
         this.hash = buf.readStringFromBuffer(40);
-        this.status = (C19PacketResourcePackStatus.Action)buf.readEnumValue(C19PacketResourcePackStatus.Action.class);
+        this.status = (C19PacketResourcePackStatus.Action) buf.readEnumValue(C19PacketResourcePackStatus.Action.class);
     }
 
     public void writePacketData(PacketBuffer buf) throws IOException
     {
         buf.writeString(this.hash);
@@ -45,11 +36,11 @@
         handler.handleResourcePackStatus(this);
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.handle((INetHandlerPlayServer)handler);
+        this.handle((INetHandlerPlayServer) handler);
     }
 
     public static enum Action
     {
         SUCCESSFULLY_LOADED,
