--- decompiled\net\minecraft\server\management\BanEntry.java
+++ src\main\java\net\minecraft\server\management\BanEntry.java
@@ -1,14 +1,16 @@
 package net.minecraft.server.management;
 
-import com.google.gson.JsonObject;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 
+import com.google.gson.JsonObject;
+
 public abstract class BanEntry extends UserListEntry
 {
+
     public static final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss Z");
     protected final Date banStartDate;
     protected final String bannedBy;
     protected final Date banEndDate;
     protected final String reason;
@@ -74,6 +76,41 @@
         data.addProperty("created", dateFormat.format(this.banStartDate));
         data.addProperty("source", this.bannedBy);
         data.addProperty("expires", this.banEndDate == null ? "forever" : dateFormat.format(this.banEndDate));
         data.addProperty("reason", this.reason);
     }
+
+    // CraftBukkit start
+    public String getSource()
+    {
+        return this.bannedBy;
+    }
+
+    public Date getCreated()
+    {
+        return this.banStartDate;
+    }
+
+    private static Object checkExpiry(Object object, JsonObject jsonobject)
+    {
+        Date expires = null;
+
+        try
+        {
+            expires = jsonobject.has("expires") ? dateFormat.parse(jsonobject.get("expires").getAsString()) : null;
+        }
+        catch (ParseException ex)
+        {
+            // Guess we don't have a date
+        }
+
+        if (expires == null || expires.after(new Date()))
+        {
+            return object;
+        }
+        else
+        {
+            return null;
+        }
+    }
+    // CraftBukkit end
 }
