--- decompiled\net\minecraft\server\management\BanList.java
+++ src\main\java\net\minecraft\server\management\BanList.java
@@ -1,13 +1,15 @@
 package net.minecraft.server.management;
 
-import com.google.gson.JsonObject;
 import java.io.File;
 import java.net.SocketAddress;
 
+import com.google.gson.JsonObject;
+
 public class BanList extends UserList
 {
+
     private static final String __OBFID = "CL_00001396";
 
     public BanList(File bansFile)
     {
         super(bansFile);
