--- decompiled\net\minecraft\server\management\IPBanEntry.java
+++ src\main\java\net\minecraft\server\management\IPBanEntry.java
@@ -1,12 +1,14 @@
 package net.minecraft.server.management;
 
-import com.google.gson.JsonObject;
 import java.util.Date;
 
+import com.google.gson.JsonObject;
+
 public class IPBanEntry extends BanEntry
 {
+
     private static final String __OBFID = "CL_00001883";
 
     public IPBanEntry(String p_i46330_1_)
     {
         this(p_i46330_1_, (Date)null, (String)null, (Date)null, (String)null);
