--- decompiled\net\minecraft\server\management\UserListOpsEntry.java
+++ src\main\java\net\minecraft\server\management\UserListOpsEntry.java
@@ -1,13 +1,15 @@
 package net.minecraft.server.management;
 
+import java.util.UUID;
+
 import com.google.gson.JsonObject;
 import com.mojang.authlib.GameProfile;
-import java.util.UUID;
 
 public class UserListOpsEntry extends UserListEntry
 {
+
     private final int field_152645_a;
     private static final String __OBFID = "CL_00001878";
 
     public UserListOpsEntry(GameProfile profile, int p_i46328_2_)
     {
