--- decompiled\net\minecraft\server\management\UserListOps.java
+++ src\main\java\net\minecraft\server\management\UserListOps.java
@@ -1,14 +1,16 @@
 package net.minecraft.server.management;
 
-import com.google.gson.JsonObject;
-import com.mojang.authlib.GameProfile;
 import java.io.File;
 import java.util.Iterator;
 
+import com.google.gson.JsonObject;
+import com.mojang.authlib.GameProfile;
+
 public class UserListOps extends UserList
 {
+
     private static final String __OBFID = "CL_00001879";
 
     public UserListOps(File p_i1152_1_)
     {
         super(p_i1152_1_);
