--- decompiled\net\minecraft\server\management\UserListWhitelist.java
+++ src\main\java\net\minecraft\server\management\UserListWhitelist.java
@@ -1,16 +1,16 @@
 package net.minecraft.server.management;
 
-import com.google.gson.JsonObject;
-import com.mojang.authlib.GameProfile;
 import java.io.File;
 import java.util.Iterator;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.gson.JsonObject;
+import com.mojang.authlib.GameProfile;
+
 public class UserListWhitelist extends UserList
 {
+
     private static final String __OBFID = "CL_00001871";
 
     public UserListWhitelist(File p_i1132_1_)
     {
         super(p_i1132_1_);
@@ -33,11 +33,10 @@
         }
 
         return astring;
     }
 
-    @SideOnly(Side.SERVER)
     public boolean func_152705_a(GameProfile p_152705_1_)
     {
         return this.hasEntry(p_152705_1_);
     }
 
