--- decompiled\net\minecraft\inventory\ContainerHorseInventory.java
+++ src\main\java\net\minecraft\inventory\ContainerHorseInventory.java
@@ -1,22 +1,45 @@
 package net.minecraft.inventory;
 
 import net.minecraft.entity.passive.EntityHorse;
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.init.Items;
 import net.minecraft.item.ItemStack;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.craftbukkit.inventory.CraftInventory;
+import org.bukkit.craftbukkit.inventory.CraftInventoryHorse;
+import org.bukkit.craftbukkit.inventory.CraftInventoryView;
+import org.bukkit.inventory.InventoryView;
+
 public class ContainerHorseInventory extends Container
 {
+
     private IInventory horseInventory;
     private EntityHorse theHorse;
     private static final String __OBFID = "CL_00001751";
 
+    // CraftBukkit start
+    CraftInventoryView bukkitEntity;
+    InventoryPlayer player;
+
+    @Override
+    public InventoryView getBukkitView()
+    {
+        if (bukkitEntity != null)
+        {
+            return bukkitEntity;
+        }
+
+        CraftInventory inventory = new CraftInventoryHorse(this.horseInventory);
+        return bukkitEntity = new CraftInventoryView(player.player.getBukkitEntity(), inventory, this);
+    }
+
     public ContainerHorseInventory(IInventory playerInventory, final IInventory horseInventoryIn, final EntityHorse horse, EntityPlayer player)
     {
+        this.player = (InventoryPlayer)playerInventory;
+        // CraftBukkit end
         this.horseInventory = horseInventoryIn;
         this.theHorse = horse;
         byte b0 = 3;
         horseInventoryIn.openInventory(player);
         int i = (b0 - 4) * 18;
@@ -33,15 +56,10 @@
             private static final String __OBFID = "CL_00001753";
             public boolean isItemValid(ItemStack stack)
             {
                 return super.isItemValid(stack) && horse.canWearArmor() && EntityHorse.func_146085_a(stack.getItem());
             }
-            @SideOnly(Side.CLIENT)
-            public boolean canBeHovered()
-            {
-                return horse.canWearArmor();
-            }
         });
         int j;
         int k;
 
         if (horse.isChested())
