--- decompiled\net\minecraft\inventory\ContainerMerchant.java
+++ src\main\java\net\minecraft\inventory\ContainerMerchant.java
@@ -3,28 +3,47 @@
 import net.minecraft.entity.IMerchant;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.craftbukkit.inventory.CraftInventoryMerchant;
+import org.bukkit.craftbukkit.inventory.CraftInventoryView;
+
 public class ContainerMerchant extends Container
 {
+
     private IMerchant theMerchant;
     private InventoryMerchant merchantInventory;
     private final World theWorld;
     private static final String __OBFID = "CL_00001757";
 
+    // CraftBukkit start
+    private CraftInventoryView bukkitEntity = null;
+    private InventoryPlayer player;
+
+    @Override
+    public org.bukkit.craftbukkit.inventory.CraftInventoryView getBukkitView()
+    {
+        if (bukkitEntity == null)
+        {
+            bukkitEntity = new CraftInventoryView(this.player.player.getBukkitEntity(), new CraftInventoryMerchant((InventoryMerchant)merchantInventory), this);
+        }
+
+        return bukkitEntity;
+    }
+    // CraftBukkit end
+
     public ContainerMerchant(InventoryPlayer playerInventory, IMerchant merchant, World worldIn)
     {
         this.theMerchant = merchant;
         this.theWorld = worldIn;
         this.merchantInventory = new InventoryMerchant(playerInventory.player, merchant);
         this.addSlotToContainer(new Slot(this.merchantInventory, 0, 36, 53));
         this.addSlotToContainer(new Slot(this.merchantInventory, 1, 62, 53));
         this.addSlotToContainer(new SlotMerchantResult(playerInventory.player, merchant, this.merchantInventory, 2, 120, 53));
+        this.player = playerInventory; // CraftBukkit - save player
         int i;
 
         for (i = 0; i < 3; ++i)
         {
             for (int j = 0; j < 9; ++j)
@@ -63,13 +82,10 @@
     public void setCurrentRecipeIndex(int p_75175_1_)
     {
         this.merchantInventory.setCurrentRecipeIndex(p_75175_1_);
     }
 
-    @SideOnly(Side.CLIENT)
-    public void updateProgressBar(int id, int data) {}
-
     public boolean canInteractWith(EntityPlayer playerIn)
     {
         return this.theMerchant.getCustomer() == playerIn;
     }
 
