--- decompiled\net\minecraft\inventory\ContainerBeacon.java
+++ src\main\java\net\minecraft\inventory\ContainerBeacon.java
@@ -1,21 +1,30 @@
 package net.minecraft.inventory;
 
 import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.init.Items;
 import net.minecraft.item.ItemStack;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraft.tileentity.TileEntityBeacon;
 
+import org.bukkit.craftbukkit.inventory.CraftInventoryBeacon;
+import org.bukkit.craftbukkit.inventory.CraftInventoryView;
+
 public class ContainerBeacon extends Container
 {
+
     private IInventory tileBeacon;
     private final ContainerBeacon.BeaconSlot beaconSlot;
+    // CraftBukkit start
+    private CraftInventoryView bukkitEntity = null;
+    private InventoryPlayer player;
+    // CraftBukkit end
     private static final String __OBFID = "CL_00001735";
 
     public ContainerBeacon(IInventory p_i45804_1_, IInventory p_i45804_2_)
     {
+        player = (InventoryPlayer)p_i45804_1_; // CraftBukkit - TODO: check this
         this.tileBeacon = p_i45804_2_;
         this.addSlotToContainer(this.beaconSlot = new ContainerBeacon.BeaconSlot(p_i45804_2_, 0, 136, 110));
         byte b0 = 36;
         short short1 = 137;
         int i;
@@ -38,23 +47,19 @@
     {
         super.addCraftingToCrafters(listener);
         listener.func_175173_a(this, this.tileBeacon);
     }
 
-    @SideOnly(Side.CLIENT)
-    public void updateProgressBar(int id, int data)
-    {
-        this.tileBeacon.setField(id, data);
-    }
-
     public IInventory func_180611_e()
     {
         return this.tileBeacon;
     }
 
     public boolean canInteractWith(EntityPlayer playerIn)
     {
+        if (!this.checkReachable) return true; // CraftBukkit
+
         return this.tileBeacon.isUseableByPlayer(playerIn);
     }
 
     public ItemStack transferStackInSlot(EntityPlayer playerIn, int index)
     {
@@ -121,10 +126,11 @@
         return itemstack;
     }
 
     class BeaconSlot extends Slot
     {
+
         private static final String __OBFID = "CL_00001736";
 
         public BeaconSlot(IInventory p_i1801_2_, int p_i1801_3_, int p_i1801_4_, int p_i1801_5_)
         {
             super(p_i1801_2_, p_i1801_3_, p_i1801_4_, p_i1801_5_);
@@ -138,6 +144,21 @@
         public int getSlotStackLimit()
         {
             return 1;
         }
     }
+
+    // CraftBukkit start
+    @Override
+    public org.bukkit.craftbukkit.inventory.CraftInventoryView getBukkitView()
+    {
+        if (bukkitEntity != null)
+        {
+            return bukkitEntity;
+        }
+
+        org.bukkit.craftbukkit.inventory.CraftInventory inventory = new CraftInventoryBeacon((TileEntityBeacon)this.tileBeacon); // TODO - check this
+        bukkitEntity = new CraftInventoryView(this.player.player.getBukkitEntity(), inventory, this);
+        return bukkitEntity;
+    }
+    // CraftBukkit end
 }
