--- decompiled\net\minecraft\inventory\ContainerBrewingStand.java
+++ src\main\java\net\minecraft\inventory\ContainerBrewingStand.java
@@ -3,22 +3,30 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.init.Items;
 import net.minecraft.item.ItemStack;
 import net.minecraft.stats.AchievementList;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.craftbukkit.inventory.CraftInventoryBrewer;
+import org.bukkit.craftbukkit.inventory.CraftInventoryView;
+
 public class ContainerBrewingStand extends Container
 {
+
     private IInventory tileBrewingStand;
     private final Slot theSlot;
     private int brewTime;
+
+    // CraftBukkit start
+    private CraftInventoryView bukkitEntity = null;
+    private InventoryPlayer player;
+    // CraftBukkit end
     private static final String __OBFID = "CL_00001737";
 
     public ContainerBrewingStand(InventoryPlayer playerInventory, IInventory p_i45802_2_)
     {
+        player = playerInventory; // CraftBukkit
         this.tileBrewingStand = p_i45802_2_;
         this.addSlotToContainer(new ContainerBrewingStand.Potion(playerInventory.player, p_i45802_2_, 0, 56, 46));
         this.addSlotToContainer(new ContainerBrewingStand.Potion(playerInventory.player, p_i45802_2_, 1, 79, 53));
         this.addSlotToContainer(new ContainerBrewingStand.Potion(playerInventory.player, p_i45802_2_, 2, 102, 46));
         this.theSlot = this.addSlotToContainer(new ContainerBrewingStand.Ingredient(p_i45802_2_, 3, 79, 17));
@@ -59,18 +67,14 @@
         }
 
         this.brewTime = this.tileBrewingStand.getField(0);
     }
 
-    @SideOnly(Side.CLIENT)
-    public void updateProgressBar(int id, int data)
-    {
-        this.tileBrewingStand.setField(id, data);
-    }
-
     public boolean canInteractWith(EntityPlayer playerIn)
     {
+        if (!this.checkReachable) return true; // CraftBukkit
+
         return this.tileBrewingStand.isUseableByPlayer(playerIn);
     }
 
     public ItemStack transferStackInSlot(EntityPlayer playerIn, int index)
     {
@@ -145,12 +149,28 @@
         }
 
         return itemstack;
     }
 
+    // CraftBukkit start
+    @Override
+    public org.bukkit.craftbukkit.inventory.CraftInventoryView getBukkitView()
+    {
+        if (bukkitEntity != null)
+        {
+            return bukkitEntity;
+        }
+
+        CraftInventoryBrewer inventory = new CraftInventoryBrewer(this.tileBrewingStand);
+        bukkitEntity = new CraftInventoryView(this.player.player.getBukkitEntity(), inventory, this);
+        return bukkitEntity;
+    }
+    // CraftBukkit end
+
     class Ingredient extends Slot
     {
+
         private static final String __OBFID = "CL_00001738";
 
         public Ingredient(IInventory p_i1803_2_, int p_i1803_3_, int p_i1803_4_, int p_i1803_5_)
         {
             super(p_i1803_2_, p_i1803_3_, p_i1803_4_, p_i1803_5_);
@@ -167,10 +187,11 @@
         }
     }
 
     static class Potion extends Slot
         {
+
             private EntityPlayer player;
             private static final String __OBFID = "CL_00001740";
 
             public Potion(EntityPlayer p_i1804_1_, IInventory p_i1804_2_, int p_i1804_3_, int p_i1804_4_, int p_i1804_5_)
             {
