--- decompiled\net\minecraft\inventory\ContainerHopper.java
+++ src\main\java\net\minecraft\inventory\ContainerHopper.java
@@ -2,18 +2,41 @@
 
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.item.ItemStack;
 
+import org.bukkit.craftbukkit.inventory.CraftInventory;
+import org.bukkit.craftbukkit.inventory.CraftInventoryView;
+
 public class ContainerHopper extends Container
 {
+
     private final IInventory hopperInventory;
     private static final String __OBFID = "CL_00001750";
 
+    // CraftBukkit start
+    private CraftInventoryView bukkitEntity = null;
+    private InventoryPlayer player;
+
+    @Override
+    public org.bukkit.craftbukkit.inventory.CraftInventoryView getBukkitView()
+    {
+        if (bukkitEntity != null)
+        {
+            return bukkitEntity;
+        }
+
+        CraftInventory inventory = new CraftInventory(this.hopperInventory);
+        bukkitEntity = new CraftInventoryView(this.player.player.getBukkitEntity(), inventory, this);
+        return bukkitEntity;
+    }
+    // CraftBukkit end
+
     public ContainerHopper(InventoryPlayer playerInventory, IInventory hopperInventoryIn, EntityPlayer player)
     {
         this.hopperInventory = hopperInventoryIn;
+        this.player = playerInventory; // CraftBukkit - save player
         hopperInventoryIn.openInventory(player);
         byte b0 = 51;
         int i;
 
         for (i = 0; i < hopperInventoryIn.getSizeInventory(); ++i)
@@ -35,10 +58,12 @@
         }
     }
 
     public boolean canInteractWith(EntityPlayer playerIn)
     {
+        if (!this.checkReachable) return true; // CraftBukkit
+
         return this.hopperInventory.isUseableByPlayer(playerIn);
     }
 
     public ItemStack transferStackInSlot(EntityPlayer playerIn, int index)
     {
