--- decompiled\net\minecraft\inventory\InventoryEnderChest.java
+++ src\main\java\net\minecraft\inventory\InventoryEnderChest.java
@@ -1,18 +1,67 @@
 package net.minecraft.inventory;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.tileentity.TileEntityEnderChest;
 
+import org.bukkit.craftbukkit.entity.CraftHumanEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.InventoryHolder;
+
 public class InventoryEnderChest extends InventoryBasic
 {
+
     private TileEntityEnderChest associatedChest;
     private static final String __OBFID = "CL_00001759";
 
+    // CraftBukkit start - add fields and methods
+    public List transaction = new ArrayList();
+    public Player player;
+    private int maxStack = MAX_STACK;
+
+    public net.minecraft.item.ItemStack[] getContents()
+    {
+        return this.inventoryContents;
+    }
+
+    public void onOpen(CraftHumanEntity who)
+    {
+        transaction.add(who);
+    }
+
+    public void onClose(CraftHumanEntity who)
+    {
+        transaction.remove(who);
+    }
+
+    public java.util.List getViewers()
+    {
+        return transaction;
+    }
+
+    public InventoryHolder getOwner()
+    {
+        return this.player;
+    }
+
+    public void setMaxStackSize(int size)
+    {
+        maxStack = size;
+    }
+
+    public int getInventoryStackLimit()
+    {
+        return maxStack;
+    }
+    // CraftBukkit end
+
     public InventoryEnderChest()
     {
         super("container.enderchest", false, 27);
     }
 
