--- decompiled\net\minecraft\inventory\SlotCrafting.java
+++ src\main\java\net\minecraft\inventory\SlotCrafting.java
@@ -11,10 +11,11 @@
 import net.minecraft.item.crafting.CraftingManager;
 import net.minecraft.stats.AchievementList;
 
 public class SlotCrafting extends Slot
 {
+
     private final InventoryCrafting craftMatrix;
     private final EntityPlayer thePlayer;
     private int amountCrafted;
     private static final String __OBFID = "CL_00001761";
 
