--- decompiled\net\minecraft\inventory\Slot.java
+++ src\main\java\net\minecraft\inventory\Slot.java
@@ -1,15 +1,14 @@
 package net.minecraft.inventory;
 
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
 public class Slot
 {
-    private final int slotIndex;
+
+    public final int slotIndex; // SpigotMCP -- Apply access transform
     public final IInventory inventory;
     public int slotNumber;
     public int xDisplayPosition;
     public int yDisplayPosition;
     private static final String __OBFID = "CL_00001762";
@@ -36,13 +35,17 @@
                 }
             }
         }
     }
 
-    protected void onCrafting(ItemStack stack, int amount) {}
+    protected void onCrafting(ItemStack stack, int amount)
+    {
+    }
 
-    protected void onCrafting(ItemStack stack) {}
+    protected void onCrafting(ItemStack stack)
+    {
+    }
 
     public void onPickupFromSlot(EntityPlayer playerIn, ItemStack stack)
     {
         this.onSlotChanged();
     }
@@ -81,16 +84,10 @@
     public int getItemStackLimit(ItemStack stack)
     {
         return this.getSlotStackLimit();
     }
 
-    @SideOnly(Side.CLIENT)
-    public String getSlotTexture()
-    {
-        return null;
-    }
-
     public ItemStack decrStackSize(int amount)
     {
         return this.inventory.decrStackSize(this.slotIndex, amount);
     }
 
@@ -101,12 +98,6 @@
 
     public boolean canTakeStack(EntityPlayer playerIn)
     {
         return true;
     }
-
-    @SideOnly(Side.CLIENT)
-    public boolean canBeHovered()
-    {
-        return true;
-    }
 }
