--- decompiled\net\minecraft\inventory\SlotMerchantResult.java
+++ src\main\java\net\minecraft\inventory\SlotMerchantResult.java
@@ -6,10 +6,11 @@
 import net.minecraft.stats.StatList;
 import net.minecraft.village.MerchantRecipe;
 
 public class SlotMerchantResult extends Slot
 {
+
     private final InventoryMerchant theMerchantInventory;
     private EntityPlayer thePlayer;
     private int field_75231_g;
     private final IMerchant theMerchant;
     private static final String __OBFID = "CL_00001758";
