--- decompiled\net\minecraft\village\MerchantRecipe.java
+++ src\main\java\net\minecraft\village\MerchantRecipe.java
@@ -1,15 +1,15 @@
 package net.minecraft.village;
 
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class MerchantRecipe
 {
+
     private ItemStack itemToBuy;
     private ItemStack secondItemToBuy;
     private ItemStack itemToSell;
     private int toolUses;
     private int maxTradeUses;
@@ -89,16 +89,10 @@
     public boolean isRecipeDisabled()
     {
         return this.toolUses >= this.maxTradeUses;
     }
 
-    @SideOnly(Side.CLIENT)
-    public void func_82785_h()
-    {
-        this.toolUses = this.maxTradeUses;
-    }
-
     public boolean getRewardsExp()
     {
         return this.rewardsExp;
     }
 
