--- decompiled\net\minecraft\village\MerchantRecipeList.java
+++ src\main\java\net\minecraft\village\MerchantRecipeList.java
@@ -1,21 +1,23 @@
 package net.minecraft.village;
 
-import java.io.IOException;
 import java.util.ArrayList;
+
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.network.PacketBuffer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class MerchantRecipeList extends ArrayList
 {
+
     private static final String __OBFID = "CL_00000127";
 
-    public MerchantRecipeList() {}
+    public MerchantRecipeList()
+    {
+    }
 
     public MerchantRecipeList(NBTTagCompound p_i1944_1_)
     {
         this.readRecipiesFromTags(p_i1944_1_);
     }
@@ -89,39 +91,6 @@
         }
 
         nbttagcompound.setTag("Recipes", nbttaglist);
         return nbttagcompound;
     }
-
-    @SideOnly(Side.CLIENT)
-    public static MerchantRecipeList readFromBuf(PacketBuffer p_151390_0_) throws IOException
-    {
-        MerchantRecipeList merchantrecipelist = new MerchantRecipeList();
-        int i = p_151390_0_.readByte() & 255;
-
-        for (int j = 0; j < i; ++j)
-        {
-            ItemStack itemstack = p_151390_0_.readItemStackFromBuffer();
-            ItemStack itemstack1 = p_151390_0_.readItemStackFromBuffer();
-            ItemStack itemstack2 = null;
-
-            if (p_151390_0_.readBoolean())
-            {
-                itemstack2 = p_151390_0_.readItemStackFromBuffer();
-            }
-
-            boolean flag = p_151390_0_.readBoolean();
-            int k = p_151390_0_.readInt();
-            int l = p_151390_0_.readInt();
-            MerchantRecipe merchantrecipe = new MerchantRecipe(itemstack, itemstack2, itemstack1, k, l);
-
-            if (flag)
-            {
-                merchantrecipe.func_82785_h();
-            }
-
-            merchantrecipelist.add(merchantrecipe);
-        }
-
-        return merchantrecipelist;
-    }
 }
