--- decompiled\net\minecraft\village\VillageDoorInfo.java
+++ src\main\java\net\minecraft\village\VillageDoorInfo.java
@@ -3,10 +3,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 
 public class VillageDoorInfo
 {
+
     private final BlockPos doorBlockPos;
     private final BlockPos insideBlock;
     private final EnumFacing insideDirection;
     private int lastActivityTimestamp;
     private boolean isDetachedFromVillageFlag;
