--- decompiled\net\minecraft\village\VillageSiege.java
+++ src\main\java\net\minecraft\village\VillageSiege.java
@@ -1,21 +1,25 @@
 package net.minecraft.village;
 
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.IEntityLivingData;
 import net.minecraft.entity.monster.EntityZombie;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.SpawnerAnimals;
 import net.minecraft.world.World;
 
+import org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason;
+
 public class VillageSiege
 {
+
     private World worldObj;
     private boolean field_75535_b;
     private int field_75536_c = -1;
     private int field_75533_d;
     private int field_75534_e;
@@ -185,11 +189,11 @@
                 exception.printStackTrace();
                 return false;
             }
 
             entityzombie.setLocationAndAngles(vec3.xCoord, vec3.yCoord, vec3.zCoord, this.worldObj.rand.nextFloat() * 360.0F, 0.0F);
-            this.worldObj.spawnEntityInWorld(entityzombie);
+            this.worldObj.addEntity(entityzombie, SpawnReason.VILLAGE_INVASION); // CraftBukkit
             BlockPos blockpos = this.theVillage.getCenter();
             entityzombie.func_175449_a(blockpos, this.theVillage.getVillageRadius());
             return true;
         }
     }
