--- decompiled\net\minecraft\network\rcon\IServer.java
+++ src\main\java\net\minecraft\network\rcon\IServer.java
@@ -1,13 +1,10 @@
 package net.minecraft.network.rcon;
 
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
-@SideOnly(Side.SERVER)
-public interface IServer
-{
+public interface IServer {
+
     int getIntProperty(String key, int defaultValue);
 
     String getStringProperty(String key, String defaultValue);
 
     void setProperty(String key, Object value);
