--- decompiled\net\minecraft\network\rcon\RConThreadClient.java
+++ src\main\java\net\minecraft\network\rcon\RConThreadClient.java
@@ -4,18 +4,17 @@
 import java.io.ByteArrayOutputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
 import java.net.Socket;
 import java.net.SocketTimeoutException;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-@SideOnly(Side.SERVER)
 public class RConThreadClient extends RConThreadBase
 {
+
     private static final Logger LOGGER = LogManager.getLogger();
     private boolean loggedIn;
     private Socket clientSocket;
     private byte[] buffer = new byte[1460];
     private String rconPassword;
@@ -70,10 +69,11 @@
                     i1 += 4;
 
                     switch (l)
                     {
                         case 2:
+
                             if (this.loggedIn)
                             {
                                 String s1 = RConUtils.getBytesAsString(this.buffer, i1, i);
 
                                 try
