--- decompiled\net\minecraft\network\rcon\RConUtils.java
+++ src\main\java\net\minecraft\network\rcon\RConUtils.java
@@ -1,14 +1,13 @@
 package net.minecraft.network.rcon;
 
+
 import com.google.common.base.Charsets;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
-@SideOnly(Side.SERVER)
 public class RConUtils
 {
+
     public static final char[] hexDigits = new char[] {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'};
     private static final String __OBFID = "CL_00001799";
 
     public static String getBytesAsString(byte[] p_72661_0_, int p_72661_1_, int p_72661_2_)
     {
