--- decompiled\net\minecraft\network\rcon\RConThreadMain.java
+++ src\main\java\net\minecraft\network\rcon\RConThreadMain.java
@@ -1,22 +1,21 @@
 package net.minecraft.network.rcon;
 
-import com.google.common.collect.Maps;
 import java.io.IOException;
 import java.net.InetAddress;
 import java.net.ServerSocket;
 import java.net.Socket;
 import java.net.SocketTimeoutException;
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Map.Entry;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
-@SideOnly(Side.SERVER)
+import com.google.common.collect.Maps;
+
 public class RConThreadMain extends RConThreadBase
 {
+
     private int rconPort;
     private int serverPort;
     private String hostname;
     private ServerSocket serverSocket;
     private String rconPassword;
