--- decompiled\net\minecraft\network\rcon\RConConsoleSource.java
+++ src\main\java\net\minecraft\network\rcon\RConConsoleSource.java
@@ -7,77 +7,70 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.ChatComponentText;
 import net.minecraft.util.IChatComponent;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
-@SideOnly(Side.SERVER)
-public class RConConsoleSource implements ICommandSender
-{
+
+public class RConConsoleSource implements ICommandSender {
+
     private static final RConConsoleSource instance = new RConConsoleSource();
     private StringBuffer buffer = new StringBuffer();
     private static final String __OBFID = "CL_00001800";
 
-    public static RConConsoleSource getInstance()
-    {
+    public static RConConsoleSource getInstance() {
         return instance;
     }
 
-    public void resetLog()
-    {
+    public void resetLog() {
         this.buffer.setLength(0);
     }
 
-    public String getLogContents()
-    {
+    public String getLogContents() {
         return this.buffer.toString();
     }
 
-    public String getName()
-    {
+    public String getName() {
         return "Rcon";
     }
 
-    public IChatComponent getDisplayName()
-    {
+    public IChatComponent getDisplayName() {
         return new ChatComponentText(this.getName());
     }
 
-    public void addChatMessage(IChatComponent message)
-    {
+    // CraftBukkit start - Send a String
+    public void sendMessage(String message) {
+        this.buffer.append(message);
+    }
+    // CraftBukkit end
+
+    public void addChatMessage(IChatComponent message) {
         this.buffer.append(message.getUnformattedText());
     }
 
-    public boolean canUseCommand(int permLevel, String commandName)
-    {
+    public boolean canUseCommand(int permLevel, String commandName) {
         return true;
     }
 
-    public BlockPos getPosition()
-    {
+    public BlockPos getPosition() {
         return new BlockPos(0, 0, 0);
     }
 
-    public Vec3 getPositionVector()
-    {
+    public Vec3 getPositionVector() {
         return new Vec3(0.0D, 0.0D, 0.0D);
     }
 
-    public World getEntityWorld()
-    {
+    public World getEntityWorld() {
         return MinecraftServer.getServer().getEntityWorld();
     }
 
-    public Entity getCommandSenderEntity()
-    {
+    public Entity getCommandSenderEntity() {
         return null;
     }
 
-    public boolean sendCommandFeedback()
-    {
+    public boolean sendCommandFeedback() {
         return true;
     }
 
-    public void setCommandStat(CommandResultStats.Type type, int amount) {}
+    public void setCommandStat(CommandResultStats.Type type, int amount) {
+    }
 }
