--- decompiled\net\minecraft\pathfinding\PathEntity.java
+++ src\main\java\net\minecraft\pathfinding\PathEntity.java
@@ -3,10 +3,11 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.util.Vec3;
 
 public class PathEntity
 {
+
     private final PathPoint[] points;
     private int currentPathIndex;
     private int pathLength;
     private static final String __OBFID = "CL_00000575";
 
