--- decompiled\net\minecraft\pathfinding\Path.java
+++ src\main\java\net\minecraft\pathfinding\Path.java
@@ -1,10 +1,11 @@
 package net.minecraft.pathfinding;
 
 public class Path
 {
-    private PathPoint[] pathPoints = new PathPoint[1024];
+
+    private PathPoint[] pathPoints = new PathPoint[128]; // CraftBukkit - reduce default size
     private int count;
     private static final String __OBFID = "CL_00000573";
 
     public PathPoint addPoint(PathPoint point)
     {
