--- decompiled\net\minecraft\pathfinding\PathFinder.java
+++ src\main\java\net\minecraft\pathfinding\PathFinder.java
@@ -5,10 +5,11 @@
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.pathfinder.NodeProcessor;
 
 public class PathFinder
 {
+
     private Path path = new Path();
     private PathPoint[] pathOptions = new PathPoint[32];
     private NodeProcessor field_176190_c;
     private static final String __OBFID = "CL_00000576";
 
