--- decompiled\net\minecraft\pathfinding\PathNavigateSwimmer.java
+++ src\main\java\net\minecraft\pathfinding\PathNavigateSwimmer.java
@@ -6,10 +6,11 @@
 import net.minecraft.world.World;
 import net.minecraft.world.pathfinder.SwimNodeProcessor;
 
 public class PathNavigateSwimmer extends PathNavigate
 {
+
     private static final String __OBFID = "CL_00002244";
 
     public PathNavigateSwimmer(EntityLiving p_i45873_1_, World worldIn)
     {
         super(p_i45873_1_, worldIn);
