--- decompiled\net\minecraft\pathfinding\PathNavigateClimber.java
+++ src\main\java\net\minecraft\pathfinding\PathNavigateClimber.java
@@ -6,10 +6,11 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
 public class PathNavigateClimber extends PathNavigateGround
 {
+
     private BlockPos field_179696_f;
     private static final String __OBFID = "CL_00002245";
 
     public PathNavigateClimber(EntityLiving p_i45874_1_, World worldIn)
     {
