--- decompiled\net\minecraft\pathfinding\PathNavigate.java
+++ src\main\java\net\minecraft\pathfinding\PathNavigate.java
@@ -10,10 +10,11 @@
 import net.minecraft.world.ChunkCache;
 import net.minecraft.world.World;
 
 public abstract class PathNavigate
 {
+
     protected EntityLiving theEntity;
     protected World worldObj;
     protected PathEntity currentPath;
     protected double speed;
     private final IAttributeInstance pathSearchRange;
@@ -249,9 +250,11 @@
     protected boolean isInLiquid()
     {
         return this.theEntity.isInWater() || this.theEntity.isInLava();
     }
 
-    protected void removeSunnyPath() {}
+    protected void removeSunnyPath()
+    {
+    }
 
     protected abstract boolean isDirectPathBetweenPoints(Vec3 p_75493_1_, Vec3 p_75493_2_, int p_75493_3_, int p_75493_4_, int p_75493_5_);
 }
