--- decompiled\net\minecraft\pathfinding\PathNavigateGround.java
+++ src\main\java\net\minecraft\pathfinding\PathNavigateGround.java
@@ -1,8 +1,9 @@
 package net.minecraft.pathfinding;
 
 import java.util.Iterator;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.monster.EntityZombie;
 import net.minecraft.entity.passive.EntityChicken;
@@ -13,10 +14,11 @@
 import net.minecraft.world.World;
 import net.minecraft.world.pathfinder.WalkNodeProcessor;
 
 public class PathNavigateGround extends PathNavigate
 {
+
     protected WalkNodeProcessor field_179695_a;
     private boolean field_179694_f;
     private static final String __OBFID = "CL_00002246";
 
     public PathNavigateGround(EntityLiving p_i45875_1_, World worldIn)
