--- decompiled\net\minecraft\world\storage\ThreadedFileIOBase.java
+++ src\main\java\net\minecraft\world\storage\ThreadedFileIOBase.java
@@ -1,13 +1,15 @@
 package net.minecraft.world.storage;
 
-import com.google.common.collect.Lists;
 import java.util.Collections;
 import java.util.List;
 
+import com.google.common.collect.Lists;
+
 public class ThreadedFileIOBase implements Runnable
 {
+
     private static final ThreadedFileIOBase threadedIOInstance = new ThreadedFileIOBase();
     private List threadedIOQueue = Collections.synchronizedList(Lists.newArrayList());
     private volatile long writeQueuedCounter;
     private volatile long savedIOCounter;
     private volatile boolean isThreadWaiting;
@@ -25,12 +27,11 @@
         return threadedIOInstance;
     }
 
     public void run()
     {
-        while (true)
-            this.processQueue();
+        while (true) this.processQueue();
     }
 
     private void processQueue()
     {
         for (int i = 0; i < this.threadedIOQueue.size(); ++i)
