--- decompiled\net\minecraft\world\storage\WorldInfo.java
+++ src\main\java\net\minecraft\world\storage\WorldInfo.java
@@ -1,21 +1,26 @@
 package net.minecraft.world.storage;
 
 import java.util.concurrent.Callable;
+
 import net.minecraft.crash.CrashReportCategory;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.GameRules;
 import net.minecraft.world.WorldSettings;
 import net.minecraft.world.WorldType;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+import org.bukkit.event.weather.ThunderChangeEvent;
+import org.bukkit.event.weather.WeatherChangeEvent;
+
 public class WorldInfo
 {
+
     public static final EnumDifficulty DEFAULT_DIFFICULTY = EnumDifficulty.NORMAL;
     private long randomSeed;
     private WorldType terrainType;
     private String generatorOptions;
     private int spawnX;
@@ -406,44 +411,20 @@
     public long getWorldTime()
     {
         return this.worldTime;
     }
 
-    @SideOnly(Side.CLIENT)
-    public long getSizeOnDisk()
-    {
-        return this.sizeOnDisk;
-    }
-
     public NBTTagCompound getPlayerNBTTagCompound()
     {
         return this.playerTag;
     }
 
-    @SideOnly(Side.CLIENT)
-    public void setSpawnX(int x)
-    {
-        this.spawnX = x;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public void setSpawnY(int y)
-    {
-        this.spawnY = y;
-    }
-
     public void incrementTotalWorldTime(long time)
     {
         this.totalTime = time;
     }
 
-    @SideOnly(Side.CLIENT)
-    public void setSpawnZ(int z)
-    {
-        this.spawnZ = z;
-    }
-
     public void setWorldTime(long time)
     {
         this.worldTime = time;
     }
 
@@ -472,16 +453,10 @@
     public void setSaveVersion(int version)
     {
         this.saveVersion = version;
     }
 
-    @SideOnly(Side.CLIENT)
-    public long getLastTimePlayed()
-    {
-        return this.lastTimePlayed;
-    }
-
     public int getCleanWeatherTime()
     {
         return this.cleanWeatherTime;
     }
 
@@ -495,10 +470,27 @@
         return this.thundering;
     }
 
     public void setThundering(boolean thunderingIn)
     {
+        // CraftBukkit start
+        World world = Bukkit.getWorld(getWorldName());
+
+        if (world != null)
+        {
+            ThunderChangeEvent thunder = new ThunderChangeEvent(world, thunderingIn);
+            Bukkit.getServer().getPluginManager().callEvent(thunder);
+
+            if (thunder.isCancelled())
+            {
+                return;
+            }
+
+            setThunderTime(0); // Will force a time reset
+        }
+
+        // CraftBukkit end
         this.thundering = thunderingIn;
     }
 
     public int getThunderTime()
     {
@@ -515,10 +507,27 @@
         return this.raining;
     }
 
     public void setRaining(boolean isRaining)
     {
+        // CraftBukkit start
+        World world = Bukkit.getWorld(getWorldName());
+
+        if (world != null)
+        {
+            WeatherChangeEvent weather = new WeatherChangeEvent(world, isRaining);
+            Bukkit.getServer().getPluginManager().callEvent(weather);
+
+            if (weather.isCancelled())
+            {
+                return;
+            }
+
+            setRainTime(0); // Will force a time reset
+        }
+
+        // CraftBukkit end
         this.raining = isRaining;
     }
 
     public int getRainTime()
     {
@@ -801,6 +810,16 @@
             {
                 return String.format("Game mode: %s (ID %d). Hardcore: %b. Cheats: %b", new Object[] {WorldInfo.this.theGameType.getName(), Integer.valueOf(WorldInfo.this.theGameType.getID()), Boolean.valueOf(WorldInfo.this.hardcore), Boolean.valueOf(WorldInfo.this.allowCommands)});
             }
         });
     }
+
+    // CraftBukkit start - Check if the name stored in NBT is the correct one
+    public void checkName(String name)
+    {
+        if (!this.levelName.equals(name))
+        {
+            this.levelName = name;
+        }
+    }
+    // CraftBukkit end
 }
