--- decompiled\net\minecraft\network\login\client\C00PacketLoginStart.java
+++ src\main\java\net\minecraft\network\login\client\C00PacketLoginStart.java
@@ -1,30 +1,35 @@
 package net.minecraft.network.login.client;
 
-import com.mojang.authlib.GameProfile;
 import java.io.IOException;
 import java.util.UUID;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.login.INetHandlerLoginServer;
 
+import com.mojang.authlib.GameProfile;
+
 public class C00PacketLoginStart implements Packet
 {
+
     private GameProfile profile;
     private static final String __OBFID = "CL_00001379";
 
-    public C00PacketLoginStart() {}
+    public C00PacketLoginStart()
+    {
+    }
 
     public C00PacketLoginStart(GameProfile profileIn)
     {
         this.profile = profileIn;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
-        this.profile = new GameProfile((UUID)null, buf.readStringFromBuffer(16));
+        this.profile = new GameProfile((UUID) null, buf.readStringFromBuffer(16));
     }
 
     public void writePacketData(PacketBuffer buf) throws IOException
     {
         buf.writeString(this.profile.getName());
@@ -40,8 +45,8 @@
         return this.profile;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerLoginServer)handler);
+        this.processPacket((INetHandlerLoginServer) handler);
     }
 }
