--- decompiled\net\minecraft\entity\ai\EntityAIBeg.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIBeg.java
@@ -6,10 +6,11 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.world.World;
 
 public class EntityAIBeg extends EntityAIBase
 {
+
     private EntityWolf theWolf;
     private EntityPlayer thePlayer;
     private World worldObject;
     private float minPlayerDistance;
     private int field_75384_e;
