--- decompiled\net\minecraft\entity\ai\EntitySenses.java
+++ src\main\java\net\minecraft\entity\ai\EntitySenses.java
@@ -1,14 +1,17 @@
 package net.minecraft.entity.ai;
 
-import com.google.common.collect.Lists;
 import java.util.List;
+
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLiving;
 
+import com.google.common.collect.Lists;
+
 public class EntitySenses
 {
+
     EntityLiving entityObj;
     List seenEntities = Lists.newArrayList();
     List unseenEntities = Lists.newArrayList();
     private static final String __OBFID = "CL_00001628";
 
