--- decompiled\net\minecraft\entity\ai\EntityAIOpenDoor.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIOpenDoor.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.entity.EntityLiving;
 
 public class EntityAIOpenDoor extends EntityAIDoorInteract
 {
+
     boolean closeDoor;
     int closeDoorTemporisation;
     private static final String __OBFID = "CL_00001603";
 
     public EntityAIOpenDoor(EntityLiving p_i1644_1_, boolean p_i1644_2_)
