--- decompiled\net\minecraft\entity\ai\EntityAIFindEntityNearestPlayer.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIFindEntityNearestPlayer.java
@@ -1,24 +1,28 @@
 package net.minecraft.entity.ai;
 
-import com.google.common.base.Predicate;
 import java.util.Collections;
 import java.util.List;
+
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityCreature;
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.entity.ai.attributes.IAttributeInstance;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.scoreboard.Team;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import com.google.common.base.Predicate;
+
 public class EntityAIFindEntityNearestPlayer extends EntityAIBase
 {
+
     private static final Logger field_179436_a = LogManager.getLogger();
     private EntityLiving field_179434_b;
     private final Predicate field_179435_c;
     private final EntityAINearestAttackableTarget.Sorter field_179432_d;
     private EntityLivingBase field_179433_e;
@@ -120,11 +124,11 @@
         }
     }
 
     public void startExecuting()
     {
-        this.field_179434_b.setAttackTarget(this.field_179433_e);
+        this.field_179434_b.setGoalTarget(this.field_179433_e, org.bukkit.event.entity.EntityTargetEvent.TargetReason.CLOSEST_PLAYER, true); // CraftBukkit - added reason
         super.startExecuting();
     }
 
     public void resetTask()
     {
