--- decompiled\net\minecraft\entity\ai\EntityAILookAtTradePlayer.java
+++ src\main\java\net\minecraft\entity\ai\EntityAILookAtTradePlayer.java
@@ -3,10 +3,11 @@
 import net.minecraft.entity.passive.EntityVillager;
 import net.minecraft.entity.player.EntityPlayer;
 
 public class EntityAILookAtTradePlayer extends EntityAIWatchClosest
 {
+
     private final EntityVillager theMerchant;
     private static final String __OBFID = "CL_00001593";
 
     public EntityAILookAtTradePlayer(EntityVillager p_i1633_1_)
     {
