--- decompiled\net\minecraft\entity\ai\EntityAIAttackOnCollide.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIAttackOnCollide.java
@@ -6,10 +6,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class EntityAIAttackOnCollide extends EntityAIBase
 {
+
     World worldObj;
     protected EntityCreature attacker;
     int attackTick;
     double speedTowardsTarget;
     boolean longMemory;
