--- decompiled\net\minecraft\entity\ai\EntityAIFollowGolem.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIFollowGolem.java
@@ -1,14 +1,16 @@
 package net.minecraft.entity.ai;
 
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.entity.monster.EntityIronGolem;
 import net.minecraft.entity.passive.EntityVillager;
 
 public class EntityAIFollowGolem extends EntityAIBase
 {
+
     private EntityVillager theVillager;
     private EntityIronGolem theGolem;
     private int takeGolemRoseTick;
     private boolean tookGolemRose;
     private static final String __OBFID = "CL_00001615";
