--- decompiled\net\minecraft\entity\ai\EntityAIFollowParent.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIFollowParent.java
@@ -1,13 +1,15 @@
 package net.minecraft.entity.ai;
 
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.entity.passive.EntityAnimal;
 
 public class EntityAIFollowParent extends EntityAIBase
 {
+
     EntityAnimal childAnimal;
     EntityAnimal parentAnimal;
     double field_75347_c;
     private int field_75345_d;
     private static final String __OBFID = "CL_00001586";
