--- decompiled\net\minecraft\entity\ai\EntityAIFollowOwner.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIFollowOwner.java
@@ -8,10 +8,11 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
 public class EntityAIFollowOwner extends EntityAIBase
 {
+
     private EntityTameable thePet;
     private EntityLivingBase theOwner;
     World theWorld;
     private double field_75336_f;
     private PathNavigate petPathfinder;
