--- decompiled\net\minecraft\entity\ai\EntityAIMoveToBlock.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIMoveToBlock.java
@@ -4,10 +4,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public abstract class EntityAIMoveToBlock extends EntityAIBase
 {
+
     private final EntityCreature field_179495_c;
     private final double field_179492_d;
     protected int field_179496_a;
     private int field_179493_e;
     private int field_179490_f;
@@ -49,11 +50,13 @@
         this.field_179495_c.getNavigator().tryMoveToXYZ((double)((float)this.destinationBlock.getX()) + 0.5D, (double)(this.destinationBlock.getY() + 1), (double)((float)this.destinationBlock.getZ()) + 0.5D, this.field_179492_d);
         this.field_179493_e = 0;
         this.field_179490_f = this.field_179495_c.getRNG().nextInt(this.field_179495_c.getRNG().nextInt(1200) + 1200) + 1200;
     }
 
-    public void resetTask() {}
+    public void resetTask()
+    {
+    }
 
     public void updateTask()
     {
         if (this.field_179495_c.getDistanceSqToCenter(this.destinationBlock.up()) > 1.0D)
         {
