--- decompiled\net\minecraft\entity\ai\EntityAIMoveTowardsTarget.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIMoveTowardsTarget.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.util.Vec3;
 
 public class EntityAIMoveTowardsTarget extends EntityAIBase
 {
+
     private EntityCreature theEntity;
     private EntityLivingBase targetEntity;
     private double movePosX;
     private double movePosY;
     private double movePosZ;
