--- decompiled\net\minecraft\entity\ai\EntityAIDefendVillage.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIDefendVillage.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.monster.EntityIronGolem;
 import net.minecraft.village.Village;
 
 public class EntityAIDefendVillage extends EntityAITarget
 {
+
     EntityIronGolem irongolem;
     EntityLivingBase villageAgressorTarget;
     private static final String __OBFID = "CL_00001618";
 
     public EntityAIDefendVillage(EntityIronGolem p_i1659_1_)
@@ -48,9 +49,9 @@
         }
     }
 
     public void startExecuting()
     {
-        this.irongolem.setAttackTarget(this.villageAgressorTarget);
+        this.irongolem.setGoalTarget(this.villageAgressorTarget, org.bukkit.event.entity.EntityTargetEvent.TargetReason.DEFEND_VILLAGE, true); // CraftBukkit - reason
         super.startExecuting();
     }
 }
