--- decompiled\net\minecraft\entity\ai\EntityAIOwnerHurtByTarget.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIOwnerHurtByTarget.java
@@ -3,10 +3,11 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.passive.EntityTameable;
 
 public class EntityAIOwnerHurtByTarget extends EntityAITarget
 {
+
     EntityTameable theDefendingTameable;
     EntityLivingBase theOwnerAttacker;
     private int field_142051_e;
     private static final String __OBFID = "CL_00001624";
 
@@ -40,11 +41,11 @@
         }
     }
 
     public void startExecuting()
     {
-        this.taskOwner.setAttackTarget(this.theOwnerAttacker);
+        this.taskOwner.setGoalTarget(this.theOwnerAttacker, org.bukkit.event.entity.EntityTargetEvent.TargetReason.TARGET_ATTACKED_OWNER, true); // CraftBukkit - reason
         EntityLivingBase entitylivingbase = this.theDefendingTameable.getOwnerEntity();
 
         if (entitylivingbase != null)
         {
             this.field_142051_e = entitylivingbase.getRevengeTimer();
