--- decompiled\net\minecraft\entity\ai\EntityAIHarvestFarmland.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIHarvestFarmland.java
@@ -11,10 +11,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class EntityAIHarvestFarmland extends EntityAIMoveToBlock
 {
+
     private final EntityVillager theVillager;
     private boolean field_179502_d;
     private boolean field_179503_e;
     private int field_179501_f;
     private static final String __OBFID = "CL_00002253";
