--- decompiled\net\minecraft\entity\ai\EntityAITasks.java
+++ src\main\java\net\minecraft\entity\ai\EntityAITasks.java
@@ -1,19 +1,24 @@
 package net.minecraft.entity.ai;
 
-import com.google.common.collect.Lists;
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.profiler.Profiler;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
+import org.bukkit.craftbukkit.util.UnsafeList;
 
 public class EntityAITasks
 {
+
     private static final Logger logger = LogManager.getLogger();
-    private List taskEntries = Lists.newArrayList();
-    private List executingTaskEntries = Lists.newArrayList();
+    // CraftBukkit start - ArrayList -> UnsafeList
+    private List taskEntries = new UnsafeList();
+    private List executingTaskEntries = new UnsafeList();
+    // CraftBukkit end
     private final Profiler theProfiler;
     private int tickCount;
     private int tickRate = 3;
     private static final String __OBFID = "CL_00001588";
 
@@ -22,10 +27,12 @@
         this.theProfiler = p_i1628_1_;
     }
 
     public void addTask(int p_75776_1_, EntityAIBase p_75776_2_)
     {
+        if (p_75776_2_ == null) return; // Spigot
+
         this.taskEntries.add(new EntityAITasks.EntityAITaskEntry(p_75776_1_, p_75776_2_));
     }
 
     public void removeTask(EntityAIBase p_85156_1_)
     {
@@ -129,15 +136,17 @@
             {
                 if (p_75775_1_.priority >= entityaitaskentry.priority)
                 {
                     if (!this.areTasksCompatible(p_75775_1_, entityaitaskentry) && this.executingTaskEntries.contains(entityaitaskentry))
                     {
+                        ((UnsafeList.Itr)iterator).valid = false; // CraftBukkit - mark iterator for reuse
                         return false;
                     }
                 }
                 else if (!entityaitaskentry.action.isInterruptible() && this.executingTaskEntries.contains(entityaitaskentry))
                 {
+                    ((UnsafeList.Itr)iterator).valid = false; // CraftBukkit - mark iterator for reuse
                     return false;
                 }
             }
         }
 
@@ -147,12 +156,13 @@
     private boolean areTasksCompatible(EntityAITasks.EntityAITaskEntry p_75777_1_, EntityAITasks.EntityAITaskEntry p_75777_2_)
     {
         return (p_75777_1_.action.getMutexBits() & p_75777_2_.action.getMutexBits()) == 0;
     }
 
-    class EntityAITaskEntry
+    public class EntityAITaskEntry
     {
+
         public EntityAIBase action;
         public int priority;
         private static final String __OBFID = "CL_00001589";
 
         public EntityAITaskEntry(int p_i1627_2_, EntityAIBase p_i1627_3_)
