--- decompiled\net\minecraft\entity\ai\EntityAIWander.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIWander.java
@@ -3,10 +3,11 @@
 import net.minecraft.entity.EntityCreature;
 import net.minecraft.util.Vec3;
 
 public class EntityAIWander extends EntityAIBase
 {
+
     private EntityCreature entity;
     private double xPosition;
     private double yPosition;
     private double zPosition;
     private double speed;
