--- decompiled\net\minecraft\entity\ai\EntityAIVillagerInteract.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIVillagerInteract.java
@@ -8,10 +8,11 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.MathHelper;
 
 public class EntityAIVillagerInteract extends EntityAIWatchClosest2
 {
+
     private int field_179478_e;
     private EntityVillager field_179477_f;
     private static final String __OBFID = "CL_00002251";
 
     public EntityAIVillagerInteract(EntityVillager p_i45886_1_)
