--- decompiled\net\minecraft\entity\ai\EntityAIWatchClosest.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIWatchClosest.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.player.EntityPlayer;
 
 public class EntityAIWatchClosest extends EntityAIBase
 {
+
     protected EntityLiving theWatcher;
     protected Entity closestEntity;
     protected float maxDistanceForPlayer;
     private int lookTime;
     private float field_75331_e;
