--- decompiled\net\minecraft\entity\ai\EntityAIPlay.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIPlay.java
@@ -1,15 +1,17 @@
 package net.minecraft.entity.ai;
 
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.passive.EntityVillager;
 import net.minecraft.util.Vec3;
 
 public class EntityAIPlay extends EntityAIBase
 {
+
     private EntityVillager villagerObj;
     private EntityLivingBase targetVillager;
     private double speed;
     private int playTime;
     private static final String __OBFID = "CL_00001605";
