--- decompiled\net\minecraft\entity\ai\EntityAIMoveIndoors.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIMoveIndoors.java
@@ -6,10 +6,11 @@
 import net.minecraft.village.Village;
 import net.minecraft.village.VillageDoorInfo;
 
 public class EntityAIMoveIndoors extends EntityAIBase
 {
+
     private EntityCreature entityObj;
     private VillageDoorInfo doorInfo;
     private int insidePosX = -1;
     private int insidePosZ = -1;
     private static final String __OBFID = "CL_00001596";
