--- decompiled\net\minecraft\entity\ai\EntityLookHelper.java
+++ src\main\java\net\minecraft\entity\ai\EntityLookHelper.java
@@ -3,12 +3,15 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.util.MathHelper;
 
+import org.bukkit.craftbukkit.TrigMath;
+
 public class EntityLookHelper
 {
+
     private EntityLiving entity;
     private float deltaLookYaw;
     private float deltaLookPitch;
     private boolean isLooking;
     private double posX;
@@ -59,12 +62,14 @@
             this.isLooking = false;
             double d0 = this.posX - this.entity.posX;
             double d1 = this.posY - (this.entity.posY + (double)this.entity.getEyeHeight());
             double d2 = this.posZ - this.entity.posZ;
             double d3 = (double)MathHelper.sqrt_double(d0 * d0 + d2 * d2);
-            float f = (float)(Math.atan2(d2, d0) * 180.0D / Math.PI) - 90.0F;
-            float f1 = (float)(-(Math.atan2(d1, d3) * 180.0D / Math.PI));
+            // CraftBukkit start - Math -> TrigMath
+            float f = (float)(TrigMath.atan2(d2, d0) * 180.0D / 3.1415927410125732D) - 90.0F;
+            float f1 = (float)(-(TrigMath.atan2(d1, d3) * 180.0D / 3.1415927410125732D));
+            // CraftBukkit end
             this.entity.rotationPitch = this.updateRotation(this.entity.rotationPitch, f1, this.deltaLookPitch);
             this.entity.rotationYawHead = this.updateRotation(this.entity.rotationYawHead, f, this.deltaLookYaw);
         }
         else
         {
