--- decompiled\net\minecraft\entity\ai\EntityAILookIdle.java
+++ src\main\java\net\minecraft\entity\ai\EntityAILookIdle.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.entity.EntityLiving;
 
 public class EntityAILookIdle extends EntityAIBase
 {
+
     private EntityLiving idleEntity;
     private double lookX;
     private double lookZ;
     private int idleTime;
     private static final String __OBFID = "CL_00001607";
