--- decompiled\net\minecraft\entity\ai\EntityAIOcelotSit.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIOcelotSit.java
@@ -10,10 +10,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class EntityAIOcelotSit extends EntityAIMoveToBlock
 {
+
     private final EntityOcelot field_151493_a;
     private static final String __OBFID = "CL_00001601";
 
     public EntityAIOcelotSit(EntityOcelot p_i45315_1_, double p_i45315_2_)
     {
