--- decompiled\net\minecraft\entity\ai\EntityAITempt.java
+++ src\main\java\net\minecraft\entity\ai\EntityAITempt.java
@@ -6,10 +6,11 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.pathfinding.PathNavigateGround;
 
 public class EntityAITempt extends EntityAIBase
 {
+
     private EntityCreature temptedEntity;
     private double speed;
     private double targetX;
     private double targetY;
     private double targetZ;
