--- decompiled\net\minecraft\entity\ai\EntityAIBreakDoor.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIBreakDoor.java
@@ -5,10 +5,11 @@
 import net.minecraft.entity.EntityLiving;
 import net.minecraft.world.EnumDifficulty;
 
 public class EntityAIBreakDoor extends EntityAIDoorInteract
 {
+
     private int breakingTime;
     private int previousBreakProgress = -1;
     private static final String __OBFID = "CL_00001577";
 
     public EntityAIBreakDoor(EntityLiving entityIn)
@@ -83,10 +84,18 @@
             this.previousBreakProgress = i;
         }
 
         if (this.breakingTime == 240 && this.theEntity.worldObj.getDifficulty() == EnumDifficulty.HARD)
         {
+            // CraftBukkit start
+            if (org.bukkit.craftbukkit.event.CraftEventFactory.callEntityBreakDoorEvent(this.theEntity, this.doorPosition.getX(), this.doorPosition.getY(), this.doorPosition.getZ()).isCancelled())
+            {
+                this.startExecuting();
+                return;
+            }
+
+            // CraftBukkit end
             this.theEntity.worldObj.setBlockToAir(this.doorPosition);
             this.theEntity.worldObj.playAuxSFX(1012, this.doorPosition, 0);
             this.theEntity.worldObj.playAuxSFX(2001, this.doorPosition, Block.getIdFromBlock(this.doorBlock));
         }
     }
