--- decompiled\net\minecraft\entity\ai\EntityAIRestrictOpenDoor.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIRestrictOpenDoor.java
@@ -6,10 +6,11 @@
 import net.minecraft.village.Village;
 import net.minecraft.village.VillageDoorInfo;
 
 public class EntityAIRestrictOpenDoor extends EntityAIBase
 {
+
     private EntityCreature entityObj;
     private VillageDoorInfo frontDoor;
     private static final String __OBFID = "CL_00001610";
 
     public EntityAIRestrictOpenDoor(EntityCreature p_i1651_1_)
