--- decompiled\net\minecraft\entity\ai\EntityAIRestrictSun.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIRestrictSun.java
@@ -3,10 +3,11 @@
 import net.minecraft.entity.EntityCreature;
 import net.minecraft.pathfinding.PathNavigateGround;
 
 public class EntityAIRestrictSun extends EntityAIBase
 {
+
     private EntityCreature theEntity;
     private static final String __OBFID = "CL_00001611";
 
     public EntityAIRestrictSun(EntityCreature p_i1652_1_)
     {
