--- decompiled\net\minecraft\entity\ai\EntityAIOwnerHurtTarget.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIOwnerHurtTarget.java
@@ -3,10 +3,11 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.passive.EntityTameable;
 
 public class EntityAIOwnerHurtTarget extends EntityAITarget
 {
+
     EntityTameable theEntityTameable;
     EntityLivingBase theTarget;
     private int field_142050_e;
     private static final String __OBFID = "CL_00001625";
 
@@ -40,11 +41,11 @@
         }
     }
 
     public void startExecuting()
     {
-        this.taskOwner.setAttackTarget(this.theTarget);
+        this.taskOwner.setGoalTarget(this.theTarget, org.bukkit.event.entity.EntityTargetEvent.TargetReason.OWNER_ATTACKED_TARGET, true); // CraftBukkit - reason);
         EntityLivingBase entitylivingbase = this.theEntityTameable.getOwnerEntity();
 
         if (entitylivingbase != null)
         {
             this.field_142050_e = entitylivingbase.getLastAttackerTime();
