--- decompiled\net\minecraft\entity\ai\EntityAIControlledByPlayer.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIControlledByPlayer.java
@@ -13,10 +13,11 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.pathfinder.WalkNodeProcessor;
 
 public class EntityAIControlledByPlayer extends EntityAIBase
 {
+
     private final EntityLiving thisEntity;
     private final float maxSpeed;
     private float currentSpeed;
     private boolean speedBoosted;
     private int speedBoostTime;
