--- decompiled\net\minecraft\entity\ai\EntityAIOcelotAttack.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIOcelotAttack.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.world.World;
 
 public class EntityAIOcelotAttack extends EntityAIBase
 {
+
     World theWorld;
     EntityLiving theEntity;
     EntityLivingBase theVictim;
     int attackCountdown;
     private static final String __OBFID = "CL_00001600";
