--- decompiled\net\minecraft\entity\ai\EntityAISit.java
+++ src\main\java\net\minecraft\entity\ai\EntityAISit.java
@@ -1,54 +1,42 @@
 package net.minecraft.entity.ai;
 
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.passive.EntityTameable;
 
-public class EntityAISit extends EntityAIBase
-{
+public class EntityAISit extends EntityAIBase {
+
     private EntityTameable theEntity;
     private boolean isSitting;
     private static final String __OBFID = "CL_00001613";
 
-    public EntityAISit(EntityTameable entityIn)
-    {
+    public EntityAISit(EntityTameable entityIn) {
         this.theEntity = entityIn;
         this.setMutexBits(5);
     }
 
-    public boolean shouldExecute()
-    {
-        if (!this.theEntity.isTamed())
-        {
+    public boolean shouldExecute() {
+        if (!this.theEntity.isTamed()) {
+            return this.isSitting && this.theEntity.getAttackTarget() == null; // CraftBukkit - Allow sitting for wild animals
+        } else if (this.theEntity.isInWater()) {
             return false;
-        }
-        else if (this.theEntity.isInWater())
-        {
+        } else if (!this.theEntity.onGround) {
             return false;
-        }
-        else if (!this.theEntity.onGround)
-        {
-            return false;
-        }
-        else
-        {
+        } else {
             EntityLivingBase entitylivingbase = this.theEntity.getOwnerEntity();
             return entitylivingbase == null ? true : (this.theEntity.getDistanceSqToEntity(entitylivingbase) < 144.0D && entitylivingbase.getAITarget() != null ? false : this.isSitting);
         }
     }
 
-    public void startExecuting()
-    {
+    public void startExecuting() {
         this.theEntity.getNavigator().clearPathEntity();
         this.theEntity.setSitting(true);
     }
 
-    public void resetTask()
-    {
+    public void resetTask() {
         this.theEntity.setSitting(false);
     }
 
-    public void setSitting(boolean sitting)
-    {
+    public void setSitting(boolean sitting) {
         this.isSitting = sitting;
     }
 }
