--- decompiled\net\minecraft\entity\ai\EntityAITarget.java
+++ src\main\java\net\minecraft\entity\ai\EntityAITarget.java
@@ -10,14 +10,16 @@
 import net.minecraft.pathfinding.PathEntity;
 import net.minecraft.pathfinding.PathPoint;
 import net.minecraft.scoreboard.Team;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
+
 import org.apache.commons.lang3.StringUtils;
 
 public abstract class EntityAITarget extends EntityAIBase
 {
+
     protected final EntityCreature taskOwner;
     protected boolean shouldCheckSight;
     private boolean nearbyOnly;
     private int targetSearchStatus;
     private int targetSearchDelay;
