--- decompiled\net\minecraft\entity\ai\EntityAILeapAtTarget.java
+++ src\main\java\net\minecraft\entity\ai\EntityAILeapAtTarget.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.util.MathHelper;
 
 public class EntityAILeapAtTarget extends EntityAIBase
 {
+
     EntityLiving leaper;
     EntityLivingBase leapTarget;
     float leapMotionY;
     private static final String __OBFID = "CL_00001591";
 
