--- decompiled\net\minecraft\entity\ai\EntityAIArrowAttack.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIArrowAttack.java
@@ -5,10 +5,11 @@
 import net.minecraft.entity.IRangedAttackMob;
 import net.minecraft.util.MathHelper;
 
 public class EntityAIArrowAttack extends EntityAIBase
 {
+
     private final EntityLiving entityHost;
     private final IRangedAttackMob rangedAttackEntityHost;
     private EntityLivingBase attackTarget;
     private int rangedAttackTime;
     private double entityMoveSpeed;
