--- decompiled\net\minecraft\entity\ai\EntityAITradePlayer.java
+++ src\main\java\net\minecraft\entity\ai\EntityAITradePlayer.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.inventory.Container;
 
 public class EntityAITradePlayer extends EntityAIBase
 {
+
     private EntityVillager villager;
     private static final String __OBFID = "CL_00001617";
 
     public EntityAITradePlayer(EntityVillager p_i1658_1_)
     {
