--- decompiled\net\minecraft\entity\ai\EntityAIVillagerMate.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIVillagerMate.java
@@ -6,10 +6,11 @@
 import net.minecraft.village.Village;
 import net.minecraft.world.World;
 
 public class EntityAIVillagerMate extends EntityAIBase
 {
+
     private EntityVillager villagerObj;
     private EntityVillager mate;
     private World worldObj;
     private int matingTimeout;
     Village villageObj;
@@ -119,9 +120,9 @@
         this.villagerObj.setGrowingAge(6000);
         this.mate.func_175549_o(false);
         this.villagerObj.func_175549_o(false);
         entityvillager.setGrowingAge(-24000);
         entityvillager.setLocationAndAngles(this.villagerObj.posX, this.villagerObj.posY, this.villagerObj.posZ, 0.0F, 0.0F);
-        this.worldObj.spawnEntityInWorld(entityvillager);
+        this.worldObj.addEntity(entityvillager, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.BREEDING); // CraftBukkit - added SpawnReason
         this.worldObj.setEntityState(entityvillager, (byte)12);
     }
 }
