--- decompiled\net\minecraft\entity\ai\EntityAIDoorInteract.java
+++ src\main\java\net\minecraft\entity\ai\EntityAIDoorInteract.java
@@ -9,10 +9,11 @@
 import net.minecraft.pathfinding.PathPoint;
 import net.minecraft.util.BlockPos;
 
 public abstract class EntityAIDoorInteract extends EntityAIBase
 {
+
     protected EntityLiving theEntity;
     protected BlockPos doorPosition;
     protected BlockDoor doorBlock;
     boolean hasStoppedDoorInteraction;
     float entityPositionX;
