--- decompiled\net\minecraft\entity\ai\EntityMinecartMobSpawner.java
+++ src\main\java\net\minecraft\entity\ai\EntityMinecartMobSpawner.java
@@ -5,15 +5,15 @@
 import net.minecraft.init.Blocks;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.tileentity.MobSpawnerBaseLogic;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class EntityMinecartMobSpawner extends EntityMinecart
 {
+
     private final MobSpawnerBaseLogic mobSpawnerLogic = new MobSpawnerBaseLogic()
     {
         private static final String __OBFID = "CL_00001679";
         public void func_98267_a(int p_98267_1_)
         {
@@ -60,23 +60,12 @@
     {
         super.writeEntityToNBT(tagCompound);
         this.mobSpawnerLogic.writeToNBT(tagCompound);
     }
 
-    @SideOnly(Side.CLIENT)
-    public void handleHealthUpdate(byte p_70103_1_)
-    {
-        this.mobSpawnerLogic.setDelayToMin(p_70103_1_);
-    }
-
     public void onUpdate()
     {
         super.onUpdate();
         this.mobSpawnerLogic.updateSpawner();
     }
 
-    @SideOnly(Side.CLIENT)
-    public MobSpawnerBaseLogic func_98039_d()
-    {
-        return this.mobSpawnerLogic;
-    }
 }
