--- decompiled\net\minecraft\block\state\BlockWorldState.java
+++ src\main\java\net\minecraft\block\state\BlockWorldState.java
@@ -1,14 +1,17 @@
 package net.minecraft.block.state;
 
-import com.google.common.base.Predicate;
+import net.minecraft.init.Blocks;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
+import com.google.common.base.Predicate;
+
 public class BlockWorldState
 {
+
     private final World world;
     private final BlockPos pos;
     private IBlockState state;
     private TileEntity tileEntity;
     private boolean tileEntityInitialized;
@@ -20,15 +23,17 @@
         this.pos = posIn;
     }
 
     public IBlockState getBlockState()
     {
-        if (this.state == null && this.world.isBlockLoaded(this.pos))
+        // CraftBukkit start - fix null pointer
+        if (this.state == null)
         {
-            this.state = this.world.getBlockState(this.pos);
+            this.state = this.world.isBlockLoaded(this.pos) ? this.world.getBlockState(this.pos) : Blocks.air.getDefaultState();
         }
 
+        // CraftBukkit end
         return this.state;
     }
 
     public TileEntity getTileEntity()
     {
