--- decompiled\net\minecraft\block\state\BlockStateBase.java
+++ src\main\java\net\minecraft\block\state\BlockStateBase.java
@@ -1,18 +1,21 @@
 package net.minecraft.block.state;
 
-import com.google.common.base.Function;
-import com.google.common.base.Joiner;
-import com.google.common.collect.Iterables;
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.Map.Entry;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.properties.IProperty;
 
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.collect.Iterables;
+
 public abstract class BlockStateBase implements IBlockState
 {
+
     private static final Joiner COMMA_JOINER = Joiner.on(',');
     private static final Function MAP_ENTRY_TO_STRING = new Function()
     {
         private static final String __OBFID = "CL_00002031";
         public String func_177225_a(Entry p_177225_1_)
