--- decompiled\net\minecraft\block\state\IBlockState.java
+++ src\main\java\net\minecraft\block\state\IBlockState.java
@@ -1,14 +1,17 @@
 package net.minecraft.block.state;
 
-import com.google.common.collect.ImmutableMap;
 import java.util.Collection;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.properties.IProperty;
 
+import com.google.common.collect.ImmutableMap;
+
 public interface IBlockState
 {
+
     Collection getPropertyNames();
 
     Comparable getValue(IProperty property);
 
     IBlockState withProperty(IProperty property, Comparable value);
