--- decompiled\net\minecraft\entity\item\EntityMinecartHopper.java
+++ src\main\java\net\minecraft\entity\item\EntityMinecartHopper.java
@@ -1,8 +1,9 @@
 package net.minecraft.entity.item;
 
 import java.util.List;
+
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.command.IEntitySelector;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.init.Blocks;
@@ -16,10 +17,11 @@
 import net.minecraft.util.DamageSource;
 import net.minecraft.world.World;
 
 public class EntityMinecartHopper extends EntityMinecartContainer implements IHopper
 {
+
     private boolean isBlocked = true;
     private int transferTicker = -1;
     private BlockPos field_174900_c;
     private static final String __OBFID = "CL_00001676";
 
