--- decompiled\net\minecraft\entity\item\EntityItemFrame.java
+++ src\main\java\net\minecraft\entity\item\EntityItemFrame.java
@@ -11,15 +11,15 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 import net.minecraft.world.storage.MapData;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class EntityItemFrame extends EntityHanging
 {
+
     private float itemDropChance = 1.0F;
     private static final String __OBFID = "CL_00001547";
 
     public EntityItemFrame(World worldIn)
     {
@@ -51,10 +51,17 @@
         }
         else if (!source.isExplosion() && this.getDisplayedItem() != null)
         {
             if (!this.worldObj.isRemote)
             {
+                // CraftBukkit start - fire EntityDamageEvent
+                if (org.bukkit.craftbukkit.event.CraftEventFactory.handleNonLivingEntityDamageEvent(this, source, amount, false) || this.isDead)
+                {
+                    return true;
+                }
+
+                // CraftBukkit end
                 this.dropItemOrSelf(source.getEntity(), false);
                 this.setDisplayedItem((ItemStack)null);
             }
 
             return true;
@@ -73,18 +80,10 @@
     public int getHeightPixels()
     {
         return 12;
     }
 
-    @SideOnly(Side.CLIENT)
-    public boolean isInRangeToRenderDist(double distance)
-    {
-        double d1 = 16.0D;
-        d1 *= 64.0D * this.renderDistanceWeight;
-        return distance < d1 * d1;
-    }
-
     public void onBroken(Entity p_110128_1_)
     {
         this.dropItemOrSelf(p_110128_1_, true);
     }
 
