--- decompiled\net\minecraft\entity\item\EntityExpBottle.java
+++ src\main\java\net\minecraft\entity\item\EntityExpBottle.java
@@ -6,10 +6,11 @@
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraft.world.World;
 
 public class EntityExpBottle extends EntityThrowable
 {
+
     private static final String __OBFID = "CL_00001726";
 
     public EntityExpBottle(World worldIn)
     {
         super(worldIn);
@@ -42,13 +43,24 @@
 
     protected void onImpact(MovingObjectPosition p_70184_1_)
     {
         if (!this.worldObj.isRemote)
         {
-            this.worldObj.playAuxSFX(2002, new BlockPos(this), 0);
+            // CraftBukkit - moved to after event
+            // this.worldObj.playAuxSFX(2002, new BlockPos(this), 0);
             int i = 3 + this.worldObj.rand.nextInt(5) + this.worldObj.rand.nextInt(5);
+            // CraftBukkit start
+            org.bukkit.event.entity.ExpBottleEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callExpBottleEvent(this, i);
+            i = event.getExperience();
 
+            if (event.getShowEffect())
+            {
+                this.worldObj.playAuxSFX(2002, new BlockPos(this), 0);
+            }
+
+            // CraftBukkit end
+
             while (i > 0)
             {
                 int j = EntityXPOrb.getXPSplit(i);
                 i -= j;
                 this.worldObj.spawnEntityInWorld(new EntityXPOrb(this.worldObj, this.posX, this.posY, this.posZ, j));
