--- decompiled\net\minecraft\entity\item\EntityEnderPearl.java
+++ src\main\java\net\minecraft\entity\item\EntityEnderPearl.java
@@ -8,26 +8,31 @@
 import net.minecraft.entity.projectile.EntityThrowable;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.Bukkit;
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.event.player.PlayerTeleportEvent;
+
 public class EntityEnderPearl extends EntityThrowable
 {
+
     private static final String __OBFID = "CL_00001725";
 
-    public EntityEnderPearl(World worldIn, EntityLivingBase p_i1783_2_)
+    // CraftBukkit start
+    // This is backported from 1.8.1
+    public EntityEnderPearl(World world)
     {
-        super(worldIn, p_i1783_2_);
+        super(world);
     }
+    // CraftBukkit end
 
-    @SideOnly(Side.CLIENT)
-    public EntityEnderPearl(World worldIn, double p_i1784_2_, double p_i1784_4_, double p_i1784_6_)
+    public EntityEnderPearl(World worldIn, EntityLivingBase p_i1783_2_)
     {
-        super(worldIn, p_i1784_2_, p_i1784_4_, p_i1784_6_);
+        super(worldIn, p_i1783_2_);
     }
 
     protected void onImpact(MovingObjectPosition p_70184_1_)
     {
         EntityLivingBase entitylivingbase = this.getThrower();
@@ -56,18 +61,33 @@
                         entityendermite.setSpawnedByPlayer(true);
                         entityendermite.setLocationAndAngles(entitylivingbase.posX, entitylivingbase.posY, entitylivingbase.posZ, entitylivingbase.rotationYaw, entitylivingbase.rotationPitch);
                         this.worldObj.spawnEntityInWorld(entityendermite);
                     }
 
-                    if (entitylivingbase.isRiding())
+                    // CraftBukkit start - Fire PlayerTeleportEvent
+                    org.bukkit.craftbukkit.entity.CraftPlayer player = entityplayermp.getBukkitEntity();
+                    org.bukkit.Location location = getBukkitEntity().getLocation();
+                    location.setPitch(player.getLocation().getPitch());
+                    location.setYaw(player.getLocation().getYaw());
+                    PlayerTeleportEvent teleEvent = new PlayerTeleportEvent(player, player.getLocation(), location, PlayerTeleportEvent.TeleportCause.ENDER_PEARL);
+                    Bukkit.getPluginManager().callEvent(teleEvent);
+
+                    if (!teleEvent.isCancelled() && !entityplayermp.playerNetServerHandler.isDisconnected())
                     {
-                        entitylivingbase.mountEntity((Entity)null);
+                        if (entitylivingbase.isRiding())
+                        {
+                            entitylivingbase.mountEntity((Entity)null);
+                        }
+
+                        entityplayermp.playerNetServerHandler.teleport(teleEvent.getTo());
+                        entitylivingbase.fallDistance = 0.0F;
+                        CraftEventFactory.entityDamage = this;
+                        entitylivingbase.attackEntityFrom(DamageSource.fall, 5.0F);
+                        CraftEventFactory.entityDamage = null;
                     }
 
-                    entitylivingbase.setPositionAndUpdate(this.posX, this.posY, this.posZ);
-                    entitylivingbase.fallDistance = 0.0F;
-                    entitylivingbase.attackEntityFrom(DamageSource.fall, 5.0F);
+                    // CraftBukkit end
                 }
             }
 
             this.setDead();
         }
