--- decompiled\net\minecraft\entity\item\EntityMinecartEmpty.java
+++ src\main\java\net\minecraft\entity\item\EntityMinecartEmpty.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.world.World;
 
 public class EntityMinecartEmpty extends EntityMinecart
 {
+
     private static final String __OBFID = "CL_00001677";
 
     public EntityMinecartEmpty(World worldIn)
     {
         super(worldIn);
