--- decompiled\net\minecraft\entity\item\EntityEnderCrystal.java
+++ src\main\java\net\minecraft\entity\item\EntityEnderCrystal.java
@@ -6,15 +6,17 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 import net.minecraft.world.WorldProviderEnd;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.event.entity.ExplosionPrimeEvent;
+
 public class EntityEnderCrystal extends Entity
 {
+
     public int innerRotation;
     public int health;
     private static final String __OBFID = "CL_00001658";
 
     public EntityEnderCrystal(World worldIn)
@@ -24,17 +26,10 @@
         this.setSize(2.0F, 2.0F);
         this.health = 5;
         this.innerRotation = this.rand.nextInt(100000);
     }
 
-    @SideOnly(Side.CLIENT)
-    public EntityEnderCrystal(World worldIn, double p_i1699_2_, double p_i1699_4_, double p_i1699_6_)
-    {
-        this(worldIn);
-        this.setPosition(p_i1699_2_, p_i1699_4_, p_i1699_6_);
-    }
-
     protected boolean canTriggerWalking()
     {
         return false;
     }
 
@@ -54,17 +49,27 @@
         int j = MathHelper.floor_double(this.posY);
         int k = MathHelper.floor_double(this.posZ);
 
         if (this.worldObj.provider instanceof WorldProviderEnd && this.worldObj.getBlockState(new BlockPos(i, j, k)).getBlock() != Blocks.fire)
         {
-            this.worldObj.setBlockState(new BlockPos(i, j, k), Blocks.fire.getDefaultState());
+            // CraftBukkit start
+            if (!CraftEventFactory.callBlockIgniteEvent(this.worldObj, i, j, k, this).isCancelled())
+            {
+                this.worldObj.setBlockState(new BlockPos(i, j, k), Blocks.fire.getDefaultState());
+            }
+
+            // CraftBukkit end
         }
     }
 
-    protected void writeEntityToNBT(NBTTagCompound tagCompound) {}
+    protected void writeEntityToNBT(NBTTagCompound tagCompound)
+    {
+    }
 
-    protected void readEntityFromNBT(NBTTagCompound tagCompund) {}
+    protected void readEntityFromNBT(NBTTagCompound tagCompund)
+    {
+    }
 
     public boolean canBeCollidedWith()
     {
         return true;
     }
@@ -77,19 +82,37 @@
         }
         else
         {
             if (!this.isDead && !this.worldObj.isRemote)
             {
+                // CraftBukkit start - All non-living entities need this
+                if (CraftEventFactory.handleNonLivingEntityDamageEvent(this, source, amount))
+                {
+                    return false;
+                }
+
+                // CraftBukkit end
                 this.health = 0;
 
                 if (this.health <= 0)
                 {
                     this.setDead();
 
                     if (!this.worldObj.isRemote)
                     {
-                        this.worldObj.createExplosion((Entity)null, this.posX, this.posY, this.posZ, 6.0F, true);
+                        // CraftBukkit start
+                        ExplosionPrimeEvent event = new ExplosionPrimeEvent(this.getBukkitEntity(), 6.0F, false);
+                        this.worldObj.getServer().getPluginManager().callEvent(event);
+
+                        if (event.isCancelled())
+                        {
+                            this.isDead = false;
+                            return false;
+                        }
+
+                        this.worldObj.newExplosion(this, this.posX, this.posY, this.posZ, event.getRadius(), event.getFire(), true);
+                        // CraftBukkit end
                     }
                 }
             }
 
             return true;
