--- decompiled\net\minecraft\entity\item\EntityMinecartFurnace.java
+++ src\main\java\net\minecraft\entity\item\EntityMinecartFurnace.java
@@ -14,10 +14,11 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
 public class EntityMinecartFurnace extends EntityMinecart
 {
+
     private int fuel;
     public double pushX;
     public double pushZ;
     private static final String __OBFID = "CL_00001675";
 
