--- decompiled\net\minecraft\world\border\IBorderListener.java
+++ src\main\java\net\minecraft\world\border\IBorderListener.java
@@ -1,9 +1,9 @@
 package net.minecraft.world.border;
 
-public interface IBorderListener
-{
+public interface IBorderListener {
+
     void onSizeChanged(WorldBorder border, double newSize);
 
     void onTransitionStarted(WorldBorder border, double oldSize, double newSize, long time);
 
     void onCenterChanged(WorldBorder border, double x, double z);
