--- decompiled\net\minecraft\world\gen\ChunkProviderHell.java
+++ src\main\java\net\minecraft\world\gen\ChunkProviderHell.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.gen;
 
 import java.util.List;
 import java.util.Random;
+
 import net.minecraft.block.BlockFalling;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.block.state.pattern.BlockHelper;
 import net.minecraft.entity.EnumCreatureType;
@@ -25,10 +26,11 @@
 import net.minecraft.world.gen.feature.WorldGenerator;
 import net.minecraft.world.gen.structure.MapGenNetherBridge;
 
 public class ChunkProviderHell implements IChunkProvider
 {
+
     private final World worldObj;
     private final boolean field_177466_i;
     private final Random hellRNG;
     private double[] slowsandNoise = new double[256];
     private double[] gravelNoise = new double[256];
@@ -421,11 +423,13 @@
     public boolean saveChunks(boolean p_73151_1_, IProgressUpdate p_73151_2_)
     {
         return true;
     }
 
-    public void saveExtraData() {}
+    public void saveExtraData()
+    {
+    }
 
     public boolean unloadQueuedChunks()
     {
         return false;
     }
