--- decompiled\net\minecraft\world\gen\ChunkProviderGenerate.java
+++ src\main\java\net\minecraft\world\gen\ChunkProviderGenerate.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.gen;
 
 import java.util.List;
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockFalling;
 import net.minecraft.entity.EnumCreatureType;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
@@ -25,10 +26,11 @@
 import net.minecraft.world.gen.structure.MapGenVillage;
 import net.minecraft.world.gen.structure.StructureOceanMonument;
 
 public class ChunkProviderGenerate implements IChunkProvider
 {
+
     private Random rand;
     private NoiseGeneratorOctaves field_147431_j;
     private NoiseGeneratorOctaves field_147432_k;
     private NoiseGeneratorOctaves field_147429_l;
     private NoiseGeneratorPerlin field_147430_m;
@@ -476,11 +478,13 @@
     public boolean saveChunks(boolean p_73151_1_, IProgressUpdate p_73151_2_)
     {
         return true;
     }
 
-    public void saveExtraData() {}
+    public void saveExtraData()
+    {
+    }
 
     public boolean unloadQueuedChunks()
     {
         return false;
     }
