--- decompiled\net\minecraft\world\gen\FlatLayerInfo.java
+++ src\main\java\net\minecraft\world\gen\FlatLayerInfo.java
@@ -4,10 +4,11 @@
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.util.ResourceLocation;
 
 public class FlatLayerInfo
 {
+
     private final int field_175902_a;
     private IBlockState field_175901_b;
     private int layerCount;
     private int layerMinimumY;
     private static final String __OBFID = "CL_00000441";
