--- decompiled\net\minecraft\world\gen\NoiseGeneratorImproved.java
+++ src\main\java\net\minecraft\world\gen\NoiseGeneratorImproved.java
@@ -2,10 +2,11 @@
 
 import java.util.Random;
 
 public class NoiseGeneratorImproved extends NoiseGenerator
 {
+
     private int[] permutations;
     public double xCoord;
     public double yCoord;
     public double zCoord;
     private static final double[] field_152381_e = new double[] {1.0D, -1.0D, 1.0D, -1.0D, 1.0D, -1.0D, 1.0D, -1.0D, 0.0D, 0.0D, 0.0D, 0.0D, 1.0D, 0.0D, -1.0D, 0.0D};
