--- decompiled\net\minecraft\world\gen\NoiseGeneratorSimplex.java
+++ src\main\java\net\minecraft\world\gen\NoiseGeneratorSimplex.java
@@ -2,10 +2,11 @@
 
 import java.util.Random;
 
 public class NoiseGeneratorSimplex
 {
+
     private static int[][] field_151611_e = new int[][] {{1, 1, 0}, { -1, 1, 0}, {1, -1, 0}, { -1, -1, 0}, {1, 0, 1}, { -1, 0, 1}, {1, 0, -1}, { -1, 0, -1}, {0, 1, 1}, {0, -1, 1}, {0, 1, -1}, {0, -1, -1}};
     public static final double field_151614_a = Math.sqrt(3.0D);
     private int[] field_151608_f;
     public double field_151612_b;
     public double field_151613_c;
