--- decompiled\net\minecraft\network\play\INetHandlerPlayServer.java
+++ src\main\java\net\minecraft\network\play\INetHandlerPlayServer.java
@@ -25,10 +25,11 @@
 import net.minecraft.network.play.client.C18PacketSpectate;
 import net.minecraft.network.play.client.C19PacketResourcePackStatus;
 
 public interface INetHandlerPlayServer extends INetHandler
 {
+
     void handleAnimation(C0APacketAnimation packetIn);
 
     void processChatMessage(C01PacketChatMessage packetIn);
 
     void processTabComplete(C14PacketTabComplete packetIn);
