--- decompiled\net\minecraft\world\DifficultyInstance.java
+++ src\main\java\net\minecraft\world\DifficultyInstance.java
@@ -1,29 +1,23 @@
 package net.minecraft.world;
 
 import net.minecraft.util.MathHelper;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class DifficultyInstance
 {
+
     private final EnumDifficulty worldDifficulty;
     private final float additionalDifficulty;
     private static final String __OBFID = "CL_00002261";
 
     public DifficultyInstance(EnumDifficulty worldDifficulty, long worldTime, long chunkInhabitedTime, float moonPhaseFactor)
     {
         this.worldDifficulty = worldDifficulty;
         this.additionalDifficulty = this.calculateAdditionalDifficulty(worldDifficulty, worldTime, chunkInhabitedTime, moonPhaseFactor);
     }
 
-    @SideOnly(Side.CLIENT)
-    public float func_180168_b()
-    {
-        return this.additionalDifficulty;
-    }
-
     public float getClampedAdditionalDifficulty()
     {
         return this.additionalDifficulty < 2.0F ? 0.0F : (this.additionalDifficulty > 4.0F ? 1.0F : (this.additionalDifficulty - 2.0F) / 2.0F);
     }
 
