--- decompiled\net\minecraft\world\GameRules.java
+++ src\main\java\net\minecraft\world\GameRules.java
@@ -1,14 +1,16 @@
 package net.minecraft.world;
 
 import java.util.Iterator;
 import java.util.Set;
 import java.util.TreeMap;
+
 import net.minecraft.nbt.NBTTagCompound;
 
 public class GameRules
 {
+
     private TreeMap theGameRules = new TreeMap();
     private static final String __OBFID = "CL_00000136";
 
     public GameRules()
     {
@@ -109,10 +111,11 @@
         return value != null && (value.getType() == otherValue || otherValue == GameRules.ValueType.ANY_VALUE);
     }
 
     static class Value
         {
+
             private String valueString;
             private boolean valueBoolean;
             private int valueInteger;
             private double valueDouble;
             private final GameRules.ValueType type;
