--- decompiled\net\minecraft\world\IInteractionObject.java
+++ src\main\java\net\minecraft\world\IInteractionObject.java
@@ -2,11 +2,11 @@
 
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.inventory.Container;
 
-public interface IInteractionObject extends IWorldNameable
-{
+public interface IInteractionObject extends IWorldNameable {
+
     Container createContainer(InventoryPlayer playerInventory, EntityPlayer playerIn);
 
     String getGuiID();
 }
