--- decompiled\net\minecraft\world\NextTickListEntry.java
+++ src\main\java\net\minecraft\world\NextTickListEntry.java
@@ -3,10 +3,11 @@
 import net.minecraft.block.Block;
 import net.minecraft.util.BlockPos;
 
 public class NextTickListEntry implements Comparable
 {
+
     private static long nextTickEntryID;
     private final Block block;
     public final BlockPos position;
     public long scheduledTime;
     public int priority;
