--- decompiled\net\minecraft\world\WorldSettings.java
+++ src\main\java\net\minecraft\world\WorldSettings.java
@@ -1,14 +1,14 @@
 package net.minecraft.world;
 
 import net.minecraft.entity.player.PlayerCapabilities;
 import net.minecraft.world.storage.WorldInfo;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public final class WorldSettings
 {
+
     private final long seed;
     private final WorldSettings.GameType theGameType;
     private final boolean mapFeaturesEnabled;
     private final boolean hardcoreEnabled;
     private final WorldType terrainType;
@@ -42,17 +42,10 @@
     {
         this.worldName = name;
         return this;
     }
 
-    @SideOnly(Side.CLIENT)
-    public WorldSettings enableCommands()
-    {
-        this.commandsAllowed = true;
-        return this;
-    }
-
     public boolean isBonusChestEnabled()
     {
         return this.bonusChestEnabled;
     }
 
@@ -180,26 +173,7 @@
                 }
             }
 
             return SURVIVAL;
         }
-
-        @SideOnly(Side.CLIENT)
-        public static WorldSettings.GameType getByName(String p_77142_0_)
-        {
-            WorldSettings.GameType[] agametype = values();
-            int i = agametype.length;
-
-            for (int j = 0; j < i; ++j)
-            {
-                WorldSettings.GameType gametype = agametype[j];
-
-                if (gametype.name.equals(p_77142_0_))
-                {
-                    return gametype;
-                }
-            }
-
-            return SURVIVAL;
-        }
     }
 }
