--- decompiled\net\minecraft\world\WorldProviderHell.java
+++ src\main\java\net\minecraft\world\WorldProviderHell.java
@@ -1,34 +1,27 @@
 package net.minecraft.world;
 
-import net.minecraft.util.Vec3;
 import net.minecraft.world.biome.BiomeGenBase;
 import net.minecraft.world.biome.WorldChunkManagerHell;
 import net.minecraft.world.border.WorldBorder;
 import net.minecraft.world.chunk.IChunkProvider;
 import net.minecraft.world.gen.ChunkProviderHell;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class WorldProviderHell extends WorldProvider
 {
+
     private static final String __OBFID = "CL_00000387";
 
     public void registerWorldChunkManager()
     {
         this.worldChunkMgr = new WorldChunkManagerHell(BiomeGenBase.hell, 0.0F);
         this.isHellWorld = true;
         this.hasNoSky = true;
         this.dimensionId = -1;
     }
 
-    @SideOnly(Side.CLIENT)
-    public Vec3 getFogColor(float p_76562_1_, float p_76562_2_)
-    {
-        return new Vec3(0.20000000298023224D, 0.029999999329447746D, 0.029999999329447746D);
-    }
-
     protected void generateLightBrightnessTable()
     {
         float f = 0.1F;
 
         for (int i = 0; i <= 15; ++i)
@@ -61,16 +54,10 @@
     public boolean canRespawnHere()
     {
         return false;
     }
 
-    @SideOnly(Side.CLIENT)
-    public boolean doesXZShowFog(int x, int z)
-    {
-        return true;
-    }
-
     public String getDimensionName()
     {
         return "Nether";
     }
 
