--- decompiled\net\minecraft\world\WorldType.java
+++ src\main\java\net\minecraft\world\WorldType.java
@@ -1,13 +1,12 @@
 package net.minecraft.world;
 
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
 public class WorldType
 {
-    public static final WorldType[] worldTypes = new WorldType[16];
+
+    public static WorldType[] worldTypes = new WorldType[16];
     public static final WorldType DEFAULT = (new WorldType(0, "default", 1)).setVersioned();
     public static final WorldType FLAT = new WorldType(1, "flat");
     public static final WorldType LARGE_BIOMES = new WorldType(2, "largeBiomes");
     public static final WorldType AMPLIFIED = (new WorldType(3, "amplified")).setNotificationData();
     public static final WorldType CUSTOMIZED = new WorldType(4, "customized");
@@ -38,22 +37,10 @@
     public String getWorldTypeName()
     {
         return this.worldType;
     }
 
-    @SideOnly(Side.CLIENT)
-    public String getTranslateName()
-    {
-        return "generator." + this.worldType;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public String func_151359_c()
-    {
-        return this.getTranslateName() + ".info";
-    }
-
     public int getGeneratorVersion()
     {
         return this.generatorVersion;
     }
 
@@ -66,16 +53,10 @@
     {
         this.canBeCreated = enable;
         return this;
     }
 
-    @SideOnly(Side.CLIENT)
-    public boolean getCanBeCreated()
-    {
-        return this.canBeCreated;
-    }
-
     private WorldType setVersioned()
     {
         this.isWorldTypeVersioned = true;
         return this;
     }
@@ -101,16 +82,10 @@
     public int getWorldTypeID()
     {
         return this.worldTypeId;
     }
 
-    @SideOnly(Side.CLIENT)
-    public boolean showWorldInfoNotice()
-    {
-        return this.hasNotificationData;
-    }
-
     private WorldType setNotificationData()
     {
         this.hasNotificationData = true;
         return this;
     }
