--- decompiled\net\minecraft\event\ClickEvent.java
+++ src\main\java\net\minecraft\event\ClickEvent.java
@@ -1,12 +1,14 @@
 package net.minecraft.event;
 
-import com.google.common.collect.Maps;
 import java.util.Map;
 
+import com.google.common.collect.Maps;
+
 public class ClickEvent
 {
+
     private final ClickEvent.Action action;
     private final String value;
     private static final String __OBFID = "CL_00001260";
 
     public ClickEvent(ClickEvent.Action theAction, String theValue)
