--- decompiled\net\minecraft\event\HoverEvent.java
+++ src\main\java\net\minecraft\event\HoverEvent.java
@@ -1,13 +1,16 @@
 package net.minecraft.event;
 
-import com.google.common.collect.Maps;
 import java.util.Map;
+
 import net.minecraft.util.IChatComponent;
 
+import com.google.common.collect.Maps;
+
 public class HoverEvent
 {
+
     private final HoverEvent.Action action;
     private final IChatComponent value;
     private static final String __OBFID = "CL_00001264";
 
     public HoverEvent(HoverEvent.Action p_i45158_1_, IChatComponent p_i45158_2_)
