--- decompiled\net\minecraft\entity\boss\EntityDragonPart.java
+++ src\main\java\net\minecraft\entity\boss\EntityDragonPart.java
@@ -5,10 +5,11 @@
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.util.DamageSource;
 
 public class EntityDragonPart extends Entity
 {
+
     public final IEntityMultiPart entityDragonObj;
     public final String partName;
     private static final String __OBFID = "CL_00001657";
 
     public EntityDragonPart(IEntityMultiPart parent, String partName, float base, float sizeHeight)
@@ -17,15 +18,21 @@
         this.setSize(base, sizeHeight);
         this.entityDragonObj = parent;
         this.partName = partName;
     }
 
-    protected void entityInit() {}
+    protected void entityInit()
+    {
+    }
 
-    protected void readEntityFromNBT(NBTTagCompound tagCompund) {}
+    protected void readEntityFromNBT(NBTTagCompound tagCompund)
+    {
+    }
 
-    protected void writeEntityToNBT(NBTTagCompound tagCompound) {}
+    protected void writeEntityToNBT(NBTTagCompound tagCompound)
+    {
+    }
 
     public boolean canBeCollidedWith()
     {
         return true;
     }
