--- decompiled\net\minecraft\network\login\INetHandlerLoginClient.java
+++ src\main\java\net\minecraft\network\login\INetHandlerLoginClient.java
@@ -6,10 +6,11 @@
 import net.minecraft.network.login.server.S02PacketLoginSuccess;
 import net.minecraft.network.login.server.S03PacketEnableCompression;
 
 public interface INetHandlerLoginClient extends INetHandler
 {
+
     void handleEncryptionRequest(S01PacketEncryptionRequest packetIn);
 
     void handleLoginSuccess(S02PacketLoginSuccess packetIn);
 
     void handleDisconnect(S00PacketDisconnect packetIn);
