--- decompiled\net\minecraft\network\status\client\C01PacketPing.java
+++ src\main\java\net\minecraft\network\status\client\C01PacketPing.java
@@ -1,26 +1,23 @@
 package net.minecraft.network.status.client;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.status.INetHandlerStatusServer;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class C01PacketPing implements Packet
 {
+
     private long clientTime;
     private static final String __OBFID = "CL_00001392";
 
-    public C01PacketPing() {}
-
-    @SideOnly(Side.CLIENT)
-    public C01PacketPing(long ping)
+    public C01PacketPing()
     {
-        this.clientTime = ping;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
         this.clientTime = buf.readLong();
@@ -41,8 +38,8 @@
         return this.clientTime;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerStatusServer)handler);
+        this.processPacket((INetHandlerStatusServer) handler);
     }
 }
