--- decompiled\net\minecraft\entity\monster\EntityIronGolem.java
+++ src\main\java\net\minecraft\entity\monster\EntityIronGolem.java
@@ -1,8 +1,7 @@
 package net.minecraft.entity.monster;
 
-import com.google.common.base.Predicate;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockFlower;
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
@@ -31,15 +30,16 @@
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.util.MathHelper;
 import net.minecraft.village.Village;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.base.Predicate;
+
 public class EntityIronGolem extends EntityGolem
 {
+
     private int homeCheckTimer;
     Village villageObj;
     private int attackTimer;
     private int holdRoseTick;
     private static final String __OBFID = "CL_00001652";
@@ -103,11 +103,11 @@
 
     protected void collideWithEntity(Entity p_82167_1_)
     {
         if (p_82167_1_ instanceof IMob && this.getRNG().nextInt(20) == 0)
         {
-            this.setAttackTarget((EntityLivingBase)p_82167_1_);
+            this.setGoalTarget((EntityLivingBase)p_82167_1_, org.bukkit.event.entity.EntityTargetLivingEntityEvent.TargetReason.COLLISION, true); // CraftBukkit - set reason);
         }
 
         super.collideWithEntity(p_82167_1_);
     }
 
@@ -171,39 +171,15 @@
 
         this.playSound("mob.irongolem.throw", 1.0F, 1.0F);
         return flag;
     }
 
-    @SideOnly(Side.CLIENT)
-    public void handleHealthUpdate(byte p_70103_1_)
-    {
-        if (p_70103_1_ == 4)
-        {
-            this.attackTimer = 10;
-            this.playSound("mob.irongolem.throw", 1.0F, 1.0F);
-        }
-        else if (p_70103_1_ == 11)
-        {
-            this.holdRoseTick = 400;
-        }
-        else
-        {
-            super.handleHealthUpdate(p_70103_1_);
-        }
-    }
-
     public Village getVillage()
     {
         return this.villageObj;
     }
 
-    @SideOnly(Side.CLIENT)
-    public int getAttackTimer()
-    {
-        return this.attackTimer;
-    }
-
     public void setHoldingRose(boolean p_70851_1_)
     {
         this.holdRoseTick = p_70851_1_ ? 400 : 0;
         this.worldObj.setEntityState(this, (byte)11);
     }
@@ -275,10 +251,11 @@
         super.onDeath(cause);
     }
 
     static class AINearestAttackableTargetNonCreeper extends EntityAINearestAttackableTarget
         {
+
             private static final String __OBFID = "CL_00002231";
 
             public AINearestAttackableTargetNonCreeper(final EntityCreature p_i45858_1_, Class p_i45858_2_, int p_i45858_3_, boolean p_i45858_4_, boolean p_i45858_5_, final Predicate p_i45858_6_)
             {
                 super(p_i45858_1_, p_i45858_2_, p_i45858_3_, p_i45858_4_, p_i45858_5_, p_i45858_6_);
