--- decompiled\net\minecraft\entity\monster\EntitySilverfish.java
+++ src\main\java\net\minecraft\entity\monster\EntitySilverfish.java
@@ -1,8 +1,9 @@
 package net.minecraft.entity.monster;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockSilverfish;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.EnumCreatureAttribute;
 import net.minecraft.entity.SharedMonsterAttributes;
@@ -21,10 +22,11 @@
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
 public class EntitySilverfish extends EntityMob
 {
+
     private EntitySilverfish.AISummonSilverfish field_175460_b;
     private static final String __OBFID = "CL_00001696";
 
     public EntitySilverfish(World worldIn)
     {
@@ -132,10 +134,11 @@
         return EnumCreatureAttribute.ARTHROPOD;
     }
 
     class AIHideInStone extends EntityAIWander
     {
+
         private EnumFacing field_179483_b;
         private boolean field_179484_c;
         private static final String __OBFID = "CL_00002205";
 
         public AIHideInStone()
@@ -193,20 +196,28 @@
                 BlockPos blockpos = (new BlockPos(EntitySilverfish.this.posX, EntitySilverfish.this.posY + 0.5D, EntitySilverfish.this.posZ)).offset(this.field_179483_b);
                 IBlockState iblockstate = world.getBlockState(blockpos);
 
                 if (BlockSilverfish.canContainSilverfish(iblockstate))
                 {
+                    // CraftBukkit start
+                    if (org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(EntitySilverfish.this, blockpos.getX(), blockpos.getY(), blockpos.getZ(), Blocks.monster_egg, Block.getIdFromBlock(BlockSilverfish.getBlockById(iblockstate.getBlock().getMetaFromState(iblockstate)))).isCancelled())
+                    {
+                        return;
+                    }
+
+                    // CraftBukkit end
                     world.setBlockState(blockpos, Blocks.monster_egg.getDefaultState().withProperty(BlockSilverfish.VARIANT, BlockSilverfish.EnumType.forModelBlock(iblockstate)), 3);
                     EntitySilverfish.this.spawnExplosionParticle();
                     EntitySilverfish.this.setDead();
                 }
             }
         }
     }
 
     class AISummonSilverfish extends EntityAIBase
     {
+
         private EntitySilverfish field_179464_a = EntitySilverfish.this;
         private int field_179463_b;
         private static final String __OBFID = "CL_00002204";
 
         public void func_179462_f()
@@ -241,10 +252,17 @@
                             BlockPos blockpos1 = blockpos.add(j, i, k);
                             IBlockState iblockstate = world.getBlockState(blockpos1);
 
                             if (iblockstate.getBlock() == Blocks.monster_egg)
                             {
+                                // CraftBukkit start
+                                if (org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(this.field_179464_a, blockpos1.getX(), blockpos1.getY(), blockpos1.getZ(), Blocks.air, 0).isCancelled())
+                                {
+                                    continue;
+                                }
+
+                                // CraftBukkit end
                                 if (world.getGameRules().getGameRuleBooleanValue("mobGriefing"))
                                 {
                                     world.destroyBlock(blockpos1, true);
                                 }
                                 else
