--- decompiled\net\minecraft\entity\monster\EntityMagmaCube.java
+++ src\main\java\net\minecraft\entity\monster\EntityMagmaCube.java
@@ -4,15 +4,15 @@
 import net.minecraft.init.Items;
 import net.minecraft.item.Item;
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class EntityMagmaCube extends EntitySlime
 {
+
     private static final String __OBFID = "CL_00001691";
 
     public EntityMagmaCube(World worldIn)
     {
         super(worldIn);
@@ -38,16 +38,10 @@
     public int getTotalArmorValue()
     {
         return this.getSlimeSize() * 3;
     }
 
-    @SideOnly(Side.CLIENT)
-    public int getBrightnessForRender(float p_70070_1_)
-    {
-        return 15728880;
-    }
-
     public float getBrightness(float p_70013_1_)
     {
         return 1.0F;
     }
 
@@ -111,11 +105,13 @@
     {
         this.motionY = (double)(0.22F + (float)this.getSlimeSize() * 0.05F);
         this.isAirBorne = true;
     }
 
-    public void fall(float distance, float damageMultiplier) {}
+    public void fall(float distance, float damageMultiplier)
+    {
+    }
 
     protected boolean canDamagePlayer()
     {
         return true;
     }
