--- decompiled\net\minecraft\entity\monster\EntityGiantZombie.java
+++ src\main\java\net\minecraft\entity\monster\EntityGiantZombie.java
@@ -4,10 +4,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class EntityGiantZombie extends EntityMob
 {
+
     private static final String __OBFID = "CL_00001690";
 
     public EntityGiantZombie(World worldIn)
     {
         super(worldIn);
