--- decompiled\net\minecraft\entity\monster\EntityPigZombie.java
+++ src\main\java\net\minecraft\entity\monster\EntityPigZombie.java
@@ -1,8 +1,9 @@
 package net.minecraft.entity.monster;
 
 import java.util.UUID;
+
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityCreature;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.IEntityLivingData;
 import net.minecraft.entity.SharedMonsterAttributes;
@@ -19,13 +20,14 @@
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.World;
 
 public class EntityPigZombie extends EntityZombie
 {
+
     private static final UUID field_110189_bq = UUID.fromString("49455A49-7EC5-45BA-B886-3B90B23A1718");
     private static final AttributeModifier field_110190_br = (new AttributeModifier(field_110189_bq, "Attacking speed boost", 0.05D, 0)).setSaved(false);
-    private int angerLevel;
+    public int angerLevel; // SpigotMCP -- Apply access transformation
     private int randomSoundDelay;
     private UUID field_175459_bn;
     private static final String __OBFID = "CL_00001693";
 
     public EntityPigZombie(World worldIn)
@@ -232,10 +234,11 @@
         return p_180482_2_;
     }
 
     class AIHurtByAggressor extends EntityAIHurtByTarget
     {
+
         private static final String __OBFID = "CL_00002206";
 
         public AIHurtByAggressor()
         {
             super(EntityPigZombie.this, true, new Class[0]);
@@ -252,10 +255,11 @@
         }
     }
 
     class AITargetAggressor extends EntityAINearestAttackableTarget
     {
+
         private static final String __OBFID = "CL_00002207";
 
         public AITargetAggressor()
         {
             super(EntityPigZombie.this, EntityPlayer.class, true);
