--- decompiled\net\minecraft\entity\monster\EntityEndermite.java
+++ src\main\java\net\minecraft\entity\monster\EntityEndermite.java
@@ -17,10 +17,11 @@
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.world.World;
 
 public class EntityEndermite extends EntityMob
 {
+
     private int lifetime = 0;
     private boolean playerSpawned = false;
     private static final String __OBFID = "CL_00002219";
 
     public EntityEndermite(World worldIn)
