--- decompiled\net\minecraft\entity\monster\EntitySpider.java
+++ src\main\java\net\minecraft\entity\monster\EntitySpider.java
@@ -1,8 +1,9 @@
 package net.minecraft.entity.monster;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.EnumCreatureAttribute;
 import net.minecraft.entity.IEntityLivingData;
 import net.minecraft.entity.SharedMonsterAttributes;
@@ -26,10 +27,11 @@
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.World;
 
 public class EntitySpider extends EntityMob
 {
+
     private static final String __OBFID = "CL_00001699";
 
     public EntitySpider(World worldIn)
     {
         super(worldIn);
@@ -113,11 +115,13 @@
     public boolean isOnLadder()
     {
         return this.isBesideClimbableBlock();
     }
 
-    public void setInWeb() {}
+    public void setInWeb()
+    {
+    }
 
     public EnumCreatureAttribute getCreatureAttribute()
     {
         return EnumCreatureAttribute.ARTHROPOD;
     }
@@ -155,11 +159,11 @@
         if (this.worldObj.rand.nextInt(100) == 0)
         {
             EntitySkeleton entityskeleton = new EntitySkeleton(this.worldObj);
             entityskeleton.setLocationAndAngles(this.posX, this.posY, this.posZ, this.rotationYaw, 0.0F);
             entityskeleton.func_180482_a(p_180482_1_, (IEntityLivingData)null);
-            this.worldObj.spawnEntityInWorld(entityskeleton);
+            this.worldObj.addEntity(entityskeleton, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.JOCKEY); // CraftBukkit - add SpawnReason);
             entityskeleton.mountEntity(this);
         }
 
         if (p_180482_2_1 == null)
         {
@@ -189,10 +193,11 @@
         return 0.65F;
     }
 
     class AISpiderAttack extends EntityAIAttackOnCollide
     {
+
         private static final String __OBFID = "CL_00002197";
 
         public AISpiderAttack(Class p_i45819_2_)
         {
             super(EntitySpider.this, p_i45819_2_, 1.0D, true);
@@ -219,10 +224,11 @@
         }
     }
 
     class AISpiderTarget extends EntityAINearestAttackableTarget
     {
+
         private static final String __OBFID = "CL_00002196";
 
         public AISpiderTarget(Class p_i45818_2_)
         {
             super(EntitySpider.this, p_i45818_2_, true);
@@ -235,10 +241,11 @@
         }
     }
 
     public static class GroupData implements IEntityLivingData
         {
+
             public int field_111105_a;
             private static final String __OBFID = "CL_00001700";
 
             public void func_111104_a(Random p_111104_1_)
             {
