--- decompiled\net\minecraft\entity\monster\EntityMob.java
+++ src\main\java\net\minecraft\entity\monster\EntityMob.java
@@ -1,36 +1,38 @@
 package net.minecraft.entity.monster;
 
-import com.google.common.base.Predicate;
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityCreature;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.SharedMonsterAttributes;
-import net.minecraft.entity.ai.EntityAIAvoidEntity;
 import net.minecraft.entity.ai.EntityAIBase;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.EnumSkyBlock;
 import net.minecraft.world.World;
 
+import org.bukkit.event.entity.EntityCombustByEntityEvent;
+
 public abstract class EntityMob extends EntityCreature implements IMob
 {
-    protected final EntityAIBase field_175455_a = new EntityAIAvoidEntity(this, new Predicate()
+
+    protected final EntityAIBase field_175455_a = null; // Spigot -- Removed ->
+    /* new EntityAIAvoidEntity(this, new Predicate()
     {
         private static final String __OBFID = "CL_00002208";
         public boolean func_179911_a(Entity p_179911_1_)
         {
             return p_179911_1_ instanceof EntityCreeper && ((EntityCreeper)p_179911_1_).getCreeperState() > 0;
         }
         public boolean apply(Object p_apply_1_)
         {
             return this.func_179911_a((Entity)p_apply_1_);
         }
-    }, 4.0F, 1.0D, 2.0D);
+    }, 4.0F, 1.0D, 2.0D); */
     private static final String __OBFID = "CL_00001692";
 
     public EntityMob(World worldIn)
     {
         super(worldIn);
@@ -126,11 +128,20 @@
 
             int j = EnchantmentHelper.getFireAspectModifier(this);
 
             if (j > 0)
             {
-                p_70652_1_.setFire(j * 4);
+                // CraftBukkit start - Call a combust event when somebody hits with a fire enchanted item
+                EntityCombustByEntityEvent combustEvent = new EntityCombustByEntityEvent(this.getBukkitEntity(), p_70652_1_.getBukkitEntity(), j * 4);
+                org.bukkit.Bukkit.getPluginManager().callEvent(combustEvent);
+
+                if (!combustEvent.isCancelled())
+                {
+                    p_70652_1_.setFire(combustEvent.getDuration());
+                }
+
+                // CraftBukkit end
             }
 
             this.func_174815_a(this, p_70652_1_);
         }
 
