--- decompiled\net\minecraft\entity\monster\EntityBlaze.java
+++ src\main\java\net\minecraft\entity\monster\EntityBlaze.java
@@ -16,15 +16,15 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class EntityBlaze extends EntityMob
 {
+
     private float heightOffset = 0.5F;
     private int heightOffsetUpdateTime;
     private static final String __OBFID = "CL_00001682";
 
     public EntityBlaze(World worldIn)
@@ -68,16 +68,10 @@
     protected String getDeathSound()
     {
         return "mob.blaze.death";
     }
 
-    @SideOnly(Side.CLIENT)
-    public int getBrightnessForRender(float p_70070_1_)
-    {
-        return 15728880;
-    }
-
     public float getBrightness(float p_70013_1_)
     {
         return 1.0F;
     }
 
@@ -128,11 +122,13 @@
         }
 
         super.updateAITasks();
     }
 
-    public void fall(float distance, float damageMultiplier) {}
+    public void fall(float distance, float damageMultiplier)
+    {
+    }
 
     protected Item getDropItem()
     {
         return Items.blaze_rod;
     }
@@ -181,10 +177,11 @@
         return true;
     }
 
     class AIFireballAttack extends EntityAIBase
     {
+
         private EntityBlaze field_179469_a = EntityBlaze.this;
         private int field_179467_b;
         private int field_179468_c;
         private static final String __OBFID = "CL_00002225";
 
