--- decompiled\net\minecraft\entity\monster\EntityCaveSpider.java
+++ src\main\java\net\minecraft\entity\monster\EntityCaveSpider.java
@@ -10,10 +10,11 @@
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.World;
 
 public class EntityCaveSpider extends EntitySpider
 {
+
     private static final String __OBFID = "CL_00001683";
 
     public EntityCaveSpider(World worldIn)
     {
         super(worldIn);
