--- decompiled\net\minecraft\entity\monster\EntitySlime.java
+++ src\main\java\net\minecraft\entity\monster\EntitySlime.java
@@ -23,12 +23,15 @@
 import net.minecraft.world.World;
 import net.minecraft.world.WorldType;
 import net.minecraft.world.biome.BiomeGenBase;
 import net.minecraft.world.chunk.Chunk;
 
+import org.bukkit.event.entity.SlimeSplitEvent;
+
 public class EntitySlime extends EntityLiving implements IMob
 {
+
     public float squishAmount;
     public float squishFactor;
     public float prevSquishFactor;
     private boolean field_175452_bi;
     private static final String __OBFID = "CL_00001698";
@@ -49,11 +52,11 @@
     {
         super.entityInit();
         this.dataWatcher.addObject(16, Byte.valueOf((byte)1));
     }
 
-    protected void setSlimeSize(int p_70799_1_)
+    public void setSlimeSize(int p_70799_1_)   // SpigotMCP -- Apply access transformation
     {
         this.dataWatcher.updateObject(16, Byte.valueOf((byte)p_70799_1_));
         this.setSize(0.51000005F * (float)p_70799_1_, 0.51000005F * (float)p_70799_1_);
         this.setPosition(this.posX, this.posY, this.posZ);
         this.getEntityAttribute(SharedMonsterAttributes.maxHealth).setBaseValue((double)(p_70799_1_ * p_70799_1_));
@@ -180,11 +183,26 @@
         int i = this.getSlimeSize();
 
         if (!this.worldObj.isRemote && i > 1 && this.getHealth() <= 0.0F)
         {
             int j = 2 + this.rand.nextInt(3);
+            // CraftBukkit start
+            SlimeSplitEvent event = new SlimeSplitEvent((org.bukkit.entity.Slime)this.getBukkitEntity(), j);
+            this.worldObj.getServer().getPluginManager().callEvent(event);
 
+            if (!event.isCancelled() && event.getCount() > 0)
+            {
+                j = event.getCount();
+            }
+            else
+            {
+                super.setDead();
+                return;
+            }
+
+            // CraftBukkit end
+
             for (int k = 0; k < j; ++k)
             {
                 float f = ((float)(k % 2) - 0.5F) * (float)i / 4.0F;
                 float f1 = ((float)(k / 2) - 0.5F) * (float)i / 4.0F;
                 EntitySlime entityslime = this.createInstance();
@@ -199,11 +217,11 @@
                     entityslime.enablePersistence();
                 }
 
                 entityslime.setSlimeSize(i / 2);
                 entityslime.setLocationAndAngles(this.posX + (double)f, this.posY + 0.5D, this.posZ + (double)f1, this.rand.nextFloat() * 360.0F, 0.0F);
-                this.worldObj.spawnEntityInWorld(entityslime);
+                this.worldObj.addEntity(entityslime, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason.SLIME_SPLIT); // CraftBukkit - SpawnReason
             }
         }
 
         super.setDead();
     }
@@ -336,10 +354,11 @@
         return super.func_180482_a(p_180482_1_, p_180482_2_);
     }
 
     class AISlimeAttack extends EntityAIBase
     {
+
         private EntitySlime field_179466_a = EntitySlime.this;
         private int field_179465_b;
         private static final String __OBFID = "CL_00002202";
 
         public AISlimeAttack()
@@ -372,10 +391,11 @@
         }
     }
 
     class AISlimeFaceRandom extends EntityAIBase
     {
+
         private EntitySlime field_179461_a = EntitySlime.this;
         private float field_179459_b;
         private int field_179460_c;
         private static final String __OBFID = "CL_00002198";
 
@@ -401,10 +421,11 @@
         }
     }
 
     class AISlimeFloat extends EntityAIBase
     {
+
         private EntitySlime field_179457_a = EntitySlime.this;
         private static final String __OBFID = "CL_00002201";
 
         public AISlimeFloat()
         {
@@ -428,10 +449,11 @@
         }
     }
 
     class AISlimeHop extends EntityAIBase
     {
+
         private EntitySlime field_179458_a = EntitySlime.this;
         private static final String __OBFID = "CL_00002200";
 
         public AISlimeHop()
         {
@@ -449,10 +471,11 @@
         }
     }
 
     class SlimeMoveHelper extends EntityMoveHelper
     {
+
         private float field_179922_g;
         private int field_179924_h;
         private EntitySlime field_179925_i = EntitySlime.this;
         private boolean field_179923_j;
         private static final String __OBFID = "CL_00002199";
