--- decompiled\net\minecraft\entity\monster\EntityGuardian.java
+++ src\main\java\net\minecraft\entity\monster\EntityGuardian.java
@@ -1,10 +1,10 @@
 package net.minecraft.entity.monster;
 
-import com.google.common.base.Predicate;
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.entity.ai.EntityAIBase;
@@ -36,24 +36,25 @@
 import net.minecraft.util.Vec3;
 import net.minecraft.util.WeightedRandom;
 import net.minecraft.util.WeightedRandomFishable;
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.base.Predicate;
+
 public class EntityGuardian extends EntityMob
 {
+
     private float field_175482_b;
     private float field_175484_c;
     private float field_175483_bk;
     private float field_175485_bl;
     private float field_175486_bm;
     private EntityLivingBase field_175478_bn;
     private int field_175479_bo;
     private boolean field_175480_bp;
-    private EntityAIWander wander;
+    public EntityAIWander wander; // CraftBukkit - private to public
     private static final String __OBFID = "CL_00002213";
 
     public EntityGuardian(World worldIn)
     {
         super(worldIn);
@@ -71,11 +72,12 @@
         this.targetTasks.addTask(1, new EntityAINearestAttackableTarget(this, EntityLivingBase.class, 10, true, false, new EntityGuardian.GuardianTargetSelector()));
         this.moveHelper = new EntityGuardian.GuardianMoveHelper();
         this.field_175484_c = this.field_175482_b = this.rand.nextFloat();
     }
 
-    protected void applyEntityAttributes()
+    // PAIL: Access + rename please
+    public void applyEntityAttributes() // CraftBukkit - protected to public
     {
         super.applyEntityAttributes();
         this.getEntityAttribute(SharedMonsterAttributes.attackDamage).setBaseValue(6.0D);
         this.getEntityAttribute(SharedMonsterAttributes.movementSpeed).setBaseValue(0.5D);
         this.getEntityAttribute(SharedMonsterAttributes.followRange).setBaseValue(16.0D);
@@ -158,17 +160,10 @@
             this.enablePersistence();
             this.wander.func_179479_b(400);
         }
     }
 
-    @SideOnly(Side.CLIENT)
-    public void func_175465_cm()
-    {
-        this.func_175467_a(true);
-        this.field_175486_bm = this.field_175485_bl = 1.0F;
-    }
-
     private void func_175463_b(int p_175463_1_)
     {
         this.dataWatcher.updateObject(17, Integer.valueOf(p_175463_1_));
     }
 
@@ -374,22 +369,10 @@
         }
 
         super.onLivingUpdate();
     }
 
-    @SideOnly(Side.CLIENT)
-    public float func_175471_a(float p_175471_1_)
-    {
-        return this.field_175484_c + (this.field_175482_b - this.field_175484_c) * p_175471_1_;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public float func_175469_o(float p_175469_1_)
-    {
-        return this.field_175486_bm + (this.field_175485_bl - this.field_175486_bm) * p_175469_1_;
-    }
-
     public float func_175477_p(float p_175477_1_)
     {
         return ((float)this.field_175479_bo + p_175477_1_) / (float)this.func_175464_ck();
     }
 
@@ -535,10 +518,11 @@
         }
     }
 
     class AIGuardianAttack extends EntityAIBase
     {
+
         private EntityGuardian field_179456_a = EntityGuardian.this;
         private int field_179455_b;
         private static final String __OBFID = "CL_00002211";
 
         public AIGuardianAttack()
@@ -619,10 +603,11 @@
         }
     }
 
     class GuardianMoveHelper extends EntityMoveHelper
     {
+
         private EntityGuardian field_179930_g = EntityGuardian.this;
         private static final String __OBFID = "CL_00002209";
 
         public GuardianMoveHelper()
         {
@@ -678,10 +663,11 @@
         }
     }
 
     class GuardianTargetSelector implements Predicate
     {
+
         private EntityGuardian field_179916_a = EntityGuardian.this;
         private static final String __OBFID = "CL_00002210";
 
         public boolean func_179915_a(EntityLivingBase p_179915_1_)
         {
