--- decompiled\net\minecraft\entity\monster\EntitySkeleton.java
+++ src\main\java\net\minecraft\entity\monster\EntitySkeleton.java
@@ -1,9 +1,9 @@
 package net.minecraft.entity.monster;
 
-import com.google.common.base.Predicate;
 import java.util.Calendar;
+
 import net.minecraft.block.Block;
 import net.minecraft.enchantment.Enchantment;
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityCreature;
@@ -38,12 +38,17 @@
 import net.minecraft.util.DamageSource;
 import net.minecraft.world.DifficultyInstance;
 import net.minecraft.world.World;
 import net.minecraft.world.WorldProviderHell;
 
+import org.bukkit.event.entity.EntityCombustEvent;
+
+import com.google.common.base.Predicate;
+
 public class EntitySkeleton extends EntityMob implements IRangedAttackMob
 {
+
     private EntityAIArrowAttack aiArrowAttack = new EntityAIArrowAttack(this, 1.0D, 20, 60, 15.0F);
     private EntityAIAttackOnCollide aiAttackOnCollide = new EntityAIAttackOnCollide(this, EntityPlayer.class, 1.2D, false);
     private static final String __OBFID = "CL_00001697";
 
     public EntitySkeleton(World worldIn)
@@ -160,11 +165,20 @@
                     flag = false;
                 }
 
                 if (flag)
                 {
-                    this.setFire(8);
+                    // CraftBukkit start
+                    EntityCombustEvent event = new EntityCombustEvent(this.getBukkitEntity(), 8);
+                    this.worldObj.getServer().getPluginManager().callEvent(event);
+
+                    if (!event.isCancelled())
+                    {
+                        this.setFire(event.getDuration());
+                    }
+
+                    // CraftBukkit end
                 }
             }
         }
 
         if (this.worldObj.isRemote && this.getSkeletonType() == 1)
@@ -326,15 +340,39 @@
             entityarrow.setKnockbackStrength(j);
         }
 
         if (EnchantmentHelper.getEnchantmentLevel(Enchantment.flame.effectId, this.getHeldItem()) > 0 || this.getSkeletonType() == 1)
         {
-            entityarrow.setFire(100);
+            // CraftBukkit start - call EntityCombustEvent
+            EntityCombustEvent event = new EntityCombustEvent(entityarrow.getBukkitEntity(), 100);
+            this.worldObj.getServer().getPluginManager().callEvent(event);
+
+            if (!event.isCancelled())
+            {
+                entityarrow.setFire(event.getDuration());
+            }
+
+            // CraftBukkit end
         }
 
+        // CraftBukkit start
+        org.bukkit.event.entity.EntityShootBowEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callEntityShootBowEvent(this, this.getHeldItem(), entityarrow, 0.8F);
+
+        if (event.isCancelled())
+        {
+            event.getProjectile().remove();
+            return;
+        }
+
+        if (event.getProjectile() == entityarrow.getBukkitEntity())
+        {
+            worldObj.spawnEntityInWorld(entityarrow);
+        }
+
+        // CraftBukkit end
         this.playSound("random.bow", 1.0F, 1.0F / (this.getRNG().nextFloat() * 0.4F + 0.8F));
-        this.worldObj.spawnEntityInWorld(entityarrow);
+        // this.worldObj.spawnEntityInWorld(entityarrow); // CraftBukkit - moved up
     }
 
     public int getSkeletonType()
     {
         return this.dataWatcher.getWatchableObjectByte(13);
