--- decompiled\net\minecraft\entity\monster\EntityGhast.java
+++ src\main\java\net\minecraft\entity\monster\EntityGhast.java
@@ -1,8 +1,9 @@
 package net.minecraft.entity.monster;
 
 import java.util.Random;
+
 import net.minecraft.entity.EntityFlying;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.entity.ai.EntityAIBase;
 import net.minecraft.entity.ai.EntityAIFindEntityNearestPlayer;
@@ -18,15 +19,15 @@
 import net.minecraft.util.DamageSource;
 import net.minecraft.util.MathHelper;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.EnumDifficulty;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class EntityGhast extends EntityFlying implements IMob
 {
+
     private int explosionStrength = 1;
     private static final String __OBFID = "CL_00001689";
 
     public EntityGhast(World worldIn)
     {
@@ -39,16 +40,10 @@
         this.tasks.addTask(7, new EntityGhast.AILookAround());
         this.tasks.addTask(7, new EntityGhast.AIFireballAttack());
         this.targetTasks.addTask(1, new EntityAIFindEntityNearestPlayer(this));
     }
 
-    @SideOnly(Side.CLIENT)
-    public boolean func_110182_bF()
-    {
-        return this.dataWatcher.getWatchableObjectByte(16) != 0;
-    }
-
     public void func_175454_a(boolean p_175454_1_)
     {
         this.dataWatcher.updateObject(16, Byte.valueOf((byte)(p_175454_1_ ? 1 : 0)));
     }
 
@@ -172,10 +167,11 @@
         return 2.6F;
     }
 
     class AIFireballAttack extends EntityAIBase
     {
+
         private EntityGhast field_179470_b = EntityGhast.this;
         public int field_179471_a;
         private static final String __OBFID = "CL_00002215";
 
         public boolean shouldExecute()
@@ -215,11 +211,12 @@
                     double d2 = entitylivingbase.posX - (this.field_179470_b.posX + vec3.xCoord * d1);
                     double d3 = entitylivingbase.getEntityBoundingBox().minY + (double)(entitylivingbase.height / 2.0F) - (0.5D + this.field_179470_b.posY + (double)(this.field_179470_b.height / 2.0F));
                     double d4 = entitylivingbase.posZ - (this.field_179470_b.posZ + vec3.zCoord * d1);
                     world.playAuxSFXAtEntity((EntityPlayer)null, 1008, new BlockPos(this.field_179470_b), 0);
                     EntityLargeFireball entitylargefireball = new EntityLargeFireball(world, this.field_179470_b, d2, d3, d4);
-                    entitylargefireball.explosionPower = this.field_179470_b.func_175453_cd();
+                    // CraftBukkit - set bukkitYield when setting explosionpower
+                    entitylargefireball.bukkitYield = entitylargefireball.explosionPower = this.field_179470_b.func_175453_cd();
                     entitylargefireball.posX = this.field_179470_b.posX + vec3.xCoord * d1;
                     entitylargefireball.posY = this.field_179470_b.posY + (double)(this.field_179470_b.height / 2.0F) + 0.5D;
                     entitylargefireball.posZ = this.field_179470_b.posZ + vec3.zCoord * d1;
                     world.spawnEntityInWorld(entitylargefireball);
                     this.field_179471_a = -40;
@@ -234,10 +231,11 @@
         }
     }
 
     class AILookAround extends EntityAIBase
     {
+
         private EntityGhast field_179472_a = EntityGhast.this;
         private static final String __OBFID = "CL_00002217";
 
         public AILookAround()
         {
@@ -270,10 +268,11 @@
         }
     }
 
     class AIRandomFly extends EntityAIBase
     {
+
         private EntityGhast field_179454_a = EntityGhast.this;
         private static final String __OBFID = "CL_00002214";
 
         public AIRandomFly()
         {
@@ -313,10 +312,11 @@
         }
     }
 
     class GhastMoveHelper extends EntityMoveHelper
     {
+
         private EntityGhast field_179927_g = EntityGhast.this;
         private int field_179928_h;
         private static final String __OBFID = "CL_00002216";
 
         public GhastMoveHelper()
