--- decompiled\net\minecraft\entity\monster\IMob.java
+++ src\main\java\net\minecraft\entity\monster\IMob.java
@@ -1,13 +1,15 @@
 package net.minecraft.entity.monster;
 
-import com.google.common.base.Predicate;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.passive.IAnimals;
 
+import com.google.common.base.Predicate;
+
 public interface IMob extends IAnimals
 {
+
     Predicate mobSelector = new Predicate()
     {
         private static final String __OBFID = "CL_00001688";
         public boolean func_179983_a(Entity p_179983_1_)
         {
