--- decompiled\net\minecraft\enchantment\EnchantmentData.java
+++ src\main\java\net\minecraft\enchantment\EnchantmentData.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.util.WeightedRandom;
 
 public class EnchantmentData extends WeightedRandom.Item
 {
+
     public final Enchantment enchantmentobj;
     public final int enchantmentLevel;
     private static final String __OBFID = "CL_00000115";
 
     public EnchantmentData(Enchantment enchantmentObj, int enchLevel)
