--- decompiled\net\minecraft\enchantment\EnchantmentDamage.java
+++ src\main\java\net\minecraft\enchantment\EnchantmentDamage.java
@@ -9,10 +9,11 @@
 import net.minecraft.potion.PotionEffect;
 import net.minecraft.util.ResourceLocation;
 
 public class EnchantmentDamage extends Enchantment
 {
+
     private static final String[] protectionName = new String[] {"all", "undead", "arthropods"};
     private static final int[] baseEnchantability = new int[] {1, 5, 5};
     private static final int[] levelEnchantability = new int[] {11, 8, 8};
     private static final int[] thresholdEnchantability = new int[] {20, 20, 20};
     public final int damageType;
