--- decompiled\net\minecraft\enchantment\EnchantmentFireAspect.java
+++ src\main\java\net\minecraft\enchantment\EnchantmentFireAspect.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.util.ResourceLocation;
 
 public class EnchantmentFireAspect extends Enchantment
 {
+
     private static final String __OBFID = "CL_00000116";
 
     protected EnchantmentFireAspect(int enchID, ResourceLocation enchName, int enchWeight)
     {
         super(enchID, enchName, enchWeight, EnumEnchantmentType.WEAPON);
