--- decompiled\net\minecraft\enchantment\EnchantmentWaterWorker.java
+++ src\main\java\net\minecraft\enchantment\EnchantmentWaterWorker.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.util.ResourceLocation;
 
 public class EnchantmentWaterWorker extends Enchantment
 {
+
     private static final String __OBFID = "CL_00000124";
 
     public EnchantmentWaterWorker(int p_i45761_1_, ResourceLocation p_i45761_2_, int p_i45761_3_)
     {
         super(p_i45761_1_, p_i45761_2_, p_i45761_3_, EnumEnchantmentType.ARMOR_HEAD);
