--- decompiled\net\minecraft\enchantment\EnchantmentDigging.java
+++ src\main\java\net\minecraft\enchantment\EnchantmentDigging.java
@@ -4,10 +4,11 @@
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.ResourceLocation;
 
 public class EnchantmentDigging extends Enchantment
 {
+
     private static final String __OBFID = "CL_00000104";
 
     protected EnchantmentDigging(int enchID, ResourceLocation enchName, int enchWeight)
     {
         super(enchID, enchName, enchWeight, EnumEnchantmentType.DIGGER);
