--- decompiled\net\minecraft\enchantment\EnchantmentProtection.java
+++ src\main\java\net\minecraft\enchantment\EnchantmentProtection.java
@@ -5,10 +5,11 @@
 import net.minecraft.util.MathHelper;
 import net.minecraft.util.ResourceLocation;
 
 public class EnchantmentProtection extends Enchantment
 {
+
     private static final String[] protectionName = new String[] {"all", "fire", "fall", "explosion", "projectile"};
     private static final int[] baseEnchantability = new int[] {1, 10, 5, 5, 3};
     private static final int[] levelEnchantability = new int[] {11, 8, 6, 8, 6};
     private static final int[] thresholdEnchantability = new int[] {20, 12, 10, 12, 15};
     public final int protectionType;
