--- decompiled\net\minecraft\enchantment\EnchantmentWaterWorker.java
+++ src\main\java\net\minecraft\enchantment\EnchantmentWaterWorker.java
@@ -1,29 +1,25 @@
 package net.minecraft.enchantment;
 
 import net.minecraft.util.ResourceLocation;
 
-public class EnchantmentWaterWorker extends Enchantment
-{
+public class EnchantmentWaterWorker extends Enchantment {
+
     private static final String __OBFID = "CL_00000124";
 
-    public EnchantmentWaterWorker(int p_i45761_1_, ResourceLocation p_i45761_2_, int p_i45761_3_)
-    {
+    public EnchantmentWaterWorker(int p_i45761_1_, ResourceLocation p_i45761_2_, int p_i45761_3_) {
         super(p_i45761_1_, p_i45761_2_, p_i45761_3_, EnumEnchantmentType.ARMOR_HEAD);
         this.setName("waterWorker");
     }
 
-    public int getMinEnchantability(int enchantmentLevel)
-    {
+    public int getMinEnchantability(int enchantmentLevel) {
         return 1;
     }
 
-    public int getMaxEnchantability(int enchantmentLevel)
-    {
+    public int getMaxEnchantability(int enchantmentLevel) {
         return this.getMinEnchantability(enchantmentLevel) + 40;
     }
 
-    public int getMaxLevel()
-    {
+    public int getMaxLevel() {
         return 1;
     }
 }
