--- decompiled\net\minecraft\enchantment\EnchantmentArrowFire.java
+++ src\main\java\net\minecraft\enchantment\EnchantmentArrowFire.java
@@ -1,29 +1,25 @@
 package net.minecraft.enchantment;
 
 import net.minecraft.util.ResourceLocation;
 
-public class EnchantmentArrowFire extends Enchantment
-{
+public class EnchantmentArrowFire extends Enchantment {
+
     private static final String __OBFID = "CL_00000099";
 
-    public EnchantmentArrowFire(int enchID, ResourceLocation enchName, int enchWeight)
-    {
+    public EnchantmentArrowFire(int enchID, ResourceLocation enchName, int enchWeight) {
         super(enchID, enchName, enchWeight, EnumEnchantmentType.BOW);
         this.setName("arrowFire");
     }
 
-    public int getMinEnchantability(int enchantmentLevel)
-    {
+    public int getMinEnchantability(int enchantmentLevel) {
         return 20;
     }
 
-    public int getMaxEnchantability(int enchantmentLevel)
-    {
+    public int getMaxEnchantability(int enchantmentLevel) {
         return 50;
     }
 
-    public int getMaxLevel()
-    {
+    public int getMaxLevel() {
         return 1;
     }
 }
