--- decompiled\net\minecraft\enchantment\EnumEnchantmentType.java
+++ src\main\java\net\minecraft\enchantment\EnumEnchantmentType.java
@@ -39,11 +39,11 @@
             {
                 return true;
             }
             else
             {
-                ItemArmor itemarmor = (ItemArmor)p_77557_1_;
+                ItemArmor itemarmor = (ItemArmor) p_77557_1_;
                 return itemarmor.armorType == 0 ? this == ARMOR_HEAD : (itemarmor.armorType == 2 ? this == ARMOR_LEGS : (itemarmor.armorType == 1 ? this == ARMOR_TORSO : (itemarmor.armorType == 3 ? this == ARMOR_FEET : false)));
             }
         }
         else
         {
