--- decompiled\net\minecraft\block\BlockWallSign.java
+++ src\main\java\net\minecraft\block\BlockWallSign.java
@@ -9,10 +9,11 @@
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
 
 public class BlockWallSign extends BlockSign
 {
+
     public static final PropertyDirection FACING = PropertyDirection.create("facing", EnumFacing.Plane.HORIZONTAL);
     private static final String __OBFID = "CL_00002047";
 
     public BlockWallSign()
     {
@@ -80,10 +81,11 @@
         return new BlockState(this, new IProperty[] {FACING});
     }
 
     static final class SwitchEnumFacing
         {
+
             static final int[] FACING_LOOKUP = new int[EnumFacing.values().length];
             private static final String __OBFID = "CL_00002046";
 
             static
             {
