--- decompiled\net\minecraft\block\BlockSign.java
+++ src\main\java\net\minecraft\block\BlockSign.java
@@ -1,8 +1,9 @@
 package net.minecraft.block;
 
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Items;
 import net.minecraft.item.Item;
@@ -11,15 +12,15 @@
 import net.minecraft.util.AxisAlignedBB;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class BlockSign extends BlockContainer
 {
+
     private static final String __OBFID = "CL_00000306";
 
     protected BlockSign()
     {
         super(Material.wood);
@@ -31,17 +32,10 @@
     public AxisAlignedBB getCollisionBoundingBox(World worldIn, BlockPos pos, IBlockState state)
     {
         return null;
     }
 
-    @SideOnly(Side.CLIENT)
-    public AxisAlignedBB getSelectedBoundingBox(World worldIn, BlockPos pos)
-    {
-        this.setBlockBoundsBasedOnState(worldIn, pos);
-        return super.getSelectedBoundingBox(worldIn, pos);
-    }
-
     public boolean isFullCube()
     {
         return false;
     }
 
@@ -75,12 +69,6 @@
         {
             TileEntity tileentity = worldIn.getTileEntity(pos);
             return tileentity instanceof TileEntitySign ? ((TileEntitySign)tileentity).func_174882_b(playerIn) : false;
         }
     }
-
-    @SideOnly(Side.CLIENT)
-    public Item getItem(World worldIn, BlockPos pos)
-    {
-        return Items.sign;
-    }
 }
