--- decompiled\net\minecraft\block\BlockPressurePlate.java
+++ src\main\java\net\minecraft\block\BlockPressurePlate.java
@@ -1,22 +1,27 @@
 package net.minecraft.block;
 
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyBool;
 import net.minecraft.block.state.BlockState;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.AxisAlignedBB;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
+import org.bukkit.event.entity.EntityInteractEvent;
+
 public class BlockPressurePlate extends BlockBasePressurePlate
 {
+
     public static final PropertyBool POWERED = PropertyBool.create("powered");
     private final BlockPressurePlate.Sensitivity sensitivity;
     private static final String __OBFID = "CL_00000289";
 
     protected BlockPressurePlate(Material p_i45693_1_, BlockPressurePlate.Sensitivity p_i45693_2_)
@@ -59,10 +64,35 @@
 
             while (iterator.hasNext())
             {
                 Entity entity = (Entity)iterator.next();
 
+                // CraftBukkit start - Call interact event when turning on a pressure plate
+                if (this.getRedstoneStrength(worldIn.getBlockState(pos)) == 0)
+                {
+                    org.bukkit.World bworld = worldIn.getWorld();
+                    org.bukkit.plugin.PluginManager manager = worldIn.getServer().getPluginManager();
+                    org.bukkit.event.Cancellable cancellable;
+
+                    if (entity instanceof EntityPlayer)
+                    {
+                        cancellable = org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerInteractEvent((EntityPlayer)entity, org.bukkit.event.block.Action.PHYSICAL, pos, null, null);
+                    }
+                    else
+                    {
+                        cancellable = new EntityInteractEvent(entity.getBukkitEntity(), bworld.getBlockAt(pos.getX(), pos.getY(), pos.getZ()));
+                        manager.callEvent((EntityInteractEvent)cancellable);
+                    }
+
+                    // We only want to block turning the plate on if all events are cancelled
+                    if (cancellable.isCancelled())
+                    {
+                        continue;
+                    }
+                }
+
+                // CraftBukkit end1
                 if (!entity.doesEntityNotTriggerPressurePlate())
                 {
                     return 15;
                 }
             }
@@ -94,10 +124,11 @@
         private static final String __OBFID = "CL_00000290";
     }
 
     static final class SwitchSensitivity
         {
+
             static final int[] SENSITIVITY_ARRAY = new int[BlockPressurePlate.Sensitivity.values().length];
             private static final String __OBFID = "CL_00002078";
 
             static
             {
