--- decompiled\net\minecraft\block\BlockRedstoneComparator.java
+++ src\main\java\net\minecraft\block\BlockRedstoneComparator.java
@@ -1,10 +1,10 @@
 package net.minecraft.block;
 
-import com.google.common.base.Predicate;
 import java.util.List;
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyBool;
 import net.minecraft.block.properties.PropertyEnum;
 import net.minecraft.block.state.BlockState;
@@ -22,15 +22,16 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.IStringSerializable;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.base.Predicate;
+
 public class BlockRedstoneComparator extends BlockRedstoneDiode implements ITileEntityProvider
 {
+
     public static final PropertyBool POWERED = PropertyBool.create("powered");
     public static final PropertyEnum MODE = PropertyEnum.create("mode", BlockRedstoneComparator.Mode.class);
     private static final String __OBFID = "CL_00000220";
 
     public BlockRedstoneComparator(boolean powered)
@@ -43,16 +44,10 @@
     public Item getItemDropped(IBlockState state, Random rand, int fortune)
     {
         return Items.comparator;
     }
 
-    @SideOnly(Side.CLIENT)
-    public Item getItem(World worldIn, BlockPos pos)
-    {
-        return Items.comparator;
-    }
-
     protected int getDelay(IBlockState state)
     {
         return 2;
     }
 
