--- decompiled\net\minecraft\block\BlockWorkbench.java
+++ src\main\java\net\minecraft\block\BlockWorkbench.java
@@ -15,10 +15,11 @@
 import net.minecraft.world.IInteractionObject;
 import net.minecraft.world.World;
 
 public class BlockWorkbench extends Block
 {
+
     private static final String __OBFID = "CL_00000221";
 
     protected BlockWorkbench()
     {
         super(Material.wood);
@@ -38,10 +39,11 @@
         }
     }
 
     public static class InterfaceCraftingTable implements IInteractionObject
         {
+
             private final World world;
             private final BlockPos position;
             private static final String __OBFID = "CL_00002127";
 
             public InterfaceCraftingTable(World worldIn, BlockPos pos)
