--- decompiled\net\minecraft\block\BlockNewLog.java
+++ src\main\java\net\minecraft\block\BlockNewLog.java
@@ -1,21 +1,19 @@
 package net.minecraft.block;
 
-import com.google.common.base.Predicate;
-import java.util.List;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyEnum;
 import net.minecraft.block.state.BlockState;
 import net.minecraft.block.state.IBlockState;
-import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.base.Predicate;
+
 public class BlockNewLog extends BlockLog
 {
+
     public static final PropertyEnum VARIANT = PropertyEnum.create("variant", BlockPlanks.EnumType.class, new Predicate()
     {
         private static final String __OBFID = "CL_00002089";
         public boolean apply(BlockPlanks.EnumType type)
         {
@@ -31,17 +29,10 @@
     public BlockNewLog()
     {
         this.setDefaultState(this.blockState.getBaseState().withProperty(VARIANT, BlockPlanks.EnumType.ACACIA).withProperty(LOG_AXIS, BlockLog.EnumAxis.Y));
     }
 
-    @SideOnly(Side.CLIENT)
-    public void getSubBlocks(Item itemIn, CreativeTabs tab, List list)
-    {
-        list.add(new ItemStack(itemIn, 1, BlockPlanks.EnumType.ACACIA.getMetadata() - 4));
-        list.add(new ItemStack(itemIn, 1, BlockPlanks.EnumType.DARK_OAK.getMetadata() - 4));
-    }
-
     public IBlockState getStateFromMeta(int meta)
     {
         IBlockState iblockstate = this.getDefaultState().withProperty(VARIANT, BlockPlanks.EnumType.byMetadata((meta & 3) + 4));
 
         switch (meta & 12)
@@ -97,10 +88,11 @@
         return ((BlockPlanks.EnumType)state.getValue(VARIANT)).getMetadata() - 4;
     }
 
     static final class SwitchEnumAxis
         {
+
             static final int[] AXIS_LOOKUP = new int[BlockLog.EnumAxis.values().length];
             private static final String __OBFID = "CL_00002088";
 
             static
             {
