--- decompiled\net\minecraft\block\BlockMobSpawner.java
+++ src\main\java\net\minecraft\block\BlockMobSpawner.java
@@ -1,67 +1,59 @@
 package net.minecraft.block;
 
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.item.Item;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.tileentity.TileEntityMobSpawner;
 import net.minecraft.util.BlockPos;
-import net.minecraft.util.EnumWorldBlockLayer;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class BlockMobSpawner extends BlockContainer
-{
+
+public class BlockMobSpawner extends BlockContainer {
+
     private static final String __OBFID = "CL_00000269";
 
-    protected BlockMobSpawner()
-    {
+    protected BlockMobSpawner() {
         super(Material.rock);
     }
 
-    public TileEntity createNewTileEntity(World worldIn, int meta)
-    {
+    public TileEntity createNewTileEntity(World worldIn, int meta) {
         return new TileEntityMobSpawner();
     }
 
-    public Item getItemDropped(IBlockState state, Random rand, int fortune)
-    {
+    public Item getItemDropped(IBlockState state, Random rand, int fortune) {
         return null;
     }
 
-    public int quantityDropped(Random random)
-    {
+    public int quantityDropped(Random random) {
         return 0;
     }
 
-    public void dropBlockAsItemWithChance(World worldIn, BlockPos pos, IBlockState state, float chance, int fortune)
-    {
+    public void dropBlockAsItemWithChance(World worldIn, BlockPos pos, IBlockState state, float chance, int fortune) {
         super.dropBlockAsItemWithChance(worldIn, pos, state, chance, fortune);
-        int j = 15 + worldIn.rand.nextInt(15) + worldIn.rand.nextInt(15);
-        this.dropXpOnBlockBreak(worldIn, pos, j);
-    }
+        // CraftBukkit start - Delegate to getExpDrop
+        /*
+        int j = 15 + world.random.nextInt(15) + world.random.nextInt(15);
 
-    public boolean isOpaqueCube()
-    {
-        return false;
+        this.dropExperience(world, blockposition, j);
+        */
     }
 
-    public int getRenderType()
-    {
-        return 3;
+    @Override
+    public int getExpDrop(World world, IBlockState iblockdata, int enchantmentLevel) {
+        int j = 15 + world.rand.nextInt(15) + world.rand.nextInt(15);
+
+        return j;
+        // CraftBukkit end
     }
 
-    @SideOnly(Side.CLIENT)
-    public EnumWorldBlockLayer getBlockLayer()
-    {
-        return EnumWorldBlockLayer.CUTOUT;
+    public boolean isOpaqueCube() {
+        return false;
     }
 
-    @SideOnly(Side.CLIENT)
-    public Item getItem(World worldIn, BlockPos pos)
-    {
-        return null;
+    public int getRenderType() {
+        return 3;
     }
 }
