--- decompiled\net\minecraft\block\BlockCake.java
+++ src\main\java\net\minecraft\block\BlockCake.java
@@ -6,11 +6,13 @@
 import net.minecraft.block.properties.PropertyInteger;
 import net.minecraft.block.state.BlockState;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Items;
+import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.item.Item;
+import net.minecraft.network.play.server.S06PacketUpdateHealth;
 import net.minecraft.util.AxisAlignedBB;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.EnumWorldBlockLayer;
 import net.minecraft.world.IBlockAccess;
@@ -82,11 +84,22 @@
 
     private void eatCake(World worldIn, BlockPos pos, IBlockState state, EntityPlayer player)
     {
         if (player.canEat(false))
         {
-            player.getFoodStats().addStats(2, 0.1F);
+            // CraftBukkit start
+            // player.getFoodStats().addStats(2, 0.1F);
+            int oldFoodLevel = player.getFoodStats().foodLevel;
+            org.bukkit.event.entity.FoodLevelChangeEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callFoodLevelChangeEvent(player, 2 + oldFoodLevel);
+
+            if (!event.isCancelled())
+            {
+                player.getFoodStats().addStats(event.getFoodLevel() - oldFoodLevel, 0.1F);
+            }
+
+            ((EntityPlayerMP)player).playerNetServerHandler.sendPacket(new S06PacketUpdateHealth(((EntityPlayerMP)player).getBukkitEntity().getScaledHealth(), player.getFoodStats().foodLevel, player.getFoodStats().foodSaturationLevel));
+            // CraftBukkit end
             int i = ((Integer)state.getValue(BITES)).intValue();
 
             if (i < 6)
             {
                 worldIn.setBlockState(pos, state.withProperty(BITES, Integer.valueOf(i + 1)), 3);
