--- decompiled\net\minecraft\block\BlockTripWireHook.java
+++ src\main\java\net\minecraft\block\BlockTripWireHook.java
@@ -1,10 +1,10 @@
 package net.minecraft.block;
 
-import com.google.common.base.Objects;
 import java.util.Iterator;
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyBool;
 import net.minecraft.block.properties.PropertyDirection;
 import net.minecraft.block.state.BlockState;
@@ -14,18 +14,20 @@
 import net.minecraft.init.Blocks;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.AxisAlignedBB;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
-import net.minecraft.util.EnumWorldBlockLayer;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.event.block.BlockRedstoneEvent;
+
+import com.google.common.base.Objects;
+
 public class BlockTripWireHook extends Block
 {
+
     public static final PropertyDirection FACING = PropertyDirection.create("facing", EnumFacing.Plane.HORIZONTAL);
     public static final PropertyBool POWERED = PropertyBool.create("powered");
     public static final PropertyBool ATTACHED = PropertyBool.create("attached");
     public static final PropertyBool SUSPENDED = PropertyBool.create("suspended");
     private static final String __OBFID = "CL_00000329";
@@ -181,10 +183,21 @@
             worldIn.setBlockState(blockpos1, iblockstate3.withProperty(FACING, enumfacing1), 3);
             this.func_176262_b(worldIn, blockpos1, enumfacing1);
             this.func_180694_a(worldIn, blockpos1, flag5, flag6, flag2, flag3);
         }
 
+        // CraftBukkit start
+        org.bukkit.block.Block block = worldIn.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ());
+        BlockRedstoneEvent eventRedstone = new BlockRedstoneEvent(block, 15, 0);
+        worldIn.getServer().getPluginManager().callEvent(eventRedstone);
+
+        if (eventRedstone.getNewCurrent() > 0)
+        {
+            return;
+        }
+
+        // CraftBukkit end
         this.func_180694_a(worldIn, pos, flag5, flag6, flag2, flag3);
 
         if (!p_176260_4_)
         {
             worldIn.setBlockState(pos, iblockstate3.withProperty(FACING, enumfacing), 3);
@@ -208,11 +221,13 @@
                 }
             }
         }
     }
 
-    public void randomTick(World worldIn, BlockPos pos, IBlockState state, Random random) {}
+    public void randomTick(World worldIn, BlockPos pos, IBlockState state, Random random)
+    {
+    }
 
     public void updateTick(World worldIn, BlockPos pos, IBlockState state, Random rand)
     {
         this.func_176260_a(worldIn, pos, state, false, true, -1, (IBlockState)null);
     }
@@ -309,16 +324,10 @@
     public boolean canProvidePower()
     {
         return true;
     }
 
-    @SideOnly(Side.CLIENT)
-    public EnumWorldBlockLayer getBlockLayer()
-    {
-        return EnumWorldBlockLayer.CUTOUT_MIPPED;
-    }
-
     public IBlockState getStateFromMeta(int meta)
     {
         return this.getDefaultState().withProperty(FACING, EnumFacing.getHorizontal(meta & 3)).withProperty(POWERED, Boolean.valueOf((meta & 8) > 0)).withProperty(ATTACHED, Boolean.valueOf((meta & 4) > 0));
     }
 
@@ -345,10 +354,11 @@
         return new BlockState(this, new IProperty[] {FACING, POWERED, ATTACHED, SUSPENDED});
     }
 
     static final class SwitchEnumFacing
         {
+
             static final int[] FACING_LOOKUP = new int[EnumFacing.values().length];
             private static final String __OBFID = "CL_00002050";
 
             static
             {
