--- decompiled\net\minecraft\block\BlockRail.java
+++ src\main\java\net\minecraft\block\BlockRail.java
@@ -7,10 +7,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class BlockRail extends BlockRailBase
 {
+
     public static final PropertyEnum SHAPE = PropertyEnum.create("shape", BlockRailBase.EnumRailDirection.class);
     private static final String __OBFID = "CL_00000293";
 
     protected BlockRail()
     {
