--- decompiled\net\minecraft\block\BlockEndPortalFrame.java
+++ src\main\java\net\minecraft\block\BlockEndPortalFrame.java
@@ -1,9 +1,10 @@
 package net.minecraft.block;
 
 import java.util.List;
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyBool;
 import net.minecraft.block.properties.PropertyDirection;
 import net.minecraft.block.state.BlockState;
@@ -16,10 +17,11 @@
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
 public class BlockEndPortalFrame extends Block
 {
+
     public static final PropertyDirection FACING = PropertyDirection.create("facing", EnumFacing.Plane.HORIZONTAL);
     public static final PropertyBool EYE = PropertyBool.create("eye");
     private static final String __OBFID = "CL_00000237";
 
     public BlockEndPortalFrame()
