--- decompiled\net\minecraft\block\BlockStandingSign.java
+++ src\main\java\net\minecraft\block\BlockStandingSign.java
@@ -5,41 +5,35 @@
 import net.minecraft.block.state.BlockState;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
-public class BlockStandingSign extends BlockSign
-{
+public class BlockStandingSign extends BlockSign {
+
     public static final PropertyInteger ROTATION = PropertyInteger.create("rotation", 0, 15);
     private static final String __OBFID = "CL_00002060";
 
-    public BlockStandingSign()
-    {
+    public BlockStandingSign() {
         this.setDefaultState(this.blockState.getBaseState().withProperty(ROTATION, Integer.valueOf(0)));
     }
 
-    public void onNeighborBlockChange(World worldIn, BlockPos pos, IBlockState state, Block neighborBlock)
-    {
-        if (!worldIn.getBlockState(pos.down()).getBlock().getMaterial().isSolid())
-        {
+    public void onNeighborBlockChange(World worldIn, BlockPos pos, IBlockState state, Block neighborBlock) {
+        if (!worldIn.getBlockState(pos.down()).getBlock().getMaterial().isSolid()) {
             this.dropBlockAsItem(worldIn, pos, state, 0);
             worldIn.setBlockToAir(pos);
         }
 
         super.onNeighborBlockChange(worldIn, pos, state, neighborBlock);
     }
 
-    public IBlockState getStateFromMeta(int meta)
-    {
+    public IBlockState getStateFromMeta(int meta) {
         return this.getDefaultState().withProperty(ROTATION, Integer.valueOf(meta));
     }
 
-    public int getMetaFromState(IBlockState state)
-    {
-        return ((Integer)state.getValue(ROTATION)).intValue();
+    public int getMetaFromState(IBlockState state) {
+        return ((Integer) state.getValue(ROTATION)).intValue();
     }
 
-    protected BlockState createBlockState()
-    {
-        return new BlockState(this, new IProperty[] {ROTATION});
+    protected BlockState createBlockState() {
+        return new BlockState(this, new IProperty[]{ROTATION});
     }
 }
