--- decompiled\net\minecraft\block\BlockTNT.java
+++ src\main\java\net\minecraft\block\BlockTNT.java
@@ -9,19 +9,21 @@
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.item.EntityTNTPrimed;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.projectile.EntityArrow;
+import net.minecraft.init.Blocks;
 import net.minecraft.init.Items;
 import net.minecraft.item.Item;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.Explosion;
 import net.minecraft.world.World;
 
 public class BlockTNT extends Block
 {
+
     public static final PropertyBool EXPLODE = PropertyBool.create("explode");
     private static final String __OBFID = "CL_00000324";
 
     public BlockTNT()
     {
@@ -111,10 +113,17 @@
         {
             EntityArrow entityarrow = (EntityArrow)entityIn;
 
             if (entityarrow.isBurning())
             {
+                // CraftBukkit start
+                if (org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(entityarrow, pos.getX(), pos.getY(), pos.getZ(), Blocks.air, 0).isCancelled())
+                {
+                    return;
+                }
+
+                // CraftBukkit end
                 this.explode(worldIn, pos, worldIn.getBlockState(pos).withProperty(EXPLODE, Boolean.valueOf(true)), entityarrow.shootingEntity instanceof EntityLivingBase ? (EntityLivingBase)entityarrow.shootingEntity : null);
                 worldIn.setBlockToAir(pos);
             }
         }
     }
