--- decompiled\net\minecraft\block\BlockSnowBlock.java
+++ src\main\java\net\minecraft\block\BlockSnowBlock.java
@@ -1,8 +1,9 @@
 package net.minecraft.block;
 
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.init.Items;
 import net.minecraft.item.Item;
@@ -10,10 +11,11 @@
 import net.minecraft.world.EnumSkyBlock;
 import net.minecraft.world.World;
 
 public class BlockSnowBlock extends Block
 {
+
     private static final String __OBFID = "CL_00000308";
 
     protected BlockSnowBlock()
     {
         super(Material.craftedSnow);
