--- decompiled\net\minecraft\block\Block.java
+++ src\main\java\net\minecraft\block\Block.java
@@ -34,10 +34,12 @@
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.spigotmc.AsyncCatcher; // Spigot
+
 public class Block
 {
     private static final ResourceLocation AIR_ID = new ResourceLocation("air");
     public static final RegistryNamespacedDefaultedByKey blockRegistry = new RegistryNamespacedDefaultedByKey(AIR_ID);
     public static final ObjectIntIdentityMap BLOCK_STATE_IDS = new ObjectIntIdentityMap();
@@ -434,13 +436,19 @@
     public int tickRate(World worldIn)
     {
         return 10;
     }
 
-    public void onBlockAdded(World worldIn, BlockPos pos, IBlockState state) {}
+    public void onBlockAdded(World worldIn, BlockPos pos, IBlockState state)
+    {
+        AsyncCatcher.catchOp("block onPlace"); // Spigot
+    }
 
-    public void breakBlock(World worldIn, BlockPos pos, IBlockState state) {}
+    public void breakBlock(World worldIn, BlockPos pos, IBlockState state)
+    {
+        AsyncCatcher.catchOp("block remove"); // Spigot
+    }
 
     public int quantityDropped(Random random)
     {
         return 1;
     }
@@ -467,11 +475,11 @@
         {
             int j = this.quantityDroppedWithBonus(fortune, worldIn.rand);
 
             for (int k = 0; k < j; ++k)
             {
-                if (worldIn.rand.nextFloat() <= chance)
+                if (worldIn.rand.nextFloat() < chance)// CraftBukkit - <= to < to allow for plugins to completely disable block drops from explosions
                 {
                     Item item = this.getItemDropped(state, worldIn.rand, fortune);
 
                     if (item != null)
                     {
@@ -786,11 +794,11 @@
             int i = EnchantmentHelper.getFortuneModifier(player);
             this.dropBlockAsItem(worldIn, pos, state, i);
         }
     }
 
-    protected boolean canSilkHarvest()
+    public boolean canSilkHarvest() // SpigotMCP -- Make public as bukkit assumes in the same package
     {
         return this.isFullCube() && !this.isBlockContainer;
     }
 
     protected ItemStack createStackedBlock(IBlockState state)
@@ -1234,10 +1242,32 @@
     private static void registerBlock(int id, String textualID, Block block_)
     {
         registerBlock(id, new ResourceLocation(textualID), block_);
     }
 
+    // CraftBukkit start
+    public int getExpDrop(World world, IBlockState data, int enchantmentLevel)
+    {
+        return 0;
+    }
+    // CraftBukkit end
+
+    // Spigot start
+    public static float range(float min, float value, float max)
+    {
+        if (value < min)
+        {
+            return min;
+        }
+        if (value > max)
+        {
+            return max;
+        }
+        return value;
+    }
+    // Spigot end
+
     @SideOnly(Side.CLIENT)
     public static enum EnumOffsetType
     {
         NONE,
         XZ,
