--- decompiled\net\minecraft\block\BlockNote.java
+++ src\main\java\net\minecraft\block\BlockNote.java
@@ -1,9 +1,9 @@
 package net.minecraft.block;
 
-import com.google.common.collect.Lists;
 import java.util.List;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.tileentity.TileEntity;
@@ -11,12 +11,15 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.world.World;
 
+import com.google.common.collect.Lists;
+
 public class BlockNote extends BlockContainer
 {
+
     private static final List INSTRUMENTS = Lists.newArrayList(new String[] {"harp", "bd", "snare", "hat", "bassattack"});
     private static final String __OBFID = "CL_00000278";
 
     public BlockNote()
     {
