--- decompiled\net\minecraft\block\BlockMycelium.java
+++ src\main\java\net\minecraft\block\BlockMycelium.java
@@ -1,25 +1,28 @@
 package net.minecraft.block;
 
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyBool;
 import net.minecraft.block.state.BlockState;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.init.Blocks;
 import net.minecraft.item.Item;
 import net.minecraft.util.BlockPos;
-import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.craftbukkit.util.CraftMagicNumbers;
+import org.bukkit.event.block.BlockFadeEvent;
+import org.bukkit.event.block.BlockSpreadEvent;
+
 public class BlockMycelium extends Block
 {
+
     public static final PropertyBool SNOWY = PropertyBool.create("snowy");
     private static final String __OBFID = "CL_00000273";
 
     protected BlockMycelium()
     {
@@ -39,11 +42,24 @@
     {
         if (!worldIn.isRemote)
         {
             if (worldIn.getLightFromNeighbors(pos.up()) < 4 && worldIn.getBlockState(pos.up()).getBlock().getLightOpacity() > 2)
             {
-                worldIn.setBlockState(pos, Blocks.dirt.getDefaultState().withProperty(BlockDirt.VARIANT, BlockDirt.DirtType.DIRT));
+                // CraftBukkit start
+                // worldIn.setBlockState(pos, Blocks.dirt.getDefaultState().withProperty(BlockDirt.VARIANT, BlockDirt.DirtType.DIRT));
+                org.bukkit.World bworld = worldIn.getWorld();
+                org.bukkit.block.BlockState blockState = bworld.getBlockAt(pos.getX(), pos.getY(), pos.getZ()).getState();
+                blockState.setType(CraftMagicNumbers.getMaterial(Blocks.dirt));
+                BlockFadeEvent event = new BlockFadeEvent(blockState.getBlock(), blockState);
+                worldIn.getServer().getPluginManager().callEvent(event);
+
+                if (!event.isCancelled())
+                {
+                    blockState.update(true);
+                }
+
+                // CraftBukkit end
             }
             else
             {
                 if (worldIn.getLightFromNeighbors(pos.up()) >= 9)
                 {
@@ -53,29 +69,31 @@
                         IBlockState iblockstate1 = worldIn.getBlockState(blockpos1);
                         Block block = worldIn.getBlockState(blockpos1.up()).getBlock();
 
                         if (iblockstate1.getBlock() == Blocks.dirt && iblockstate1.getValue(BlockDirt.VARIANT) == BlockDirt.DirtType.DIRT && worldIn.getLightFromNeighbors(blockpos1.up()) >= 4 && block.getLightOpacity() <= 2)
                         {
-                            worldIn.setBlockState(blockpos1, this.getDefaultState());
+                            // CraftBukkit start
+                            // worldIn.setBlockState(blockpos1, this.getDefaultState());
+                            org.bukkit.World bworld = worldIn.getWorld();
+                            org.bukkit.block.BlockState blockState = bworld.getBlockAt(blockpos1.getX(), blockpos1.getY(), blockpos1.getZ()).getState();
+                            blockState.setType(CraftMagicNumbers.getMaterial(this));
+                            BlockSpreadEvent event = new BlockSpreadEvent(blockState.getBlock(), bworld.getBlockAt(pos.getX(), pos.getY(), pos.getZ()), blockState);
+                            worldIn.getServer().getPluginManager().callEvent(event);
+
+                            if (!event.isCancelled())
+                            {
+                                blockState.update(true);
+                            }
+
+                            // CraftBukkit end
                         }
                     }
                 }
             }
         }
     }
 
-    @SideOnly(Side.CLIENT)
-    public void randomDisplayTick(World worldIn, BlockPos pos, IBlockState state, Random rand)
-    {
-        super.randomDisplayTick(worldIn, pos, state, rand);
-
-        if (rand.nextInt(10) == 0)
-        {
-            worldIn.spawnParticle(EnumParticleTypes.TOWN_AURA, (double)((float)pos.getX() + rand.nextFloat()), (double)((float)pos.getY() + 1.1F), (double)((float)pos.getZ() + rand.nextFloat()), 0.0D, 0.0D, 0.0D, new int[0]);
-        }
-    }
-
     public Item getItemDropped(IBlockState state, Random rand, int fortune)
     {
         return Blocks.dirt.getItemDropped(Blocks.dirt.getDefaultState().withProperty(BlockDirt.VARIANT, BlockDirt.DirtType.DIRT), rand, fortune);
     }
 
