--- decompiled\net\minecraft\block\BlockDragonEgg.java
+++ src\main\java\net\minecraft\block\BlockDragonEgg.java
@@ -1,23 +1,23 @@
 package net.minecraft.block;
 
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.item.EntityFallingBlock;
 import net.minecraft.entity.player.EntityPlayer;
-import net.minecraft.item.Item;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.EnumParticleTypes;
-import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.event.block.BlockFromToEvent;
+
 public class BlockDragonEgg extends Block
 {
+
     private static final String __OBFID = "CL_00000232";
 
     public BlockDragonEgg()
     {
         super(Material.dragonEgg);
@@ -88,10 +88,24 @@
             {
                 BlockPos blockpos1 = pos.add(worldIn.rand.nextInt(16) - worldIn.rand.nextInt(16), worldIn.rand.nextInt(8) - worldIn.rand.nextInt(8), worldIn.rand.nextInt(16) - worldIn.rand.nextInt(16));
 
                 if (worldIn.getBlockState(blockpos1).getBlock().blockMaterial == Material.air)
                 {
+                    // CraftBukkit start
+                    org.bukkit.block.Block from = worldIn.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ());
+                    org.bukkit.block.Block to = worldIn.getWorld().getBlockAt(blockpos1.getX(), blockpos1.getY(), blockpos1.getZ());
+                    BlockFromToEvent event = new BlockFromToEvent(from, to);
+                    org.bukkit.Bukkit.getPluginManager().callEvent(event);
+
+                    if (event.isCancelled())
+                    {
+                        return;
+                    }
+
+                    blockpos1 = new BlockPos(event.getToBlock().getX(), event.getToBlock().getY(), event.getToBlock().getZ());
+
+                    // CraftBukkit end
                     if (worldIn.isRemote)
                     {
                         for (int j = 0; j < 128; ++j)
                         {
                             double d0 = worldIn.rand.nextDouble();
@@ -128,18 +142,6 @@
 
     public boolean isFullCube()
     {
         return false;
     }
-
-    @SideOnly(Side.CLIENT)
-    public boolean shouldSideBeRendered(IBlockAccess worldIn, BlockPos pos, EnumFacing side)
-    {
-        return true;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public Item getItem(World worldIn, BlockPos pos)
-    {
-        return null;
-    }
 }
