--- decompiled\net\minecraft\block\BlockCompressed.java
+++ src\main\java\net\minecraft\block\BlockCompressed.java
@@ -5,10 +5,11 @@
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 
 public class BlockCompressed extends Block
 {
+
     private final MapColor mapColor;
     private static final String __OBFID = "CL_00000268";
 
     public BlockCompressed(MapColor mapColorIn)
     {
