--- decompiled\net\minecraft\block\BlockRailBase.java
+++ src\main\java\net\minecraft\block\BlockRailBase.java
@@ -1,29 +1,29 @@
 package net.minecraft.block;
 
-import com.google.common.collect.Lists;
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.AxisAlignedBB;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
-import net.minecraft.util.EnumWorldBlockLayer;
 import net.minecraft.util.IStringSerializable;
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.collect.Lists;
+
 public abstract class BlockRailBase extends Block
 {
+
     protected final boolean isPowered;
     private static final String __OBFID = "CL_00000195";
 
     public static boolean isRailBlock(World worldIn, BlockPos pos)
     {
@@ -137,11 +137,13 @@
                 this.onNeighborChangedInternal(worldIn, pos, state, neighborBlock);
             }
         }
     }
 
-    protected void onNeighborChangedInternal(World worldIn, BlockPos pos, IBlockState state, Block neighborBlock) {}
+    protected void onNeighborChangedInternal(World worldIn, BlockPos pos, IBlockState state, Block neighborBlock)
+    {
+    }
 
     protected IBlockState func_176564_a(World worldIn, BlockPos p_176564_2_, IBlockState p_176564_3_, boolean p_176564_4_)
     {
         return worldIn.isRemote ? p_176564_3_ : (new BlockRailBase.Rail(worldIn, p_176564_2_, p_176564_3_)).func_180364_a(worldIn.isBlockPowered(p_176564_2_), p_176564_4_).getBlockState();
     }
@@ -149,16 +151,10 @@
     public int getMobilityFlag()
     {
         return 0;
     }
 
-    @SideOnly(Side.CLIENT)
-    public EnumWorldBlockLayer getBlockLayer()
-    {
-        return EnumWorldBlockLayer.CUTOUT;
-    }
-
     public void breakBlock(World worldIn, BlockPos pos, IBlockState state)
     {
         super.breakBlock(worldIn, pos, state);
 
         if (((BlockRailBase.EnumRailDirection)state.getValue(this.getShapeProperty())).isAscending())
@@ -242,10 +238,11 @@
         }
     }
 
     public class Rail
     {
+
         private final World world;
         private final BlockPos pos;
         private final BlockRailBase block;
         private IBlockState state;
         private final boolean isPowered;
@@ -667,10 +664,11 @@
         }
     }
 
     static final class SwitchEnumRailDirection
         {
+
             static final int[] DIRECTION_LOOKUP = new int[BlockRailBase.EnumRailDirection.values().length];
             private static final String __OBFID = "CL_00002138";
 
             static
             {
