--- decompiled\net\minecraft\block\BlockHardenedClay.java
+++ src\main\java\net\minecraft\block\BlockHardenedClay.java
@@ -5,10 +5,11 @@
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 
 public class BlockHardenedClay extends Block
 {
+
     private static final String __OBFID = "CL_00000255";
 
     public BlockHardenedClay()
     {
         super(Material.rock);
