--- decompiled\net\minecraft\block\BlockStaticLiquid.java
+++ src\main\java\net\minecraft\block\BlockStaticLiquid.java
@@ -1,17 +1,21 @@
 package net.minecraft.block;
 
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+
 public class BlockStaticLiquid extends BlockLiquid
 {
+
     private static final String __OBFID = "CL_00000315";
 
     protected BlockStaticLiquid(Material materialIn)
     {
         super(materialIn);
@@ -57,10 +61,20 @@
 
                         if (block.blockMaterial == Material.air)
                         {
                             if (this.isSurroundingBlockFlammable(worldIn, blockpos1))
                             {
+                                // CraftBukkit start - Prevent lava putting something on fire
+                                if (worldIn.getBlockState(blockpos1) != Blocks.fire)
+                                {
+                                    if (CraftEventFactory.callBlockIgniteEvent(worldIn, blockpos1.getX(), blockpos1.getY(), blockpos1.getZ(), pos.getX(), pos.getY(), pos.getZ()).isCancelled())
+                                    {
+                                        continue;
+                                    }
+                                }
+
+                                // CraftBukkit end
                                 worldIn.setBlockState(blockpos1, Blocks.fire.getDefaultState());
                                 return;
                             }
                         }
                         else if (block.blockMaterial.blocksMovement())
@@ -75,10 +89,22 @@
                     {
                         BlockPos blockpos2 = pos.add(rand.nextInt(3) - 1, 0, rand.nextInt(3) - 1);
 
                         if (worldIn.isAirBlock(blockpos2.up()) && this.getCanBlockBurn(worldIn, blockpos2))
                         {
+                            // CraftBukkit start - Prevent lava putting something on fire
+                            BlockPos up = blockpos2.up();
+
+                            if (worldIn.getBlockState(up) != Blocks.fire)
+                            {
+                                if (CraftEventFactory.callBlockIgniteEvent(worldIn, up.getX(), up.getY(), up.getZ(), pos.getX(), pos.getY(), pos.getZ()).isCancelled())
+                                {
+                                    continue;
+                                }
+                            }
+
+                            // CraftBukkit end
                             worldIn.setBlockState(blockpos2.up(), Blocks.fire.getDefaultState());
                         }
                     }
                 }
             }
