--- decompiled\net\minecraft\block\BlockFlower.java
+++ src\main\java\net\minecraft\block\BlockFlower.java
@@ -1,27 +1,24 @@
 package net.minecraft.block;
 
-import com.google.common.base.Predicate;
-import com.google.common.collect.Collections2;
-import com.google.common.collect.Lists;
 import java.util.Collection;
-import java.util.List;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyEnum;
 import net.minecraft.block.state.BlockState;
 import net.minecraft.block.state.IBlockState;
-import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.init.Blocks;
-import net.minecraft.item.Item;
-import net.minecraft.item.ItemStack;
 import net.minecraft.util.IStringSerializable;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.base.Predicate;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.Lists;
+
 public abstract class BlockFlower extends BlockBush
 {
+
     protected PropertyEnum type;
     private static final String __OBFID = "CL_00000246";
 
     protected BlockFlower()
     {
@@ -32,23 +29,10 @@
     public int damageDropped(IBlockState state)
     {
         return ((BlockFlower.EnumFlowerType)state.getValue(this.getTypeProperty())).getMeta();
     }
 
-    @SideOnly(Side.CLIENT)
-    public void getSubBlocks(Item itemIn, CreativeTabs tab, List list)
-    {
-        BlockFlower.EnumFlowerType[] aenumflowertype = BlockFlower.EnumFlowerType.getTypes(this.getBlockType());
-        int i = aenumflowertype.length;
-
-        for (int j = 0; j < i; ++j)
-        {
-            BlockFlower.EnumFlowerType enumflowertype = aenumflowertype[j];
-            list.add(new ItemStack(itemIn, 1, enumflowertype.getMeta()));
-        }
-    }
-
     public IBlockState getStateFromMeta(int meta)
     {
         return this.getDefaultState().withProperty(this.getTypeProperty(), BlockFlower.EnumFlowerType.getType(this.getBlockType(), meta));
     }
 
@@ -83,20 +67,14 @@
     protected BlockState createBlockState()
     {
         return new BlockState(this, new IProperty[] {this.getTypeProperty()});
     }
 
-    @SideOnly(Side.CLIENT)
-    public Block.EnumOffsetType getOffsetType()
-    {
-        return Block.EnumOffsetType.XZ;
-    }
-
     public static enum EnumFlowerColor
     {
         YELLOW,
-        RED;
+        RED, CraftMagicNumbers;
 
         private static final String __OBFID = "CL_00002117";
 
         public BlockFlower getBlock()
         {
@@ -157,16 +135,10 @@
             }
 
             return aenumflowertype[meta];
         }
 
-        @SideOnly(Side.CLIENT)
-        public static BlockFlower.EnumFlowerType[] getTypes(BlockFlower.EnumFlowerColor p_176966_0_)
-        {
-            return TYPES_FOR_BLOCK[p_176966_0_.ordinal()];
-        }
-
         public String toString()
         {
             return this.name;
         }
 
