--- decompiled\net\minecraft\block\BlockBush.java
+++ src\main\java\net\minecraft\block\BlockBush.java
@@ -10,10 +10,12 @@
 import net.minecraft.util.EnumWorldBlockLayer;
 import net.minecraft.world.World;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
+import org.bukkit.event.block.BlockPhysicsEvent; // CraftBukkit
+
 public class BlockBush extends Block
 {
     private static final String __OBFID = "CL_00000208";
 
     protected BlockBush(Material materialIn)
@@ -53,10 +55,20 @@
 
     protected void checkAndDropBlock(World worldIn, BlockPos pos, IBlockState state)
     {
         if (!this.canBlockStay(worldIn, pos, state))
         {
+            // CraftBukkit Start
+            org.bukkit.block.Block block = worldIn.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ());
+            BlockPhysicsEvent event = new BlockPhysicsEvent(block, block.getTypeId());
+            worldIn.getServer().getPluginManager().callEvent(event);
+
+            if (event.isCancelled())
+            {
+                return;
+            }
+            // CraftBukkit end
             this.dropBlockAsItem(worldIn, pos, state, 0);
             worldIn.setBlockState(pos, Blocks.air.getDefaultState(), 3);
         }
     }
 
