--- decompiled\net\minecraft\block\BlockPressurePlateWeighted.java
+++ src\main\java\net\minecraft\block\BlockPressurePlateWeighted.java
@@ -4,16 +4,20 @@
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyInteger;
 import net.minecraft.block.state.BlockState;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
+import org.bukkit.event.entity.EntityInteractEvent;
+
 public class BlockPressurePlateWeighted extends BlockBasePressurePlate
 {
+
     public static final PropertyInteger POWER = PropertyInteger.create("power", 0, 15);
     private final int field_150068_a;
     private static final String __OBFID = "CL_00000334";
 
     protected BlockPressurePlateWeighted(String p_i45436_1_, Material p_i45436_2_, int p_i45436_3_)
@@ -23,12 +27,40 @@
         this.field_150068_a = p_i45436_3_;
     }
 
     protected int computeRedstoneStrength(World worldIn, BlockPos pos)
     {
-        int i = Math.min(worldIn.getEntitiesWithinAABB(Entity.class, this.getSensitiveAABB(pos)).size(), this.field_150068_a);
+        // CraftBukkit start
+        //int i = Math.min(world.a(Entity.class, this.a(blockposition)).size(), this.b);
+        int i = 0;
+        java.util.Iterator iterator = worldIn.getEntitiesWithinAABB(net.minecraft.entity.Entity.class, this.getSensitiveAABB(pos)).iterator();
 
+        while (iterator.hasNext())
+        {
+            Entity entity = (Entity)iterator.next();
+            org.bukkit.event.Cancellable cancellable;
+
+            if (entity instanceof EntityPlayer)
+            {
+                cancellable = org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerInteractEvent((EntityPlayer)entity, org.bukkit.event.block.Action.PHYSICAL, pos, null, null);
+            }
+            else
+            {
+                cancellable = new EntityInteractEvent(entity.getBukkitEntity(), worldIn.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ()));
+                worldIn.getServer().getPluginManager().callEvent((EntityInteractEvent)cancellable);
+            }
+
+            // We only want to block turning the plate on if all events are cancelled
+            if (!cancellable.isCancelled())
+            {
+                i++;
+            }
+        }
+
+        i = Math.min(i, this.field_150068_a);
+
+        // CraftBukkit end
         if (i > 0)
         {
             float f = (float)Math.min(this.field_150068_a, i) / (float)this.field_150068_a;
             return MathHelper.ceiling_float_int(f * 15.0F);
         }
