--- decompiled\net\minecraft\block\BlockDoublePlant.java
+++ src\main\java\net\minecraft\block\BlockDoublePlant.java
@@ -1,15 +1,14 @@
 package net.minecraft.block;
 
-import java.util.List;
 import java.util.Random;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.IProperty;
 import net.minecraft.block.properties.PropertyEnum;
 import net.minecraft.block.state.BlockState;
 import net.minecraft.block.state.IBlockState;
-import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Blocks;
 import net.minecraft.init.Items;
 import net.minecraft.item.Item;
@@ -18,16 +17,15 @@
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.IStringSerializable;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.World;
-import net.minecraft.world.biome.BiomeColorHelper;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class BlockDoublePlant extends BlockBush implements IGrowable
 {
+
     public static final PropertyEnum VARIANT = PropertyEnum.create("variant", BlockDoublePlant.EnumPlantType.class);
     public static final PropertyEnum HALF = PropertyEnum.create("half", BlockDoublePlant.EnumBlockHalf.class);
     private static final String __OBFID = "CL_00000231";
 
     public BlockDoublePlant()
@@ -89,22 +87,27 @@
             Object object = flag ? this : worldIn.getBlockState(blockpos1).getBlock();
             Object object1 = flag ? worldIn.getBlockState(blockpos2).getBlock() : this;
 
             if (object == this)
             {
-                worldIn.setBlockState(blockpos1, Blocks.air.getDefaultState(), 3);
+                worldIn.setBlockState(blockpos1, Blocks.air.getDefaultState(), 2); // CraftBukkit
             }
 
             if (object1 == this)
             {
-                worldIn.setBlockState(blockpos2, Blocks.air.getDefaultState(), 3);
+                worldIn.setBlockState(blockpos2, Blocks.air.getDefaultState(), 2); // CraftBukkit
 
                 if (!flag)
                 {
                     this.dropBlockAsItem(worldIn, blockpos2, state, 0);
                 }
             }
+
+            // CraftBukkit start - delay updates
+            worldIn.notifyNeighborsRespectDebug(blockpos1, this);
+            worldIn.notifyNeighborsRespectDebug(blockpos2, this);
+            // CraftBukkit end
         }
     }
 
     public boolean canBlockStay(World worldIn, BlockPos pos, IBlockState state)
     {
@@ -135,17 +138,10 @@
     public int damageDropped(IBlockState state)
     {
         return state.getValue(HALF) != BlockDoublePlant.EnumBlockHalf.UPPER && state.getValue(VARIANT) != BlockDoublePlant.EnumPlantType.GRASS ? ((BlockDoublePlant.EnumPlantType)state.getValue(VARIANT)).getMeta() : 0;
     }
 
-    @SideOnly(Side.CLIENT)
-    public int colorMultiplier(IBlockAccess worldIn, BlockPos pos, int renderPass)
-    {
-        BlockDoublePlant.EnumPlantType enumplanttype = this.getVariant(worldIn, pos);
-        return enumplanttype != BlockDoublePlant.EnumPlantType.GRASS && enumplanttype != BlockDoublePlant.EnumPlantType.FERN ? 16777215 : BiomeColorHelper.getGrassColorAtPos(worldIn, pos);
-    }
-
     public void placeAt(World worldIn, BlockPos lowerPos, BlockDoublePlant.EnumPlantType variant, int flags)
     {
         worldIn.setBlockState(lowerPos, this.getDefaultState().withProperty(HALF, BlockDoublePlant.EnumBlockHalf.LOWER).withProperty(VARIANT, variant), flags);
         worldIn.setBlockState(lowerPos.up(), this.getDefaultState().withProperty(HALF, BlockDoublePlant.EnumBlockHalf.UPPER), flags);
     }
@@ -224,23 +220,10 @@
             spawnAsEntity(worldIn, pos, new ItemStack(Blocks.tallgrass, 2, i));
             return true;
         }
     }
 
-    @SideOnly(Side.CLIENT)
-    public void getSubBlocks(Item itemIn, CreativeTabs tab, List list)
-    {
-        BlockDoublePlant.EnumPlantType[] aenumplanttype = BlockDoublePlant.EnumPlantType.values();
-        int i = aenumplanttype.length;
-
-        for (int j = 0; j < i; ++j)
-        {
-            BlockDoublePlant.EnumPlantType enumplanttype = aenumplanttype[j];
-            list.add(new ItemStack(itemIn, 1, enumplanttype.getMeta()));
-        }
-    }
-
     public int getDamageValue(World worldIn, BlockPos pos)
     {
         return this.getVariant(worldIn, pos).getMeta();
     }
 
@@ -288,16 +271,10 @@
     protected BlockState createBlockState()
     {
         return new BlockState(this, new IProperty[] {HALF, VARIANT});
     }
 
-    @SideOnly(Side.CLIENT)
-    public Block.EnumOffsetType getOffsetType()
-    {
-        return Block.EnumOffsetType.XZ;
-    }
-
     static enum EnumBlockHalf implements IStringSerializable
     {
         UPPER,
         LOWER;
 
