--- decompiled\net\minecraft\block\BlockContainer.java
+++ src\main\java\net\minecraft\block\BlockContainer.java
@@ -6,10 +6,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public abstract class BlockContainer extends Block implements ITileEntityProvider
 {
+
     private static final String __OBFID = "CL_00000193";
 
     protected BlockContainer(Material materialIn)
     {
         super(materialIn);
