--- decompiled\net\minecraft\block\BlockSourceImpl.java
+++ src\main\java\net\minecraft\block\BlockSourceImpl.java
@@ -6,10 +6,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class BlockSourceImpl implements IBlockSource
 {
+
     private final World worldObj;
     private final BlockPos pos;
     private static final String __OBFID = "CL_00001194";
 
     public BlockSourceImpl(World worldIn, BlockPos p_i46023_2_)
