--- decompiled\net\minecraft\block\BlockLog.java
+++ src\main\java\net\minecraft\block\BlockLog.java
@@ -1,8 +1,9 @@
 package net.minecraft.block;
 
 import java.util.Iterator;
+
 import net.minecraft.block.material.Material;
 import net.minecraft.block.properties.PropertyEnum;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.EntityLivingBase;
@@ -11,10 +12,11 @@
 import net.minecraft.util.IStringSerializable;
 import net.minecraft.world.World;
 
 public abstract class BlockLog extends BlockRotatedPillar
 {
+
     public static final PropertyEnum LOG_AXIS = PropertyEnum.create("axis", BlockLog.EnumAxis.class);
     private static final String __OBFID = "CL_00000266";
 
     public BlockLog()
     {
@@ -92,10 +94,11 @@
         }
     }
 
     static final class SwitchAxis
         {
+
             static final int[] AXIS_LOOKUP = new int[EnumFacing.Axis.values().length];
             private static final String __OBFID = "CL_00002101";
 
             static
             {
