--- decompiled\net\minecraft\block\BlockNetherBrick.java
+++ src\main\java\net\minecraft\block\BlockNetherBrick.java
@@ -5,10 +5,11 @@
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 
 public class BlockNetherBrick extends Block
 {
+
     private static final String __OBFID = "CL_00002091";
 
     public BlockNetherBrick()
     {
         super(Material.rock);
