--- decompiled\net\minecraft\potion\PotionHealthBoost.java
+++ src\main\java\net\minecraft\potion\PotionHealthBoost.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.ai.attributes.BaseAttributeMap;
 import net.minecraft.util.ResourceLocation;
 
 public class PotionHealthBoost extends Potion
 {
+
     private static final String __OBFID = "CL_00001526";
 
     public PotionHealthBoost(int p_i45899_1_, ResourceLocation p_i45899_2_, boolean p_i45899_3_, int p_i45899_4_)
     {
         super(p_i45899_1_, p_i45899_2_, p_i45899_3_, p_i45899_4_);
