--- decompiled\net\minecraft\potion\PotionAbsorption.java
+++ src\main\java\net\minecraft\potion\PotionAbsorption.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.ai.attributes.BaseAttributeMap;
 import net.minecraft.util.ResourceLocation;
 
 public class PotionAbsorption extends Potion
 {
+
     private static final String __OBFID = "CL_00001524";
 
     protected PotionAbsorption(int p_i45901_1_, ResourceLocation p_i45901_2_, boolean p_i45901_3_, int p_i45901_4_)
     {
         super(p_i45901_1_, p_i45901_2_, p_i45901_3_, p_i45901_4_);
