--- decompiled\net\minecraft\potion\PotionEffect.java
+++ src\main\java\net\minecraft\potion\PotionEffect.java
@@ -1,24 +1,22 @@
 package net.minecraft.potion;
 
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.nbt.NBTTagCompound;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class PotionEffect
 {
+
     private static final Logger LOGGER = LogManager.getLogger();
     private int potionID;
     private int duration;
     private int amplifier;
     private boolean isSplashPotion;
     private boolean isAmbient;
-    @SideOnly(Side.CLIENT)
-    private boolean isPotionDurationMax;
     private boolean showParticles;
     private static final String __OBFID = "CL_00001529";
 
     public PotionEffect(int id, int effectDuration)
     {
@@ -210,18 +208,6 @@
         else
         {
             return null;
         }
     }
-
-    @SideOnly(Side.CLIENT)
-    public void setPotionDurationMax(boolean maxDuration)
-    {
-        this.isPotionDurationMax = maxDuration;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public boolean getIsPotionDurationMax()
-    {
-        return this.isPotionDurationMax;
-    }
 }
