--- decompiled\net\minecraft\command\server\CommandPublishLocalServer.java
+++ src\main\java\net\minecraft\command\server\CommandPublishLocalServer.java
@@ -6,10 +6,11 @@
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.world.WorldSettings;
 
 public class CommandPublishLocalServer extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000799";
 
     public String getName()
     {
         return "publish";
