--- decompiled\net\minecraft\command\server\CommandBanIp.java
+++ src\main\java\net\minecraft\command\server\CommandBanIp.java
@@ -3,10 +3,11 @@
 import java.util.Date;
 import java.util.Iterator;
 import java.util.List;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.PlayerNotFoundException;
 import net.minecraft.command.WrongUsageException;
@@ -16,10 +17,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.IChatComponent;
 
 public class CommandBanIp extends CommandBase
 {
+
     public static final Pattern field_147211_a = Pattern.compile("^([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.([01]?\\d\\d?|2[0-4]\\d|25[0-5])$");
     private static final String __OBFID = "CL_00000139";
 
     public String getName()
     {
