--- decompiled\net\minecraft\command\server\CommandWhitelist.java
+++ src\main\java\net\minecraft\command\server\CommandWhitelist.java
@@ -1,20 +1,23 @@
 package net.minecraft.command.server;
 
-import com.mojang.authlib.GameProfile;
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.ChatComponentText;
 import net.minecraft.util.ChatComponentTranslation;
 
+import com.mojang.authlib.GameProfile;
+
 public class CommandWhitelist extends CommandBase
 {
+
     private static final String __OBFID = "CL_00001186";
 
     public String getName()
     {
         return "whitelist";
