--- decompiled\net\minecraft\command\server\CommandAchievement.java
+++ src\main\java\net\minecraft\command\server\CommandAchievement.java
@@ -1,13 +1,11 @@
 package net.minecraft.command.server;
 
-import com.google.common.base.Predicate;
-import com.google.common.collect.Iterators;
-import com.google.common.collect.Lists;
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.entity.player.EntityPlayerMP;
@@ -16,12 +14,17 @@
 import net.minecraft.stats.AchievementList;
 import net.minecraft.stats.StatBase;
 import net.minecraft.stats.StatList;
 import net.minecraft.util.BlockPos;
 
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Lists;
+
 public class CommandAchievement extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000113";
 
     public String getName()
     {
         return "achievement";
