--- decompiled\net\minecraft\command\server\CommandListBans.java
+++ src\main\java\net\minecraft\command\server\CommandListBans.java
@@ -1,18 +1,20 @@
 package net.minecraft.command.server;
 
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.ChatComponentText;
 import net.minecraft.util.ChatComponentTranslation;
 
 public class CommandListBans extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000596";
 
     public String getName()
     {
         return "banlist";
@@ -47,8 +49,8 @@
         }
     }
 
     public List addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos)
     {
-        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"players", "ips"}): null;
+        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"players", "ips"}) : null;
     }
 }
