--- decompiled\net\minecraft\command\server\CommandSaveAll.java
+++ src\main\java\net\minecraft\command\server\CommandSaveAll.java
@@ -9,10 +9,11 @@
 import net.minecraft.world.MinecraftException;
 import net.minecraft.world.WorldServer;
 
 public class CommandSaveAll extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000826";
 
     public String getName()
     {
         return "save-all";
