--- decompiled\net\minecraft\command\server\CommandListPlayers.java
+++ src\main\java\net\minecraft\command\server\CommandListPlayers.java
@@ -8,10 +8,11 @@
 import net.minecraft.util.ChatComponentText;
 import net.minecraft.util.ChatComponentTranslation;
 
 public class CommandListPlayers extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000615";
 
     public String getName()
     {
         return "list";
