--- decompiled\net\minecraft\command\server\CommandEmote.java
+++ src\main\java\net\minecraft\command\server\CommandEmote.java
@@ -1,8 +1,9 @@
 package net.minecraft.command.server;
 
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.entity.player.EntityPlayer;
@@ -11,10 +12,11 @@
 import net.minecraft.util.ChatComponentTranslation;
 import net.minecraft.util.IChatComponent;
 
 public class CommandEmote extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000351";
 
     public String getName()
     {
         return "me";
