--- decompiled\net\minecraft\command\server\CommandStop.java
+++ src\main\java\net\minecraft\command\server\CommandStop.java
@@ -5,10 +5,11 @@
 import net.minecraft.command.ICommandSender;
 import net.minecraft.server.MinecraftServer;
 
 public class CommandStop extends CommandBase
 {
+
     private static final String __OBFID = "CL_00001132";
 
     public String getName()
     {
         return "stop";
