--- decompiled\net\minecraft\command\server\CommandSummon.java
+++ src\main\java\net\minecraft\command\server\CommandSummon.java
@@ -1,8 +1,9 @@
 package net.minecraft.command.server;
 
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.entity.Entity;
@@ -18,10 +19,11 @@
 import net.minecraft.util.Vec3;
 import net.minecraft.world.World;
 
 public class CommandSummon extends CommandBase
 {
+
     private static final String __OBFID = "CL_00001158";
 
     public String getName()
     {
         return "summon";
