--- decompiled\net\minecraft\command\server\CommandTeleport.java
+++ src\main\java\net\minecraft\command\server\CommandTeleport.java
@@ -1,11 +1,13 @@
 package net.minecraft.command.server;
 
 import java.util.EnumSet;
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommand;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.network.play.server.S08PacketPlayerPosLook;
@@ -13,10 +15,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
 
 public class CommandTeleport extends CommandBase
 {
+
     private static final String __OBFID = "CL_00001180";
 
     public String getName()
     {
         return "tp";
@@ -30,11 +33,11 @@
     public String getCommandUsage(ICommandSender sender)
     {
         return "commands.tp.usage";
     }
 
-    public void execute(ICommandSender sender, String[] args) throws CommandException
+    public void execute(ICommandSender sender, String[] args) throws CommandException // CraftBukkit - add throws // SpigotMCP NOTE -- Added by MCP
     {
         if (args.length < 1)
         {
             throw new WrongUsageException("commands.tp.usage", new Object[0]);
         }
@@ -142,28 +145,16 @@
             }
             else
             {
                 Entity entity = func_175768_b(sender, args[args.length - 1]);
 
-                if (entity.worldObj != ((Entity)object).worldObj)
+                // CraftBukkit Start
+                // Use Bukkit teleport method in all cases. It has cross dimensional handling, events
+                if (((Entity)object).getBukkitEntity().teleport(entity.getBukkitEntity(), org.bukkit.event.player.PlayerTeleportEvent.TeleportCause.COMMAND))
                 {
-                    throw new CommandException("commands.tp.notSameDimension", new Object[0]);
-                }
-                else
-                {
-                    ((Entity)object).mountEntity((Entity)null);
-
-                    if (object instanceof EntityPlayerMP)
-                    {
-                        ((EntityPlayerMP)object).playerNetServerHandler.setPlayerLocation(entity.posX, entity.posY, entity.posZ, entity.rotationYaw, entity.rotationPitch);
-                    }
-                    else
-                    {
-                        ((Entity)object).setLocationAndAngles(entity.posX, entity.posY, entity.posZ, entity.rotationYaw, entity.rotationPitch);
-                    }
-
                     notifyOperators(sender, this, "commands.tp.success", new Object[] {((Entity)object).getName(), entity.getName()});
+                    // CraftBukkit End
                 }
             }
         }
     }
 
@@ -174,6 +165,15 @@
 
     public boolean isUsernameIndex(String[] args, int index)
     {
         return index == 0;
     }
+
+
+    // CraftBukkit start - fix decompile error
+    @Override
+    public int compareTo(Object o)
+    {
+        return compareTo((ICommand)o);
+    }
+    // CraftBukkit end
 }
