--- decompiled\net\minecraft\command\server\CommandSetBlock.java
+++ src\main\java\net\minecraft\command\server\CommandSetBlock.java
@@ -1,8 +1,9 @@
 package net.minecraft.command.server;
 
 import java.util.List;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.CommandResultStats;
@@ -17,10 +18,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class CommandSetBlock extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000949";
 
     public String getName()
     {
         return "setblock";
@@ -139,8 +141,8 @@
         }
     }
 
     public List addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos)
     {
-        return args.length > 0 && args.length <= 3 ? func_175771_a(args, 0, pos) : (args.length == 4 ? func_175762_a(args, Block.blockRegistry.getKeys()) : (args.length == 6 ? getListOfStringsMatchingLastWord(args, new String[] {"replace", "destroy", "keep"}): null));
+        return args.length > 0 && args.length <= 3 ? func_175771_a(args, 0, pos) : (args.length == 4 ? func_175762_a(args, Block.blockRegistry.getKeys()) : (args.length == 6 ? getListOfStringsMatchingLastWord(args, new String[] {"replace", "destroy", "keep"}) : null));
     }
 }
