--- decompiled\net\minecraft\command\server\CommandTestFor.java
+++ src\main\java\net\minecraft\command\server\CommandTestFor.java
@@ -1,8 +1,9 @@
 package net.minecraft.command.server;
 
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.entity.Entity;
@@ -12,10 +13,11 @@
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 
 public class CommandTestFor extends CommandBase
 {
+
     private static final String __OBFID = "CL_00001182";
 
     public String getName()
     {
         return "testfor";
