--- decompiled\net\minecraft\command\server\CommandPardonIp.java
+++ src\main\java\net\minecraft\command\server\CommandPardonIp.java
@@ -1,19 +1,21 @@
 package net.minecraft.command.server;
 
 import java.util.List;
 import java.util.regex.Matcher;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.SyntaxErrorException;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 
 public class CommandPardonIp extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000720";
 
     public String getName()
     {
         return "pardon-ip";
