--- decompiled\net\minecraft\command\server\CommandBanPlayer.java
+++ src\main\java\net\minecraft\command\server\CommandBanPlayer.java
@@ -1,21 +1,24 @@
 package net.minecraft.command.server;
 
-import com.mojang.authlib.GameProfile;
 import java.util.Date;
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.management.UserListBansEntry;
 import net.minecraft.util.BlockPos;
 
+import com.mojang.authlib.GameProfile;
+
 public class CommandBanPlayer extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000165";
 
     public String getName()
     {
         return "ban";
