--- decompiled\net\minecraft\command\server\CommandTestForBlock.java
+++ src\main\java\net\minecraft\command\server\CommandTestForBlock.java
@@ -1,9 +1,10 @@
 package net.minecraft.command.server;
 
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.CommandResultStats;
@@ -19,10 +20,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class CommandTestForBlock extends CommandBase
 {
+
     private static final String __OBFID = "CL_00001181";
 
     public String getName()
     {
         return "testforblock";
