--- decompiled\net\minecraft\command\server\CommandSetDefaultSpawnpoint.java
+++ src\main\java\net\minecraft\command\server\CommandSetDefaultSpawnpoint.java
@@ -1,18 +1,20 @@
 package net.minecraft.command.server;
 
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.network.play.server.S05PacketSpawnPosition;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 
 public class CommandSetDefaultSpawnpoint extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000973";
 
     public String getName()
     {
         return "setworldspawn";
