--- decompiled\net\minecraft\command\server\CommandBroadcast.java
+++ src\main\java\net\minecraft\command\server\CommandBroadcast.java
@@ -1,8 +1,9 @@
 package net.minecraft.command.server;
 
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.server.MinecraftServer;
@@ -10,10 +11,11 @@
 import net.minecraft.util.ChatComponentTranslation;
 import net.minecraft.util.IChatComponent;
 
 public class CommandBroadcast extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000191";
 
     public String getName()
     {
         return "say";
