--- decompiled\net\minecraft\command\server\CommandMessageRaw.java
+++ src\main\java\net\minecraft\command\server\CommandMessageRaw.java
@@ -1,23 +1,27 @@
 package net.minecraft.command.server;
 
-import com.google.gson.JsonParseException;
 import java.util.List;
+
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.SyntaxErrorException;
 import net.minecraft.command.WrongUsageException;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.ChatComponentProcessor;
 import net.minecraft.util.IChatComponent;
+
 import org.apache.commons.lang3.exception.ExceptionUtils;
 
+import com.google.gson.JsonParseException;
+
 public class CommandMessageRaw extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000667";
 
     public String getName()
     {
         return "tellraw";
