--- decompiled\net\minecraft\item\crafting\IRecipe.java
+++ src\main\java\net\minecraft\item\crafting\IRecipe.java
@@ -1,18 +1,27 @@
 package net.minecraft.item.crafting;
 
+import java.util.List;
+
 import net.minecraft.inventory.InventoryCrafting;
 import net.minecraft.item.ItemStack;
 import net.minecraft.world.World;
 
+import org.bukkit.inventory.Recipe;
+
 public interface IRecipe
 {
+
     boolean matches(InventoryCrafting p_77569_1_, World worldIn);
 
     ItemStack getCraftingResult(InventoryCrafting p_77572_1_);
 
     int getRecipeSize();
 
     ItemStack getRecipeOutput();
 
     ItemStack[] getRemainingItems(InventoryCrafting p_179532_1_);
+
+    Recipe toBukkitRecipe(); // CraftBukkit
+
+    List getIngredients(); // Spigot
 }
