--- decompiled\net\minecraft\item\crafting\RecipesArmor.java
+++ src\main\java\net\minecraft\item\crafting\RecipesArmor.java
@@ -4,10 +4,11 @@
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 
 public class RecipesArmor
 {
+
     private String[][] recipePatterns = new String[][] {{"XXX", "X X"}, {"X X", "XXX", "XXX"}, {"XXX", "X X", "X X"}, {"X X", "X X"}};
     private Item[][] recipeItems;
     private static final String __OBFID = "CL_00000080";
 
     public RecipesArmor()
