--- decompiled\net\minecraft\item\crafting\RecipesFood.java
+++ src\main\java\net\minecraft\item\crafting\RecipesFood.java
@@ -5,10 +5,11 @@
 import net.minecraft.item.EnumDyeColor;
 import net.minecraft.item.ItemStack;
 
 public class RecipesFood
 {
+
     private static final String __OBFID = "CL_00000084";
 
     public void addRecipes(CraftingManager p_77608_1_)
     {
         p_77608_1_.addShapelessRecipe(new ItemStack(Items.mushroom_stew), new Object[] {Blocks.brown_mushroom, Blocks.red_mushroom, Items.bowl});
