--- decompiled\net\minecraft\item\crafting\RecipesCrafting.java
+++ src\main\java\net\minecraft\item\crafting\RecipesCrafting.java
@@ -15,10 +15,11 @@
 import net.minecraft.item.EnumDyeColor;
 import net.minecraft.item.ItemStack;
 
 public class RecipesCrafting
 {
+
     private static final String __OBFID = "CL_00000095";
 
     public void addRecipes(CraftingManager p_77589_1_)
     {
         p_77589_1_.addRecipe(new ItemStack(Blocks.chest), new Object[] {"###", "# #", "###", '#', Blocks.planks});
