--- decompiled\net\minecraft\item\crafting\RecipesBanners.java
+++ src\main\java\net\minecraft\item\crafting\RecipesBanners.java
@@ -10,10 +10,11 @@
 import net.minecraft.tileentity.TileEntityBanner;
 import net.minecraft.world.World;
 
 public class RecipesBanners
 {
+
     private static final String __OBFID = "CL_00002160";
 
     void func_179534_a(CraftingManager p_179534_1_)
     {
         EnumDyeColor[] aenumdyecolor = EnumDyeColor.values();
@@ -27,15 +28,21 @@
 
         p_179534_1_.addRecipe(new RecipesBanners.RecipeDuplicatePattern(null));
         p_179534_1_.addRecipe(new RecipesBanners.RecipeAddPattern(null));
     }
 
-    static class RecipeAddPattern implements IRecipe
+    public static class RecipeAddPattern extends ShapelessRecipes implements IRecipe   // CraftBukkit - added extends
         {
+
             private static final String __OBFID = "CL_00002158";
 
-            private RecipeAddPattern() {}
+            // CraftBukkit start - Delegate to new parent class with bogus info
+            private RecipeAddPattern()
+            {
+                super(new ItemStack(Items.banner, 0, 0), java.util.Arrays.asList(new ItemStack(Items.dye, 0, 5)));
+            }
+            // CraftBukkit end
 
             public boolean matches(InventoryCrafting p_77569_1_, World worldIn)
             {
                 boolean flag = false;
 
@@ -264,15 +271,21 @@
             {
                 this();
             }
         }
 
-    static class RecipeDuplicatePattern implements IRecipe
+    public static class RecipeDuplicatePattern extends ShapelessRecipes implements IRecipe   // CraftBukkit - added extends
         {
+
             private static final String __OBFID = "CL_00002157";
 
-            private RecipeDuplicatePattern() {}
+            // CraftBukkit start - Delegate to new parent class with bogus info
+            private RecipeDuplicatePattern()
+            {
+                super(new ItemStack(Items.banner, 0, 0), java.util.Arrays.asList(new ItemStack(Items.banner)));
+            }
+            // CraftBukkit end
 
             public boolean matches(InventoryCrafting p_77569_1_, World worldIn)
             {
                 ItemStack itemstack = null;
                 ItemStack itemstack1 = null;
