--- decompiled\net\minecraft\item\crafting\RecipesMapExtending.java
+++ src\main\java\net\minecraft\item\crafting\RecipesMapExtending.java
@@ -7,10 +7,11 @@
 import net.minecraft.world.World;
 import net.minecraft.world.storage.MapData;
 
 public class RecipesMapExtending extends ShapedRecipes
 {
+
     private static final String __OBFID = "CL_00000088";
 
     public RecipesMapExtending()
     {
         super(3, 3, new ItemStack[] {new ItemStack(Items.paper), new ItemStack(Items.paper), new ItemStack(Items.paper), new ItemStack(Items.paper), new ItemStack(Items.filled_map, 0, 32767), new ItemStack(Items.paper), new ItemStack(Items.paper), new ItemStack(Items.paper), new ItemStack(Items.paper)}, new ItemStack(Items.map, 0, 0));
