--- decompiled\net\minecraft\util\BlockPos.java
+++ src\main\java\net\minecraft\util\BlockPos.java
@@ -1,15 +1,16 @@
 package net.minecraft.util;
 
-import com.google.common.collect.AbstractIterator;
 import java.util.Iterator;
+
 import net.minecraft.entity.Entity;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.collect.AbstractIterator;
+
 public class BlockPos extends Vec3i
 {
+
     public static final BlockPos ORIGIN = new BlockPos(0, 0, 0);
     private static final int NUM_X_BITS = 1 + MathHelper.calculateLogBaseTwo(MathHelper.roundUpToPowerOfTwo(30000000));
     private static final int NUM_Z_BITS = NUM_X_BITS;
     private static final int NUM_Y_BITS = 64 - NUM_X_BITS - NUM_Z_BITS;
     private static final int Y_SHIFT = 0 + NUM_Z_BITS;
@@ -67,16 +68,10 @@
     public BlockPos up()
     {
         return this.up(1);
     }
 
-    @SideOnly(Side.CLIENT)
-    public BlockPos subtract(Vec3i vec)
-    {
-        return new BlockPos(this.getX() - vec.getX(), this.getY() - vec.getY(), this.getZ() - vec.getZ());
-    }
-
     public BlockPos up(int n)
     {
         return this.offset(EnumFacing.UP, n);
     }
 
@@ -283,10 +278,11 @@
         return this.crossProductBP(vec);
     }
 
     public static final class MutableBlockPos extends BlockPos
         {
+
             public int x;
             public int y;
             public int z;
             private static final String __OBFID = "CL_00002329";
 
