--- decompiled\net\minecraft\util\Cartesian.java
+++ src\main\java\net\minecraft\util\Cartesian.java
@@ -1,21 +1,23 @@
 package net.minecraft.util;
 
-import com.google.common.base.Function;
-import com.google.common.collect.Iterables;
-import com.google.common.collect.Lists;
-import com.google.common.collect.UnmodifiableIterator;
 import java.lang.reflect.Array;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
 import java.util.NoSuchElementException;
 
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.UnmodifiableIterator;
+
 public class Cartesian
 {
+
     private static final String __OBFID = "CL_00002327";
 
     public static Iterable cartesianProduct(Class clazz, Iterable sets)
     {
         return new Cartesian.Product(clazz, (Iterable[])toArray(Iterable.class, sets), null);
@@ -50,13 +52,16 @@
         return (Object[])((Object[])Array.newInstance(p_179319_0_, p_179319_1_));
     }
 
     static class GetList implements Function
         {
+
             private static final String __OBFID = "CL_00002325";
 
-            private GetList() {}
+            private GetList()
+            {
+            }
 
             public List apply(Object[] array)
             {
                 return Arrays.asList((Object[])array);
             }
@@ -72,10 +77,11 @@
             }
         }
 
     static class Product implements Iterable
         {
+
             private final Class clazz;
             private final Iterable[] iterables;
             private static final String __OBFID = "CL_00002324";
 
             private Product(Class clazz, Iterable[] iterables)
@@ -94,10 +100,11 @@
                 this(p_i46021_1_, p_i46021_2_);
             }
 
             static class ProductIterator extends UnmodifiableIterator
                 {
+
                     private int index;
                     private final Iterable[] iterables;
                     private final Iterator[] iterators;
                     private final Object[] results;
                     private static final String __OBFID = "CL_00002323";
