--- decompiled\net\minecraft\util\ChatComponentScore.java
+++ src\main\java\net\minecraft\util\ChatComponentScore.java
@@ -1,15 +1,17 @@
 package net.minecraft.util;
 
 import java.util.Iterator;
+
 import net.minecraft.scoreboard.Score;
 import net.minecraft.scoreboard.ScoreObjective;
 import net.minecraft.scoreboard.Scoreboard;
 import net.minecraft.server.MinecraftServer;
 
 public class ChatComponentScore extends ChatComponentStyle
 {
+
     private final String field_179999_b;
     private final String field_180000_c;
     private String field_179998_d = "";
     private static final String __OBFID = "CL_00002309";
 
