--- decompiled\net\minecraft\util\ChatAllowedCharacters.java
+++ src\main\java\net\minecraft\util\ChatAllowedCharacters.java
@@ -1,9 +1,10 @@
 package net.minecraft.util;
 
 public class ChatAllowedCharacters
 {
+
     public static final char[] allowedCharactersArray = new char[] {'/', '\n', '\r', '\t', '\u0000', '\f', '`', '?', '*', '\\', '<', '>', '|', '\"', ':'};
     private static final String __OBFID = "CL_00001606";
 
     public static boolean isAllowedCharacter(char character)
     {
