--- decompiled\net\minecraft\util\ChatComponentProcessor.java
+++ src\main\java\net\minecraft\util\ChatComponentProcessor.java
@@ -1,17 +1,19 @@
 package net.minecraft.util;
 
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.command.CommandException;
 import net.minecraft.command.EntityNotFoundException;
 import net.minecraft.command.ICommandSender;
 import net.minecraft.command.PlayerSelector;
 import net.minecraft.entity.Entity;
 
 public class ChatComponentProcessor
 {
+
     private static final String __OBFID = "CL_00002310";
 
     public static IChatComponent func_179985_a(ICommandSender commandSender, IChatComponent component, Entity p_179985_2_) throws CommandException
     {
         Object object = null;
