--- decompiled\net\minecraft\util\ChatStyle.java
+++ src\main\java\net\minecraft\util\ChatStyle.java
@@ -1,22 +1,23 @@
 package net.minecraft.util;
 
+import java.lang.reflect.Type;
+
+import net.minecraft.event.ClickEvent;
+import net.minecraft.event.HoverEvent;
+
 import com.google.gson.JsonDeserializationContext;
 import com.google.gson.JsonDeserializer;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonPrimitive;
 import com.google.gson.JsonSerializationContext;
 import com.google.gson.JsonSerializer;
-import java.lang.reflect.Type;
-import net.minecraft.event.ClickEvent;
-import net.minecraft.event.HoverEvent;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
 public class ChatStyle
 {
+
     private ChatStyle parentStyle;
     private EnumChatFormatting color;
     private Boolean bold;
     private Boolean italic;
     private Boolean underlined;
@@ -110,15 +111,10 @@
         }
         public ChatStyle createDeepCopy()
         {
             return this;
         }
-        @SideOnly(Side.CLIENT)
-        public String getFormattingCode()
-        {
-            return "";
-        }
     };
     private static final String __OBFID = "CL_00001266";
 
     public EnumChatFormatting getColor()
     {
@@ -228,55 +224,10 @@
     {
         this.parentStyle = parent;
         return this;
     }
 
-    @SideOnly(Side.CLIENT)
-    public String getFormattingCode()
-    {
-        if (this.isEmpty())
-        {
-            return this.parentStyle != null ? this.parentStyle.getFormattingCode() : "";
-        }
-        else
-        {
-            StringBuilder stringbuilder = new StringBuilder();
-
-            if (this.getColor() != null)
-            {
-                stringbuilder.append(this.getColor());
-            }
-
-            if (this.getBold())
-            {
-                stringbuilder.append(EnumChatFormatting.BOLD);
-            }
-
-            if (this.getItalic())
-            {
-                stringbuilder.append(EnumChatFormatting.ITALIC);
-            }
-
-            if (this.getUnderlined())
-            {
-                stringbuilder.append(EnumChatFormatting.UNDERLINE);
-            }
-
-            if (this.getObfuscated())
-            {
-                stringbuilder.append(EnumChatFormatting.OBFUSCATED);
-            }
-
-            if (this.getStrikethrough())
-            {
-                stringbuilder.append(EnumChatFormatting.STRIKETHROUGH);
-            }
-
-            return stringbuilder.toString();
-        }
-    }
-
     private ChatStyle getParent()
     {
         return this.parentStyle == null ? rootStyle : this.parentStyle;
     }
 
@@ -350,19 +301,21 @@
         }
     }
 
     public int hashCode()
     {
-        int i = this.color.hashCode();
-        i = 31 * i + this.bold.hashCode();
-        i = 31 * i + this.italic.hashCode();
-        i = 31 * i + this.underlined.hashCode();
-        i = 31 * i + this.strikethrough.hashCode();
-        i = 31 * i + this.obfuscated.hashCode();
-        i = 31 * i + this.chatClickEvent.hashCode();
-        i = 31 * i + this.chatHoverEvent.hashCode();
-        i = 31 * i + this.insertion.hashCode();
+        // CraftBukkit start - fix npe
+        int i = color == null ? 0 : this.color.hashCode();
+        i = 31 * i + (bold == null ? 0 : this.bold.hashCode());
+        i = 31 * i + (italic == null ? 0 : this.italic.hashCode());
+        i = 31 * i + (underlined == null ? 0 : this.underlined.hashCode());
+        i = 31 * i + (strikethrough == null ? 0 : this.strikethrough.hashCode());
+        i = 31 * i + (obfuscated == null ? 0 : this.obfuscated.hashCode());
+        i = 31 * i + (chatClickEvent == null ? 0 : this.chatClickEvent.hashCode());
+        i = 31 * i + (chatHoverEvent == null ? 0 : this.chatHoverEvent.hashCode());
+        i = 31 * i + (insertion == null ? 0 : this.insertion.hashCode());
+        // CraftBukkit end
         return i;
     }
 
     public ChatStyle createShallowCopy()
     {
@@ -395,10 +348,11 @@
         return chatstyle;
     }
 
     public static class Serializer implements JsonDeserializer, JsonSerializer
         {
+
             private static final String __OBFID = "CL_00001268";
 
             public ChatStyle deserialize(JsonElement p_deserialize_1_, Type p_deserialize_2_, JsonDeserializationContext p_deserialize_3_)
             {
                 if (p_deserialize_1_.isJsonObject())
