--- decompiled\net\minecraft\util\ChatComponentSelector.java
+++ src\main\java\net\minecraft\util\ChatComponentSelector.java
@@ -2,10 +2,11 @@
 
 import java.util.Iterator;
 
 public class ChatComponentSelector extends ChatComponentStyle
 {
+
     private final String field_179993_b;
     private static final String __OBFID = "CL_00002308";
 
     public ChatComponentSelector(String p_i45996_1_)
     {
