--- decompiled\net\minecraft\util\ChatComponentTranslationFormatException.java
+++ src\main\java\net\minecraft\util\ChatComponentTranslationFormatException.java
@@ -1,9 +1,10 @@
 package net.minecraft.util;
 
 public class ChatComponentTranslationFormatException extends IllegalArgumentException
 {
+
     private static final String __OBFID = "CL_00001271";
 
     public ChatComponentTranslationFormatException(ChatComponentTranslation component, String message)
     {
         super(String.format("Error parsing: %s: %s", new Object[] {component, message}));
