--- decompiled\net\minecraft\util\ClassInheritanceMultiMap.java
+++ src\main\java\net\minecraft\util\ClassInheritanceMultiMap.java
@@ -1,20 +1,24 @@
 package net.minecraft.util;
 
-import com.google.common.collect.AbstractIterator;
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.Iterators;
-import com.google.common.collect.Multimap;
-import com.google.common.collect.Sets;
 import java.util.AbstractSet;
+import java.util.Collections;
 import java.util.Iterator;
 import java.util.Set;
+
 import org.apache.commons.lang3.ClassUtils;
 import org.apache.commons.lang3.ClassUtils.Interfaces;
 
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+
 public class ClassInheritanceMultiMap extends AbstractSet
 {
+
     private final Multimap field_180218_a = HashMultimap.create();
     private final Set field_180216_b = Sets.newIdentityHashSet();
     private final Class field_180217_c;
     private static final String __OBFID = "CL_00002266";
 
@@ -101,10 +105,11 @@
         return flag;
     }
 
     public Iterable func_180215_b(final Class p_180215_1_)
     {
+        // PAIL: Access transform // SpigotMCP -- Unnecessary
         return new Iterable()
         {
             private static final String __OBFID = "CL_00002265";
             public Iterator iterator()
             {
@@ -114,11 +119,19 @@
         };
     }
 
     public Iterator iterator()
     {
+        // CraftBukkit start
+        if (!this.field_180218_a.containsKey(this.field_180217_c))
+        {
+            return Collections.EMPTY_LIST.iterator();
+        }
+
+        // CraftBukkit end
         final Iterator iterator = this.field_180218_a.get(this.field_180217_c).iterator();
+        // PAIL: Access transform // SpigotMCP -- Unnecessary
         return new AbstractIterator()
         {
             private static final String __OBFID = "CL_00002264";
             protected Object computeNext()
             {
