--- decompiled\net\minecraft\util\CombatEntry.java
+++ src\main\java\net\minecraft\util\CombatEntry.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.entity.EntityLivingBase;
 
 public class CombatEntry
 {
+
     private final DamageSource damageSrc;
     private final int field_94567_b;
     private final float field_94568_c;
     private final float field_94565_d;
     private final String field_94566_e;
