--- decompiled\net\minecraft\util\DamageSource.java
+++ src\main\java\net\minecraft\util\DamageSource.java
@@ -7,10 +7,11 @@
 import net.minecraft.entity.projectile.EntityFireball;
 import net.minecraft.world.Explosion;
 
 public class DamageSource
 {
+
     public static DamageSource inFire = (new DamageSource("inFire")).setFireDamage();
     public static DamageSource lightningBolt = new DamageSource("lightningBolt");
     public static DamageSource onFire = (new DamageSource("onFire")).setDamageBypassesArmor().setFireDamage();
     public static DamageSource lava = (new DamageSource("lava")).setFireDamage();
     public static DamageSource inWall = (new DamageSource("inWall")).setDamageBypassesArmor();
@@ -116,11 +117,11 @@
     public boolean isDamageAbsolute()
     {
         return this.damageIsAbsolute;
     }
 
-    protected DamageSource(String damageTypeIn)
+    public DamageSource(String damageTypeIn)
     {
         this.damageType = damageTypeIn;
     }
 
     public Entity getSourceOfDamage()
@@ -131,42 +132,42 @@
     public Entity getEntity()
     {
         return null;
     }
 
-    protected DamageSource setDamageBypassesArmor()
+    public DamageSource setDamageBypassesArmor()
     {
         this.isUnblockable = true;
         this.hungerDamage = 0.0F;
         return this;
     }
 
-    protected DamageSource setDamageAllowedInCreativeMode()
+    public DamageSource setDamageAllowedInCreativeMode()
     {
         this.isDamageAllowedInCreativeMode = true;
         return this;
     }
 
-    protected DamageSource setDamageIsAbsolute()
+    public DamageSource setDamageIsAbsolute()
     {
         this.damageIsAbsolute = true;
         this.hungerDamage = 0.0F;
         return this;
     }
 
-    protected DamageSource setFireDamage()
+    public DamageSource setFireDamage()
     {
         this.fireDamage = true;
         return this;
     }
 
     public IChatComponent getDeathMessage(EntityLivingBase p_151519_1_)
     {
         EntityLivingBase entitylivingbase1 = p_151519_1_.func_94060_bK();
         String s = "death.attack." + this.damageType;
         String s1 = s + ".player";
-        return entitylivingbase1 != null && StatCollector.canTranslate(s1) ? new ChatComponentTranslation(s1, new Object[] {p_151519_1_.getDisplayName(), entitylivingbase1.getDisplayName()}): new ChatComponentTranslation(s, new Object[] {p_151519_1_.getDisplayName()});
+        return entitylivingbase1 != null && StatCollector.canTranslate(s1) ? new ChatComponentTranslation(s1, new Object[] {p_151519_1_.getDisplayName(), entitylivingbase1.getDisplayName()}) : new ChatComponentTranslation(s, new Object[] {p_151519_1_.getDisplayName()});
     }
 
     public boolean isFireDamage()
     {
         return this.fireDamage;
