--- decompiled\net\minecraft\util\EntityDamageSourceIndirect.java
+++ src\main\java\net\minecraft\util\EntityDamageSourceIndirect.java
@@ -4,10 +4,11 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.item.ItemStack;
 
 public class EntityDamageSourceIndirect extends EntityDamageSource
 {
+
     private Entity indirectEntity;
     private static final String __OBFID = "CL_00001523";
 
     public EntityDamageSourceIndirect(String p_i1568_1_, Entity p_i1568_2_, Entity indirectEntityIn)
     {
@@ -29,8 +30,16 @@
     {
         IChatComponent ichatcomponent = this.indirectEntity == null ? this.damageSourceEntity.getDisplayName() : this.indirectEntity.getDisplayName();
         ItemStack itemstack = this.indirectEntity instanceof EntityLivingBase ? ((EntityLivingBase)this.indirectEntity).getHeldItem() : null;
         String s = "death.attack." + this.damageType;
         String s1 = s + ".item";
-        return itemstack != null && itemstack.hasDisplayName() && StatCollector.canTranslate(s1) ? new ChatComponentTranslation(s1, new Object[] {p_151519_1_.getDisplayName(), ichatcomponent, itemstack.getChatComponent()}): new ChatComponentTranslation(s, new Object[] {p_151519_1_.getDisplayName(), ichatcomponent});
+        return itemstack != null && itemstack.hasDisplayName() && StatCollector.canTranslate(s1) ? new ChatComponentTranslation(s1, new Object[] {p_151519_1_.getDisplayName(), ichatcomponent, itemstack.getChatComponent()}) : new ChatComponentTranslation(s, new Object[] {p_151519_1_.getDisplayName(), ichatcomponent});
     }
+
+
+    // CraftBukkit start
+    public Entity getProximateDamageSource()
+    {
+        return super.getEntity();
+    }
+    // CraftBukkit end
 }
