--- decompiled\net\minecraft\util\EnumTypeAdapterFactory.java
+++ src\main\java\net\minecraft\util\EnumTypeAdapterFactory.java
@@ -1,21 +1,23 @@
 package net.minecraft.util;
 
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Locale;
+
 import com.google.common.collect.Maps;
 import com.google.gson.Gson;
 import com.google.gson.TypeAdapter;
 import com.google.gson.TypeAdapterFactory;
 import com.google.gson.reflect.TypeToken;
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonToken;
 import com.google.gson.stream.JsonWriter;
-import java.io.IOException;
-import java.util.HashMap;
-import java.util.Locale;
 
 public class EnumTypeAdapterFactory implements TypeAdapterFactory
 {
+
     private static final String __OBFID = "CL_00001494";
 
     public TypeAdapter create(Gson p_create_1_, TypeToken p_create_2_)
     {
         Class oclass = p_create_2_.getRawType();
