--- decompiled\net\minecraft\util\IChatComponent.java
+++ src\main\java\net\minecraft\util\IChatComponent.java
@@ -1,7 +1,12 @@
 package net.minecraft.util;
 
+import java.lang.reflect.Type;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map.Entry;
+
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
 import com.google.gson.JsonArray;
 import com.google.gson.JsonDeserializationContext;
 import com.google.gson.JsonDeserializer;
@@ -9,19 +14,14 @@
 import com.google.gson.JsonObject;
 import com.google.gson.JsonParseException;
 import com.google.gson.JsonPrimitive;
 import com.google.gson.JsonSerializationContext;
 import com.google.gson.JsonSerializer;
-import java.lang.reflect.Type;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map.Entry;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
 public interface IChatComponent extends Iterable
 {
+
     IChatComponent setChatStyle(ChatStyle style);
 
     ChatStyle getChatStyle();
 
     IChatComponent appendText(String text);
@@ -30,19 +30,17 @@
 
     String getUnformattedTextForChat();
 
     String getUnformattedText();
 
-    @SideOnly(Side.CLIENT)
-    String getFormattedText();
-
     List getSiblings();
 
     IChatComponent createCopy();
 
     public static class Serializer implements JsonDeserializer, JsonSerializer
         {
+
             private static final Gson GSON;
             private static final String __OBFID = "CL_00001263";
 
             public IChatComponent deserialize(JsonElement p_deserialize_1_, Type p_deserialize_2_, JsonDeserializationContext p_deserialize_3_)
             {
