--- decompiled\net\minecraft\util\IntHashMap.java
+++ src\main\java\net\minecraft\util\IntHashMap.java
@@ -1,9 +1,10 @@
 package net.minecraft.util;
 
 public class IntHashMap
 {
+
     private transient IntHashMap.Entry[] slots = new IntHashMap.Entry[16];
     private transient int count;
     private int threshold = 12;
     private final float growFactor = 0.75F;
     private static final String __OBFID = "CL_00001490";
@@ -179,10 +180,11 @@
         }
     }
 
     static class Entry
         {
+
             final int hashEntry;
             Object valueEntry;
             IntHashMap.Entry nextEntry;
             final int slotHash;
             private static final String __OBFID = "CL_00001491";
