--- decompiled\net\minecraft\util\IProgressUpdate.java
+++ src\main\java\net\minecraft\util\IProgressUpdate.java
@@ -1,19 +1,11 @@
 package net.minecraft.util;
 
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
-public interface IProgressUpdate
-{
+public interface IProgressUpdate {
+
     void displaySavingString(String message);
 
-    @SideOnly(Side.CLIENT)
-    void resetProgressAndMessage(String p_73721_1_);
-
     void displayLoadingString(String message);
 
     void setLoadingProgress(int progress);
-
-    @SideOnly(Side.CLIENT)
-    void setDoneWorking();
 }
