--- decompiled\net\minecraft\util\IThreadListener.java
+++ src\main\java\net\minecraft\util\IThreadListener.java
@@ -1,10 +1,10 @@
 package net.minecraft.util;
 
 import com.google.common.util.concurrent.ListenableFuture;
 
-public interface IThreadListener
-{
+public interface IThreadListener {
+
     ListenableFuture addScheduledTask(Runnable runnableToSchedule);
 
     boolean isCallingFromMinecraftThread();
 }
