--- decompiled\net\minecraft\util\JsonSerializableSet.java
+++ src\main\java\net\minecraft\util\JsonSerializableSet.java
@@ -1,18 +1,19 @@
 package net.minecraft.util;
 
+import java.util.Iterator;
+import java.util.Set;
+
 import com.google.common.collect.ForwardingSet;
 import com.google.common.collect.Sets;
 import com.google.gson.JsonArray;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonPrimitive;
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.Set;
 
 public class JsonSerializableSet extends ForwardingSet implements IJsonSerializable
 {
+
     private final Set underlyingSet = Sets.newHashSet();
     private static final String __OBFID = "CL_00001482";
 
     public void func_152753_a(JsonElement p_152753_1_)
     {
