--- decompiled\net\minecraft\util\LoggingPrintStream.java
+++ src\main\java\net\minecraft\util\LoggingPrintStream.java
@@ -1,14 +1,16 @@
 package net.minecraft.util;
 
 import java.io.OutputStream;
 import java.io.PrintStream;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class LoggingPrintStream extends PrintStream
 {
+
     private static final Logger LOGGER = LogManager.getLogger();
     private final String domain;
     private static final String __OBFID = "CL_00002275";
 
     public LoggingPrintStream(String domainIn, OutputStream outStream)
