--- decompiled\net\minecraft\util\LongHashMap.java
+++ src\main\java\net\minecraft\util\LongHashMap.java
@@ -1,9 +1,10 @@
 package net.minecraft.util;
 
 public class LongHashMap
 {
+
     private transient LongHashMap.Entry[] hashArray = new LongHashMap.Entry[4096];
     private transient int numHashElements;
     private int mask;
     private int capacity = 3072;
     private final float percentUseable = 0.75F;
@@ -187,10 +188,11 @@
         }
     }
 
     static class Entry
         {
+
             final long key;
             Object value;
             LongHashMap.Entry nextEntry;
             final int hash;
             private static final String __OBFID = "CL_00001493";
