--- decompiled\net\minecraft\util\MapPopulator.java
+++ src\main\java\net\minecraft\util\MapPopulator.java
@@ -1,14 +1,16 @@
 package net.minecraft.util;
 
-import com.google.common.collect.Maps;
 import java.util.Iterator;
 import java.util.Map;
 import java.util.NoSuchElementException;
 
+import com.google.common.collect.Maps;
+
 public class MapPopulator
 {
+
     private static final String __OBFID = "CL_00002318";
 
     public static Map createMap(Iterable keys, Iterable values)
     {
         return populateMap(keys, values, Maps.newLinkedHashMap());
