--- decompiled\net\minecraft\util\MessageDeserializer2.java
+++ src\main\java\net\minecraft\util\MessageDeserializer2.java
@@ -1,17 +1,20 @@
 package net.minecraft.util;
 
+import java.util.List;
+
+import net.minecraft.network.PacketBuffer;
+
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.Unpooled;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.ByteToMessageDecoder;
 import io.netty.handler.codec.CorruptedFrameException;
-import java.util.List;
-import net.minecraft.network.PacketBuffer;
 
 public class MessageDeserializer2 extends ByteToMessageDecoder
 {
+
     private static final String __OBFID = "CL_00001255";
 
     protected void decode(ChannelHandlerContext p_decode_1_, ByteBuf p_decode_2_, List p_decode_3_)
     {
         p_decode_2_.markReaderIndex();
@@ -33,17 +36,17 @@
 
                 try
                 {
                     int j = packetbuffer.readVarIntFromBuffer();
 
-                    if (p_decode_2_.readableBytes() < j)
+                    if (p_decode_2_.readableBytes() >= j)
                     {
-                        p_decode_2_.resetReaderIndex();
+                        p_decode_3_.add(p_decode_2_.readBytes(j));
                         return;
                     }
 
-                    p_decode_3_.add(p_decode_2_.readBytes(j));
+                    p_decode_2_.resetReaderIndex();
                 }
                 finally
                 {
                     packetbuffer.release();
                 }
