--- decompiled\net\minecraft\util\MovingObjectPosition.java
+++ src\main\java\net\minecraft\util\MovingObjectPosition.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.entity.Entity;
 
 public class MovingObjectPosition
 {
+
     private BlockPos blockPos;
     public MovingObjectPosition.MovingObjectType typeOfHit;
     public EnumFacing sideHit;
     public Vec3 hitVec;
     public Entity entityHit;
