--- decompiled\net\minecraft\util\RegistryNamespacedDefaultedByKey.java
+++ src\main\java\net\minecraft\util\RegistryNamespacedDefaultedByKey.java
@@ -2,10 +2,11 @@
 
 import org.apache.commons.lang3.Validate;
 
 public class RegistryNamespacedDefaultedByKey extends RegistryNamespaced
 {
+
     private final Object field_148760_d;
     private Object field_148761_e;
     private static final String __OBFID = "CL_00001196";
 
     public RegistryNamespacedDefaultedByKey(Object p_i46017_1_)
