--- decompiled\net\minecraft\util\StatCollector.java
+++ src\main\java\net\minecraft\util\StatCollector.java
@@ -1,19 +1,20 @@
 package net.minecraft.util;
 
 public class StatCollector
 {
+
     private static StringTranslate localizedName = StringTranslate.getInstance();
     private static StringTranslate fallbackTranslator = new StringTranslate();
     private static final String __OBFID = "CL_00001211";
 
     public static String translateToLocal(String key)
     {
         return localizedName.translateKey(key);
     }
 
-    public static String translateToLocalFormatted(String key, Object ... format)
+    public static String translateToLocalFormatted(String key, Object... format)
     {
         return localizedName.translateKeyFormat(key, format);
     }
 
     public static String translateToFallback(String key)
