--- decompiled\net\minecraft\util\StringTranslate.java
+++ src\main\java\net\minecraft\util\StringTranslate.java
@@ -1,23 +1,24 @@
 package net.minecraft.util;
 
-import com.google.common.base.Splitter;
-import com.google.common.collect.Iterables;
-import com.google.common.collect.Maps;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.IllegalFormatException;
 import java.util.Iterator;
 import java.util.Map;
 import java.util.regex.Pattern;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
+
 import org.apache.commons.io.Charsets;
 import org.apache.commons.io.IOUtils;
 
+import com.google.common.base.Splitter;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+
 public class StringTranslate
 {
+
     private static final Pattern numericVariablePattern = Pattern.compile("%(\\d+\\$)?[\\d\\.]*[df]");
     private static final Splitter equalSignSplitter = Splitter.on('=').limit(2);
     private static StringTranslate instance = new StringTranslate();
     private final Map languageList = Maps.newHashMap();
     private long lastUpdateTimeInMilliseconds;
@@ -58,25 +59,16 @@
     static StringTranslate getInstance()
     {
         return instance;
     }
 
-    @SideOnly(Side.CLIENT)
-
-    public static synchronized void replaceWith(Map p_135063_0_)
-    {
-        instance.languageList.clear();
-        instance.languageList.putAll(p_135063_0_);
-        instance.lastUpdateTimeInMilliseconds = System.currentTimeMillis();
-    }
-
     public synchronized String translateKey(String key)
     {
         return this.tryTranslateKey(key);
     }
 
-    public synchronized String translateKeyFormat(String key, Object ... format)
+    public synchronized String translateKeyFormat(String key, Object... format)
     {
         String s1 = this.tryTranslateKey(key);
 
         try
         {
