--- decompiled\net\minecraft\util\TupleIntJsonSerializable.java
+++ src\main\java\net\minecraft\util\TupleIntJsonSerializable.java
@@ -1,9 +1,10 @@
 package net.minecraft.util;
 
 public class TupleIntJsonSerializable
 {
+
     private int integerValue;
     private IJsonSerializable jsonSerializableValue;
     private static final String __OBFID = "CL_00001478";
 
     public int getIntegerValue()
