--- decompiled\net\minecraft\util\Vec3i.java
+++ src\main\java\net\minecraft\util\Vec3i.java
@@ -2,10 +2,11 @@
 
 import com.google.common.base.Objects;
 
 public class Vec3i implements Comparable
 {
+
     public static final Vec3i NULL_VECTOR = new Vec3i(0, 0, 0);
     private final int x;
     private final int y;
     private final int z;
     private static final String __OBFID = "CL_00002315";
