--- decompiled\net\minecraft\util\WeightedRandomChestContent.java
+++ src\main\java\net\minecraft\util\WeightedRandomChestContent.java
@@ -1,22 +1,25 @@
 package net.minecraft.util;
 
-import com.google.common.collect.Lists;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Random;
+
 import net.minecraft.inventory.IInventory;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.tileentity.TileEntityDispenser;
 
+import com.google.common.collect.Lists;
+
 public class WeightedRandomChestContent extends WeightedRandom.Item
 {
-    private ItemStack theItemId;
-    private int theMinimumChanceToGenerateItem;
-    private int theMaximumChanceToGenerateItem;
+
+    public ItemStack theItemId;
+    public int theMinimumChanceToGenerateItem;
+    public int theMaximumChanceToGenerateItem;
     private static final String __OBFID = "CL_00001505";
 
     public WeightedRandomChestContent(Item p_i45311_1_, int p_i45311_2_, int minimumChance, int maximumChance, int itemWeightIn)
     {
         super(itemWeightIn);
@@ -81,11 +84,11 @@
                 }
             }
         }
     }
 
-    public static List func_177629_a(List p_177629_0_, WeightedRandomChestContent ... p_177629_1_)
+    public static List func_177629_a(List p_177629_0_, WeightedRandomChestContent... p_177629_1_)
     {
         ArrayList arraylist = Lists.newArrayList(p_177629_0_);
         Collections.addAll(arraylist, p_177629_1_);
         return arraylist;
     }
