--- decompiled\net\minecraft\util\WeightedRandom.java
+++ src\main\java\net\minecraft\util\WeightedRandom.java
@@ -4,10 +4,11 @@
 import java.util.Iterator;
 import java.util.Random;
 
 public class WeightedRandom
 {
+
     private static final String __OBFID = "CL_00001503";
 
     public static int getTotalWeight(Collection collection)
     {
         int i = 0;
@@ -59,11 +60,12 @@
         return getRandomItem(random, collection, getTotalWeight(collection));
     }
 
     public static class Item
         {
-            protected int itemWeight;
+
+            public int itemWeight;
             private static final String __OBFID = "CL_00001504";
 
             public Item(int itemWeightIn)
             {
                 this.itemWeight = itemWeightIn;
