--- decompiled\net\minecraft\util\WeightedRandomFishable.java
+++ src\main\java\net\minecraft\util\WeightedRandomFishable.java
@@ -1,13 +1,15 @@
 package net.minecraft.util;
 
 import java.util.Random;
+
 import net.minecraft.enchantment.EnchantmentHelper;
 import net.minecraft.item.ItemStack;
 
 public class WeightedRandomFishable extends WeightedRandom.Item
 {
+
     private final ItemStack returnStack;
     private float maxDamagePercent;
     private boolean enchantable;
     private static final String __OBFID = "CL_00001664";
 
