--- decompiled\net\minecraft\network\EnumConnectionState.java
+++ src\main\java\net\minecraft\network\EnumConnectionState.java
@@ -1,14 +1,10 @@
 package net.minecraft.network;
 
-import com.google.common.collect.BiMap;
-import com.google.common.collect.HashBiMap;
-import com.google.common.collect.Maps;
-import gnu.trove.map.TIntObjectMap;
-import gnu.trove.map.hash.TIntObjectHashMap;
 import java.util.Iterator;
 import java.util.Map;
+
 import net.minecraft.network.handshake.client.C00Handshake;
 import net.minecraft.network.login.client.C00PacketLoginStart;
 import net.minecraft.network.login.client.C01PacketEncryptionResponse;
 import net.minecraft.network.login.server.S00PacketDisconnect;
 import net.minecraft.network.login.server.S01PacketEncryptionRequest;
@@ -110,12 +106,20 @@
 import net.minecraft.network.play.server.S49PacketUpdateEntityNBT;
 import net.minecraft.network.status.client.C00PacketServerQuery;
 import net.minecraft.network.status.client.C01PacketPing;
 import net.minecraft.network.status.server.S00PacketServerInfo;
 import net.minecraft.network.status.server.S01PacketPong;
+
 import org.apache.logging.log4j.LogManager;
 
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.google.common.collect.Maps;
+
+import gnu.trove.map.TIntObjectMap;
+import gnu.trove.map.hash.TIntObjectHashMap;
+
 public enum EnumConnectionState
 {
     HANDSHAKING(-1)
     {
         private static final String __OBFID = "CL_00001246";
