--- decompiled\net\minecraft\network\NettyEncryptingDecoder.java
+++ src\main\java\net\minecraft\network\NettyEncryptingDecoder.java
@@ -1,16 +1,19 @@
 package net.minecraft.network;
 
-import io.netty.buffer.ByteBuf;
-import io.netty.channel.ChannelHandlerContext;
-import io.netty.handler.codec.MessageToMessageDecoder;
 import java.util.List;
+
 import javax.crypto.Cipher;
 import javax.crypto.ShortBufferException;
 
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToMessageDecoder;
+
 public class NettyEncryptingDecoder extends MessageToMessageDecoder
 {
+
     private final NettyEncryptionTranslator decryptionCodec;
     private static final String __OBFID = "CL_00001238";
 
     public NettyEncryptingDecoder(Cipher cipher)
     {
