--- decompiled\net\minecraft\network\NettyCompressionEncoder.java
+++ src\main\java\net\minecraft\network\NettyCompressionEncoder.java
@@ -1,14 +1,16 @@
 package net.minecraft.network;
 
+import java.util.zip.Deflater;
+
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.MessageToByteEncoder;
-import java.util.zip.Deflater;
 
 public class NettyCompressionEncoder extends MessageToByteEncoder
 {
+
     private final byte[] buffer = new byte[8192];
     private final Deflater deflater;
     private int treshold;
     private static final String __OBFID = "CL_00002313";
 
