--- decompiled\net\minecraft\network\PacketThreadUtil.java
+++ src\main\java\net\minecraft\network\PacketThreadUtil.java
@@ -2,10 +2,11 @@
 
 import net.minecraft.util.IThreadListener;
 
 public class PacketThreadUtil
 {
+
     private static final String __OBFID = "CL_00002306";
 
     public static void checkThreadAndEnqueue(final Packet p_180031_0_, final INetHandler p_180031_1_, IThreadListener p_180031_2_)
     {
         if (!p_180031_2_.isCallingFromMinecraftThread())
