--- decompiled\net\minecraft\network\Packet.java
+++ src\main\java\net\minecraft\network\Packet.java
@@ -2,11 +2,12 @@
 
 import java.io.IOException;
 
 public interface Packet
 {
-    void readPacketData(PacketBuffer buf) throws IOException;
 
-    void writePacketData(PacketBuffer buf) throws IOException;
+    void readPacketData(PacketBuffer buf) throws IOException; // CraftBukkit - added throws // SpigotMCP NOTE -- MCP added this for us
 
+    void writePacketData(PacketBuffer buf) throws IOException; // CraftBukkit - added throws // SpigotMCP NOTE -- MCP added this for us
+
     void processPacket(INetHandler handler);
 }
