--- decompiled\net\minecraft\network\ServerStatusResponse.java
+++ src\main\java\net\minecraft\network\ServerStatusResponse.java
@@ -1,22 +1,25 @@
 package net.minecraft.network;
 
+import java.lang.reflect.Type;
+import java.util.UUID;
+
+import net.minecraft.util.IChatComponent;
+import net.minecraft.util.JsonUtils;
+
 import com.google.gson.JsonArray;
 import com.google.gson.JsonDeserializationContext;
 import com.google.gson.JsonDeserializer;
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonSerializationContext;
 import com.google.gson.JsonSerializer;
 import com.mojang.authlib.GameProfile;
-import java.lang.reflect.Type;
-import java.util.UUID;
-import net.minecraft.util.IChatComponent;
-import net.minecraft.util.JsonUtils;
 
 public class ServerStatusResponse
 {
+
     private IChatComponent serverMotd;
     private ServerStatusResponse.PlayerCountData playerCount;
     private ServerStatusResponse.MinecraftProtocolVersionIdentifier protocolVersion;
     private String favicon;
     private static final String __OBFID = "CL_00001385";
@@ -61,10 +64,11 @@
         return this.favicon;
     }
 
     public static class MinecraftProtocolVersionIdentifier
         {
+
             private final String name;
             private final int protocol;
             private static final String __OBFID = "CL_00001389";
 
             public MinecraftProtocolVersionIdentifier(String nameIn, int protocolIn)
@@ -83,10 +87,11 @@
                 return this.protocol;
             }
 
             public static class Serializer implements JsonDeserializer, JsonSerializer
                 {
+
                     private static final String __OBFID = "CL_00001390";
 
                     public ServerStatusResponse.MinecraftProtocolVersionIdentifier deserialize(JsonElement p_deserialize_1_, Type p_deserialize_2_, JsonDeserializationContext p_deserialize_3_)
                     {
                         JsonObject jsonobject = JsonUtils.getElementAsJsonObject(p_deserialize_1_, "version");
@@ -108,10 +113,11 @@
                 }
         }
 
     public static class PlayerCountData
         {
+
             private final int maxPlayers;
             private final int onlinePlayerCount;
             private GameProfile[] players;
             private static final String __OBFID = "CL_00001386";
 
@@ -141,10 +147,11 @@
                 this.players = playersIn;
             }
 
             public static class Serializer implements JsonDeserializer, JsonSerializer
                 {
+
                     private static final String __OBFID = "CL_00001387";
 
                     public ServerStatusResponse.PlayerCountData deserialize(JsonElement p_deserialize_1_, Type p_deserialize_2_, JsonDeserializationContext p_deserialize_3_)
                     {
                         JsonObject jsonobject = JsonUtils.getElementAsJsonObject(p_deserialize_1_, "players");
@@ -204,10 +211,11 @@
                 }
         }
 
     public static class Serializer implements JsonDeserializer, JsonSerializer
         {
+
             private static final String __OBFID = "CL_00001388";
 
             public ServerStatusResponse deserialize(JsonElement p_deserialize_1_, Type p_deserialize_2_, JsonDeserializationContext p_deserialize_3_)
             {
                 JsonObject jsonobject = JsonUtils.getElementAsJsonObject(p_deserialize_1_, "status");
