--- decompiled\net\minecraft\network\ThreadQuickExitException.java
+++ src\main\java\net\minecraft\network\ThreadQuickExitException.java
@@ -1,9 +1,10 @@
 package net.minecraft.network;
 
 public final class ThreadQuickExitException extends RuntimeException
 {
+
     public static final ThreadQuickExitException field_179886_a = new ThreadQuickExitException();
     private static final String __OBFID = "CL_00002274";
 
     private ThreadQuickExitException()
     {
