--- decompiled\net\minecraft\item\ItemCoal.java
+++ src\main\java\net\minecraft\item\ItemCoal.java
@@ -1,14 +1,13 @@
 package net.minecraft.item;
 
-import java.util.List;
 import net.minecraft.creativetab.CreativeTabs;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class ItemCoal extends Item
 {
+
     private static final String __OBFID = "CL_00000002";
 
     public ItemCoal()
     {
         this.setHasSubtypes(true);
@@ -18,13 +17,6 @@
 
     public String getUnlocalizedName(ItemStack stack)
     {
         return stack.getMetadata() == 1 ? "item.charcoal" : "item.coal";
     }
-
-    @SideOnly(Side.CLIENT)
-    public void getSubItems(Item itemIn, CreativeTabs tab, List subItems)
-    {
-        subItems.add(new ItemStack(itemIn, 1, 0));
-        subItems.add(new ItemStack(itemIn, 1, 1));
-    }
 }
