--- decompiled\net\minecraft\item\ItemBook.java
+++ src\main\java\net\minecraft\item\ItemBook.java
@@ -1,16 +1,14 @@
 package net.minecraft.item;
 
-public class ItemBook extends Item
-{
+public class ItemBook extends Item {
+
     private static final String __OBFID = "CL_00001775";
 
-    public boolean isItemTool(ItemStack stack)
-    {
+    public boolean isItemTool(ItemStack stack) {
         return stack.stackSize == 1;
     }
 
-    public int getItemEnchantability()
-    {
+    public int getItemEnchantability() {
         return 1;
     }
 }
