--- decompiled\net\minecraft\item\ItemExpBottle.java
+++ src\main\java\net\minecraft\item\ItemExpBottle.java
@@ -3,28 +3,22 @@
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.item.EntityExpBottle;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.stats.StatList;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class ItemExpBottle extends Item
 {
+
     private static final String __OBFID = "CL_00000028";
 
     public ItemExpBottle()
     {
         this.setCreativeTab(CreativeTabs.tabMisc);
     }
 
-    @SideOnly(Side.CLIENT)
-    public boolean hasEffect(ItemStack stack)
-    {
-        return true;
-    }
-
     public ItemStack onItemRightClick(ItemStack itemStackIn, World worldIn, EntityPlayer playerIn)
     {
         if (!playerIn.capabilities.isCreativeMode)
         {
             --itemStackIn.stackSize;
