--- decompiled\net\minecraft\item\ItemLead.java
+++ src\main\java\net\minecraft\item\ItemLead.java
@@ -1,9 +1,10 @@
 package net.minecraft.item;
 
 import java.util.Iterator;
 import java.util.List;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockFence;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.EntityLeashKnot;
 import net.minecraft.entity.EntityLiving;
@@ -11,12 +12,15 @@
 import net.minecraft.util.AxisAlignedBB;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
+import org.bukkit.event.hanging.HangingPlaceEvent;
+
 public class ItemLead extends Item
 {
+
     private static final String __OBFID = "CL_00000045";
 
     public ItemLead()
     {
         this.setCreativeTab(CreativeTabs.tabTools);
@@ -62,12 +66,30 @@
             if (entityliving.getLeashed() && entityliving.getLeashedToEntity() == player)
             {
                 if (entityleashknot == null)
                 {
                     entityleashknot = EntityLeashKnot.createKnot(worldIn, fence);
+                    // CraftBukkit start - fire HangingPlaceEvent
+                    HangingPlaceEvent event = new HangingPlaceEvent((org.bukkit.entity.Hanging)entityleashknot.getBukkitEntity(), player != null ? (org.bukkit.entity.Player)player.getBukkitEntity() : null, worldIn.getWorld().getBlockAt(i, j, k), org.bukkit.block.BlockFace.SELF);
+                    worldIn.getServer().getPluginManager().callEvent(event);
+
+                    if (event.isCancelled())
+                    {
+                        entityleashknot.setDead();
+                        return false;
+                    }
+
+                    // CraftBukkit end
                 }
 
+                // CraftBukkit start
+                if (org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerLeashEntityEvent(entityliving, entityleashknot, player).isCancelled())
+                {
+                    continue;
+                }
+
+                // CraftBukkit end
                 entityliving.setLeashedToEntity(entityleashknot, true);
                 flag = true;
             }
         }
 
