--- decompiled\net\minecraft\item\ItemRecord.java
+++ src\main\java\net\minecraft\item\ItemRecord.java
@@ -1,24 +1,23 @@
 package net.minecraft.item;
 
-import com.google.common.collect.Maps;
-import java.util.List;
 import java.util.Map;
+
 import net.minecraft.block.BlockJukebox;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
-import net.minecraft.util.StatCollector;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.collect.Maps;
+
 public class ItemRecord extends Item
 {
+
     private static final Map RECORDS = Maps.newHashMap();
     public final String recordName;
     private static final String __OBFID = "CL_00000057";
 
     protected ItemRecord(String name)
@@ -51,28 +50,11 @@
         {
             return false;
         }
     }
 
-    @SideOnly(Side.CLIENT)
-    public void addInformation(ItemStack stack, EntityPlayer playerIn, List tooltip, boolean advanced)
-    {
-        tooltip.add(this.getRecordNameLocal());
-    }
-
-    @SideOnly(Side.CLIENT)
-    public String getRecordNameLocal()
-    {
-        return StatCollector.translateToLocal("item.record." + this.recordName + ".desc");
-    }
-
     public EnumRarity getRarity(ItemStack stack)
     {
         return EnumRarity.RARE;
     }
 
-    @SideOnly(Side.CLIENT)
-    public static ItemRecord getRecord(String name)
-    {
-        return (ItemRecord)RECORDS.get(name);
-    }
 }
