--- decompiled\net\minecraft\item\ItemSign.java
+++ src\main\java\net\minecraft\item\ItemSign.java
@@ -10,64 +10,46 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
-public class ItemSign extends Item
-{
+public class ItemSign extends Item {
+
     private static final String __OBFID = "CL_00000064";
 
-    public ItemSign()
-    {
+    public ItemSign() {
         this.maxStackSize = 16;
         this.setCreativeTab(CreativeTabs.tabDecorations);
     }
 
-    public boolean onItemUse(ItemStack stack, EntityPlayer playerIn, World worldIn, BlockPos pos, EnumFacing side, float hitX, float hitY, float hitZ)
-    {
-        if (side == EnumFacing.DOWN)
-        {
+    public boolean onItemUse(ItemStack stack, EntityPlayer playerIn, World worldIn, BlockPos pos, EnumFacing side, float hitX, float hitY, float hitZ) {
+        if (side == EnumFacing.DOWN) {
             return false;
-        }
-        else if (!worldIn.getBlockState(pos).getBlock().getMaterial().isSolid())
-        {
+        } else if (!worldIn.getBlockState(pos).getBlock().getMaterial().isSolid()) {
             return false;
-        }
-        else
-        {
+        } else {
             pos = pos.offset(side);
 
-            if (!playerIn.canPlayerEdit(pos, side, stack))
-            {
+            if (!playerIn.canPlayerEdit(pos, side, stack)) {
                 return false;
-            }
-            else if (!Blocks.standing_sign.canPlaceBlockAt(worldIn, pos))
-            {
+            } else if (!Blocks.standing_sign.canPlaceBlockAt(worldIn, pos)) {
                 return false;
-            }
-            else if (worldIn.isRemote)
-            {
+            } else if (worldIn.isRemote) {
                 return true;
-            }
-            else
-            {
-                if (side == EnumFacing.UP)
-                {
-                    int i = MathHelper.floor_double((double)((playerIn.rotationYaw + 180.0F) * 16.0F / 360.0F) + 0.5D) & 15;
+            } else {
+                if (side == EnumFacing.UP) {
+                    int i = MathHelper.floor_double((double) ((playerIn.rotationYaw + 180.0F) * 16.0F / 360.0F) + 0.5D) & 15;
                     worldIn.setBlockState(pos, Blocks.standing_sign.getDefaultState().withProperty(BlockStandingSign.ROTATION, Integer.valueOf(i)), 3);
-                }
-                else
-                {
+                } else {
                     worldIn.setBlockState(pos, Blocks.wall_sign.getDefaultState().withProperty(BlockWallSign.FACING, side), 3);
                 }
 
                 --stack.stackSize;
                 TileEntity tileentity = worldIn.getTileEntity(pos);
 
-                if (tileentity instanceof TileEntitySign && !ItemBlock.setTileEntityNBT(worldIn, pos, stack))
-                {
-                    playerIn.openEditSign((TileEntitySign)tileentity);
+                if (tileentity instanceof TileEntitySign && !ItemBlock.setTileEntityNBT(worldIn, pos, stack)) {
+                    playerIn.openEditSign((TileEntitySign) tileentity);
                 }
 
                 return true;
             }
         }
