--- decompiled\net\minecraft\item\ItemMultiTexture.java
+++ src\main\java\net\minecraft\item\ItemMultiTexture.java
@@ -1,12 +1,14 @@
 package net.minecraft.item;
 
-import com.google.common.base.Function;
 import net.minecraft.block.Block;
 
+import com.google.common.base.Function;
+
 public class ItemMultiTexture extends ItemBlock
 {
+
     protected final Block theBlock;
     protected final Function nameFunction;
     private static final String __OBFID = "CL_00000051";
 
     public ItemMultiTexture(Block block, Block block2, Function nameFunction)
