--- decompiled\net\minecraft\item\ItemShears.java
+++ src\main\java\net\minecraft\item\ItemShears.java
@@ -6,39 +6,32 @@
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
-public class ItemShears extends Item
-{
+public class ItemShears extends Item {
+
     private static final String __OBFID = "CL_00000062";
 
-    public ItemShears()
-    {
+    public ItemShears() {
         this.setMaxStackSize(1);
         this.setMaxDamage(238);
         this.setCreativeTab(CreativeTabs.tabTools);
     }
 
-    public boolean onBlockDestroyed(ItemStack stack, World worldIn, Block blockIn, BlockPos pos, EntityLivingBase playerIn)
-    {
-        if (blockIn.getMaterial() != Material.leaves && blockIn != Blocks.web && blockIn != Blocks.tallgrass && blockIn != Blocks.vine && blockIn != Blocks.tripwire && blockIn != Blocks.wool)
-        {
+    public boolean onBlockDestroyed(ItemStack stack, World worldIn, Block blockIn, BlockPos pos, EntityLivingBase playerIn) {
+        if (blockIn.getMaterial() != Material.leaves && blockIn != Blocks.web && blockIn != Blocks.tallgrass && blockIn != Blocks.vine && blockIn != Blocks.tripwire && blockIn != Blocks.wool) {
             return super.onBlockDestroyed(stack, worldIn, blockIn, pos, playerIn);
-        }
-        else
-        {
+        } else {
             stack.damageItem(1, playerIn);
             return true;
         }
     }
 
-    public boolean canHarvestBlock(Block blockIn)
-    {
+    public boolean canHarvestBlock(Block blockIn) {
         return blockIn == Blocks.web || blockIn == Blocks.redstone_wire || blockIn == Blocks.tripwire;
     }
 
-    public float getStrVsBlock(ItemStack stack, Block block)
-    {
+    public float getStrVsBlock(ItemStack stack, Block block) {
         return block != Blocks.web && block.getMaterial() != Material.leaves ? (block == Blocks.wool ? 5.0F : super.getStrVsBlock(stack, block)) : 15.0F;
     }
 }
