--- decompiled\net\minecraft\item\ItemBoat.java
+++ src\main\java\net\minecraft\item\ItemBoat.java
@@ -1,8 +1,9 @@
 package net.minecraft.item;
 
 import java.util.List;
+
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.item.EntityBoat;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.init.Blocks;
@@ -14,10 +15,11 @@
 import net.minecraft.util.Vec3;
 import net.minecraft.world.World;
 
 public class ItemBoat extends Item
 {
+
     private static final String __OBFID = "CL_00001774";
 
     public ItemBoat()
     {
         this.maxStackSize = 1;
@@ -77,11 +79,20 @@
             else
             {
                 if (movingobjectposition.typeOfHit == MovingObjectPosition.MovingObjectType.BLOCK)
                 {
                     BlockPos blockpos = movingobjectposition.getBlockPos();
+                    // CraftBukkit start - Boat placement
+                    org.bukkit.event.player.PlayerInteractEvent event = org.bukkit.craftbukkit.event.CraftEventFactory.callPlayerInteractEvent(playerIn, org.bukkit.event.block.Action.RIGHT_CLICK_BLOCK, blockpos, movingobjectposition.sideHit, itemStackIn);
 
+                    if (event.isCancelled())
+                    {
+                        return itemStackIn;
+                    }
+
+                    // CraftBukkit end
+
                     if (worldIn.getBlockState(blockpos).getBlock() == Blocks.snow_layer)
                     {
                         blockpos = blockpos.down();
                     }
 
