--- decompiled\net\minecraft\item\ItemFireball.java
+++ src\main\java\net\minecraft\item\ItemFireball.java
@@ -8,10 +8,11 @@
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
 public class ItemFireball extends Item
 {
+
     private static final String __OBFID = "CL_00000029";
 
     public ItemFireball()
     {
         this.setCreativeTab(CreativeTabs.tabMisc);
@@ -33,10 +34,22 @@
             }
             else
             {
                 if (worldIn.getBlockState(pos).getBlock().getMaterial() == Material.air)
                 {
+                    // CraftBukkit start - fire BlockIgniteEvent
+                    if (org.bukkit.craftbukkit.event.CraftEventFactory.callBlockIgniteEvent(worldIn, pos.getX(), pos.getY(), pos.getZ(), org.bukkit.event.block.BlockIgniteEvent.IgniteCause.FIREBALL, playerIn).isCancelled())
+                    {
+                        if (!playerIn.capabilities.isCreativeMode)
+                        {
+                            --stack.stackSize;
+                        }
+
+                        return false;
+                    }
+
+                    // CraftBukkit end
                     worldIn.playSoundEffect((double)pos.getX() + 0.5D, (double)pos.getY() + 0.5D, (double)pos.getZ() + 0.5D, "item.fireCharge.use", 1.0F, (itemRand.nextFloat() - itemRand.nextFloat()) * 0.2F + 1.0F);
                     worldIn.setBlockState(pos, Blocks.fire.getDefaultState());
                 }
 
                 if (!playerIn.capabilities.isCreativeMode)
