--- decompiled\net\minecraft\item\ItemAppleGold.java
+++ src\main\java\net\minecraft\item\ItemAppleGold.java
@@ -1,32 +1,24 @@
 package net.minecraft.item;
 
-import java.util.List;
-import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.potion.Potion;
 import net.minecraft.potion.PotionEffect;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class ItemAppleGold extends ItemFood
 {
+
     private static final String __OBFID = "CL_00000037";
 
     public ItemAppleGold(int amount, float saturation, boolean isWolfFood)
     {
         super(amount, saturation, isWolfFood);
         this.setHasSubtypes(true);
     }
 
-    @SideOnly(Side.CLIENT)
-    public boolean hasEffect(ItemStack stack)
-    {
-        return stack.getMetadata() > 0;
-    }
-
     public EnumRarity getRarity(ItemStack stack)
     {
         return stack.getMetadata() == 0 ? EnumRarity.RARE : EnumRarity.EPIC;
     }
 
@@ -49,13 +41,6 @@
         else
         {
             super.onFoodEaten(stack, worldIn, player);
         }
     }
-
-    @SideOnly(Side.CLIENT)
-    public void getSubItems(Item itemIn, CreativeTabs tab, List subItems)
-    {
-        subItems.add(new ItemStack(itemIn, 1, 0));
-        subItems.add(new ItemStack(itemIn, 1, 1));
-    }
 }
