--- decompiled\net\minecraft\item\ItemTool.java
+++ src\main\java\net\minecraft\item\ItemTool.java
@@ -1,21 +1,22 @@
 package net.minecraft.item;
 
-import com.google.common.collect.Multimap;
 import java.util.Set;
+
 import net.minecraft.block.Block;
 import net.minecraft.creativetab.CreativeTabs;
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.entity.SharedMonsterAttributes;
 import net.minecraft.entity.ai.attributes.AttributeModifier;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.collect.Multimap;
+
 public class ItemTool extends Item
 {
+
     private Set effectiveBlocks;
     protected float efficiencyOnProperMaterial = 4.0F;
     private float damageVsEntity;
     protected Item.ToolMaterial toolMaterial;
     private static final String __OBFID = "CL_00000019";
@@ -50,16 +51,10 @@
         }
 
         return true;
     }
 
-    @SideOnly(Side.CLIENT)
-    public boolean isFull3D()
-    {
-        return true;
-    }
-
     public Item.ToolMaterial getToolMaterial()
     {
         return this.toolMaterial;
     }
 
