--- decompiled\net\minecraft\item\ItemHangingEntity.java
+++ src\main\java\net\minecraft\item\ItemHangingEntity.java
@@ -7,12 +7,17 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
+import org.bukkit.entity.Player;
+import org.bukkit.event.hanging.HangingPlaceEvent;
+import org.bukkit.event.painting.PaintingPlaceEvent;
+
 public class ItemHangingEntity extends Item
 {
+
     private final Class hangingEntityClass;
     private static final String __OBFID = "CL_00000038";
 
     public ItemHangingEntity(Class entityClass)
     {
@@ -44,10 +49,32 @@
 
                 if (entityhanging != null && entityhanging.onValidSurface())
                 {
                     if (!worldIn.isRemote)
                     {
+                        // CraftBukkit start - fire HangingPlaceEvent
+                        Player who = (playerIn == null) ? null : (Player)playerIn.getBukkitEntity();
+                        org.bukkit.block.Block blockClicked = worldIn.getWorld().getBlockAt(pos.getX(), pos.getY(), pos.getZ());
+                        org.bukkit.block.BlockFace blockFace = org.bukkit.craftbukkit.block.CraftBlock.notchToBlockFace(side);
+                        HangingPlaceEvent event = new HangingPlaceEvent((org.bukkit.entity.Hanging)entityhanging.getBukkitEntity(), who, blockClicked, blockFace);
+                        worldIn.getServer().getPluginManager().callEvent(event);
+                        PaintingPlaceEvent paintingEvent = null;
+
+                        if (entityhanging instanceof EntityPainting)
+                        {
+                            // Fire old painting event until it can be removed
+                            paintingEvent = new PaintingPlaceEvent((org.bukkit.entity.Painting)entityhanging.getBukkitEntity(), who, blockClicked, blockFace);
+                            paintingEvent.setCancelled(event.isCancelled());
+                            worldIn.getServer().getPluginManager().callEvent(paintingEvent);
+                        }
+
+                        if (event.isCancelled() || (paintingEvent != null && paintingEvent.isCancelled()))
+                        {
+                            return false;
+                        }
+
+                        // CraftBukkit end
                         worldIn.spawnEntityInWorld(entityhanging);
                     }
 
                     --stack.stackSize;
                 }
