--- decompiled\net\minecraft\item\ItemEnderEye.java
+++ src\main\java\net\minecraft\item\ItemEnderEye.java
@@ -13,10 +13,11 @@
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraft.world.World;
 
 public class ItemEnderEye extends Item
 {
+
     private static final String __OBFID = "CL_00000026";
 
     public ItemEnderEye()
     {
         this.setCreativeTab(CreativeTabs.tabMisc);
