--- decompiled\net\minecraft\item\ItemGlassBottle.java
+++ src\main\java\net\minecraft\item\ItemGlassBottle.java
@@ -9,10 +9,11 @@
 import net.minecraft.util.MovingObjectPosition;
 import net.minecraft.world.World;
 
 public class ItemGlassBottle extends Item
 {
+
     private static final String __OBFID = "CL_00001776";
 
     public ItemGlassBottle()
     {
         this.setCreativeTab(CreativeTabs.tabBrewing);
