--- decompiled\net\minecraft\item\ItemMinecart.java
+++ src\main\java\net\minecraft\item\ItemMinecart.java
@@ -12,12 +12,19 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
+import org.bukkit.craftbukkit.event.CraftEventFactory;
+import org.bukkit.craftbukkit.inventory.CraftItemStack;
+import org.bukkit.event.block.BlockDispenseEvent;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.util.Vector;
+
 public class ItemMinecart extends Item
 {
+
     private static final IBehaviorDispenseItem dispenserMinecartBehavior = new BehaviorDefaultDispenseItem()
     {
         private final BehaviorDefaultDispenseItem behaviourDefaultDispenseItem = new BehaviorDefaultDispenseItem();
         private static final String __OBFID = "CL_00000050";
         public ItemStack dispenseStack(IBlockSource source, ItemStack stack)
@@ -61,19 +68,52 @@
                 {
                     d3 = -0.9D;
                 }
             }
 
-            EntityMinecart entityminecart = EntityMinecart.func_180458_a(world, d0, d1 + d3, d2, ((ItemMinecart)stack.getItem()).minecartType);
+            // CraftBukkit startx
+            // EntityMinecart entityminecart = EntityMinecart.func_180458_a(world, d0, d1 + d3, d2, ((ItemMinecart) stack.getItem()).minecartType);
+            ItemStack itemstack1 = stack.splitStack(1);
+            org.bukkit.block.Block block2 = world.getWorld().getBlockAt(source.getBlockPos().getX(), source.getBlockPos().getY(), source.getBlockPos().getZ());
+            CraftItemStack craftItem = CraftItemStack.asCraftMirror(itemstack1);
+            BlockDispenseEvent event = new BlockDispenseEvent(block2, craftItem.clone(), new Vector(d0, d1 + d3, d2));
 
+            if (!BlockDispenser.eventFired)
+            {
+                world.getServer().getPluginManager().callEvent(event);
+            }
+
+            if (event.isCancelled())
+            {
+                stack.stackSize++;
+                return stack;
+            }
+
+            if (!event.getItem().equals(craftItem))
+            {
+                stack.stackSize++;
+                // Chain to handler for new item
+                ItemStack eventStack = CraftItemStack.asNMSCopy(event.getItem());
+                IBehaviorDispenseItem idispensebehavior = (IBehaviorDispenseItem)BlockDispenser.dispenseBehaviorRegistry.getObject(eventStack.getItem());
+
+                if (idispensebehavior != IBehaviorDispenseItem.itemDispenseBehaviorProvider && idispensebehavior != this)
+                {
+                    idispensebehavior.dispense(source, eventStack);
+                    return stack;
+                }
+            }
+
+            itemstack1 = CraftItemStack.asNMSCopy(event.getItem());
+            EntityMinecart entityminecartabstract = EntityMinecart.func_180458_a(world, event.getVelocity().getX(), event.getVelocity().getY(), event.getVelocity().getZ(), ((ItemMinecart)itemstack1.getItem()).minecartType); // SpigotMCP -- Fix a
+
             if (stack.hasDisplayName())
             {
-                entityminecart.setCustomNameTag(stack.getDisplayName());
+                entityminecartabstract.setCustomNameTag(stack.getDisplayName());
             }
 
-            world.spawnEntityInWorld(entityminecart);
-            stack.splitStack(1);
+            world.spawnEntityInWorld(entityminecartabstract);
+            // stack.splitStack(1); // CraftBukkit - handled during event processing
             return stack;
         }
         protected void playDispenseSound(IBlockSource source)
         {
             source.getWorld().playAuxSFX(1000, source.getBlockPos(), 0);
@@ -104,10 +144,19 @@
                 if (enumraildirection.isAscending())
                 {
                     d0 = 0.5D;
                 }
 
+                // CraftBukkit start - Minecarts
+                PlayerInteractEvent event = CraftEventFactory.callPlayerInteractEvent(playerIn, org.bukkit.event.block.Action.RIGHT_CLICK_BLOCK, pos, side, stack);
+
+                if (event.isCancelled())
+                {
+                    return false;
+                }
+
+                // CraftBukkit end
                 EntityMinecart entityminecart = EntityMinecart.func_180458_a(worldIn, (double)pos.getX() + 0.5D, (double)pos.getY() + 0.0625D + d0, (double)pos.getZ() + 0.5D, this.minecartType);
 
                 if (stack.hasDisplayName())
                 {
                     entityminecart.setCustomNameTag(stack.getDisplayName());
