--- decompiled\net\minecraft\item\ItemFood.java
+++ src\main\java\net\minecraft\item\ItemFood.java
@@ -6,10 +6,11 @@
 import net.minecraft.stats.StatList;
 import net.minecraft.world.World;
 
 public class ItemFood extends Item
 {
+
     public final int itemUseDuration;
     private final int healAmount;
     private final float saturationModifier;
     private final boolean isWolfsFavoriteMeat;
     private boolean alwaysEdible;
