--- decompiled\net\minecraft\item\ItemDoor.java
+++ src\main\java\net\minecraft\item\ItemDoor.java
@@ -9,10 +9,11 @@
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
 public class ItemDoor extends Item
 {
+
     private Block block;
     private static final String __OBFID = "CL_00000020";
 
     public ItemDoor(Block block)
     {
@@ -68,11 +69,13 @@
             flag2 = true;
         }
 
         BlockPos blockpos3 = pos.up();
         IBlockState iblockstate = door.getDefaultState().withProperty(BlockDoor.FACING, facing).withProperty(BlockDoor.HINGE, flag2 ? BlockDoor.EnumHingePosition.RIGHT : BlockDoor.EnumHingePosition.LEFT);
-        worldIn.setBlockState(pos, iblockstate.withProperty(BlockDoor.HALF, BlockDoor.EnumDoorHalf.LOWER), 2);
-        worldIn.setBlockState(blockpos3, iblockstate.withProperty(BlockDoor.HALF, BlockDoor.EnumDoorHalf.UPPER), 2);
-        worldIn.notifyNeighborsOfStateChange(pos, door);
-        worldIn.notifyNeighborsOfStateChange(blockpos3, door);
+        // Spigot start - update physics after the block multi place event
+        worldIn.setBlockState(pos, iblockstate.withProperty(BlockDoor.HALF, BlockDoor.EnumDoorHalf.LOWER), 3);
+        worldIn.setBlockState(blockpos3, iblockstate.withProperty(BlockDoor.HALF, BlockDoor.EnumDoorHalf.UPPER), 3);
+        // worldIn.notifyNeighborsOfStateChange(pos, door);
+        // worldIn.notifyNeighborsOfStateChange(blockpos3, door);
+        // Spigot end
     }
 }
