--- decompiled\net\minecraft\item\ItemSeedFood.java
+++ src\main\java\net\minecraft\item\ItemSeedFood.java
@@ -6,10 +6,11 @@
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
 public class ItemSeedFood extends ItemFood
 {
+
     private Block crops;
     private Block soilId;
     private static final String __OBFID = "CL_00000060";
 
     public ItemSeedFood(int healAmount, float saturation, Block crops, Block soil)
