--- decompiled\net\minecraft\item\ItemFishFood.java
+++ src\main\java\net\minecraft\item\ItemFishFood.java
@@ -1,21 +1,20 @@
 package net.minecraft.item;
 
-import com.google.common.collect.Maps;
-import java.util.List;
 import java.util.Map;
-import net.minecraft.creativetab.CreativeTabs;
+
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.potion.Potion;
 import net.minecraft.potion.PotionEffect;
 import net.minecraft.potion.PotionHelper;
 import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.collect.Maps;
+
 public class ItemFishFood extends ItemFood
 {
+
     private final boolean cooked;
     private static final String __OBFID = "CL_00000032";
 
     public ItemFishFood(boolean cooked)
     {
@@ -52,27 +51,10 @@
         }
 
         super.onFoodEaten(stack, worldIn, player);
     }
 
-    @SideOnly(Side.CLIENT)
-    public void getSubItems(Item itemIn, CreativeTabs tab, List subItems)
-    {
-        ItemFishFood.FishType[] afishtype = ItemFishFood.FishType.values();
-        int i = afishtype.length;
-
-        for (int j = 0; j < i; ++j)
-        {
-            ItemFishFood.FishType fishtype = afishtype[j];
-
-            if (!this.cooked || fishtype.canCook())
-            {
-                subItems.add(new ItemStack(this, 1, fishtype.getMetadata()));
-            }
-        }
-    }
-
     public String getUnlocalizedName(ItemStack stack)
     {
         ItemFishFood.FishType fishtype = ItemFishFood.FishType.byItemStack(stack);
         return this.getUnlocalizedName() + "." + fishtype.getUnlocalizedName() + "." + (this.cooked && fishtype.canCook() ? "cooked" : "raw");
     }
