--- decompiled\net\minecraft\profiler\IPlayerUsage.java
+++ src\main\java\net\minecraft\profiler\IPlayerUsage.java
@@ -1,9 +1,9 @@
 package net.minecraft.profiler;
 
-public interface IPlayerUsage
-{
+public interface IPlayerUsage {
+
     void addServerStatsToSnooper(PlayerUsageSnooper playerSnooper);
 
     void addServerTypeToSnooper(PlayerUsageSnooper playerSnooper);
 
     boolean isSnooperEnabled();
