--- decompiled\net\minecraft\command\common\CommandReplaceItem.java
+++ src\main\java\net\minecraft\command\common\CommandReplaceItem.java
@@ -1,10 +1,10 @@
 package net.minecraft.command.common;
 
-import com.google.common.collect.Maps;
 import java.util.List;
 import java.util.Map;
+
 import net.minecraft.block.Block;
 import net.minecraft.command.CommandBase;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.CommandResultStats;
 import net.minecraft.command.ICommandSender;
@@ -21,12 +21,15 @@
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
+import com.google.common.collect.Maps;
+
 public class CommandReplaceItem extends CommandBase
 {
+
     private static final Map SHORTCUTS = Maps.newHashMap();
     private static final String __OBFID = "CL_00002340";
 
     public String getName()
     {
@@ -187,11 +190,12 @@
         }
     }
 
     public List addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos)
     {
-        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"entity", "block"}): (args.length == 2 && args[0].equals("entity") ? getListOfStringsMatchingLastWord(args, this.getUsernames()) : ((args.length != 3 || !args[0].equals("entity")) && (args.length != 5 || !args[0].equals("block")) ? ((args.length != 4 || !args[0].equals("entity")) && (args.length != 6 || !args[0].equals("block")) ? null : func_175762_a(args, Item.itemRegistry.getKeys())) : func_175762_a(args, SHORTCUTS.keySet())));
+        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"entity", "block"}) : (args.length == 2 && args[0].equals("entity") ? getListOfStringsMatchingLastWord(args, this.getUsernames()) : ((args.length != 3 || !args[0].equals("entity")) && (args.length != 5 || !args[0].equals("block")) ? ((args.length != 4 || !args[0].equals("entity")) && (args.length != 6 || !args[0].equals("block")) ? null : func_175762_a(args, Item.itemRegistry
+                .getKeys())) : func_175762_a(args, SHORTCUTS.keySet())));
     }
 
     protected String[] getUsernames()
     {
         return MinecraftServer.getServer().getAllUsernames();
