--- decompiled\net\minecraft\command\CommandToggleDownfall.java
+++ src\main\java\net\minecraft\command\CommandToggleDownfall.java
@@ -1,36 +1,31 @@
 package net.minecraft.command;
 
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.world.storage.WorldInfo;
 
-public class CommandToggleDownfall extends CommandBase
-{
+public class CommandToggleDownfall extends CommandBase {
+
     private static final String __OBFID = "CL_00001184";
 
-    public String getName()
-    {
+    public String getName() {
         return "toggledownfall";
     }
 
-    public int getRequiredPermissionLevel()
-    {
+    public int getRequiredPermissionLevel() {
         return 2;
     }
 
-    public String getCommandUsage(ICommandSender sender)
-    {
+    public String getCommandUsage(ICommandSender sender) {
         return "commands.downfall.usage";
     }
 
-    public void execute(ICommandSender sender, String[] args) throws CommandException
-    {
+    public void execute(ICommandSender sender, String[] args) throws CommandException {
         this.toggleDownfall();
         notifyOperators(sender, this, "commands.downfall.success", new Object[0]);
     }
 
-    protected void toggleDownfall()
-    {
+    protected void toggleDownfall() {
         WorldInfo worldinfo = MinecraftServer.getServer().worldServers[0].getWorldInfo();
         worldinfo.setRaining(!worldinfo.isRaining());
     }
 }
