--- decompiled\net\minecraft\command\CommandPlaySound.java
+++ src\main\java\net\minecraft\command\CommandPlaySound.java
@@ -1,16 +1,18 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.network.play.server.S29PacketSoundEffect;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.Vec3;
 
 public class CommandPlaySound extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000774";
 
     public String getName()
     {
         return "playsound";
