--- decompiled\net\minecraft\command\CommandParticle.java
+++ src\main\java\net\minecraft\command\CommandParticle.java
@@ -1,16 +1,18 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumParticleTypes;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.World;
 import net.minecraft.world.WorldServer;
 
 public class CommandParticle extends CommandBase
 {
+
     private static final String __OBFID = "CL_00002341";
 
     public String getName()
     {
         return "particle";
@@ -120,8 +122,8 @@
         }
     }
 
     public List addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos)
     {
-        return args.length == 1 ? getListOfStringsMatchingLastWord(args, EnumParticleTypes.func_179349_a()) : (args.length > 1 && args.length <= 4 ? func_175771_a(args, 1, pos) : (args.length == 9 ? getListOfStringsMatchingLastWord(args, new String[] {"normal", "force"}): null));
+        return args.length == 1 ? getListOfStringsMatchingLastWord(args, EnumParticleTypes.func_179349_a()) : (args.length > 1 && args.length <= 4 ? func_175771_a(args, 1, pos) : (args.length == 9 ? getListOfStringsMatchingLastWord(args, new String[] {"normal", "force"}) : null));
     }
 }
