--- decompiled\net\minecraft\command\CommandXP.java
+++ src\main\java\net\minecraft\command\CommandXP.java
@@ -1,14 +1,16 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 
 public class CommandXP extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000398";
 
     public String getName()
     {
         return "xp";
