--- decompiled\net\minecraft\command\CommandHelp.java
+++ src\main\java\net\minecraft\command\CommandHelp.java
@@ -3,20 +3,22 @@
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.event.ClickEvent;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.ChatComponentTranslation;
 import net.minecraft.util.EnumChatFormatting;
 import net.minecraft.util.MathHelper;
 
 public class CommandHelp extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000529";
 
     public String getName()
     {
         return "help";
