--- decompiled\net\minecraft\command\CommandDebug.java
+++ src\main\java\net\minecraft\command\CommandDebug.java
@@ -3,18 +3,21 @@
 import java.io.File;
 import java.io.FileWriter;
 import java.text.SimpleDateFormat;
 import java.util.Date;
 import java.util.List;
+
 import net.minecraft.profiler.Profiler;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
+
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
 public class CommandDebug extends CommandBase
 {
+
     private static final Logger logger = LogManager.getLogger();
     private long field_147206_b;
     private int field_147207_c;
     private static final String __OBFID = "CL_00000270";
 
@@ -151,11 +154,27 @@
         }
     }
 
     private static String func_147203_d()
     {
-        String[] astring = new String[] {"Shiny numbers!", "Am I not running fast enough? :(", "I\'m working as hard as I can!", "Will I ever be good enough for you? :(", "Speedy. Zoooooom!", "Hello world", "40% better than a crash report.", "Now with extra numbers", "Now with less numbers", "Now with the same numbers", "You should add flames to things, it makes them go faster!", "Do you feel the need for... optimization?", "*cracks redstone whip*", "Maybe if you treated it better then it\'ll have more motivation to work faster! Poor server."};
+        String[] astring = new String[]
+        {
+            "Shiny numbers!",
+            "Am I not running fast enough? :(",
+            "I\'m working as hard as I can!",
+            "Will I ever be good enough for you? :(",
+            "Speedy. Zoooooom!",
+            "Hello world",
+            "40% better than a crash report.",
+            "Now with extra numbers",
+            "Now with less numbers",
+            "Now with the same numbers",
+            "You should add flames to things, it makes them go faster!",
+            "Do you feel the need for... optimization?",
+            "*cracks redstone whip*",
+            "Maybe if you treated it better then it\'ll have more motivation to work faster! Poor server."
+        };
 
         try
         {
             return astring[(int)(System.nanoTime() % (long)astring.length)];
         }
@@ -165,8 +184,8 @@
         }
     }
 
     public List addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos)
     {
-        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"start", "stop"}): null;
+        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"start", "stop"}) : null;
     }
 }
