--- decompiled\net\minecraft\command\CommandWorldBorder.java
+++ src\main\java\net\minecraft\command\CommandWorldBorder.java
@@ -1,16 +1,18 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.ChatComponentTranslation;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.border.WorldBorder;
 
 public class CommandWorldBorder extends CommandBase
 {
+
     private static final String __OBFID = "CL_00002336";
 
     public String getName()
     {
         return "worldborder";
@@ -54,21 +56,21 @@
                 {
                     worldborder.setTransition(d0, d2, i);
 
                     if (d0 > d2)
                     {
-                        notifyOperators(sender, this, "commands.worldborder.setSlowly.shrink.success", new Object[] {String.format("%.1f", new Object[]{Double.valueOf(d2)}), String.format("%.1f", new Object[]{Double.valueOf(d0)}), Long.toString(i / 1000L)});
+                        notifyOperators(sender, this, "commands.worldborder.setSlowly.shrink.success", new Object[] {String.format("%.1f", new Object[] {Double.valueOf(d2)}), String.format("%.1f", new Object[] {Double.valueOf(d0)}), Long.toString(i / 1000L)});
                     }
                     else
                     {
-                        notifyOperators(sender, this, "commands.worldborder.setSlowly.grow.success", new Object[] {String.format("%.1f", new Object[]{Double.valueOf(d2)}), String.format("%.1f", new Object[]{Double.valueOf(d0)}), Long.toString(i / 1000L)});
+                        notifyOperators(sender, this, "commands.worldborder.setSlowly.grow.success", new Object[] {String.format("%.1f", new Object[] {Double.valueOf(d2)}), String.format("%.1f", new Object[] {Double.valueOf(d0)}), Long.toString(i / 1000L)});
                     }
                 }
                 else
                 {
                     worldborder.setTransition(d2);
-                    notifyOperators(sender, this, "commands.worldborder.set.success", new Object[] {String.format("%.1f", new Object[]{Double.valueOf(d2)}), String.format("%.1f", new Object[]{Double.valueOf(d0)})});
+                    notifyOperators(sender, this, "commands.worldborder.set.success", new Object[] {String.format("%.1f", new Object[] {Double.valueOf(d2)}), String.format("%.1f", new Object[] {Double.valueOf(d0)})});
                 }
             }
             else if (args[0].equals("add"))
             {
                 if (args.length != 2 && args.length != 3)
@@ -84,21 +86,21 @@
                 {
                     worldborder.setTransition(d0, d2, i);
 
                     if (d0 > d2)
                     {
-                        notifyOperators(sender, this, "commands.worldborder.setSlowly.shrink.success", new Object[] {String.format("%.1f", new Object[]{Double.valueOf(d2)}), String.format("%.1f", new Object[]{Double.valueOf(d0)}), Long.toString(i / 1000L)});
+                        notifyOperators(sender, this, "commands.worldborder.setSlowly.shrink.success", new Object[] {String.format("%.1f", new Object[] {Double.valueOf(d2)}), String.format("%.1f", new Object[] {Double.valueOf(d0)}), Long.toString(i / 1000L)});
                     }
                     else
                     {
-                        notifyOperators(sender, this, "commands.worldborder.setSlowly.grow.success", new Object[] {String.format("%.1f", new Object[]{Double.valueOf(d2)}), String.format("%.1f", new Object[]{Double.valueOf(d0)}), Long.toString(i / 1000L)});
+                        notifyOperators(sender, this, "commands.worldborder.setSlowly.grow.success", new Object[] {String.format("%.1f", new Object[] {Double.valueOf(d2)}), String.format("%.1f", new Object[] {Double.valueOf(d0)}), Long.toString(i / 1000L)});
                     }
                 }
                 else
                 {
                     worldborder.setTransition(d2);
-                    notifyOperators(sender, this, "commands.worldborder.set.success", new Object[] {String.format("%.1f", new Object[]{Double.valueOf(d2)}), String.format("%.1f", new Object[]{Double.valueOf(d0)})});
+                    notifyOperators(sender, this, "commands.worldborder.set.success", new Object[] {String.format("%.1f", new Object[] {Double.valueOf(d2)}), String.format("%.1f", new Object[] {Double.valueOf(d0)})});
                 }
             }
             else if (args[0].equals("center"))
             {
                 if (args.length != 3)
@@ -127,11 +129,11 @@
                     }
 
                     d0 = parseDouble(args[2], 0.0D);
                     d2 = worldborder.getDamageBuffer();
                     worldborder.setDamageBuffer(d0);
-                    notifyOperators(sender, this, "commands.worldborder.damage.buffer.success", new Object[] {String.format("%.1f", new Object[]{Double.valueOf(d0)}), String.format("%.1f", new Object[]{Double.valueOf(d2)})});
+                    notifyOperators(sender, this, "commands.worldborder.damage.buffer.success", new Object[] {String.format("%.1f", new Object[] {Double.valueOf(d0)}), String.format("%.1f", new Object[] {Double.valueOf(d2)})});
                 }
                 else if (args[1].equals("amount"))
                 {
                     if (args.length != 3)
                     {
@@ -139,11 +141,11 @@
                     }
 
                     d0 = parseDouble(args[2], 0.0D);
                     d2 = worldborder.getDamageAmount();
                     worldborder.setDamageAmount(d0);
-                    notifyOperators(sender, this, "commands.worldborder.damage.amount.success", new Object[] {String.format("%.2f", new Object[]{Double.valueOf(d0)}), String.format("%.2f", new Object[]{Double.valueOf(d2)})});
+                    notifyOperators(sender, this, "commands.worldborder.damage.amount.success", new Object[] {String.format("%.2f", new Object[] {Double.valueOf(d0)}), String.format("%.2f", new Object[] {Double.valueOf(d2)})});
                 }
             }
             else if (args[0].equals("warning"))
             {
                 if (args.length < 2)
@@ -179,11 +181,11 @@
             }
             else if (args[0].equals("get"))
             {
                 d0 = worldborder.getDiameter();
                 sender.setCommandStat(CommandResultStats.Type.QUERY_RESULT, MathHelper.floor_double(d0 + 0.5D));
-                sender.addChatMessage(new ChatComponentTranslation("commands.worldborder.get.success", new Object[] {String.format("%.0f", new Object[]{Double.valueOf(d0)})}));
+                sender.addChatMessage(new ChatComponentTranslation("commands.worldborder.get.success", new Object[] {String.format("%.0f", new Object[] {Double.valueOf(d0)})}));
             }
         }
     }
 
     protected WorldBorder getWorldBorder()
@@ -191,8 +193,8 @@
         return MinecraftServer.getServer().worldServers[0].getWorldBorder();
     }
 
     public List addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos)
     {
-        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"set", "center", "damage", "warning", "add", "get"}): (args.length == 2 && args[0].equals("damage") ? getListOfStringsMatchingLastWord(args, new String[] {"buffer", "amount"}): (args.length == 2 && args[0].equals("warning") ? getListOfStringsMatchingLastWord(args, new String[] {"time", "distance"}): null));
+        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"set", "center", "damage", "warning", "add", "get"}) : (args.length == 2 && args[0].equals("damage") ? getListOfStringsMatchingLastWord(args, new String[] {"buffer", "amount"}) : (args.length == 2 && args[0].equals("warning") ? getListOfStringsMatchingLastWord(args, new String[] {"time", "distance"}) : null));
     }
 }
