--- decompiled\net\minecraft\command\CommandToggleDownfall.java
+++ src\main\java\net\minecraft\command\CommandToggleDownfall.java
@@ -3,10 +3,11 @@
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.world.storage.WorldInfo;
 
 public class CommandToggleDownfall extends CommandBase
 {
+
     private static final String __OBFID = "CL_00001184";
 
     public String getName()
     {
         return "toggledownfall";
