--- decompiled\net\minecraft\command\CommandGive.java
+++ src\main\java\net\minecraft\command\CommandGive.java
@@ -1,8 +1,9 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.entity.item.EntityItem;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.item.Item;
 import net.minecraft.item.ItemStack;
 import net.minecraft.nbt.JsonToNBT;
@@ -10,10 +11,11 @@
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 
 public class CommandGive extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000502";
 
     public String getName()
     {
         return "give";
