--- decompiled\net\minecraft\command\CommandTitle.java
+++ src\main\java\net\minecraft\command\CommandTitle.java
@@ -1,21 +1,25 @@
 package net.minecraft.command;
 
-import com.google.gson.JsonParseException;
 import java.util.List;
+
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.network.play.server.S45PacketTitle;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.ChatComponentProcessor;
 import net.minecraft.util.IChatComponent;
+
 import org.apache.commons.lang3.exception.ExceptionUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import com.google.gson.JsonParseException;
+
 public class CommandTitle extends CommandBase
 {
+
     private static final Logger LOGGER = LogManager.getLogger();
     private static final String __OBFID = "CL_00002338";
 
     public String getName()
     {
