--- decompiled\net\minecraft\command\CommandTime.java
+++ src\main\java\net\minecraft\command\CommandTime.java
@@ -1,14 +1,16 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.WorldServer;
 
 public class CommandTime extends CommandBase
 {
+
     private static final String __OBFID = "CL_00001183";
 
     public String getName()
     {
         return "time";
@@ -81,11 +83,11 @@
         throw new WrongUsageException("commands.time.usage", new Object[0]);
     }
 
     public List addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos)
     {
-        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"set", "add", "query"}): (args.length == 2 && args[0].equals("set") ? getListOfStringsMatchingLastWord(args, new String[] {"day", "night"}): (args.length == 2 && args[0].equals("query") ? getListOfStringsMatchingLastWord(args, new String[] {"daytime", "gametime"}): null));
+        return args.length == 1 ? getListOfStringsMatchingLastWord(args, new String[] {"set", "add", "query"}) : (args.length == 2 && args[0].equals("set") ? getListOfStringsMatchingLastWord(args, new String[] {"day", "night"}) : (args.length == 2 && args[0].equals("query") ? getListOfStringsMatchingLastWord(args, new String[] {"daytime", "gametime"}) : null));
     }
 
     protected void setTime(ICommandSender p_71552_1_, int p_71552_2_)
     {
         for (int j = 0; j < MinecraftServer.getServer().worldServers.length; ++j)
