--- decompiled\net\minecraft\command\CommandClearInventory.java
+++ src\main\java\net\minecraft\command\CommandClearInventory.java
@@ -1,8 +1,9 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.item.Item;
 import net.minecraft.nbt.JsonToNBT;
 import net.minecraft.nbt.NBTException;
 import net.minecraft.nbt.NBTTagCompound;
@@ -10,10 +11,11 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.ChatComponentTranslation;
 
 public class CommandClearInventory extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000218";
 
     public String getName()
     {
         return "clear";
