--- decompiled\net\minecraft\command\CommandEntityData.java
+++ src\main\java\net\minecraft\command\CommandEntityData.java
@@ -1,18 +1,20 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.nbt.JsonToNBT;
 import net.minecraft.nbt.NBTException;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 
 public class CommandEntityData extends CommandBase
 {
+
     private static final String __OBFID = "CL_00002345";
 
     public String getName()
     {
         return "entitydata";
