--- decompiled\net\minecraft\command\CommandDefaultGameMode.java
+++ src\main\java\net\minecraft\command\CommandDefaultGameMode.java
@@ -1,15 +1,17 @@
 package net.minecraft.command;
 
 import java.util.Iterator;
+
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.ChatComponentTranslation;
 import net.minecraft.world.WorldSettings;
 
 public class CommandDefaultGameMode extends CommandGameMode
 {
+
     private static final String __OBFID = "CL_00000296";
 
     public String getName()
     {
         return "defaultgamemode";
