--- decompiled\net\minecraft\command\CommandResultStats.java
+++ src\main\java\net\minecraft\command\CommandResultStats.java
@@ -5,10 +5,11 @@
 import net.minecraft.scoreboard.ScoreObjective;
 import net.minecraft.scoreboard.Scoreboard;
 
 public class CommandResultStats
 {
+
     private static final int NUM_RESULT_TYPES = CommandResultStats.Type.values().length;
     private static final String[] field_179674_b = new String[NUM_RESULT_TYPES];
     private String[] field_179675_c;
     private String[] field_179673_d;
     private static final String __OBFID = "CL_00002364";
