--- decompiled\net\minecraft\command\CommandNotFoundException.java
+++ src\main\java\net\minecraft\command\CommandNotFoundException.java
@@ -1,16 +1,17 @@
 package net.minecraft.command;
 
 public class CommandNotFoundException extends CommandException
 {
+
     private static final String __OBFID = "CL_00001191";
 
     public CommandNotFoundException()
     {
         this("commands.generic.notFound", new Object[0]);
     }
 
-    public CommandNotFoundException(String p_i1363_1_, Object ... p_i1363_2_)
+    public CommandNotFoundException(String p_i1363_1_, Object... p_i1363_2_)
     {
         super(p_i1363_1_, p_i1363_2_);
     }
 }
