--- decompiled\net\minecraft\command\CommandEffect.java
+++ src\main\java\net\minecraft\command\CommandEffect.java
@@ -1,17 +1,19 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.potion.Potion;
 import net.minecraft.potion.PotionEffect;
 import net.minecraft.server.MinecraftServer;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.ChatComponentTranslation;
 
 public class CommandEffect extends CommandBase
 {
+
     private static final String __OBFID = "CL_00000323";
 
     public String getName()
     {
         return "effect";
@@ -131,11 +133,11 @@
         }
     }
 
     public List addTabCompletionOptions(ICommandSender sender, String[] args, BlockPos pos)
     {
-        return args.length == 1 ? getListOfStringsMatchingLastWord(args, this.getAllUsernames()) : (args.length == 2 ? getListOfStringsMatchingLastWord(args, Potion.getPotionMapAsArray()) : (args.length == 5 ? getListOfStringsMatchingLastWord(args, new String[] {"true", "false"}): null));
+        return args.length == 1 ? getListOfStringsMatchingLastWord(args, this.getAllUsernames()) : (args.length == 2 ? getListOfStringsMatchingLastWord(args, Potion.getPotionMapAsArray()) : (args.length == 5 ? getListOfStringsMatchingLastWord(args, new String[] {"true", "false"}) : null));
     }
 
     protected String[] getAllUsernames()
     {
         return MinecraftServer.getServer().getAllUsernames();
