--- decompiled\net\minecraft\command\CommandShowSeed.java
+++ src\main\java\net\minecraft\command\CommandShowSeed.java
@@ -5,10 +5,11 @@
 import net.minecraft.util.ChatComponentTranslation;
 import net.minecraft.world.World;
 
 public class CommandShowSeed extends CommandBase
 {
+
     private static final String __OBFID = "CL_00001053";
 
     public boolean canCommandSenderUse(ICommandSender sender)
     {
         return MinecraftServer.getServer().isSinglePlayer() || super.canCommandSenderUse(sender);
