--- decompiled\net\minecraft\command\ICommandSender.java
+++ src\main\java\net\minecraft\command\ICommandSender.java
@@ -4,12 +4,12 @@
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.IChatComponent;
 import net.minecraft.util.Vec3;
 import net.minecraft.world.World;
 
-public interface ICommandSender
-{
+public interface ICommandSender {
+
     String getName();
 
     IChatComponent getDisplayName();
 
     void addChatMessage(IChatComponent message);
