--- decompiled\net\minecraft\command\ICommandManager.java
+++ src\main\java\net\minecraft\command\ICommandManager.java
@@ -1,13 +1,14 @@
 package net.minecraft.command;
 
 import java.util.List;
 import java.util.Map;
+
 import net.minecraft.util.BlockPos;
 
-public interface ICommandManager
-{
+public interface ICommandManager {
+
     int executeCommand(ICommandSender sender, String rawCommand);
 
     List getTabCompletionOptions(ICommandSender sender, String input, BlockPos pos);
 
     List getPossibleCommands(ICommandSender sender);
