--- decompiled\net\minecraft\command\ServerCommandManager.java
+++ src\main\java\net\minecraft\command\ServerCommandManager.java
@@ -1,8 +1,9 @@
 package net.minecraft.command;
 
 import java.util.Iterator;
+
 import net.minecraft.command.common.CommandReplaceItem;
 import net.minecraft.command.server.CommandAchievement;
 import net.minecraft.command.server.CommandBanIp;
 import net.minecraft.command.server.CommandBanPlayer;
 import net.minecraft.command.server.CommandBlockLogic;
@@ -34,10 +35,11 @@
 import net.minecraft.util.ChatComponentTranslation;
 import net.minecraft.util.EnumChatFormatting;
 
 public class ServerCommandManager extends CommandHandler implements IAdminCommand
 {
+
     private static final String __OBFID = "CL_00000922";
 
     public ServerCommandManager()
     {
         this.registerCommand(new CommandTime());
@@ -108,11 +110,11 @@
         }
 
         CommandBase.setAdminCommander(this);
     }
 
-    public void notifyOperators(ICommandSender sender, ICommand command, int p_152372_3_, String msgFormat, Object ... msgParams)
+    public void notifyOperators(ICommandSender sender, ICommand command, int p_152372_3_, String msgFormat, Object... msgParams)
     {
         boolean flag = true;
         MinecraftServer minecraftserver = MinecraftServer.getServer();
 
         if (!sender.sendCommandFeedback())
@@ -137,11 +139,11 @@
                     entityplayer.addChatMessage(chatcomponenttranslation);
                 }
             }
         }
 
-        if (sender != minecraftserver && minecraftserver.worldServers[0].getGameRules().getGameRuleBooleanValue("logAdminCommands"))
+        if (sender != minecraftserver && minecraftserver.worldServers[0].getGameRules().getGameRuleBooleanValue("logAdminCommands") && !org.spigotmc.SpigotConfig.silentCommandBlocks)   // Spigot        {
         {
             minecraftserver.addChatMessage(chatcomponenttranslation);
         }
 
         boolean flag1 = minecraftserver.worldServers[0].getGameRules().getGameRuleBooleanValue("sendCommandFeedback");
