--- decompiled\net\minecraft\network\play\server\S00PacketKeepAlive.java
+++ src\main\java\net\minecraft\network\play\server\S00PacketKeepAlive.java
@@ -1,21 +1,24 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class S00PacketKeepAlive implements Packet
 {
+
     private int field_149136_a;
     private static final String __OBFID = "CL_00001303";
 
-    public S00PacketKeepAlive() {}
+    public S00PacketKeepAlive()
+    {
+    }
 
     public S00PacketKeepAlive(int p_i45195_1_)
     {
         this.field_149136_a = p_i45195_1_;
     }
@@ -33,16 +36,10 @@
     public void writePacketData(PacketBuffer buf) throws IOException
     {
         buf.writeVarIntToBuffer(this.field_149136_a);
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_149134_c()
-    {
-        return this.field_149136_a;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
 }
