--- decompiled\net\minecraft\network\play\server\S03PacketTimeUpdate.java
+++ src\main\java\net\minecraft\network\play\server\S03PacketTimeUpdate.java
@@ -1,22 +1,25 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class S03PacketTimeUpdate implements Packet
 {
+
     private long field_149369_a;
     private long field_149368_b;
     private static final String __OBFID = "CL_00001337";
 
-    public S03PacketTimeUpdate() {}
+    public S03PacketTimeUpdate()
+    {
+    }
 
     public S03PacketTimeUpdate(long p_i45230_1_, long p_i45230_3_, boolean p_i45230_5_)
     {
         this.field_149369_a = p_i45230_1_;
         this.field_149368_b = p_i45230_3_;
@@ -47,22 +50,11 @@
     public void processPacket(INetHandlerPlayClient handler)
     {
         handler.handleTimeUpdate(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public long func_149366_c()
-    {
-        return this.field_149369_a;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
 
-    @SideOnly(Side.CLIENT)
-    public long func_149365_d()
-    {
-        return this.field_149368_b;
-    }
 }
