--- decompiled\net\minecraft\network\play\server\S0APacketUseBed.java
+++ src\main\java\net\minecraft\network\play\server\S0APacketUseBed.java
@@ -1,61 +1,44 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraft.util.BlockPos;
-import net.minecraft.world.World;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
-public class S0APacketUseBed implements Packet
-{
+
+public class S0APacketUseBed implements Packet {
+
     private int playerID;
     private BlockPos bedPos;
     private static final String __OBFID = "CL_00001319";
 
-    public S0APacketUseBed() {}
+    public S0APacketUseBed() {
+    }
 
-    public S0APacketUseBed(EntityPlayer p_i45964_1_, BlockPos p_i45964_2_)
-    {
+    public S0APacketUseBed(EntityPlayer p_i45964_1_, BlockPos p_i45964_2_) {
         this.playerID = p_i45964_1_.getEntityId();
         this.bedPos = p_i45964_2_;
     }
 
-    public void readPacketData(PacketBuffer buf) throws IOException
-    {
+    public void readPacketData(PacketBuffer buf) throws IOException {
         this.playerID = buf.readVarIntFromBuffer();
         this.bedPos = buf.readBlockPos();
     }
 
-    public void writePacketData(PacketBuffer buf) throws IOException
-    {
+    public void writePacketData(PacketBuffer buf) throws IOException {
         buf.writeVarIntToBuffer(this.playerID);
         buf.writeBlockPos(this.bedPos);
     }
 
-    public void func_180744_a(INetHandlerPlayClient p_180744_1_)
-    {
+    public void func_180744_a(INetHandlerPlayClient p_180744_1_) {
         p_180744_1_.handleUseBed(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public EntityPlayer getPlayer(World worldIn)
-    {
-        return (EntityPlayer)worldIn.getEntityByID(this.playerID);
+    public void processPacket(INetHandler handler) {
+        this.func_180744_a((INetHandlerPlayClient) handler);
     }
-
-    @SideOnly(Side.CLIENT)
-    public BlockPos func_179798_a()
-    {
-        return this.bedPos;
-    }
-
-    public void processPacket(INetHandler handler)
-    {
-        this.func_180744_a((INetHandlerPlayClient)handler);
-    }
 }
