--- decompiled\net\minecraft\network\play\server\S0EPacketSpawnObject.java
+++ src\main\java\net\minecraft\network\play\server\S0EPacketSpawnObject.java
@@ -1,19 +1,20 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.entity.Entity;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraft.util.MathHelper;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class S0EPacketSpawnObject implements Packet
 {
+
     private int field_149018_a;
     private int field_149016_b;
     private int field_149017_c;
     private int field_149014_d;
     private int field_149015_e;
@@ -23,11 +24,13 @@
     private int field_149022_i;
     private int field_149019_j;
     private int field_149020_k;
     private static final String __OBFID = "CL_00001276";
 
-    public S0EPacketSpawnObject() {}
+    public S0EPacketSpawnObject()
+    {
+    }
 
     public S0EPacketSpawnObject(Entity p_i45165_1_, int p_i45165_2_)
     {
         this(p_i45165_1_, p_i45165_2_, 0);
     }
@@ -127,112 +130,41 @@
     public void processPacket(INetHandlerPlayClient handler)
     {
         handler.handleSpawnObject(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_149001_c()
-    {
-        return this.field_149018_a;
-    }
-
     public void func_148996_a(int p_148996_1_)
     {
         this.field_149016_b = p_148996_1_;
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_148997_d()
-    {
-        return this.field_149016_b;
-    }
-
     public void func_148995_b(int p_148995_1_)
     {
         this.field_149017_c = p_148995_1_;
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_148998_e()
-    {
-        return this.field_149017_c;
-    }
-
     public void func_149005_c(int p_149005_1_)
     {
         this.field_149014_d = p_149005_1_;
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_148994_f()
-    {
-        return this.field_149014_d;
-    }
-
     public void func_149003_d(int p_149003_1_)
     {
         this.field_149015_e = p_149003_1_;
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_149010_g()
-    {
-        return this.field_149015_e;
-    }
-
     public void func_149000_e(int p_149000_1_)
     {
         this.field_149012_f = p_149000_1_;
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_149004_h()
-    {
-        return this.field_149012_f;
-    }
 
     public void func_149007_f(int p_149007_1_)
     {
         this.field_149013_g = p_149007_1_;
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_148999_i()
-    {
-        return this.field_149013_g;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149008_j()
-    {
-        return this.field_149021_h;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149006_k()
-    {
-        return this.field_149022_i;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_148993_l()
-    {
-        return this.field_149019_j;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149009_m()
-    {
-        return this.field_149020_k;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
-
-    @SideOnly(Side.CLIENT)
-    public void func_149002_g(int p_149002_1_)
-    {
-        this.field_149020_k = p_149002_1_;
-    }
 }
