--- decompiled\net\minecraft\network\play\server\S13PacketDestroyEntities.java
+++ src\main\java\net\minecraft\network\play\server\S13PacketDestroyEntities.java
@@ -1,23 +1,26 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class S13PacketDestroyEntities implements Packet
 {
+
     private int[] field_149100_a;
     private static final String __OBFID = "CL_00001320";
 
-    public S13PacketDestroyEntities() {}
+    public S13PacketDestroyEntities()
+    {
+    }
 
-    public S13PacketDestroyEntities(int ... p_i45211_1_)
+    public S13PacketDestroyEntities(int... p_i45211_1_)
     {
         this.field_149100_a = p_i45211_1_;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
@@ -43,16 +46,10 @@
     public void processPacket(INetHandlerPlayClient handler)
     {
         handler.handleDestroyEntities(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public int[] func_149098_c()
-    {
-        return this.field_149100_a;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
 }
