--- decompiled\net\minecraft\network\play\server\S18PacketEntityTeleport.java
+++ src\main\java\net\minecraft\network\play\server\S18PacketEntityTeleport.java
@@ -1,29 +1,32 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.entity.Entity;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraft.util.MathHelper;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class S18PacketEntityTeleport implements Packet
 {
+
     private int field_149458_a;
     private int field_149456_b;
     private int field_149457_c;
     private int field_149454_d;
     private byte field_149455_e;
     private byte field_149453_f;
     private boolean field_179698_g;
     private static final String __OBFID = "CL_00001340";
 
-    public S18PacketEntityTeleport() {}
+    public S18PacketEntityTeleport()
+    {
+    }
 
     public S18PacketEntityTeleport(Entity p_i45233_1_)
     {
         this.field_149458_a = p_i45233_1_.getEntityId();
         this.field_149456_b = MathHelper.floor_double(p_i45233_1_.posX * 32.0D);
@@ -70,52 +73,10 @@
     public void processPacket(INetHandlerPlayClient handler)
     {
         handler.handleEntityTeleport(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_149451_c()
-    {
-        return this.field_149458_a;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149449_d()
-    {
-        return this.field_149456_b;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149448_e()
-    {
-        return this.field_149457_c;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149446_f()
-    {
-        return this.field_149454_d;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public byte func_149450_g()
-    {
-        return this.field_149455_e;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public byte func_149447_h()
-    {
-        return this.field_149453_f;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public boolean func_179697_g()
-    {
-        return this.field_179698_g;
-    }
 }
