--- decompiled\net\minecraft\network\play\server\S21PacketChunkData.java
+++ src\main\java\net\minecraft\network\play\server\S21PacketChunkData.java
@@ -1,29 +1,32 @@
 package net.minecraft.network.play.server;
 
-import com.google.common.collect.Lists;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Iterator;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraft.world.chunk.Chunk;
 import net.minecraft.world.chunk.storage.ExtendedBlockStorage;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.collect.Lists;
+
 public class S21PacketChunkData implements Packet
 {
+
     private int chunkX;
     private int chunkZ;
     private S21PacketChunkData.Extracted field_179758_c;
     private boolean field_149279_g;
     private static final String __OBFID = "CL_00001304";
 
-    public S21PacketChunkData() {}
+    public S21PacketChunkData()
+    {
+    }
 
     public S21PacketChunkData(Chunk p_i45196_1_, boolean p_i45196_2_, int p_i45196_3_)
     {
         this.chunkX = p_i45196_1_.xPosition;
         this.chunkZ = p_i45196_1_.zPosition;
@@ -53,16 +56,10 @@
     public void processPacket(INetHandlerPlayClient handler)
     {
         handler.handleChunkData(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public byte[] func_149272_d()
-    {
-        return this.field_179758_c.data;
-    }
-
     protected static int func_180737_a(int p_180737_0_, boolean p_180737_1_, boolean p_180737_2_)
     {
         int j = p_180737_0_ * 2 * 16 * 16 * 16;
         int k = p_180737_0_ * 16 * 16 * 16 / 2;
         int l = p_180737_1_ ? p_180737_0_ * 16 * 16 * 16 / 2 : 0;
@@ -93,11 +90,11 @@
         Iterator iterator = arraylist.iterator();
         ExtendedBlockStorage extendedblockstorage1;
 
         while (iterator.hasNext())
         {
-            extendedblockstorage1 = (ExtendedBlockStorage)iterator.next();
+            extendedblockstorage1 = (ExtendedBlockStorage) iterator.next();
             char[] achar = extendedblockstorage1.getData();
             char[] achar1 = achar;
             int k = achar.length;
 
             for (int l = 0; l < k; ++l)
@@ -108,18 +105,18 @@
             }
         }
 
         for (iterator = arraylist.iterator(); iterator.hasNext(); j = func_179757_a(extendedblockstorage1.getBlocklightArray().getData(), extracted.data, j))
         {
-            extendedblockstorage1 = (ExtendedBlockStorage)iterator.next();
+            extendedblockstorage1 = (ExtendedBlockStorage) iterator.next();
         }
 
         if (p_179756_2_)
         {
             for (iterator = arraylist.iterator(); iterator.hasNext(); j = func_179757_a(extendedblockstorage1.getSkylightArray().getData(), extracted.data, j))
             {
-                extendedblockstorage1 = (ExtendedBlockStorage)iterator.next();
+                extendedblockstorage1 = (ExtendedBlockStorage) iterator.next();
             }
         }
 
         if (p_179756_1_)
         {
@@ -135,39 +132,16 @@
         return p_179757_2_ + p_179757_0_.length;
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
 
-    @SideOnly(Side.CLIENT)
-    public int func_149273_e()
-    {
-        return this.chunkX;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149271_f()
-    {
-        return this.chunkZ;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149276_g()
-    {
-        return this.field_179758_c.dataSize;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public boolean func_149274_i()
-    {
-        return this.field_149279_g;
-    }
-
     public static class Extracted
         {
+
             public byte[] data;
             public int dataSize;
             private static final String __OBFID = "CL_00001305";
         }
 }
