--- decompiled\net\minecraft\network\play\server\S2DPacketOpenWindow.java
+++ src\main\java\net\minecraft\network\play\server\S2DPacketOpenWindow.java
@@ -1,26 +1,29 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraft.util.IChatComponent;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class S2DPacketOpenWindow implements Packet
 {
+
     private int windowId;
     private String inventoryType;
     private IChatComponent windowTitle;
     private int slotCount;
     private int entityId;
     private static final String __OBFID = "CL_00001293";
 
-    public S2DPacketOpenWindow() {}
+    public S2DPacketOpenWindow()
+    {
+    }
 
     public S2DPacketOpenWindow(int incomingWindowId, String incomingWindowTitle, IChatComponent windowTitleIn)
     {
         this(incomingWindowId, incomingWindowTitle, windowTitleIn, 0);
     }
@@ -68,46 +71,10 @@
         {
             buf.writeInt(this.entityId);
         }
     }
 
-    @SideOnly(Side.CLIENT)
-    public int getWindowId()
-    {
-        return this.windowId;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
-
-    @SideOnly(Side.CLIENT)
-    public String getGuiId()
-    {
-        return this.inventoryType;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public IChatComponent getWindowTitle()
-    {
-        return this.windowTitle;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int getSlotCount()
-    {
-        return this.slotCount;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int getEntityId()
-    {
-        return this.entityId;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public boolean hasSlots()
-    {
-        return this.slotCount > 0;
-    }
 }
