--- decompiled\net\minecraft\network\play\server\S2EPacketCloseWindow.java
+++ src\main\java\net\minecraft\network\play\server\S2EPacketCloseWindow.java
@@ -1,19 +1,23 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 
 public class S2EPacketCloseWindow implements Packet
 {
+
     private int field_148896_a;
     private static final String __OBFID = "CL_00001292";
 
-    public S2EPacketCloseWindow() {}
+    public S2EPacketCloseWindow()
+    {
+    }
 
     public S2EPacketCloseWindow(int p_i45183_1_)
     {
         this.field_148896_a = p_i45183_1_;
     }
@@ -33,8 +37,8 @@
         buf.writeByte(this.field_148896_a);
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.func_180731_a((INetHandlerPlayClient)handler);
+        this.func_180731_a((INetHandlerPlayClient) handler);
     }
 }
