--- decompiled\net\minecraft\network\play\server\S36PacketSignEditorOpen.java
+++ src\main\java\net\minecraft\network\play\server\S36PacketSignEditorOpen.java
@@ -1,22 +1,25 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraft.util.BlockPos;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class S36PacketSignEditorOpen implements Packet
 {
+
     private BlockPos field_179778_a;
     private static final String __OBFID = "CL_00001316";
 
-    public S36PacketSignEditorOpen() {}
+    public S36PacketSignEditorOpen()
+    {
+    }
 
     public S36PacketSignEditorOpen(BlockPos p_i45971_1_)
     {
         this.field_179778_a = p_i45971_1_;
     }
@@ -34,16 +37,10 @@
     public void writePacketData(PacketBuffer buf) throws IOException
     {
         buf.writeBlockPos(this.field_179778_a);
     }
 
-    @SideOnly(Side.CLIENT)
-    public BlockPos func_179777_a()
-    {
-        return this.field_179778_a;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
 }
