--- decompiled\net\minecraft\network\play\server\S3APacketTabComplete.java
+++ src\main\java\net\minecraft\network\play\server\S3APacketTabComplete.java
@@ -1,21 +1,24 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class S3APacketTabComplete implements Packet
 {
+
     private String[] field_149632_a;
     private static final String __OBFID = "CL_00001288";
 
-    public S3APacketTabComplete() {}
+    public S3APacketTabComplete()
+    {
+    }
 
     public S3APacketTabComplete(String[] p_i45178_1_)
     {
         this.field_149632_a = p_i45178_1_;
     }
@@ -46,16 +49,10 @@
     public void processPacket(INetHandlerPlayClient handler)
     {
         handler.handleTabComplete(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public String[] func_149630_c()
-    {
-        return this.field_149632_a;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
 }
