--- decompiled\net\minecraft\network\play\server\S3EPacketTeams.java
+++ src\main\java\net\minecraft\network\play\server\S3EPacketTeams.java
@@ -1,22 +1,23 @@
 package net.minecraft.network.play.server;
 
-import com.google.common.collect.Lists;
 import java.io.IOException;
 import java.util.Collection;
 import java.util.Iterator;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraft.scoreboard.ScorePlayerTeam;
 import net.minecraft.scoreboard.Team;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import com.google.common.collect.Lists;
+
 public class S3EPacketTeams implements Packet
 {
+
     private String field_149320_a = "";
     private String field_149318_b = "";
     private String field_149319_c = "";
     private String field_149316_d = "";
     private String field_179816_e;
@@ -125,75 +126,21 @@
             buf.writeVarIntToBuffer(this.field_149317_e.size());
             Iterator iterator = this.field_149317_e.iterator();
 
             while (iterator.hasNext())
             {
-                String s = (String)iterator.next();
+                String s = (String) iterator.next();
                 buf.writeString(s);
             }
         }
     }
 
     public void processPacket(INetHandlerPlayClient handler)
     {
         handler.handleTeams(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public String func_149312_c()
-    {
-        return this.field_149320_a;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.processPacket((INetHandlerPlayClient)handler);
+        this.processPacket((INetHandlerPlayClient) handler);
     }
-
-    @SideOnly(Side.CLIENT)
-    public String func_149306_d()
-    {
-        return this.field_149318_b;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public String func_149311_e()
-    {
-        return this.field_149319_c;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public String func_149309_f()
-    {
-        return this.field_149316_d;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public Collection func_149310_g()
-    {
-        return this.field_149317_e;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149307_h()
-    {
-        return this.field_149314_f;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_149308_i()
-    {
-        return this.field_149315_g;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_179813_h()
-    {
-        return this.field_179815_f;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public String func_179814_i()
-    {
-        return this.field_179816_e;
-    }
 }
