--- decompiled\net\minecraft\network\play\server\S3FPacketCustomPayload.java
+++ src\main\java\net\minecraft\network\play\server\S3FPacketCustomPayload.java
@@ -1,23 +1,26 @@
 package net.minecraft.network.play.server;
 
-import io.netty.buffer.ByteBuf;
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+import io.netty.buffer.ByteBuf;
+
 public class S3FPacketCustomPayload implements Packet
 {
+
     private String channel;
     private PacketBuffer data;
     private static final String __OBFID = "CL_00001297";
 
-    public S3FPacketCustomPayload() {}
+    public S3FPacketCustomPayload()
+    {
+    }
 
     public S3FPacketCustomPayload(String channelName, PacketBuffer dataIn)
     {
         this.channel = channelName;
         this.data = dataIn;
@@ -44,30 +47,19 @@
     }
 
     public void writePacketData(PacketBuffer buf) throws IOException
     {
         buf.writeString(this.channel);
-        buf.writeBytes((ByteBuf)this.data);
+        buf.writeBytes((ByteBuf) this.data);
     }
 
     public void process(INetHandlerPlayClient p_180734_1_)
     {
         p_180734_1_.handleCustomPayload(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public String getChannelName()
-    {
-        return this.channel;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.process((INetHandlerPlayClient)handler);
+        this.process((INetHandlerPlayClient) handler);
     }
 
-    @SideOnly(Side.CLIENT)
-    public PacketBuffer getBufferData()
-    {
-        return this.data;
-    }
 }
