--- decompiled\net\minecraft\network\play\server\S42PacketCombatEvent.java
+++ src\main\java\net\minecraft\network\play\server\S42PacketCombatEvent.java
@@ -1,25 +1,29 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.entity.EntityLivingBase;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraft.util.CombatTracker;
 
 public class S42PacketCombatEvent implements Packet
 {
+
     public S42PacketCombatEvent.Event field_179776_a;
     public int field_179774_b;
     public int field_179775_c;
     public int field_179772_d;
     public String field_179773_e;
     private static final String __OBFID = "CL_00002299";
 
-    public S42PacketCombatEvent() {}
+    public S42PacketCombatEvent()
+    {
+    }
 
     public S42PacketCombatEvent(CombatTracker p_i45970_1_, S42PacketCombatEvent.Event p_i45970_2_)
     {
         this.field_179776_a = p_i45970_2_;
         EntityLivingBase entitylivingbase = p_i45970_1_.func_94550_c();
@@ -37,11 +41,11 @@
         }
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
-        this.field_179776_a = (S42PacketCombatEvent.Event)buf.readEnumValue(S42PacketCombatEvent.Event.class);
+        this.field_179776_a = (S42PacketCombatEvent.Event) buf.readEnumValue(S42PacketCombatEvent.Event.class);
 
         if (this.field_179776_a == S42PacketCombatEvent.Event.END_COMBAT)
         {
             this.field_179772_d = buf.readVarIntFromBuffer();
             this.field_179775_c = buf.readInt();
@@ -76,11 +80,11 @@
         p_179771_1_.handleCombatEvent(this);
     }
 
     public void processPacket(INetHandler handler)
     {
-        this.func_179771_a((INetHandlerPlayClient)handler);
+        this.func_179771_a((INetHandlerPlayClient) handler);
     }
 
     public static enum Event
     {
         ENTER_COMBAT,
@@ -90,10 +94,11 @@
         private static final String __OBFID = "CL_00002297";
     }
 
     static final class SwitchEvent
         {
+
             static final int[] field_179944_a = new int[S42PacketCombatEvent.Event.values().length];
             private static final String __OBFID = "CL_00002298";
 
             static
             {
