--- decompiled\net\minecraft\network\play\server\S45PacketTitle.java
+++ src\main\java\net\minecraft\network\play\server\S45PacketTitle.java
@@ -1,35 +1,38 @@
 package net.minecraft.network.play.server;
 
 import java.io.IOException;
+
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
 import net.minecraft.network.PacketBuffer;
 import net.minecraft.network.play.INetHandlerPlayClient;
 import net.minecraft.util.IChatComponent;
-import net.minecraftforge.fml.relauncher.Side;
-import net.minecraftforge.fml.relauncher.SideOnly;
 
+
 public class S45PacketTitle implements Packet
 {
+
     private S45PacketTitle.Type type;
     private IChatComponent message;
     private int fadeInTime;
     private int displayTime;
     private int fadeOutTime;
     private static final String __OBFID = "CL_00002287";
 
-    public S45PacketTitle() {}
+    public S45PacketTitle()
+    {
+    }
 
     public S45PacketTitle(S45PacketTitle.Type type, IChatComponent message)
     {
         this(type, message, -1, -1, -1);
     }
 
     public S45PacketTitle(int fadeInTime, int displayTime, int fadeOutTime)
     {
-        this(S45PacketTitle.Type.TIMES, (IChatComponent)null, fadeInTime, displayTime, fadeOutTime);
+        this(S45PacketTitle.Type.TIMES, (IChatComponent) null, fadeInTime, displayTime, fadeOutTime);
     }
 
     public S45PacketTitle(S45PacketTitle.Type type, IChatComponent message, int fadeInTime, int displayTime, int fadeOutTime)
     {
         this.type = type;
@@ -39,11 +42,11 @@
         this.fadeOutTime = fadeOutTime;
     }
 
     public void readPacketData(PacketBuffer buf) throws IOException
     {
-        this.type = (S45PacketTitle.Type)buf.readEnumValue(S45PacketTitle.Type.class);
+        this.type = (S45PacketTitle.Type) buf.readEnumValue(S45PacketTitle.Type.class);
 
         if (this.type == S45PacketTitle.Type.TITLE || this.type == S45PacketTitle.Type.SUBTITLE)
         {
             this.message = buf.readChatComponent();
         }
@@ -76,45 +79,15 @@
     public void process(INetHandlerPlayClient handler)
     {
         handler.handleTitle(this);
     }
 
-    @SideOnly(Side.CLIENT)
-    public S45PacketTitle.Type func_179807_a()
-    {
-        return this.type;
-    }
-
     public void processPacket(INetHandler handler)
     {
-        this.process((INetHandlerPlayClient)handler);
+        this.process((INetHandlerPlayClient) handler);
     }
 
-    @SideOnly(Side.CLIENT)
-    public IChatComponent func_179805_b()
-    {
-        return this.message;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_179806_c()
-    {
-        return this.fadeInTime;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_179804_d()
-    {
-        return this.displayTime;
-    }
-
-    @SideOnly(Side.CLIENT)
-    public int func_179803_e()
-    {
-        return this.fadeOutTime;
-    }
-
     public static enum Type
     {
         TITLE,
         SUBTITLE,
         TIMES,
