--- decompiled\net\minecraft\world\gen\feature\WorldGenCanopyTree.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenCanopyTree.java
@@ -1,18 +1,20 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockPlanks;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
 public class WorldGenCanopyTree extends WorldGenAbstractTree
 {
+
     private static final String __OBFID = "CL_00000430";
 
     public WorldGenCanopyTree(boolean p_i45461_1_)
     {
         super(p_i45461_1_);
@@ -157,11 +159,30 @@
                                 int k2 = p_180709_2_.nextInt(3) + 2;
                                 int l2;
 
                                 for (l2 = 0; l2 < k2; ++l2)
                                 {
-                                    this.func_175905_a(worldIn, new BlockPos(p_180709_3_.getX() + l1, k1 - l2 - 1, p_180709_3_.getZ() + i2), Blocks.log2, BlockPlanks.EnumType.DARK_OAK.getMetadata() - 4);
+                                    /* -- SpigotMCP NOTE --
+                                     * Paramater mappings:
+                                     * blockposition -> p_180709_3_
+                                     * world -> worldIn
+                                     * k2 -> l2
+                                     * -- Exact Same --
+                                     * l1 == l1
+                                     * k1 == k1
+                                     * i2 == i2
+                                     */
+                                    // Spigot start
+                                    BlockPos position = new BlockPos(p_180709_3_.getX() + l1, k1 - l2 - 1, p_180709_3_.getZ() + i2);
+                                    Block bl = worldIn.getBlockState(position).getBlock();
+
+                                    if (bl.getMaterial() == Material.air || bl.getMaterial() == Material.leaves)
+                                    {
+                                        this.func_175905_a(worldIn, position, Blocks.log2, BlockPlanks.EnumType.DARK_OAK.getMetadata() - 4);
+                                    }
+
+                                    // Spigot end
                                 }
 
                                 int j2;
 
                                 for (l2 = -1; l2 <= 1; ++l2)
