--- decompiled\net\minecraft\world\gen\feature\WorldGenTrees.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenTrees.java
@@ -1,18 +1,20 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockVine;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
 public class WorldGenTrees extends WorldGenAbstractTree
 {
+
     private final int minTreeHeight;
     private final boolean vinesGrow;
     private final int metaWood;
     private final int metaLeaves;
     private static final String __OBFID = "CL_00000438";
