--- decompiled\net\minecraft\world\gen\feature\WorldGenDoublePlant.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenDoublePlant.java
@@ -1,15 +1,17 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.BlockDoublePlant;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class WorldGenDoublePlant extends WorldGenerator
 {
+
     private BlockDoublePlant.EnumPlantType field_150549_a;
     private static final String __OBFID = "CL_00000408";
 
     public void func_180710_a(BlockDoublePlant.EnumPlantType p_180710_1_)
     {
