--- decompiled\net\minecraft\world\gen\feature\WorldGenIceSpike.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenIceSpike.java
@@ -1,17 +1,19 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
 import net.minecraft.world.World;
 
 public class WorldGenIceSpike extends WorldGenerator
 {
+
     private static final String __OBFID = "CL_00000417";
 
     public boolean generate(World worldIn, Random p_180709_2_, BlockPos p_180709_3_)
     {
         while (worldIn.isAirBlock(p_180709_3_) && p_180709_3_.getY() > 2)
@@ -54,20 +56,20 @@
                         {
                             Block block = worldIn.getBlockState(p_180709_3_.add(i1, k, j1)).getBlock();
 
                             if (block.getMaterial() == Material.air || block == Blocks.dirt || block == Blocks.snow || block == Blocks.ice)
                             {
-                                this.func_175906_a(worldIn, p_180709_3_.add(i1, k, j1), Blocks.packed_ice);
+                                worldIn.setBlockState(p_180709_3_.add(i1, k, j1), Blocks.packed_ice.getDefaultState()); // Spigot
                             }
 
                             if (k != 0 && l > 1)
                             {
                                 block = worldIn.getBlockState(p_180709_3_.add(i1, -k, j1)).getBlock();
 
                                 if (block.getMaterial() == Material.air || block == Blocks.dirt || block == Blocks.snow || block == Blocks.ice)
                                 {
-                                    this.func_175906_a(worldIn, p_180709_3_.add(i1, -k, j1), Blocks.packed_ice);
+                                    worldIn.setBlockState(p_180709_3_.add(i1, -k, j1), Blocks.packed_ice.getDefaultState()); // Spigot
                                 }
                             }
                         }
                     }
                 }
@@ -104,11 +106,11 @@
                         {
                             Block block1 = worldIn.getBlockState(blockpos1).getBlock();
 
                             if (block1.getMaterial() == Material.air || block1 == Blocks.dirt || block1 == Blocks.snow || block1 == Blocks.ice || block1 == Blocks.packed_ice)
                             {
-                                this.func_175906_a(worldIn, blockpos1, Blocks.packed_ice);
+                                worldIn.setBlockState(blockpos1, Blocks.packed_ice.getDefaultState()); // Spigot
                                 blockpos1 = blockpos1.down();
                                 --l1;
 
                                 if (l1 <= 0)
                                 {
