--- decompiled\net\minecraft\world\gen\feature\WorldGenHellLava.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenHellLava.java
@@ -1,16 +1,18 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class WorldGenHellLava extends WorldGenerator
 {
+
     private final Block field_150553_a;
     private final boolean field_94524_b;
     private static final String __OBFID = "CL_00000414";
 
     public WorldGenHellLava(Block p_i45453_1_, boolean p_i45453_2_)
