--- decompiled\net\minecraft\world\gen\feature\WorldGenDesertWells.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenDesertWells.java
@@ -1,22 +1,25 @@
 package net.minecraft.world.gen.feature;
 
-import com.google.common.base.Predicates;
 import java.util.Iterator;
 import java.util.Random;
+
 import net.minecraft.block.BlockSand;
 import net.minecraft.block.BlockSlab;
 import net.minecraft.block.BlockStoneSlab;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.block.state.pattern.BlockStateHelper;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.world.World;
 
+import com.google.common.base.Predicates;
+
 public class WorldGenDesertWells extends WorldGenerator
 {
+
     private static final BlockStateHelper field_175913_a = BlockStateHelper.forBlock(Blocks.sand).where(BlockSand.VARIANT, Predicates.equalTo(BlockSand.EnumType.SAND));
     private final IBlockState field_175911_b;
     private final IBlockState field_175912_c;
     private final IBlockState field_175910_d;
     private static final String __OBFID = "CL_00000407";
