--- decompiled\net\minecraft\world\gen\feature\WorldGenHugeTrees.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenHugeTrees.java
@@ -1,16 +1,18 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public abstract class WorldGenHugeTrees extends WorldGenAbstractTree
 {
+
     protected final int baseHeight;
     protected final int woodMetadata;
     protected final int leavesMetadata;
     protected int field_150538_d;
     private static final String __OBFID = "CL_00000423";
@@ -57,11 +59,11 @@
 
                 for (int k = -b0; k <= b0 && flag; ++k)
                 {
                     for (int l = -b0; l <= b0 && flag; ++l)
                     {
-                        if (p_175926_2_.getY() + j < 0 || p_175926_2_.getY() + j >= 256 || !this.func_150523_a(worldIn.getBlockState(p_175926_2_.add(k, j, l)).getBlock()))
+                        if (p_175926_2_.getY() + j < 0 || p_175926_2_.getY() + j >= 256 || (!this.func_150523_a(worldIn.getBlockState(p_175926_2_.add(k, j, l)).getBlock()) && worldIn.getBlockState(p_175926_2_.add(k, j, l)).getBlock() != Blocks.sapling))   // CraftBukkit - ignore our own saplings
                         {
                             flag = false;
                         }
                     }
                 }
