--- decompiled\net\minecraft\world\gen\feature\WorldGenBigMushroom.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenBigMushroom.java
@@ -1,16 +1,18 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class WorldGenBigMushroom extends WorldGenerator
 {
+
     private int mushroomType = -1;
     private static final String __OBFID = "CL_00000415";
 
     public WorldGenBigMushroom(int p_i2017_1_)
     {
