--- decompiled\net\minecraft\world\gen\feature\WorldGenShrub.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenShrub.java
@@ -1,16 +1,18 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class WorldGenShrub extends WorldGenTrees
 {
+
     private int field_150528_a;
     private int field_150527_b;
     private static final String __OBFID = "CL_00000411";
 
     public WorldGenShrub(int p_i2015_1_, int p_i2015_2_)
@@ -59,10 +61,17 @@
                             }
                         }
                     }
                 }
             }
+
+            // CraftBukkit start - Return false if gen was unsuccessful
         }
+        else
+        {
+            return false;
+        }
 
+        // CraftBukkit end
         return true;
     }
 }
