--- decompiled\net\minecraft\world\gen\feature\WorldGenLakes.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenLakes.java
@@ -1,18 +1,20 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.EnumSkyBlock;
 import net.minecraft.world.World;
 import net.minecraft.world.biome.BiomeGenBase;
 
 public class WorldGenLakes extends WorldGenerator
 {
+
     private Block field_150556_a;
     private static final String __OBFID = "CL_00000418";
 
     public WorldGenLakes(Block p_i45455_1_)
     {
