--- decompiled\net\minecraft\world\gen\feature\WorldGenLiquids.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGenLiquids.java
@@ -1,16 +1,18 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.world.World;
 
 public class WorldGenLiquids extends WorldGenerator
 {
+
     private Block field_150521_a;
     private static final String __OBFID = "CL_00000434";
 
     public WorldGenLiquids(Block p_i45465_1_)
     {
