--- decompiled\net\minecraft\world\gen\feature\WorldGeneratorBonusChest.java
+++ src\main\java\net\minecraft\world\gen\feature\WorldGeneratorBonusChest.java
@@ -1,9 +1,10 @@
 package net.minecraft.world.gen.feature;
 
 import java.util.List;
 import java.util.Random;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.material.Material;
 import net.minecraft.init.Blocks;
 import net.minecraft.tileentity.TileEntity;
 import net.minecraft.tileentity.TileEntityChest;
@@ -11,10 +12,11 @@
 import net.minecraft.util.WeightedRandomChestContent;
 import net.minecraft.world.World;
 
 public class WorldGeneratorBonusChest extends WorldGenerator
 {
+
     private final List field_175909_a;
     private final int itemsToGenerateInBonusChest;
     private static final String __OBFID = "CL_00000403";
 
     public WorldGeneratorBonusChest(List p_i45634_1_, int p_i45634_2_)
