--- decompiled\net\minecraft\block\properties\PropertyBool.java
+++ src\main\java\net\minecraft\block\properties\PropertyBool.java
@@ -1,12 +1,14 @@
 package net.minecraft.block.properties;
 
-import com.google.common.collect.ImmutableSet;
 import java.util.Collection;
 
+import com.google.common.collect.ImmutableSet;
+
 public class PropertyBool extends PropertyHelper
 {
+
     private final ImmutableSet allowedValues = ImmutableSet.of(Boolean.valueOf(true), Boolean.valueOf(false));
     private static final String __OBFID = "CL_00002017";
 
     protected PropertyBool(String name)
     {
