--- decompiled\net\minecraft\block\properties\PropertyInteger.java
+++ src\main\java\net\minecraft\block\properties\PropertyInteger.java
@@ -1,14 +1,16 @@
 package net.minecraft.block.properties;
 
-import com.google.common.collect.ImmutableSet;
-import com.google.common.collect.Sets;
 import java.util.Collection;
 import java.util.HashSet;
 
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+
 public class PropertyInteger extends PropertyHelper
 {
+
     private final ImmutableSet allowedValues;
     private static final String __OBFID = "CL_00002014";
 
     protected PropertyInteger(String name, int min, int max)
     {
