--- decompiled\net\minecraft\block\properties\PropertyDirection.java
+++ src\main\java\net\minecraft\block\properties\PropertyDirection.java
@@ -1,16 +1,19 @@
 package net.minecraft.block.properties;
 
+import java.util.Collection;
+
+import net.minecraft.util.EnumFacing;
+
 import com.google.common.base.Predicate;
 import com.google.common.base.Predicates;
 import com.google.common.collect.Collections2;
 import com.google.common.collect.Lists;
-import java.util.Collection;
-import net.minecraft.util.EnumFacing;
 
 public class PropertyDirection extends PropertyEnum
 {
+
     private static final String __OBFID = "CL_00002016";
 
     protected PropertyDirection(String name, Collection values)
     {
         super(name, EnumFacing.class, values);
